% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
\PassOptionsToPackage{dvipsnames,svgnames,x11names}{xcolor}
%
\documentclass[
  letterpaper,
  DIV=11,
  numbers=noendperiod]{scrreprt}

\usepackage{amsmath,amssymb}
\usepackage{lmodern}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\setlength{\emergencystretch}{3em} % prevent overfull lines
\setcounter{secnumdepth}{5}
% Make \paragraph and \subparagraph free-standing
\ifx\paragraph\undefined\else
  \let\oldparagraph\paragraph
  \renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
  \let\oldsubparagraph\subparagraph
  \renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi


\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}\usepackage{longtable,booktabs,array}
\usepackage{calc} % for calculating minipage widths
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother

\KOMAoption{captions}{tableheading}
\makeatletter
\makeatother
\makeatletter
\@ifpackageloaded{bookmark}{}{\usepackage{bookmark}}
\makeatother
\makeatletter
\@ifpackageloaded{caption}{}{\usepackage{caption}}
\AtBeginDocument{%
\ifdefined\contentsname
  \renewcommand*\contentsname{Table of contents}
\else
  \newcommand\contentsname{Table of contents}
\fi
\ifdefined\listfigurename
  \renewcommand*\listfigurename{List of Figures}
\else
  \newcommand\listfigurename{List of Figures}
\fi
\ifdefined\listtablename
  \renewcommand*\listtablename{List of Tables}
\else
  \newcommand\listtablename{List of Tables}
\fi
\ifdefined\figurename
  \renewcommand*\figurename{Figure}
\else
  \newcommand\figurename{Figure}
\fi
\ifdefined\tablename
  \renewcommand*\tablename{Table}
\else
  \newcommand\tablename{Table}
\fi
}
\@ifpackageloaded{float}{}{\usepackage{float}}
\floatstyle{ruled}
\@ifundefined{c@chapter}{\newfloat{codelisting}{h}{lop}}{\newfloat{codelisting}{h}{lop}[chapter]}
\floatname{codelisting}{Listing}
\newcommand*\listoflistings{\listof{codelisting}{List of Listings}}
\makeatother
\makeatletter
\@ifpackageloaded{caption}{}{\usepackage{caption}}
\@ifpackageloaded{subcaption}{}{\usepackage{subcaption}}
\makeatother
\makeatletter
\@ifpackageloaded{tcolorbox}{}{\usepackage[many]{tcolorbox}}
\makeatother
\makeatletter
\@ifundefined{shadecolor}{\definecolor{shadecolor}{rgb}{.97, .97, .97}}
\makeatother
\makeatletter
\makeatother
\ifLuaTeX
  \usepackage{selnolig}  % disable illegal ligatures
\fi
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\urlstyle{same} % disable monospaced font for URLs
\hypersetup{
  pdftitle={lotterhos-lab-protocols},
  pdfauthor={Katie Lotterhos},
  colorlinks=true,
  linkcolor={blue},
  filecolor={Maroon},
  citecolor={Blue},
  urlcolor={Blue},
  pdfcreator={LaTeX via pandoc}}

\title{lotterhos-lab-protocols}
\author{Katie Lotterhos}
\date{10/19/2022}

\begin{document}
\maketitle
\ifdefined\Shaded\renewenvironment{Shaded}{\begin{tcolorbox}[borderline west={3pt}{0pt}{shadecolor}, enhanced, breakable, interior hidden, frame hidden, sharp corners, boxrule=0pt]}{\end{tcolorbox}}\fi

\renewcommand*\contentsname{Table of contents}
{
\hypersetup{linkcolor=}
\setcounter{tocdepth}{2}
\tableofcontents
}
\bookmarksetup{startatroot}

\hypertarget{lotterhos-lab-protocols}{%
\chapter*{Lotterhos Lab Protocols}\label{lotterhos-lab-protocols}}
\addcontentsline{toc}{chapter}{Lotterhos Lab Protocols}

This site is dedicated to the protocols used in the Lotterhos Lab at the
Northeastern University Marine Science Center.

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

If you are new to the lab you may want to jump over to the
\href{https://drk-lo.github.io/lotterhoslabprotocols/welcome/}{welcome
page}. This page will help you navigate the lab, protocols and the MSC.

\part{General}

\hypertarget{expectations}{%
\chapter{Expectations}\label{expectations}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{these-are-expectations-for-everyone-in-the-lab}{%
\section*{\texorpdfstring{\textbf{These are expectations for everyone in
the
lab}}{These are expectations for everyone in the lab}}\label{these-are-expectations-for-everyone-in-the-lab}}
\addcontentsline{toc}{section}{\textbf{These are expectations for
everyone in the lab}}

\begin{itemize}
\item
  have fun!
\item
  respect yourself and commit to self care
\item
  respect each other
\item
  remember that cutting-edge science is not usually a linear process
\item
  practice inclusive communication
\item
  commit to documenting research progress and meeting notes in notebooks
  on a regular basis
\item
  commit to file management organization and communicate about the
  system with the team
\item
  commit to eliminating distractions during work hours 9am-5pm on
  weekdays. Distractions include personal texts, push notifications, and
  personal social media. Ways to eliminate them include turning off
  notifications and leaving your phone in another location.
\item
  commit to keeping a task management system, and communicating your
  priorities and milestones to your advisor(s)
\item
  commit to keeping a calendar for meetings and plans
\item
  recognize that shared space is shared resources, keep the lab bench
  clean, log out of shared computers so others can use them
\item
  finish what we start (within reason)
\item
  We all support each other when sh*t happens.
\end{itemize}

\hypertarget{welcome}{%
\chapter{Welcome}\label{welcome}}

An introduction to the lab for those that are new. This will help you
orient yourself to where things can be found and how to access them.

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{welcome-to-the-lotterhos-lab}{%
\section*{\texorpdfstring{\textbf{Welcome to the Lotterhos
Lab!}}{Welcome to the Lotterhos Lab!}}\label{welcome-to-the-lotterhos-lab}}
\addcontentsline{toc}{section}{\textbf{Welcome to the Lotterhos Lab!}}

This page will be a good start to getting yourself settled in the lab.
If you have any questions after reading through this, don't be afraid to
ask other members of the lab (and write the questions down so we can add
it to this page :D).

\hypertarget{safety-first}{%
\subsection*{\texorpdfstring{\textbf{Safety
First}}{Safety First}}\label{safety-first}}
\addcontentsline{toc}{subsection}{\textbf{Safety First}}

This will help you know where certain items are in the lab that will
help keep you and others safe:

\begin{itemize}
\item
  \textbf{Safety Data Sheets and Hazardous Waste}:

  \begin{itemize}
  \tightlist
  \item
    All safety data sheets, chemical inventory and hazardous waste
    labels are in this binder\ldots{}
    \includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/labSafetyBinder.jpg}
  \end{itemize}
\item
  \textbf{Hazardous Waste}:

  \begin{itemize}
  \item
    Clean Harbors now does pick up and transportation of chemical waste
    containers. They will come around to each of the labs that have
    filled out the \href{https://ehsa.nunet.neu.edu/neu_wp}{online pick
    up sheet} on \textbf{the last Tuesday of every month}.
  \item
    At this same time, Clean harbors should bring the extra hazardous
    waste containers that have been requested by members of the lab.
  \item
    If our secondary containment is overflowing in the middle of the
    month, notify Ryan and he will transport the containers to the
    central accumulation area.
  \item
    Lastly, do your best to fill out the hazardous waste labels
    correctly. Use the SDS of the chemical that you have been working
    with to help fill out the label.
  \end{itemize}
\end{itemize}

\hypertarget{shared-lab-space}{%
\subsection*{\texorpdfstring{\textbf{Shared Lab
Space}}{Shared Lab Space}}\label{shared-lab-space}}
\addcontentsline{toc}{subsection}{\textbf{Shared Lab Space}}

\begin{itemize}
\item
  The first door on the right as you enter the hall where labs are
  located is the shared lab space. Here is where you will be able to
  find most of the equipment that you need that is not in the lab.

  \begin{itemize}
  \item
    \textbf{OpenTrons pipetting robot}
  \item
    \textbf{BluePippin} - DNA Size Selection for Next-Gen Sequencing
  \item
    \textbf{Sonicator} - DNA Shearing
  \item
    \textbf{Agilent TapeStation} - Quantification and quality assessment
    of RNA, DNA (including genomic DNA) and protein samples.
  \item
    \textbf{Synergy HTX (`Plate reader')} - DNA Quantification of a
    large amount of samples.
  \item
    \textbf{Centrifuge} - Centrifuging large volume containers (Falcon
    tubes).
  \end{itemize}
\item
  While using the equipment is encouraged whenever it is needed, it is
  necessary that you log your use on the
  \href{https://docs.google.com/spreadsheets/d/1M0RQYTqQwLS7puzn443S4CVAufBNgKyocfamQ179edU/edit\#gid=0}{common
  lab space spreadsheet} for auditing purposes.

  \begin{itemize}
  \tightlist
  \item
    It is also important to log your uses of other shared equipment such
    as the autoclave, gel imager, NanoDrop, etc.
  \end{itemize}
\end{itemize}

\hypertarget{computing}{%
\subsection*{\texorpdfstring{\textbf{Computing}}{Computing}}\label{computing}}
\addcontentsline{toc}{subsection}{\textbf{Computing}}

\begin{itemize}
\tightlist
\item
  Our lab and some other labs at the MSC have access to a computing
  cluster (defiance) so we can run scripts and simulations that may take
  days to run. Refer to
  \href{https://drk-lo.github.io/lotterhoslabprotocols/defiance/}{this
  page} in order to gain access and learn how to use defiance.
\end{itemize}

\hypertarget{getting-started-in-the-lab}{%
\chapter{Getting Started in the Lab}\label{getting-started-in-the-lab}}

General Rules and Set up for Starting Out in the Lotterhos Lab

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{required-trainings-to-work-in-molecular-lab}{%
\section*{\texorpdfstring{\textbf{Required Trainings to Work In
Molecular
Lab}}{Required Trainings to Work In Molecular Lab}}\label{required-trainings-to-work-in-molecular-lab}}
\addcontentsline{toc}{section}{\textbf{Required Trainings to Work In
Molecular Lab}}

\begin{itemize}
\item
  Laboratory Safety Program Orientation - See Ryan Hill (Facilities
  Manager) for setting this up
\item
  Fundamentals of Laboratory Safety - See Ryan Hill (Facilities Manager)
  for setting this up
\item
  Autoclave Training - See Technicians in Ocean Genome Legacy for
  training
\item
  Shared Equipment Training - See Technicians in Ocean Genome Legacy for
  training
\item
  Layout and Tour/Labelling of the Lotterhos Lab - See current Grad
  Student (Sara Schaal) or Lab Manager (Thais Bittar)
\end{itemize}

\hypertarget{molecular-lab-layout-rules}{%
\section*{\texorpdfstring{\textbf{Molecular Lab Layout \&
Rules}}{Molecular Lab Layout \& Rules}}\label{molecular-lab-layout-rules}}
\addcontentsline{toc}{section}{\textbf{Molecular Lab Layout \& Rules}}

\begin{itemize}
\item
  Benches, squirt bottles, markers \& pipettes have color coded tapes to
  outline where and with which pipettes specific work is done.

  \begin{itemize}
  \item
    Yellow tape indicates pre-PCR molecular work.
  \item
    Light Blue tape indicates Post-PCR molecular work.
  \item
    Orange tape indicates RNA molecular work and is all done in the UV
    hood.
  \item
    No colored tape indicates ``dirty'' benches for phones, computers,
    lab notebooks, etc.
  \item
    Exceptions to these strict rules are the multichannel pipettes and
    the magnetic plate because we only have one of each.
  \end{itemize}
\item
  Tip boxes whenever a box is taken off the shelf to be used in the
  post-PCR area that box stays in that area and does not get placed back
  on the shelf. All boxes on the shelf if opened can only be used on the
  pre-PCR bench.
\item
  All benches that will be used for your molecular work for the day
  should be cleaned with 10\% bleach then 70\% ethanol. You should also
  do this after your done working for the day.
\item
  If using the UV hood for RNA work, all supplies that will be used for
  said work need to be placed in the UV hood and then light should be
  turned on for 1 hour to sterilize everything. DO NOT PUT YOUR SAMPLES
  IN THE UV HOOD WHEN THE LIGHT IS ON!
\item
  Gloves should be used when touching anything on all molecular benches,
  shelves above those benches (except the first shelf labeled with pink
  tape),fridge, -20oC freezer.

  \begin{itemize}
  \item
    NO GLOVES IN HALLS OR TOUCHING DOOR HANDLES. Can have one glove on
    but the other hand free to open doors.
  \item
    Keyboards in shared molecular space have labels for whether you
    should operate with or without gloves. Pay attention to these
    labels.
  \end{itemize}
\item
  Filtered tips are used for all molecular work (as budget allows).
  Fisherbrand tips are to be used ONLY ONLY ONLY with the multichannel
  pipettes.
\item
  -80oC freezer is linked to a alarm system to alert us if it goes below
  -70oC. If you have to take samples/reagents out or in please work
  quickly and efficiently to ensure it does not drop below that
  temperature. If you need to do a lot of reorganizing the -80, take
  breaks to allow the temperature to come back down to \textasciitilde{}
  -80oC.
\item
  The layout of the -80oC freezer can be found
  https://docs.google.com/spreadsheets/d/12TA0x9hPe6B8G34A2ID93ebjxdaGcEiMHAAajCyuQqo/edit\#gid=0
  please see current Grad Student (Sara Schaal or Alan Downey-Wall) to
  get training on how to fill out this document.
\item
  -20oC freezer has a general layout with the top shelf for main stocks,
  second shelf aliquots of kits and other reagents, third shelf is for
  DNA/RNA extractions, fourth shelf for tissue samples for DNA/RNA
  extraction.
\end{itemize}

\hypertarget{working-in-the-molecular-lab}{%
\section*{\texorpdfstring{\textbf{Working in the Molecular
Lab}}{Working in the Molecular Lab}}\label{working-in-the-molecular-lab}}
\addcontentsline{toc}{section}{\textbf{Working in the Molecular Lab}}

\begin{itemize}
\item
  If you need an aliquot of anything, see current lab manager (Thais
  Bittar). Main stocks of primers and adapters are in the -80oC and
  should ONLY ONLY ONLY be handled by the lab manager.

  \begin{itemize}
  \tightlist
  \item
    Label your aliquot clearly with a date and keep track of which
    aliquot you use for a given days protocol. This is to ensure that if
    you get DNA in your negative control you can track which aliquots
    you need to test for contamination. ** THIS IS VERY IMPORTANT **
  \end{itemize}
\item
  Always use a negative control in all steps from extraction, to
  quantification and to library prep.
\item
  Aliquots of positive controls for multiple species are kept in the
  -80oC freezer. See lab manager for aliquot. Use if you have work that
  is failing and you want to make sure its not the samples themselves.
\item
  ADD BUFFER AND FREEZER ID INFO FOR SAMPLES
\item
  -20oC for brief use of RNA -80oC for long term storage\\
  \strut \\
\end{itemize}

\hypertarget{freezer-alarm-system}{%
\chapter{Freezer Alarm System}\label{freezer-alarm-system}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

NU contact: Dennis Todd

Here is the link for the Rees system: http://NOR926.ReesCloud.net

Your logins are first initial, last name/names. All lowercase with no
spaces or extra characters, just letters.

\hypertarget{phone-alarm-handling-instructions}{%
\section*{\texorpdfstring{\href{https://drk-lo.github.io/lotterhoslabprotocols/img/PhoneAlarmHandlingInstructionsCentronNUcontactDennisTodd.pdf}{\textbf{Phone
alarm handling
instructions}}}{Phone alarm handling instructions}}\label{phone-alarm-handling-instructions}}
\addcontentsline{toc}{section}{\href{https://drk-lo.github.io/lotterhoslabprotocols/img/PhoneAlarmHandlingInstructionsCentronNUcontactDennisTodd.pdf}{\textbf{Phone
alarm handling instructions}}}

\begin{itemize}
\item
  Your pin codes to acknowledge alarms received on your phone is last 4
  digits of your contact numbers
\item
  Add the alarm phone to your list of contacts: the phone number that
  would be calling is Node 2 Phone 16176081423
\item
  When you log in, our freezers are labeled 120 (the -20C in Edwards)
  and 121 (the -80C in Edwards)
\end{itemize}

\textbf{The alarm will go off when the freezer rises above -70C}

\hypertarget{if-you-plan-on-loading-samples-in-the-freezer}{%
\section*{\texorpdfstring{\textbf{If you plan on loading samples in the
freezer}}{If you plan on loading samples in the freezer}}\label{if-you-plan-on-loading-samples-in-the-freezer}}
\addcontentsline{toc}{section}{\textbf{If you plan on loading samples in
the freezer}}

\begin{itemize}
\item
  Click on ``Manage Alarm''
\item
  Disable the alarm for an hour or so (there is a max that is allowable,
  but not sure what it is).
\item
  Please be careful not to let the temperature drop while you are
  loading samples. If the temperature reaches -65C, wait until it gets
  back up to -75C before you load more samples. It puts a lot of stress
  on the freezer.
\end{itemize}

\hypertarget{if-you-plan-to-defrost-the-freezer}{%
\section*{\texorpdfstring{\textbf{If you plan to defrost the
freezer}}{If you plan to defrost the freezer}}\label{if-you-plan-to-defrost-the-freezer}}
\addcontentsline{toc}{section}{\textbf{If you plan to defrost the
freezer}}

We need to contact Dennis Todd to request a 24hr+ freezer alarm shutoff.

\includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/freezerimage.png}

\includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/freezerinhibit.png}

\hypertarget{defrosting-the--80c-freezer}{%
\section*{\texorpdfstring{\textbf{Defrosting the -80C
freezer}}{Defrosting the -80C freezer}}\label{defrosting-the--80c-freezer}}
\addcontentsline{toc}{section}{\textbf{Defrosting the -80C freezer}}

\href{https://github.com/DrK-Lo/lotterhoslabprotocols/files/7305953/Freezers_Operating-manual_CryoCube-F570n-h-hw.pdf}{CryoCube-F570
operating manual}

\hypertarget{to-do-ahead-of-time}{%
\subsection*{\texorpdfstring{\textbf{To do ahead of
time:}}{To do ahead of time:}}\label{to-do-ahead-of-time}}
\addcontentsline{toc}{subsection}{\textbf{To do ahead of time:}}

1) Find a backup freezer for the samples and confirm that there's enough
space for all contents of our freezer; check the dimensions of the
available shelves in the backup freezer to see if it is the same as ours
(ideally, transfer whole racks to the equivalent shelf \& rack position
on the backup freezer to save time and avoid messing up the inventory).

2) Find help - defrosting can be done by one person, but having a second
pair of hands is very helpful and strongly recommended.

3) Familiarize yourself with the protocol to disable freezer alarms (see
above), with the materials that need to be on hand to defrost \& clean
the freezer and the overall protocol/timeline described below.

\hypertarget{the-day-of-transferdefrost}{%
\subsection*{\texorpdfstring{\textbf{The day of
transfer/defrost:}}{The day of transfer/defrost:}}\label{the-day-of-transferdefrost}}
\addcontentsline{toc}{subsection}{\textbf{The day of transfer/defrost:}}

4) Before you begin, double-check that the backup freezer(s)' alarm was
disabled (see above instructions); request that our alarm is disabled
remotely for 24h by emailing Dennis Todd @ northeastern; gather:

\begin{itemize}
\item
  freezer mittens
\item
  lab cart
\item
  a mop or large towel
\item
  `wet floor' sign
\item
  a cooler or large styrofoam box for lose boxes/bags
\item
  a bucket to collect water
\item
  clean cloths to wipe the freezer
\end{itemize}

5) \textbf{WEARING FREEZER MITTENS}, transfer whole racks to the backup
freezer: place all 5 racks from one shelf on our lab cart; roll the cart
to the backup freezer and place them on \textbf{the equivalent shelf and
in the same order as they were on our freezer}. Do this one shelf at the
time until the freezer is empty.

\begin{quote}
transfer racks in and out of freezers carefully but quickly to minimize
temp rising - note that the drawers slide open, so keep the racks
horizontal; monitor both freezer's temp as you transfer - shouldn't rise
above -65oC.
\end{quote}

6) Once empty:

\begin{itemize}
\item
  put up a ``wet floor'' sign,
\item
  close the CO2 cylinder valve,
\item
  switch off the local alarm box located on top of the freezer,
\item
  unlock the switch locking plate (front-bottom-right side of the
  freezer) and switch off the battery (top) and the main (bottom)
  switches,
\item
  unplug the freezer from the wall,
\item
  open all doors and let the freezer defrost, use a bucket to catch
  water dripping from the doors, use a mop to dry up the puddles.
\end{itemize}

7) Once defrosted: wipe the whole interior of the freezer, the inner and
outer doors and the seal using clean microfiber cloths (under the lab
sink) until completely dry.

8) Reverse to bring freezer back up: close all doors, plug freezer back
up, switch on main switch/battery switch and lock up the locking plate,
switch on alarm box up on top of the freezer. Monitor temperature to
ensure that it is appropriately decreasing all the way to -80oC.

\begin{quote}
during this stage, the freezer alarm may go off (not the alarms
associated with the remote system), just mute it using the door pannel
keys.
\end{quote}

\hypertarget{refill-our-freezer-at-least-24h-after-temp-is-back-to--80oc-on-the-empty-freezer}{%
\subsection*{\texorpdfstring{\textbf{Refill our freezer (at least 24h
after temp is back to -80oC on the empty
freezer)}}{Refill our freezer (at least 24h after temp is back to -80oC on the empty freezer)}}\label{refill-our-freezer-at-least-24h-after-temp-is-back-to--80oc-on-the-empty-freezer}}
\addcontentsline{toc}{subsection}{\textbf{Refill our freezer (at least
24h after temp is back to -80oC on the empty freezer)}}

9) Disable our freezer alarm for 1h (instructions above). Set a timer
for yourself for 55 min because you might need to extend alarm
cancellation. Check with the host lab on whether the alarm on the backup
freezer needs to be disabled.

10) \textbf{WEARING FREEZER MITTENS} transfer freezer racks back and
ensure they are placed back on the freezer position (shelf \& row) where
they before - so it continues to match the freezer database. Leave one
or two empty racks in the freezer for new boxes. \textbf{Monitor} the
temperature of both freezers during retrieval to ensure it does not
raise above 65oC.

11) If you need to take a break to allow the temperature to drop,
remember to \textbf{extend the alarm cancellation for another hour}.

12) Once all samples are back, wait around and make sure that the
freezer temp drops down to -80oC.

\hypertarget{example-timeline}{%
\subsection*{\texorpdfstring{\textbf{\emph{Example
timeline:}}}{Example timeline:}}\label{example-timeline}}
\addcontentsline{toc}{subsection}{\textbf{\emph{Example timeline:}}}

October 2021 defrost (Thais, Sara and Alan):

\begin{itemize}
\item
  OGL was able to hold our samples (\textbf{MUST} coordinate with OGL
  personnel ahead of time); OGL's freezers was completely empty, and
  shelves have the same dimensions as ours (plus an extra shelf); their
  freezer was not hooked up to an alarme system at the time.
\item
  We were able to transfer all racks to the equivalent positions into
  the backup freezer (one shelf at the time) without needing to wait too
  long for the temperature to stabilize. Because the backup freezer was
  empty, the temp rose to -58oC when we transferred the first shelf/5
  racks.
\item
  Our freezer was defrosted, cleaned/dried and turned back on within 2
  hours.
\item
  Transfer-defrost-cleaning took from 11am to 2pm.
\item
  Temp down to -80oC within another 2-3 hours (all done by 5pm).
\item
  The CO2 valve was reopened the following day.
\item
  The following week, we transferred the racks back. Temp went up to
  65oC after 3 shelves were tranferred so we took 1h break, then
  tranferred the remaining racks and samples.
\item
  Took about 2h for the temp to go back down to -80oC (freezer alarm
  went off once).
\end{itemize}

\hypertarget{freezer-maintenance}{%
\section*{\texorpdfstring{\textbf{Freezer
maintenance}}{Freezer maintenance}}\label{freezer-maintenance}}
\addcontentsline{toc}{section}{\textbf{Freezer maintenance}}

Maintenance has been done with: https://alertscientific.com/

\hypertarget{autoclave-ogl}{%
\chapter{Autoclave (OGL)}\label{autoclave-ogl}}

The protocol for using the autoclave in OGL

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{overview}{%
\section*{\texorpdfstring{\textbf{Overview}}{Overview}}\label{overview}}
\addcontentsline{toc}{section}{\textbf{Overview}}

An autoclave chamber is used to sterilize laboratory equipment,
including liquids, by subjecting them to steam created at a high
temperature and pressure. Autoclaving should be performed on the
following items:

\begin{itemize}
\item
  Containers used while working with microorganisms
\item
  Glassware used to store food for experimental organisms
\item
  Biohazard waste containers
\item
  Pipette tip boxes
\end{itemize}

\hypertarget{tips-before-starting}{%
\section*{\texorpdfstring{\textbf{Tips before
starting}}{Tips before starting}}\label{tips-before-starting}}
\addcontentsline{toc}{section}{\textbf{Tips before starting}}

\begin{itemize}
\item
  Review or take the autoclave training on the Northeastern
  Environmental Health and Safety
  \href{https://northeastern.bioraft.com/node/1895187}{website}.
\item
  It is mandatory to scheudle a 20 minute in-person training with Rosie
  Falco Poulin (r.falco@northeastern.edu) at the MSC to follow up on the
  online course.
\item
  Always unscrew jars or vials as the high pressure may cause them to
  explode if tightened.
\end{itemize}

\hypertarget{protocol}{%
\section*{\texorpdfstring{\textbf{Protocol}}{Protocol}}\label{protocol}}
\addcontentsline{toc}{section}{\textbf{Protocol}}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Fill out autoclave sign up in binder
\item
  Make sure all knobs are at zero
\item
  Load samples onto metal sheet inside the autoclave
\item
  Reset red arm to zero (top gauge). This arm keeps track of temperature
  maximum during the run
\item
  Turn ON autoclave with green switch
\item
  Turn bottom knob to fill water - wait until water reaches the front of
  the groove (might already have water in it)
\item
  Switch knob to sterilize
\item
  Seal the door by turning clockwise (not too tight)
\item
  Look on the chart:

  \begin{itemize}
  \item
    determine what you are autoclaving (wrapped vs unwrapped)
  \item
    Set temperature to the desired setting
  \item
    Example: Unwrapped = 3850 model, cold start, 32 min, 134 degrees
  \end{itemize}
\item
  Set timer (NOTE: Setting timer starts the run)
\item
  Come back 52 minutes later (32 in autoclave + 20 to depressurize) - If
  the red gauge reads lower than your desired temperature, it didn't get
  hot enough (due to too much water or too much glassware). Run it
  again, if needed, for slightly longer (\textasciitilde40 mins) and
  make sure it reaches the appropriate temperature (indicated by red
  arm)
\item
  Crack open door to boil off water
\item
  OPTIONAL: Set bottom knob to extra dry (if there is still a lot of
  moisture
\item
  Run for 20-30 min on dry
\item
  Otherwise, set knobs back to zero and turn off autoclave
\item
  Make sure the red arm is back to 0\\
\end{enumerate}

\hypertarget{ordering-supplies-and-equipment}{%
\chapter{Ordering Supplies and
Equipment}\label{ordering-supplies-and-equipment}}

The protocol for ordering supplies and equipment for the lab

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{placing-orders-on-mymarketplace}{%
\section*{\texorpdfstring{\textbf{Placing Orders on
myMarketplace}}{Placing Orders on myMarketplace}}\label{placing-orders-on-mymarketplace}}
\addcontentsline{toc}{section}{\textbf{Placing Orders on myMarketplace}}

Most supplies for the lab are ordered through companies that
Northeastern has accounts with, such as Fisher, Sigma Aldrich or VWR To
order from these companies:

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Log on to myNortheastern
\item
  Go to Services \& Links -\textgreater{} myMarketplace (Under `Finance
  Services for Faculty\textbackslash Staff')
\item
  Log on to myMarketplace
\item
  Scroll down and select the company you want to order from (e.g.,
  Fisher Scientific)
\item
  Search for your desired item
\item
  Choose price and quantity
\item
  Add to cart
\item
  Continue adding items to cart
\item
  Once all items are in the cart, go to cart symbol in upper right hand
  corner and select `View Cart'
\item
  Review order. If everything is correct go to `Return Cart to
  Purchasing Application'
\item
  Click `Submit'. You will return to myMarketplace
\item
  From here, choose the index (the budget code, depending on the
  project), the account code (typically Scientific Supplies), and how
  much to cover with this budget code (typically 100\%)
\item
  Click `Send to Approval' \textbf{DO NOT FORGET STEPS BELOW}
\item
  Once you receive a confirmation e-mail with the receipt of the
  purchase forward it to the Business and Operations Manager (currently
  Liz Magee) and David Dawson. CC Katie on the e-mail
\item
  \textbf{In the body of the e-mail put the index number it was charged
  to}
\item
  Download PDF of the receipt, rename to date\_company\_item
  (e.g.~20180505\_Fisher\_Gloves) and add it to the Dropbox Folder :
  LotterhosLab/orders and receipts \textbf{Katie will need these
  receipts to do budgets at the end of the month}
\end{enumerate}

\hypertarget{placing-orders-through-external-website}{%
\section*{\texorpdfstring{\textbf{Placing Orders through External
Website}}{Placing Orders through External Website}}\label{placing-orders-through-external-website}}
\addcontentsline{toc}{section}{\textbf{Placing Orders through External
Website}}

Sometimes orders will have to be placed outside these companies. Many
sites will have a lab login (get from Katie), if there isn't one, you'll
create one \textbf{Make sure the account is tax-free; you can get the
tax numbers from Katie to set this up}

To order from external companies:

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Login with lab login
\item
  Place order, making sure no tax is applied (it's hard to get
  reimbursed)
\item
  Pay with American Express card
\item
  If using your e-mail, and you receive receipt, rename PDF with same
  convention as above and add to Dropbox
\item
  Typically lab login will be through Katie's e-mail; confirm with Katie
  that she has received the receipt
\end{enumerate}

\hypertarget{receiving-supplies-and-equipment}{%
\chapter{Receiving Supplies and
Equipment}\label{receiving-supplies-and-equipment}}

The protocol for what to do when orders come in

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{receiving-orders}{%
\section*{\texorpdfstring{\textbf{Receiving
Orders}}{Receiving Orders}}\label{receiving-orders}}
\addcontentsline{toc}{section}{\textbf{Receiving Orders}}

Typically, Sonya will e-mail you when you or the lab has received a
package. Try to pick them up as soon as possible, \textbf{especially if
it needs to be refrigerated}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  When you receive the package, unbox, and make sure you have the
  packing slip
\item
  \textbf{LABEL EVERYTHING WITH `LOTTERHOS'}
\item
  Put items in correct location
\item
  Add order to lab Quartzy inventory, specifying location. \textbf{Tip:
  if you put in company and correct item number, it will autofill
  details}
\item
  Cardboard boxes can be placed in hall for Roberto to pick up or can be
  walked to dumpster
\item
  Bring packing slip to Sonya's office
\item
  If item is important, you can also send message on \#orders slack
  channel saying that it has been received
\end{enumerate}

\hypertarget{receiving-chemical-orders}{%
\section*{\texorpdfstring{\textbf{Receiving Chemical
Orders}}{Receiving Chemical Orders}}\label{receiving-chemical-orders}}
\addcontentsline{toc}{section}{\textbf{Receiving Chemical Orders}}

There are extra steps required when receiving chemicals

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  When you receive the package, unbox the chemical, and make sure you
  have the packing slip
\item
  \textbf{LABEL WITH `LOTTERHOS'}
\item
  Add pre-printed barcode to chemical container
\end{enumerate}

\begin{figure}

{\centering \includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/chemicallabels.JPG}

}

\caption{``\,''}

\end{figure}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Log into myNortheastern
\item
  Go to Services \& Links -\textgreater{} EH\&S Assistant (under
  `Environmental Health and Safety')
\item
  Login (If you don't have a username and password, e-mail Ryan Hill)
\item
  Choose `Inventory'
\item
  Choose `Add Chemical'
\item
  Fill in information. Inventory\# is the barcode number 10 Click
  `Save/Return'
\item
  \textbf{Check compatability by looking up Safety Data Sheet} and put
  items in correct location
\item
  Add order to lab Quartzy inventory, specifying location. \textbf{Tip:
  if you put in company and correct item number, it will autofill
  details}
\item
  Cardboard boxes can be placed in hall for Roberto to pick up or can be
  walked to dumpster
\item
  Bring packing slip to Sonya's office
\item
  If item is important, you can also send message on \#orders slack
  channel saying that it has been received
\end{enumerate}

\hypertarget{purchasing}{%
\chapter{Purchasing}\label{purchasing}}

Any purchases made with grant funds above \$10K need to be routed
through the University's procurement department. To be ahead of the
process, make sure you have at least 3 quotes from different vendors for
each \textgreater\$10K purchase.

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{purchases-under-10k}{%
\section*{\texorpdfstring{\textbf{Purchases under
\$10K:}}{Purchases under \$10K:}}\label{purchases-under-10k}}
\addcontentsline{toc}{section}{\textbf{Purchases under \$10K:}}

\textbf{Amazon Business Account:}

\begin{itemize}
\item
  Search and make purchases like you normally do through Amazon (billing
  and shipping address should be already saved in the account settings).
\item
  Print order to pdf and save it using the file name format below, then
  place the pdf file into the folder Order and Receipts (located in
  Google Drive, LotLab\_Shared\_Resources)
\end{itemize}

\textbf{Northeastern My Marketplace:}

\begin{itemize}
\item
  Fill out and submit your
  \href{https://github.com/thais-neu/Thais_lab_protocols/files/7541425/myMarketplace-Access-Form.pdf}{Access
  Form} to gain access and purchase through My Marketplace.
\item
  Log in to your MyNortheastern portal, go to Services and Links, under
  Finance Serces for Faculty \& Staff, select myMarketplace.
\item
  Save myMarketplace to you favorites so it's easier to find next time.
\item
  We typically look through the Fisher Scientific catalog first - using
  key works or product/catalog number.
\item
  Add your items to cart; when done shopping, click on the cart (view
  cart) and then on ``Return Carto to Purchasing Application'' and then
  ``Submit''.
\item
  On the next page, double-check Shipping Information and select:

  \begin{itemize}
  \item
    Index to charge from (Ask Katie if not sure),
  \item
    Account code: `Scientific Supplies',
  \item
    Click ``Send to Approval'',
  \item
    Print that page (or the order confirmation you should receive by
    email) to a pdf and save it to Orders and Receipts using the format
    below for file name.
  \end{itemize}
\end{itemize}

\textbf{Save orders/invoces to a pdf file and into the folder `Orders
and Receipts' using the following file name:}

YYYYMMDD\_Amazon\_whatpurchase.pdf

\begin{itemize}
\item
  example: 20210602\_Amazon\_labnotebooks.pdf
\item
  example: 20211115\_Marketplace\_dissectingscissors.pdf
\end{itemize}

\hypertarget{reimbursement}{%
\chapter{Reimbursement}\label{reimbursement}}

How to get reimbursed (with example for mileage)

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{missing-a-receipt}{%
\section*{\texorpdfstring{\textbf{Missing a
receipt?}}{Missing a receipt?}}\label{missing-a-receipt}}
\addcontentsline{toc}{section}{\textbf{Missing a receipt?}}

You will need to fill out a
\href{https://drive.google.com/open?id=15O6iVj2p2Qf_em6hiHeyg3co1UQEIbzU\&authuser=k.lotterhos\%40gmail.com\&usp=drive_fs}{missing
receipt affidavit}.

\hypertarget{reimbursement-form-for-gradsundergrads}{%
\section*{\texorpdfstring{\textbf{Reimbursement form for
grads/undergrads}}{Reimbursement form for grads/undergrads}}\label{reimbursement-form-for-gradsundergrads}}
\addcontentsline{toc}{section}{\textbf{Reimbursement form for
grads/undergrads}}

\includegraphics{https://user-images.githubusercontent.com/6870125/184234551-393ff3b2-7bf7-42df-b7d8-9ff7ca0fb58c.png}

\hypertarget{getting-reimbursed-through-concur-postdocs-staff}{%
\section*{\texorpdfstring{\textbf{Getting Reimbursed Through Concur
(postdocs,
staff)}}{Getting Reimbursed Through Concur (postdocs, staff)}}\label{getting-reimbursed-through-concur-postdocs-staff}}
\addcontentsline{toc}{section}{\textbf{Getting Reimbursed Through Concur
(postdocs, staff)}}

There may be times when you have to get reimbursed for something you
purchased with your own funds. Below is an outline of how to get
reimbursed, with an example for auto mileage.

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Login to myNortheastern
\item
  Go to Services \& Linkes -\textgreater{} Concur (under `Finance
  Services for Faculty/Staff')
\item
  Go to the `Expense' tab at the top
\item
  Create a New Report
\item
  Fill in `Report Name' (e.g., ``Bio-Oce\_Fieldwork\_March'') which
  should include grant name
\item
  Fill in `Business Purpose' (``fieldwork for research'')
\item
  Fill in grant information with index number (depending on project)
\item
  And choose domestic or international for `Type of Expense' (typically
  domestic)
\item
  Click `Next'
\item
  \textbf{You'll likely get a warning message}. `Next' through it
\item
  In `New Expense' menu, choose ``73025 Travel-Auto Mileage''
\item
  Put in waypoints; you can put in multiple points if you went to
  multiple field sites
\item
  Click `Add Mileage to Expense'
\item
  Click Save and then Submit Report
\end{enumerate}

\hypertarget{scope-camera-mvi-nikon-smz-1000}{%
\chapter{Scope Camera MVI Nikon SMZ
1000}\label{scope-camera-mvi-nikon-smz-1000}}

Installing and configuring MVI Scope Camera

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{install}{%
\section*{\texorpdfstring{\textbf{Install}}{Install}}\label{install}}
\addcontentsline{toc}{section}{\textbf{Install}}

Download software from Google Drive folder (Lotterhos Lab \textgreater{}
equipment \textgreater{} MVI Nikon).

Run as Administrator (PC) to install.

\emph{note - a message might pop up saying the key is missing; ignore
and continue installing without the key}

Under NIS-Elements Package, select `Local Option'

Under Cameras, select Nikon Fi3

Under Modules, select EDF, Local options \textgreater{} Language Pack

Under Devices, select Manual Microscope and Nikon SMZ manual

\hypertarget{calibration-settings}{%
\section*{\texorpdfstring{\textbf{Calibration
settings}}{Calibration settings}}\label{calibration-settings}}
\addcontentsline{toc}{section}{\textbf{Calibration settings}}

Plug dongle and camera to laptop.

At start up: first power up camera, second run application software
(reverse to shut down).

Acquire \textgreater{} Select driver \textgreater{} Nikon DS Fi3

Acquire Camera light path \textgreater{} relay lens = 0.70 x
\textgreater{} ok

View \textgreater{} Acquisition controls \textgreater{} SMZ Pad
\textgreater{} Nosepiece \textgreater{} Plan Apo1x

\emph{note - for zoom calibration, double check the the value on the
software matches what you're using on the scope (knob on the right side
of the ocular) and update the software each time you use a different
zoom before taking a photo (zoom cal can not be corrected after an image
has been captured).}

\hypertarget{some-recommended-settings}{%
\section*{\texorpdfstring{\textbf{Some recommended
settings}}{Some recommended settings}}\label{some-recommended-settings}}
\addcontentsline{toc}{section}{\textbf{Some recommended settings}}

\textbf{Continuous exposure (AE)}: will automatically adjust to optimal
exposure as you move the sample around.

\textbf{Auto-white}: place a white sheet of paper under the scope or
select a known-white spot on the image and hit auto-white to define it
as white (helps with calibrating color).

\textbf{Auto Calibrated grid (this is shown by default but can also be
found in View \textgreater{} Layers)}: will add a calibrated grid,
several `shapes' available (round, squared, etc). Lines and legend in
this grid can be edited for size, color, etc.

\textbf{Show Scale}: displays a calibrated (remember to double-check the
zoom) scale that can be edited (font, color, etc).

\textbf{Annotations and Measurements (View \textgreater{} Analysis
Controls)}: this is a good window to have open and docked on the
right-side panel. Many resources to measure the image/sample, all
calibrated (remember to check zoom). A table with measurements is
automatically created and can be exported as an excel file.

\textbf{Capture and store}: if you need to take multiple pictures and
want them automatically assigned a given file name. Choose file name
under `Prefix' and photos will be saved sequentially 001, 002, etc.

\textbf{Manual large image (Acquire \textgreater{} Manual large image)}:
will compose an image of an object larger than the field of view. Turn
off auto-shading (`on' is the default). Move the object around in any
direction and snap a photo as you place different parts of the object in
focus. The software will stich the frames together and create a photo of
the larger object.

\hypertarget{creating-barcoded-labels}{%
\chapter{Creating Barcoded Labels}\label{creating-barcoded-labels}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\emph{note - TSC printer is in the shared equipment lab, hooked up to
the Plate Reader laptop.}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Create an excel spreadsheet with the sample IDs for your labels. Type
  ``ID'' on cell A1 and list your label IDs on the following rows (you
  can use more than one column).
\item
  Launch the software BarTender Designer (a shortcut should be pinned to
  the taskbar).
\item
  In the welcome screen, `Open a recently used BarTender document' or
  `Open an existing BarTender document'.

  \emph{a. If you accidentally close the welcome screen, go to File
  \textgreater{} New \textgreater{} New Document Wizard \textgreater{}
  Existing BarTender Document \textgreater{} Select \textgreater{}
  Select a template document \textgreater{} Open \textgreater{} Finish
  (not Next).}

  \emph{b. This will define the printer model (we only have one) and the
  type of stock label (we only have one) so you don't need to enter
  these paramenters manually (saves time).}
\item
  File \textgreater{} Save as \textgreater{} `yourfilename'.
\item
  On the document you just created (`yourfilename'), click on the
  barcode on the screen and delete it, so you have a blank label.
\item
  Create \textgreater{} Barcode \textgreater{} code 93 \textgreater{}
  then click anywhere on the label to place the barcode.

  \emph{a. This will place a barcode on the label on the screen. Roughly
  adjust the size of the barcode to fit the label.}
\item
  On the left side panel, right-click on Database Fields \textgreater{}
  Database Connection Setup \textgreater{} Configure Connection
  \textgreater{} Browse and select the Excel file with your sample IDs
  \textgreater{} Finish (not Next) \textgreater{} Ok.

  \emph{a. This will add your sample ID next to the barcode on the
  label.}

  \emph{b. If the window ``The format of the database in `yourexcelfile'
  does not match the old database'' pops up, click yes. Then click to
  highlight Sheet\$1 and hit Finish.}
\item
  On the left side panel, click on `Barcode' to highlight it. Then go to
  Edit \textgreater{} Properties.
\item
  Under Human Readability \textgreater{} Placement, select ``Right''
  (this is recommended because of the shape of labels we currently have;
  if using another label, edit barcode placement and sample ID placement
  accordingly). Horizontal offset: ``0.03'' is recommended, but will
  depend on the length of the barcode and sample ID.

  \emph{a. Symbology and Size: the type, size and placement of the
  barcode on the label can be edited here, if needed.}

  \emph{b. Font, Text Format, Border and Position can be edited here, if
  needed.}
\item
  Under Datasource, right-click the embedded data (should appear as
  12345678), and delete.
\item
  Right-click on Datasource \textgreater{} New \textgreater{} Database
  field \textgreater{} Next \textgreater{} Select field name (``ID'', or
  whatever else is in the first row of you excel file) and delete
  whatever is in `Sample Field Data' \textgreater{} Finish.

  \emph{a. This should show the first sample ID next to the barcode on
  the label. You might need to edit font size to fit the label. (Don't
  exit barcode properties). In Font, select the appropriate size under
  Point Size \textgreater{} Close.}
\item
  Re-size the barcode and sample ID within the margins of the label if
  necessary, by adjusting the barcode window on the label and/or editing
  barcode properties until your label looks the way you want (keep in
  mind the labels are small so try to maximize font size to fill out
  most of the label.

  \emph{a. At the bottom of the screen, you should see the number of
  labels you're creating -- should match the rows on the excel file. Use
  the arrows to scroll through the labels and make sure they all fit
  within the margins of the labels. Adjust accordingly, if needed.}
\item
  Hit Print on top of the page or File \textgreater{} Print. It's
  recommended to print the first \textasciitilde5 labels first to see
  how the come out. To do that, select ``Selected'' under ``Queried
  Records'' and type 1-5 in ``Selected Records''. Hit Print. If looks
  ok, print the rest. If you want to print all at once, select ``All''
  under ``Queried Records''.
\item
  Cut off your labels from the printer, close document, close software.
\end{enumerate}

\hypertarget{shipping-of-hazardous-materials}{%
\chapter{Shipping of Hazardous
Materials}\label{shipping-of-hazardous-materials}}

This is applicable in cases when hazardous materials are shipped in
`small quantity'. Double-check with NU's Environmental and Health Safety
(EHS) what volume is considered `small quantity' before deciding that
your shipment qualifies as excepted quantities.

Example: sample vials with 95\% ethanol -- up to 100 mL (total ethanol
volume).

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{packing-the-materials}{%
\subsection*{\texorpdfstring{\textbf{1) Packing the
materials:}}{1) Packing the materials:}}\label{packing-the-materials}}
\addcontentsline{toc}{subsection}{\textbf{1) Packing the materials:}}

a) The primary container(s), e.g.~ethanol vials, must be leak-proof and
allow space for liquid expansion,

b) If shipping multiple vials, use a grid box or rack to stabilize them,

c) Primary containers must be placed inside a secondary container (e.g.,
a ziplock bag) filled with enough appropriate absorbert material (e.g.,
paper towels) in case primary container spills.

\hypertarget{what-northeasterns-ehs-needs-from-you}{%
\subsection*{\texorpdfstring{\textbf{2) What Northeastern's EHS needs
from
you:}}{2) What Northeastern's EHS needs from you:}}\label{what-northeasterns-ehs-needs-from-you}}
\addcontentsline{toc}{subsection}{\textbf{2) What Northeastern's EHS
needs from you:}}

a) Northeastern University, Hazardous Material Shipping Declaration
Form, filled out and signed,

b) A statement that the package contents qualify as hazardous goods
\textbf{in excepted quantities},

c) A letter informing the (1) package contents, (2) name and location of
the person receiving the package and (3) your (the shipper's) name,
contact info (phone and address) and job title.

\begin{quote}
• I incorporated b) into c) for brevity.
\end{quote}

\textbf{Then make a pdf file of these documents combined and email to
EHS: biosafety@northeastern.edu and to lab manager:
t.bittar@northeastern.edu}

Example of the combined letter/statement:

\begin{verbatim}
To whom it may concern,
  
This package is intended for Dr. Jane Smith, in San Diego, CA. Inside this package you will find 1 grid box with 50 vials containing 1.9 mL of ethanol each at 95% concentration (total volume of ethanol in the package is 95 mL).

The content of this package qualifies as “Dangerous Goods in Excepted Quantities”.

If you have any questions regarding this package, please contact Dr. Thais Bittar at 781-581-XXXX or t.bittar@northeastern.edu
   
Best regards,

T. Bittar, Lotterhos Lab Manager, Northeastern University
\end{verbatim}

\hypertarget{packing-the-shipping-box}{%
\subsection*{\texorpdfstring{\textbf{3) Packing the shipping
box:}}{3) Packing the shipping box:}}\label{packing-the-shipping-box}}
\addcontentsline{toc}{subsection}{\textbf{3) Packing the shipping box:}}

\emph{Do not use FedEx boxes, envelopes, sleeves or other non-rigid
mailers for liquids; you will need a sturdy container, such as a
cardboard box, that is at least 100 x 100 mm or 4 x 4 in.}

\hypertarget{what-goes-inside-the-box}{%
\subsubsection*{\texorpdfstring{\textbf{What goes inside the
box:}}{What goes inside the box:}}\label{what-goes-inside-the-box}}
\addcontentsline{toc}{subsubsection}{\textbf{What goes inside the box:}}

a) Your packed vials;

b) MSDS, if applicable (for ethanol, in this example);

c) A copy of the letter/statement listed items 2 b) and 2 c).

\hypertarget{what-goes-on-the-outside-of-the-box}{%
\subsubsection*{\texorpdfstring{\textbf{What goes on the outside of the
box:}}{What goes on the outside of the box:}}\label{what-goes-on-the-outside-of-the-box}}
\addcontentsline{toc}{subsubsection}{\textbf{What goes on the outside of
the box:}}

a) The ``Dangerous Goods in Excepted Quantity'' label

\begin{itemize}
\item
  must be at least 100 x 100 mm or 4 x 4 in,
\item
  must conform with shape, color, format and text of the official label
  (see lab manager),
\item
  must be filled out with DOT hazmat class and your full name (class 3
  for DOT for flammable ignitable liquids in this example; see lab
  manager).
\end{itemize}

b) The shipping label (from FedEx, UPS, etc) -- you do not need to
inform the carrier that you are shipping hazmats as long as the package
is labeled correctly (see front desk associate to create shipping label
and schedule pickup).

\hypertarget{fedex-account}{%
\chapter{FedEx Account}\label{fedex-account}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{lotterhos-lab-fedex-account-info}{%
\section*{\texorpdfstring{\textbf{Lotterhos Lab FedEx account
info}}{Lotterhos Lab FedEx account info}}\label{lotterhos-lab-fedex-account-info}}
\addcontentsline{toc}{section}{\textbf{Lotterhos Lab FedEx account
info}}

\begin{itemize}
\item
  Account number: ---------432
\item
  User Id: Lotterhos
\item
  Pw: see private file
\item
  Place of birth: see private file
\item
  My account: 432
\end{itemize}

Set up the account to receive notifications such as invoice ready to
pay, package picked up, package delivered, etc.

\hypertarget{create-shipping-labels}{%
\section*{\texorpdfstring{\textbf{Create shipping
labels}}{Create shipping labels}}\label{create-shipping-labels}}
\addcontentsline{toc}{section}{\textbf{Create shipping labels}}

Go to the
\href{https://www.fedex.com/shipping/shipEntryAction.do?method=doEntry\&link=1\&locale=en_US\&urlparams=us\&sType=FShipping}{account
page} \& login.

\begin{itemize}
\item
  Shipping \textgreater{} Create Shipment
\item
  Fill out the shipping info and hit Ship - this will create a label
  (print or send it to the shipper).
\item
  Drop off labelled package by front office.
\end{itemize}

The account will be billed later, once the package is picked up.

\hypertarget{paying-invoices}{%
\section*{\texorpdfstring{\textbf{Paying
invoices}}{Paying invoices}}\label{paying-invoices}}
\addcontentsline{toc}{section}{\textbf{Paying invoices}}

Go to the \href{https://www.fedex.com/en-us/home.html}{account page} \&
login, click on your name (top right corner of page) then select ``View
and pay bill'' from the dropdown.

Any due invoices will be under `All-Open', select the open invoice(s)
and hit `Pay'.

Select the payment method (Lotterhos credit card) and complete payment
(\emph{payment info should be already saved in the account}).

Print the receipt to a pdf and place it `Order and Receipts' (Lotterhos
Shared Lab Resources) using the following format for file name:
YYYMMDD\_FedEx\_whatshipped.pdf (example: for ethanol vials shipped to
Maine to collect BSB fin clips, 20210712\_FedEx\_BSBvials.pdf)\\

\hypertarget{clearing-lab-waste}{%
\chapter{Clearing Lab Waste}\label{clearing-lab-waste}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{solid-waste}{%
\section*{\texorpdfstring{\textbf{Solid
waste}}{Solid waste}}\label{solid-waste}}
\addcontentsline{toc}{section}{\textbf{Solid waste}}

Most solid waste in our lab consists of pipette tips, small tubes
(microcentrifuge tubes, cryovials, tube strips) and, less frequently, a
larger tube or a plate used with samples/chemical solutions.\\
These used materials are discarded inside waste containers called `mayo
jars'.\\
If you are going to generate any solid waste, \textbf{label a mayo jar
BEFORE you start} using the standard
\href{https://github.com/DrK-Lo/lotterhoslabprotocols/blob/gh-pages/notebook/Haz\%20Waste\%20labels.pdf}{Hazardous
Waste Label}.\\
\textbf{NEVER dispose of solid waste inside a unlabelled mayo jar.}

\textbf{Sharps}: we also use disposable razor blades frequently when
dicing up animal tissue for extractions; all used razor blades MUST be
disposed of inside the RED sharps containters in our lab; close the
containers' lid when done for the day. Sharps containers are typically
not labelled with the Haz Waste Label. \textbf{NEVER leave razor blades
on the benches}

\hypertarget{liquid-waste}{%
\section*{\texorpdfstring{\textbf{Liquid
waste}}{Liquid waste}}\label{liquid-waste}}
\addcontentsline{toc}{section}{\textbf{Liquid waste}}

Our liquid waste typically consists of extraction/library prep isolation
and wash solutions, preservative solutions, such as ethanol and
formalin, and plastic-glassware cleaning solutions, such as HCl.\\
Liquid waste should \textbf{NEVER be mixed indiscriminately} - this can
cause serios issues such as thermal reactions/explosions and production
of harmful fumes.\\
Therefore, keep one waste bottle for each type of waste solution you are
generating.\\
If you know your work will be generating liquid waste, label each waste
bottle \textbf{BEFORE you start}, using the standard
\href{https://github.com/DrK-Lo/lotterhoslabprotocols/blob/gh-pages/notebook/Haz\%20Waste\%20labels.pdf}{Hazardous
Waste Label}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Managing open stream waste containers:

  \begin{itemize}
  \item
    label the waste containers: fill out all fields EXCEPT the ``date
    container filled'' - to keep an open stream bottle (meaning you
    still expect to produce more of that same waste in the upcoming
    days/weeks), leave the `date container filled' field empty;
  \item
    keep all waste containers in their secondary containment (gray
    trays) and in the satelitte accumulation areas in our lab as much as
    possible
  \item
    make sure that tops are closed tightly once you are done using the
    waste bottle.
  \end{itemize}
\item
  Haz waste disposal: once a waste bottle is at maximum waste volume
  (3/4 of the total volume) or you are done generating that particular
  waste

  \begin{itemize}
  \item
    date the bottle (``date container filled'')
  \item
    immediately schedule pickup:

    \begin{itemize}
    \item
      on BioRAFT: Lotterhos Lab \textgreater{} Waste \textgreater{}
      Submit New Waste Pickup Request
    \item
      email Ryan with number and location of items to be picked up.
    \end{itemize}
  \end{itemize}
\end{enumerate}

Note: Once the label has been dated, pickup request \textbf{must} be
submitted within 3 days; to be sure it gets done, we recommend that
pickup is scheduled immediately (or at leaste before the end of the
day). Because we are not on the main campus, Ryan Hill (MSC facilities)
does all of our waste pickup so \textbf{you must email him to ask him to
come to the lab for that specific purpose}; don't hesitate to follow up
if he hasn't responded or you see that waste has not been collected
within a day or 2. For liquid waste, let him know specifically which
containers to pickup (for example, put all containers to be picke up in
a separate tray inside the Satelite Accumulation Area and leave a note
on the tray saying - ``please pickup all 3 bottles inside this tray'').
For mayo jars or sharps container, I usually place them on the bench
right across from the lab door, inside a cardboard box if possible, with
a note as well.

\hypertarget{managing-chemical-inventory}{%
\chapter{Managing Chemical
Inventory}\label{managing-chemical-inventory}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

Chemicals (including kits) purchased by any lab member, for any purpose,
and stored in one of our labs (Edwards bldg and bunker lab)
\textbf{MUST} be barcoded and added to our chemical inventory which is
hosted in the BioRAFT platform.

\begin{itemize}
\tightlist
\item
  This is done at the container/individual kit level (i.e., not at the
  chemical substance or class level) - for example, if we have 3 bottles
  of ethanol in the inventory and buy a new one, that individual new
  bottle must be assigned its own barcode and added to the inventory.
\end{itemize}

\hypertarget{as-soon-as-a-new-chemicalkit-is-delivered-to-our-lab-assign-it-a-barcoded-label.-you-must-use-the-barcoded-labels-provided-by-northeasterns-ehs.-they-start-with-nu000-followed-by-a-unique-id-number.}{%
\subsection*{\texorpdfstring{\textbf{1. As soon as a new chemical/kit is
delivered to our lab, assign it a barcoded label. You MUST use the
barcoded labels provided by Northeastern's EHS. They start with
``NU000'' followed by a unique ID
number.}}{1. As soon as a new chemical/kit is delivered to our lab, assign it a barcoded label. You MUST use the barcoded labels provided by Northeastern's EHS. They start with ``NU000'' followed by a unique ID number.}}\label{as-soon-as-a-new-chemicalkit-is-delivered-to-our-lab-assign-it-a-barcoded-label.-you-must-use-the-barcoded-labels-provided-by-northeasterns-ehs.-they-start-with-nu000-followed-by-a-unique-id-number.}}
\addcontentsline{toc}{subsection}{\textbf{1. As soon as a new
chemical/kit is delivered to our lab, assign it a barcoded label. You
MUST use the barcoded labels provided by Northeastern's EHS. They start
with ``NU000'' followed by a unique ID number.}}

Always place the label where it is easier to see (i.e., not on the
bottom of the container, where you'd have to lift the bottle to find
it). I typically place it right above the main label.

\hypertarget{add-chemicalkit-to-the-labs-chemical-inventory.}{%
\subsection*{\texorpdfstring{\textbf{2. Add chemical/kit to the lab's
chemical
inventory.}}{2. Add chemical/kit to the lab's chemical inventory.}}\label{add-chemicalkit-to-the-labs-chemical-inventory.}}
\addcontentsline{toc}{subsection}{\textbf{2. Add chemical/kit to the
lab's chemical inventory.}}

\begin{itemize}
\item
  Go to the lab's BioRAFT page: Lotterhos Lab \textgreater{} Lab Profile
\item
  Click on the ChemTracker tab \textgreater{} Add inventory
\item
  Search chemical/kit by name, name or CAS
  number\textsuperscript{\href{https://drk-lo.github.io/lotterhoslabprotocols/general_chemical_inventory/\#fn:1}{1}}
\item
  Fill out all required fields
\item
  \textbf{Unique container ID} is where you enter the unique barcode ID
\item
  Click Submit
\end{itemize}

\hypertarget{remove-items-from-the-inventory-when-chemicalskits-are-used-up-expired-or-disposed-of-for-any-reason.}{%
\subsection*{\texorpdfstring{\textbf{3. Remove items from the inventory
when chemicals/kits are used up, expired or disposed of for any
reason.}}{3. Remove items from the inventory when chemicals/kits are used up, expired or disposed of for any reason.}}\label{remove-items-from-the-inventory-when-chemicalskits-are-used-up-expired-or-disposed-of-for-any-reason.}}
\addcontentsline{toc}{subsection}{\textbf{3. Remove items from the
inventory when chemicals/kits are used up, expired or disposed of for
any reason.}}

\begin{itemize}
\item
  Remove the barcode label and discard it in the trash (i.e., don't
  stick it to any other surface and \textbf{don't reuse it} for another
  chemical).
\item
  Go to the lab's BioRAFT page: Lotterhos Lab \textgreater{} Lab Profile
\item
  Click on the ChemTracker tab \textgreater{} Chem Tracker
\item
  At the bottom of the page, scroll right until you find the ``Container
  ID'' and the ``Remove'' columns
\item
  Click ``Remove'' and confirm ``Remove'' to exclude the entry from the
  inventory.
\item
  Rinse and deface the bottle, then discard or reuse to store chemical
  waste.

  \begin{itemize}
  \item
    Discard a defaced bottle: label the bottle using an ``empty
    container'' label.
  \item
    Reuse a defaced bottle: label the bottle with a ``empty container''
    label until you start using it for waste and replace it with a
    ``Chemical Waste Label'' once you star using it for waste.
  \end{itemize}
\end{itemize}

Go to Chem Tracker (Lotterhos Lab \textgreater{} Lab Profile
\textgreater{} ChemTracker tab \textgreater{} ChemTracker) to
\textbf{View} and/or \textbf{Edit} the inventory; you can also export
the inventory to an excel spreadsheet if you need to sort or print the
full list - this is useful when doing self-inspections and inventory
checks.

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  CAS registry numbers are unique, unmistakable identifiers for chemical
  substances. A CAS Registry Number itself has no inherent chemical
  significance but provides an unambiguous way to identify a chemical
  substance or molecular structure when there are many possible
  systematic, generic, proprietary or trivial names. CAS Registry
  Numbers are used in many other public and private databases as well as
  chemical inventory listings and, of course, are included in all
  CAS-produced
  databases.~\href{https://drk-lo.github.io/lotterhoslabprotocols/general_chemical_inventory/\#fnref:1}{↩}\\
  \strut \\
  \strut \\
  \strut \\
\end{enumerate}

\part{Professional Development}

\hypertarget{reducing-cognitive-load}{%
\chapter{Reducing Cognitive Load}\label{reducing-cognitive-load}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{reducing-load-on-ourselves}{%
\section*{\texorpdfstring{\textbf{Reducing load on
ourselves}}{Reducing load on ourselves}}\label{reducing-load-on-ourselves}}
\addcontentsline{toc}{section}{\textbf{Reducing load on ourselves}}

\begin{itemize}
\item
  \href{https://hamberg.no/gtd/}{GTD Getting Things Done}

  \begin{itemize}
  \tightlist
  \item
    How to organize your lists - see website for details
  \end{itemize}
\item
  Organize tasks into short term critical, short term not critical, long
  term less critical, not critical
\item
  \href{https://www.fastcompany.com/40507663/the-7-step-guide-to-achieving-inbox-zero-and-staying-there-in-2018}{Declutter
  your inbox}

  \begin{itemize}
  \item
    archive immediately no action items
  \item
    do it if one minute or less
  \item
    if need more time, set a ``snooze'' to get it later
  \item
    keep emails 5 sentences or less
  \item
    make templates
  \item
    unsubscribe from all lists or bundle promos
  \item
    stop organizing your email
  \item
    turn off notifications and create custom notifications for specific
    people

    \begin{itemize}
    \tightlist
    \item
      only check a few times a day
    \end{itemize}
  \end{itemize}
\item
  \href{https://productivityist.com/efficient-thinking/}{Art of
  efficient thinking}

  \begin{itemize}
  \item
    Download your mind

    \begin{itemize}
    \item
      Dictation service, voice recording, copytalk
    \item
      Idea journal - storage space for ideas - not time sensitive
    \item
      Google keep - writing things down
    \item
      Avoid straining your brain by reducing decisions made each day
      (prepare lunch/clothes day before). Reduce amount of unimportant
      information.
    \end{itemize}
  \end{itemize}
\item
  {[}Bullet Journals{]}

  \begin{itemize}
  \item
    Lots of You Tube videos
  \item
    We discussed whether this would actually reduce cognitive load
    because it takes so much time
  \end{itemize}
\item
  \href{https://hackernoon.com/minimizing-the-cognitive-load-in-your-life-bc14f4309fb2}{Hacker
  Noon}

  \begin{itemize}
  \item
    If something is incomplete, it's in your mind
  \item
    Slight positive things have a small mental positive effect, while
    slight negative things have a large negative effect
  \item
    Read books over articles, avoid newspapers, facebook, etc.
  \item
    Writing alot reduces cognitive load
  \item
    ``decision exhaustion'' has not held up in replicated studies
  \item
    \texttt{deep\ work} and \texttt{flow\ states} - as you do something
    more often, you need a shorter buildup to the point to become super
    productive
  \end{itemize}
\end{itemize}

\hypertarget{unsuscribe-glei}{%
\section*{\texorpdfstring{\textbf{Unsuscribe
(Glei)}}{Unsuscribe (Glei)}}\label{unsuscribe-glei}}
\addcontentsline{toc}{section}{\textbf{Unsuscribe (Glei)}}

\begin{itemize}
\item
  \emph{Negativity bias} (Goleman research)

  \begin{itemize}
  \item
    If we perceive our email as positive, the receiver perceives it as
    neutral
  \item
    If we perceive our email as neutral, the receiver perceives it as
    negative
  \item
    This happens because of absence of normal social cues
  \item
    Play close attention to phrasing
  \end{itemize}
\item
  \emph{Busy bias}

  \begin{itemize}
  \item
    every email is colored by how busy we are and if we feel we have
    time for it
  \item
    10 years ago we spent 0 time on social media because it didn't
    exist. Now we spend 1.72 hours a day.
  \item
    corrosive choice anxiety
  \end{itemize}
\item
  Map work onto your calendar

  \begin{itemize}
  \item
    Start your day with meaningful work
  \item
    Daily routine - check email once at noon and once at end of day

    \begin{itemize}
    \tightlist
    \item
      \emph{Switching cost} - creative work requires at least 25 minutes
      to get into the task after an interruption
    \end{itemize}
  \item
    Outline your goals
  \item
    Signal to noise - thinking on the decade span instead of the day
    span ``Days are long but the decades are short''
  \end{itemize}
\item
  \emph{Close the loop} - if you are asking a question, propose an
  answer
\item
  Attention to detail -

  \begin{itemize}
  \item
    Compose first, add address later
  \item
    Any grammatical errors show lack of attention to detail
  \item
    To remove someone from a chain, bcc them and make sure to mention it
    in the email
  \end{itemize}
\end{itemize}

\hypertarget{reducing-the-load-we-impose-on-others}{%
\section*{\texorpdfstring{\textbf{Reducing the load we impose on
others}}{Reducing the load we impose on others}}\label{reducing-the-load-we-impose-on-others}}
\addcontentsline{toc}{section}{\textbf{Reducing the load we impose on
others}}

\begin{itemize}
\item
  \href{http://theelearningcoach.com/learning/reduce-cognitive-load/}{Six
  strategies}

  \begin{itemize}
  \item
    maintain signal to noise ratio - get rid of all the noise
  \item
    promote generative strategies - get learner to interact and generate
    their own ideas
  \item
    write concisely - .
  \item
    supplantive strategies - small steps to bigger problem/tasks
  \item
    collaborative learning - divide cognitive process across several
    individuals
  \item
    provide cognitive aids - lists, tools, strategies to offload task
    complexity

    \begin{itemize}
    \tightlist
    \item
      mindmap on kumu.com
    \end{itemize}
  \end{itemize}
\item
  \href{https://blog.bananatag.com/internal-comms/cognitive-load-impact-on-internal-emails}{Use
  F-pattern emails}

  \begin{itemize}
  \item
    5-second test - can someone else understand it in 5 sec?
  \item
    Predictability - email templates
  \item
    Visible flow - heading organization and white space
  \item
    Ease of skimming/scanning -

    \begin{itemize}
    \item
      people skim emails in an F pattern - use pointed headings to
      convey message and short paragraphs
    \item
      most important information at the top
    \end{itemize}
  \end{itemize}
\item
  \href{https://hbr.org/2016/11/how-to-write-email-with-military-precision}{Write
  an email with military precision}

  \begin{itemize}
  \item
    informative subject line with keywords

    \begin{itemize}
    \tightlist
    \item
      what do I want by emailing this person and distill it to one word
    \end{itemize}
  \item
    bottom line up front (BLUF)

    \begin{itemize}
    \tightlist
    \item
      most important information at the top
    \end{itemize}
  \item
    Don't make recipient scroll - keep it short
  \item
    Use active voice\\
  \end{itemize}
\end{itemize}

\hypertarget{how-to-publish-a-paper}{%
\chapter{How to Publish a Paper}\label{how-to-publish-a-paper}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

The average time it takes to publish a paper after you first submit it
for review is 6-9 months. Each round of review takes 1-4 months, each
round of revisions take 1-2 months, and then it can take a month or more
to get the proofs and approve the final version for publication. Plan
accordingly. If I want a paper published in a particular calendar year,
I try to get it submitted by March.

\href{https://www.nature.com/articles/d41586-019-02918-5}{Read this
great ariticle with tips from a famous fiction author on how to write a
great paper}

\hypertarget{books-that-helped-me-learn-how-to-write}{%
\section*{\texorpdfstring{\textbf{Books that helped me learn how to
write}}{Books that helped me learn how to write}}\label{books-that-helped-me-learn-how-to-write}}
\addcontentsline{toc}{section}{\textbf{Books that helped me learn how to
write}}

Writing has two facets: building good writing habits and actually
putting words together in a way that makes sense to other people.
Knowing the difference between \emph{e.g.} and \emph{i.e.} saves your
co-authors time explaining it to you. Grammatical errors frustrate
reviewers and panelists, and make them skeptical of attention to detail
and the quality of the work.

\begin{itemize}
\tightlist
\item
  Check out this Nature article
  \href{https://www.nature.com/articles/d41586-020-00917-5}{``Kick-Start
  Writing''}
\end{itemize}

\hypertarget{building-good-writing-habits}{%
\subsubsection*{\texorpdfstring{\textbf{Building good writing
habits}}{Building good writing habits}}\label{building-good-writing-habits}}
\addcontentsline{toc}{subsubsection}{\textbf{Building good writing
habits}}

\begin{itemize}
\item
  Writing Your Dissertation in 15min/day
\item
  How to Write Alot
\end{itemize}

\hypertarget{putting-words-together}{%
\subsubsection*{\texorpdfstring{\textbf{Putting words
together}}{Putting words together}}\label{putting-words-together}}
\addcontentsline{toc}{subsubsection}{\textbf{Putting words together}}

\begin{itemize}
\item
  Style: the elements of clarity and grace
\item
  Funk and White
\end{itemize}

\hypertarget{thinking-backwards-from-a-deadline}{%
\section*{\texorpdfstring{\textbf{Thinking backwards from a
deadline}}{Thinking backwards from a deadline}}\label{thinking-backwards-from-a-deadline}}
\addcontentsline{toc}{section}{\textbf{Thinking backwards from a
deadline}}

This timeline assumes you have a full rough draft 3 months before a
deadline. Prior to this time, make sure you have a target journal and
stay within the word limit as you write.

\begin{itemize}
\item
  Week -12: Rough draft ready. A rough draft should not be missing many
  citations, and should have near-final versions of all the figures and
  tables.
\item
  Week -9 to -10: Meeting with co-authors to go over current draft of
  manuscript, approve methods, decide what figures to include, etc. Keep
  in mind this meeting could take a couple of weeks to get on the books
  (so should be planned in week -11).
\item
  Week -8: Full draft ready, with high quality figures. Make sure this
  draft is formatted for submission (e.g.~word counts, figure number,
  etc.).
\item
  Week - 6 to 7: Two weeks for comments from co-authors. Continue to
  improve manuscript while waiting for comments.
\item
  Week -5: Incorporate comments from co-authors. Plan another meeting
  with co-authors if needed.
\item
  Week - 4: Make sure final draft is formatted for submission (e.g.~word
  counts, figure number, etc.)
\item
  Week -2 to -3: Two weeks for final comments from all co-authors.
\item
  Week -1: Fine editing, grammar, figure references, bibliography edits.
  Leave 2-3 hours for submission.
\end{itemize}

\hypertarget{manuscript-figures-and-supplement}{%
\section*{\texorpdfstring{\textbf{Manuscript, Figures, and
Supplement}}{Manuscript, Figures, and Supplement}}\label{manuscript-figures-and-supplement}}
\addcontentsline{toc}{section}{\textbf{Manuscript, Figures, and
Supplement}}

Every manuscript typically has 3 parts that I submit, the text of the
manuscript (written collaboratively in google docs), the figures with
legends (in latex), and the supplement (in latex). I use LaTex for
figures and supplements because it makes them high quality, and is
easier to control the size of the figures as they are displayed on the
page.

\hypertarget{use-this-template-to-kick-start-your-manuscript-in-google-docs}{%
\subsection*{\texorpdfstring{\textbf{Use this template to kick-start
your manuscript in google
docs}}{Use this template to kick-start your manuscript in google docs}}\label{use-this-template-to-kick-start-your-manuscript-in-google-docs}}
\addcontentsline{toc}{subsection}{\textbf{Use this template to
kick-start your manuscript in google docs}}

\href{https://docs.google.com/document/d/15FYgxaSEQ6krNOlOq0qHmz0mD2No_tcRdVpG-xG8EXw/edit}{Here
is a link to a template, with important things to remember about
writing} * Having links to relevant documents at the top of the
manuscript can be helpful, including a link to the editor letter,
response to reviewers, notes and unused text, etc.

\begin{itemize}
\item
  Google docs offers a platform for collaboration that does not involve
  emailing each other many versions, which is inefficient. All changes
  are available immediately to everyone. It also has a good version
  control system (Click on ``All changes saved in Drive''). I use this
  system to name specific drafts that I may want to go back to.

  \begin{itemize}
  \tightlist
  \item
    Use ``Outlines''. Labeling different sections with headers and
    sub-headers will automatically create an outline. ``View
    ---\textgreater{} Show Document Outline'' can make it fast to
    navigate.
  \end{itemize}
\item
  Have a separate document for notes, logs, deleted text, and posting
  old versions of the figures or trial versions of figures.
\item
  Make sure to use text to create tables, because typesetters will not
  be happy if you create images or LaTeX tables
\item
  I like to have a shared folder with pdfs of all the references
\item
  Have a target journal and stay within the word limit as you write.
\end{itemize}

\hypertarget{make-figures-high-resolution-for-submission-with-latex}{%
\subsection*{\texorpdfstring{\textbf{Make figures high resolution for
submission with
Latex}}{Make figures high resolution for submission with Latex}}\label{make-figures-high-resolution-for-submission-with-latex}}
\addcontentsline{toc}{subsection}{\textbf{Make figures high resolution
for submission with Latex}}

\begin{itemize}
\item
  A good way to frustrate reviewers is to submit high resolution figures
  in an online system that appends them to the end of your manuscript
  with no Figure ID or legend.
\item
  To make my reviewer's life easier, I use LaTeX to create on figures
  document with figure legends on the same page as the figures. When the
  final (final) document is accepted for publication, then I just upload
  the high resolution figures.
  \href{https://drive.google.com/drive/folders/1mbvkizpPnM4CjkTTj2ARENtEqDwhXYDU?usp=sharing}{Template}
\item
  It will be important to double check that the Figure Legends in the
  Manuscript match the Figure Legends in the LaTeX document
\end{itemize}

\hypertarget{make-high-quality-supplemental-materials-with-latex}{%
\subsection*{\texorpdfstring{\textbf{Make high quality Supplemental
Materials with
Latex}}{Make high quality Supplemental Materials with Latex}}\label{make-high-quality-supplemental-materials-with-latex}}
\addcontentsline{toc}{subsection}{\textbf{Make high quality Supplemental
Materials with Latex}}

\begin{itemize}
\tightlist
\item
  I use the same LaTex template to produce high-quality Supplementary
  Materials.
  \href{https://drive.google.com/drive/folders/1mbvkizpPnM4CjkTTj2ARENtEqDwhXYDU?usp=sharing}{Template}
\end{itemize}

\hypertarget{inform-all-authors-of-the-authorship-policy}{%
\section*{\texorpdfstring{\textbf{Inform all authors of the authorship
policy}}{Inform all authors of the authorship policy}}\label{inform-all-authors-of-the-authorship-policy}}
\addcontentsline{toc}{section}{\textbf{Inform all authors of the
authorship policy}}

\begin{itemize}
\tightlist
\item
  I use the
  \href{http://www.icmje.org/recommendations/browse/roles-and-responsibilities/defining-the-role-of-authors-and-contributors.html}{ICMJE
  Vancouver Protocol}. It is important to run this by all authors at the
  beginning of a collaboration, and again at publication.
\end{itemize}

\hypertarget{using-paperpile-with-google-docs}{%
\section*{\texorpdfstring{\textbf{Using Paperpile with Google
Docs}}{Using Paperpile with Google Docs}}\label{using-paperpile-with-google-docs}}
\addcontentsline{toc}{section}{\textbf{Using Paperpile with Google
Docs}}

\begin{itemize}
\item
  Addons --\textgreater{} Get add ons --\textgreater{} search for
  Paperpile
\item
  It works best when you search for the webpage or the title of the
  paper.
\item
  This system is more efficient than keeping EndNote on your computer,
  where you have to download the citation and then add it to EndNote to
  cite it in a word processor.
\item
  At the bottom of the app is an ``Open Paperpile'' link, which provides
  some extra functionality
\item
  To add all your citations in a document to Paperpile, you can click on
  a reference in the Bibliography, click ``Open'', then click on ``View
  all references from this Document'', then in the upper right hand
  corner click on the icon and ``Add all references to my library''
\end{itemize}

\hypertarget{write-as-you-go}{%
\section*{\texorpdfstring{\textbf{Write as you
go}}{Write as you go}}\label{write-as-you-go}}
\addcontentsline{toc}{section}{\textbf{Write as you go}}

\begin{itemize}
\item
  Set aside 15 minutes first thing every morning to engage with the
  manuscript.
\item
  Create an outline (through discussions with all authors) and
  \emph{stick to it}
\item
  Start with methods, and be sure to update them every time you do the
  actual method. It is important to be as specific as possible (e.g.~kit
  manufacturer, name of the piece of equipment you used) to ensure the
  results are reproducible.
\item
  A well-framed research question should allow you to write the
  introduction before the results are known. In the Introduction you
  should lay out the rationale for the study, including the questions or
  hypotheses to be addressed.
\item
  I find it helpful to draw text from the papers I am citing or reading
  if I get stuck in a writing block, and then going back to edit it
  later.
\item
  As you start to work on a section, the most important thing is to
  maintain communication and discussions with all authors, especially if
  what you write strays from the outline. You may be wasting effort and
  taking the manuscript in a direction that is different from what
  others envision.
\item
  Adding comments to sections that you are working on and how you plan
  to improve them, or questions you have on how to improve them, can
  help co-authors understand where your thought process is.
\end{itemize}

\hypertarget{make-figures-perfect}{%
\section*{\texorpdfstring{\textbf{Make figures
perfect}}{Make figures perfect}}\label{make-figures-perfect}}
\addcontentsline{toc}{section}{\textbf{Make figures perfect}}

\begin{itemize}
\tightlist
\item
  Every detail matters. Be prepared to spend 4-8 hours or more refining
  a figure, even after you get a general figure that everyone agrees on.
\end{itemize}

Common mistakes:

\begin{itemize}
\item
  All axes labels are centered. There are no redundant axis labels in
  multi-panel plots.
\item
  y-axis tick labels are not sideways
\item
  The color pattern works in B\&W and for color blind. There are
  color-blind palettes in R, and there are websites online where you can
  see how your figure works for a color blind person.
\item
  No 3-D plots.
\item
  y-axis scale does not accentuate differences in the effect size (in
  other words not zoomed in)
\item
  Base R plotting options:

  \begin{itemize}
  \item
    \texttt{bty\ =\ "l"} box type is ``L'' shaped and not a square
    outline
  \item
    \texttt{las\ =\ 1} y-axis tick labels are upright
  \end{itemize}
\end{itemize}

\hypertarget{after-you-get-a-semi-complete-draft}{%
\section*{\texorpdfstring{\textbf{After you get a semi-complete
draft}}{After you get a semi-complete draft}}\label{after-you-get-a-semi-complete-draft}}
\addcontentsline{toc}{section}{\textbf{After you get a semi-complete
draft}}

\begin{itemize}
\tightlist
\item
  Share the google doc with co-authors and colleagues for feedback. It
  can take a couple weeks to get it back from them, and don't be afraid
  to follow up if they are taking too long. After that it can take a few
  or several weeks to make edits and changes. While you are waiting on
  them, you can work on some of the points below.
\end{itemize}

\hypertarget{getting-to-the-final-draft}{%
\section*{\texorpdfstring{\textbf{Getting to the final
draft}}{Getting to the final draft}}\label{getting-to-the-final-draft}}
\addcontentsline{toc}{section}{\textbf{Getting to the final draft}}

The final stages prior to submission can take about a month or more.

\begin{itemize}
\item
  I like to go through all the references in my folder and skim them
  over again. As I read them over, I double check that they are cited
  correctly, and often I find other places to cite them, which helps to
  weave common threads throughout. Especially if I am first author, I
  want to be familiar with the papers that others cited but I haven't
  read before. This process takes me a couple weeks of 4-5 hours per
  day.
\item
  Write your letter to the editor.
  \href{https://docs.google.com/document/d/1ye4hs0gYP4l4m27o9V_nPZqXD-EPNBrm9UWj-vBv_Zw/edit}{Template}
\item
  Check the author guidelines for the journal you are submitting to and
  make sure all sections of the manuscript fit those guidelines.
\item
  In the ``Data Accessibility'' section I often put ``All datasets and
  scripts used to create results will be archived on Dryad upon final
  acceptance of the manuscript. Scripts can currently be seen at the
  github site XXXX.'' (Beware if double blind review, you may not want
  to put the github site)
\item
  Do not count on co-authors to find mistakes and grammatical errors. I
  have read over manuscripts 10+ times and there are still errors.
\item
  Edit for unneccessary words or long sentences.
\item
  Attention to detail is paramount. Every error that a reviewer finds is
  a mental check that other important details may have been overlooked
  in the research.
\item
  \textbf{Writing Author Contributions.} Communication here is
  paramount. It can help to send this list to each author and let them
  tell you where they think they should have their initials by. Who:

  \begin{itemize}
  \item
    conceptualized and administered the project,
  \item
    obtained funding
  \item
    developed methods
  \item
    collected data
  \item
    did the formal analysis
  \item
    visualized the data
  \item
    wrote the software
  \item
    performed data curation
  \item
    wrote the original draft of the manuscript
  \item
    All authors reviewed and edited the manuscript.
  \end{itemize}
\item
  Be sure to have page and line numbers. There is an ``Line Numbers for
  Google Docs Add On''
\item
  Edit all references in the Bibliography (this is the very last thing
  you do). Paperpile often has errors, especially with capitalization.
  You do not want to have a reviewer find their name was spelled wrong
  in your reference list, so check also for special characters. Species
  italics is another thing that often needs to be edited.
\end{itemize}

\hypertarget{submitting-the-manuscript}{%
\section*{\texorpdfstring{\textbf{Submitting the
manuscript}}{Submitting the manuscript}}\label{submitting-the-manuscript}}
\addcontentsline{toc}{section}{\textbf{Submitting the manuscript}}

\begin{itemize}
\item
  Prior to submission, get final approval (in writing) from all authors
  and agreement that all authors are accountable for mistakes or errors
  per ICJMES policy.
\item
  When I first started submitting manuscripts, I had to go through
  multiple attempts because I kept making edits or changes to the
  manuscript. Just wait until everything is perfect and do it once.
\item
  After you submit, name your draft in google drive. (Click on ``All
  changes saved in Drive''). Save the draft as a Word Document, which
  will be used below if a ``track changes'' document is needed in the
  response to reviews.
\end{itemize}

\hypertarget{responding-to-reviews}{%
\section*{\texorpdfstring{\textbf{Responding to
reviews}}{Responding to reviews}}\label{responding-to-reviews}}
\addcontentsline{toc}{section}{\textbf{Responding to reviews}}

\begin{itemize}
\item
  Your baseline expectation should be a rejection with extensive
  comments from reviewers.
\item
  Involve all authors in responding to reviews, communication is
  important here. Take every review seriously, even if they are being
  ridiculous. A good editor will emphasize the good reviews and
  de-emphasize the ridiculous ones.
\item
  Be respectful when responding to reviews, and be sure to note which
  comments were helpful. Also be sure to note where there is
  disagreement among reviewers when responding to comments.
\item
  Be sure to acknowledge the Editor and anonymous reviewers in your
  ``Acknowledgments'' section
\item
  It is good to submit a ``track changes'' document. Microsoft word has
  a great system for comparing 2 documents. After all authors approve
  the final revisions with all changes accepted, export the manuscript
  to .docx. In Word, go to ``Tools --\textgreater{} Track Changes
  --\textgreater{} Compare Documents''. Upload the draft that was
  submitted the first time around. Word will produce a document showing
  all changes between the two documents, which you can submit with your
  review.
\end{itemize}

\hypertarget{the-final-acceptance}{%
\section*{\texorpdfstring{\textbf{The final
acceptance}}{The final acceptance}}\label{the-final-acceptance}}
\addcontentsline{toc}{section}{\textbf{The final acceptance}}

\begin{itemize}
\item
  \textbf{Archiving data and scripts on Dryad.} Ensure that the scripts
  are still reproducible by running them on your computer and someone
  else's. Once you upload to Dryad, that's it - you can't go back and
  edit or change something.

  \begin{itemize}
  \item
    Dryad has specific instructions for compressing and transferring
    large files, so initiate communication with them before you start
    this process. Make sure everything is perfect, then compress the
    folder with all data and scripts per Dryad instructions.
  \item
    The process of getting a DOI can take about a week. After you get
    the DOI, add the data citation to your Bibliography per Dryad
    instructions.
  \end{itemize}
\end{itemize}

\hypertarget{proofs}{%
\section*{\texorpdfstring{\textbf{Proofs}}{Proofs}}\label{proofs}}
\addcontentsline{toc}{section}{\textbf{Proofs}}

\begin{itemize}
\item
  When you receive the proofs, you typically have 48 hours to make
  corrections and send them back. It is good to share with co-authors if
  they have the time, but typically it is up to the first author to make
  corrections.
\item
  Read through the entire proofs, and don't just focus on the questions
  from the typesetters. There will still be mistakes and grammatical
  errors to be found.
\end{itemize}

\hypertarget{publication}{%
\section*{\texorpdfstring{\textbf{Publication}}{Publication}}\label{publication}}
\addcontentsline{toc}{section}{\textbf{Publication}}

\begin{itemize}
\tightlist
\item
  Celebrate!
\end{itemize}

\hypertarget{professional-development-resources}{%
\chapter{Professional Development
Resources}\label{professional-development-resources}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{collecting-data-and-metadata}{%
\section*{\texorpdfstring{\textbf{Collecting data and
metadata}}{Collecting data and metadata}}\label{collecting-data-and-metadata}}
\addcontentsline{toc}{section}{\textbf{Collecting data and metadata}}

\begin{itemize}
\item
  \href{https://datacarpentry.org/spreadsheet-ecology-lesson/02-common-mistakes/index.html}{How
  to organize data in spreadsheets}
\item
  \href{https://www.tandfonline.com/doi/full/10.1080/00031305.2017.1375989}{Data
  organization in spreadsheets}
\item
  \href{https://data.research.cornell.edu/content/readme}{Guide to
  writing a README file}
\end{itemize}

\hypertarget{time-management}{%
\section*{\texorpdfstring{\textbf{Time
Management}}{Time Management}}\label{time-management}}
\addcontentsline{toc}{section}{\textbf{Time Management}}

\begin{itemize}
\tightlist
\item
  \href{https://docs.google.com/document/d/13xiuKlbF0ao1CSZoeDg5RmtGibtNRclVZeQpAYcLIFo/edit}{Time
  Management, Productivity, and Organization}
\end{itemize}

\hypertarget{discrimination}{%
\section*{\texorpdfstring{\textbf{Discrimination}}{Discrimination}}\label{discrimination}}
\addcontentsline{toc}{section}{\textbf{Discrimination}}

\begin{itemize}
\item
  Discrimination is wrong. NU has guidelines for
  \href{https://www.northeastern.edu/ouec/reporting-options/discrimination/}{reporting
  discrimination}
\item
  See the related page for
  \href{https://www.northeastern.edu/ouec/reporting-options/titleix-prohibited-offenses/}{Title
  IX offenses}
\end{itemize}

\hypertarget{microaggresion}{%
\section*{\texorpdfstring{\textbf{Microaggresion}}{Microaggresion}}\label{microaggresion}}
\addcontentsline{toc}{section}{\textbf{Microaggresion}}

\begin{itemize}
\tightlist
\item
  \href{https://github.com/DrK-Lo/lotterhoslabprotocols/blob/gh-pages/Recognizing\%20and\%20Addressing\%20Microaggressions\%20Takewaways\%202021.pdf}{Recognizing
  and addressing microaggression}
\end{itemize}

\hypertarget{reviewing-manuscripts-and-writing-rejections}{%
\section*{\texorpdfstring{\textbf{Reviewing manuscripts and writing
rejections}}{Reviewing manuscripts and writing rejections}}\label{reviewing-manuscripts-and-writing-rejections}}
\addcontentsline{toc}{section}{\textbf{Reviewing manuscripts and writing
rejections}}

\begin{itemize}
\tightlist
\item
  \href{https://docs.google.com/document/d/1BGYn66Q0mkXOwL5S_woHOLh7__dTHQzQDxRc5EnKLNw/edit\#}{Lab
  discussion 2020}
\end{itemize}

\hypertarget{statistics}{%
\section*{\texorpdfstring{\textbf{Statistics}}{Statistics}}\label{statistics}}
\addcontentsline{toc}{section}{\textbf{Statistics}}

\begin{itemize}
\tightlist
\item
  {[}2020 Lab discussion how to mix
  model{]}\{https://docs.google.com/document/d/1jQZkbZbZuWnElYmLUntg5qgh1NG2-yAbsT3zzvmbkZ0/edit\#heading=h.btb80w3ha1pf\}
\end{itemize}

\hypertarget{grant-writing}{%
\section*{\texorpdfstring{\textbf{Grant
writing}}{Grant writing}}\label{grant-writing}}
\addcontentsline{toc}{section}{\textbf{Grant writing}}

\begin{itemize}
\tightlist
\item
  \href{https://docs.google.com/document/d/10TXWztCiID1-wNTGZ37rvYoIVHbf2kWmQMHoqN6pHQk/edit?pli=1}{2020
  phrasebank and advice}
\end{itemize}

\hypertarget{how-to-start-a-collaboration-and-keep-it-going}{%
\section*{\texorpdfstring{\textbf{How to start a collaboration and keep
it
going}}{How to start a collaboration and keep it going}}\label{how-to-start-a-collaboration-and-keep-it-going}}
\addcontentsline{toc}{section}{\textbf{How to start a collaboration and
keep it going}}

\begin{itemize}
\tightlist
\item
  \href{https://docs.google.com/document/d/1mgWle41KPHnlCevE5IB9S1AaWQcYDFm_7riQU7WTfOM/edit}{2020
  lab discussion}
\end{itemize}

\hypertarget{tips-for-aksing-for-extensions}{%
\section*{\texorpdfstring{\textbf{Tips for aksing for
extensions}}{Tips for aksing for extensions}}\label{tips-for-aksing-for-extensions}}
\addcontentsline{toc}{section}{\textbf{Tips for aksing for extensions}}

\begin{itemize}
\tightlist
\item
  \href{https://docs.google.com/document/d/1o87Ua79CeVIQstDpy9m9DtGsZTL9ttFmVB-Kqus1Iig/edit}{Tips
  for aksing for extensions}
\end{itemize}

\hypertarget{errors-and-mistakes}{%
\section*{\texorpdfstring{\textbf{Errors and
mistakes}}{Errors and mistakes}}\label{errors-and-mistakes}}
\addcontentsline{toc}{section}{\textbf{Errors and mistakes}}

\begin{itemize}
\item
  Publishing a correction because of a mistake:
  \href{https://dynamicecology.wordpress.com/2016/12/02/on-finding-errors-in-ones-published-analyses/}{On
  finding errors in one's published papers}
\item
  Publishing a retraction because of a mistake:
  \href{http://ecoevoevoeco.blogspot.com/2016/12/wrong-lot.html\%20http://ecoevoevoeco.blogspot.com/2016/12/wrong-lot.html}{Wrong
  alot?}
\item
  Publishing a retraction because of what appears to be fraud:
  \href{https://laskowskilab.faculty.ucdavis.edu/2020/01/29/retractions/}{\#PruittGate}
\item
  Katie's overview of her career mistakes (need to add link to
  presentation)
\end{itemize}

\hypertarget{interviewing}{%
\section*{\texorpdfstring{\textbf{Interviewing}}{Interviewing}}\label{interviewing}}
\addcontentsline{toc}{section}{\textbf{Interviewing}}

\begin{itemize}
\item
  \href{https://docs.google.com/document/d/1SM3PiIfo4f_Pd_NZxC4Br1g5E0mnqCUlwilrleatNWA/edit}{Interview
  questions for a graduate or postdoc application}
\item
  \href{https://docs.google.com/document/d/1OvvB6yl_5l05dUn3frL5GEJeksc5xjy8a_keq8sp0fc/edit}{Interview
  for a faculty position}
\end{itemize}

\hypertarget{skimming-a-paper}{%
\section*{\texorpdfstring{\textbf{Skimming a
paper}}{Skimming a paper}}\label{skimming-a-paper}}
\addcontentsline{toc}{section}{\textbf{Skimming a paper}}

\begin{itemize}
\tightlist
\item
  \href{https://docs.google.com/document/d/17wIUVwRGpfRMG-riBjCj9EyCnPMb-K8RX7ww5C0kzf8/edit}{How
  to skim a paper}
\end{itemize}

\hypertarget{lotterhos-lab-notebook-discussion}{%
\section*{\texorpdfstring{\textbf{Lotterhos Lab notebook
discussion}}{Lotterhos Lab notebook discussion}}\label{lotterhos-lab-notebook-discussion}}
\addcontentsline{toc}{section}{\textbf{Lotterhos Lab notebook
discussion}}

\begin{itemize}
\tightlist
\item
  \href{https://drive.google.com/file/d/1HIVVvMbaF8XW0Jxyh708EYBTegkneJ_C/view?usp=sharing}{Lotterhos
  Lab notebook discussion}
\end{itemize}

\hypertarget{how-to-write-about-mixed-models}{%
\section*{\texorpdfstring{\textbf{How to write about mixed
models}}{How to write about mixed models}}\label{how-to-write-about-mixed-models}}
\addcontentsline{toc}{section}{\textbf{How to write about mixed models}}

\begin{itemize}
\tightlist
\item
  \href{https://docs.google.com/document/d/1jQZkbZbZuWnElYmLUntg5qgh1NG2-yAbsT3zzvmbkZ0/edit\#heading=h.btb80w3ha1pf}{How
  to write about mixed models}
\end{itemize}

\hypertarget{anxiety}{%
\section*{\texorpdfstring{\textbf{Anxiety}}{Anxiety}}\label{anxiety}}
\addcontentsline{toc}{section}{\textbf{Anxiety}}

We all suffer from anxiety sometimes.

\begin{itemize}
\item
  NU resources for confidential support and guidance for students can be
  found through:

  \begin{itemize}
  \item
    \href{https://www.northeastern.edu/uhcs/}{University Health and
    Counseling Services}
  \item
    \href{https://www.northeastern.edu/spirituallife/}{The Northeastern
    Center for Spirituality, Dialogue, and Service}
  \item
    \href{https://studentlife.northeastern.edu/open/}{the Office of
    Prevention and Education at Northeastern (OPEN)}
  \item
    \href{https://provost.northeastern.edu/oidi/}{Office of
    Institutional Diversity and Inclusion}
  \item
    Northeastern has added Find@Northeastern, which is a 24/7 mental
    health consulting line and can be reached at 1-877-223-9477
  \item
    In addition, there are graduate student support groups
  \end{itemize}
\item
  NU resources for staff

  \begin{itemize}
  \tightlist
  \item
    To do, but also good benefits here and free services
  \end{itemize}
\item
  \emph{Anxiety relief for kids} has techniques designed for kids, but I
  found really helpful in my own life
\item
  This webpage has a good overview of
  \href{https://positivepsychology.com/cognitive-distortions/}{cognitive
  distortions} that we all suffer from
\end{itemize}

\hypertarget{nutrition-and-anxiety}{%
\subsection*{\texorpdfstring{\textbf{Nutrition and
anxiety}}{Nutrition and anxiety}}\label{nutrition-and-anxiety}}
\addcontentsline{toc}{subsection}{\textbf{Nutrition and anxiety}}

One time I had severe fatigue. My doctor did a vitamin panel and it
turned out to be a vitamin D deficiency.

\begin{itemize}
\item
  \href{https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2738337/}{Understanding
  nutrition, depression and mental illnesses}
\item
  \href{https://www.health.harvard.edu/mind-and-mood/could-a-vitamin-or-mineral-deficiency-be-behind-your-fatigue}{Could
  a vitamin or mineral deficiency be behind your fatigue?}
\item
  \href{https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7019700/}{A review
  of vitamins and minerals for energy, fatigue, and cognition}
\item
  \href{https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5207540/}{Effect of
  vitamin D3 on self-perceived fatigue}
\item
  \href{https://www.myonemedicalsource.com/2020/06/18/nutritional-testing/}{Types
  of nutritional deficiencies you can be tested for by your doctor}
\end{itemize}

\hypertarget{climate-grief-and-earth-optimism}{%
\subsection*{\texorpdfstring{\textbf{Climate grief and Earth
optimism}}{Climate grief and Earth optimism}}\label{climate-grief-and-earth-optimism}}
\addcontentsline{toc}{subsection}{\textbf{Climate grief and Earth
optimism}}

\begin{itemize}
\item
  https://www.climateandmind.org/
\item
  https://earthoptimism.si.edu/resources/educational-resources/
\item
  Here's a faculty resource from American Sociological Association on
  dealing emotional reactions to climate grief:
  https://cpb-us-e1.wpmucdn.com/sites.ucsc.edu/dist/1/1169/files/2021/06/When-students-have-powerful-reactions.pdf
\end{itemize}

\hypertarget{conflicts-of-interest}{%
\section*{\texorpdfstring{\textbf{Conflicts of
interest}}{Conflicts of interest}}\label{conflicts-of-interest}}
\addcontentsline{toc}{section}{\textbf{Conflicts of interest}}

\begin{itemize}
\tightlist
\item
  \href{https://sustainablefisheries-uw.org/flawed-mpa-science-retracted/}{Flawed
  MPA science retracted}
\end{itemize}

\hypertarget{antiracism}{%
\section*{\texorpdfstring{\textbf{Antiracism}}{Antiracism}}\label{antiracism}}
\addcontentsline{toc}{section}{\textbf{Antiracism}}

\begin{itemize}
\tightlist
\item
  \href{https://www.nature.com/articles/s41559-021-01522-z}{Anti-racist
  interventions to transform ecology, evolution and conservation biology
  departments}
\end{itemize}

\hypertarget{lab-meeting-recordings}{%
\section*{\texorpdfstring{\textbf{Lab meeting
recordings}}{Lab meeting recordings}}\label{lab-meeting-recordings}}
\addcontentsline{toc}{section}{\textbf{Lab meeting recordings}}

Please ask Katie for the password to view

\begin{itemize}
\item
  \href{https://northeastern.zoom.us/rec/share/enHUqKE7sr-bq6gpBMBOsu2-vPz3YsobBGADU2wNAgl80B-O9ijm86bx9kxOiboj.nYjVibVUxw51caKb}{June
  1 2021}
\item
  \href{https://northeastern.zoom.us/rec/share/8Z6w6mkq-XTPgyDJU_zcLIp0M5U9tTh5rGo5bhlXdIfKR7vbV4bZsA2Nm3rxvbfv.d3krLgGquGjjO5CP}{Aug
  17, 2021 Discussion of organizing data in spreadsheets}
\item
  \href{https://northeastern.zoom.us/rec/share/9JS5ZfB-eySzp6VqK24pHh4bNft2uPhd8wLx3b7J2ycq1IQd4hxxt8bvcXs8WfJs.3dgAcoKa9fQZArZR}{Aug
  24 2021 Discussion of Price Equation (first half of Chap 6 of Rices'
  Evolutionary Theory}
\item
  \href{https://northeastern.zoom.us/rec/share/L4GY5DKi6XCi1nf8_RxkzyGyu6NAojZd4nsNsL1WpxiEfNp5qe_PHd5An9ngPANS.49r97rNHCjo5PNSH}{Aug
  31 2021 Data Viz discussion from the NU specialist}
\item
  \href{https://drive.google.com/file/d/1PGqdpTqKoK9ShX6yj6hNiqI7VVO43gOv/view?usp=sharing}{April
  4, 2022 Genomics Pipeline Discussion}
\end{itemize}

\hypertarget{northeastern-travel-and-conference-funding-for-phds}{%
\section*{\texorpdfstring{\textbf{Northeastern Travel and Conference
Funding for
PhDs}}{Northeastern Travel and Conference Funding for PhDs}}\label{northeastern-travel-and-conference-funding-for-phds}}
\addcontentsline{toc}{section}{\textbf{Northeastern Travel and
Conference Funding for PhDs}}

\begin{itemize}
\item
  \href{https://drive.google.com/file/d/1Lb_1kdAgfSQqCOqkyCWnVshKLY8Y7MHT/view?usp=sharing}{MSC
  Department-Level Funding Application} The MSC grants up to \$300 once
  per academic year. Intended to supplement advisor funding. Apply at
  least 30 days before conference.
\item
  \href{https://forms.office.com/pages/responsepage.aspx?id=gcLuqKOqrk2sm5o5i5IV5yDZoNSxGVNDmDTVJk5ayVFURVRETTY1MUZUUDIzNzRHVFRGQkpaWTBZTi4u}{College
  of Science Funding Application} The College of Science grants up to
  \$350 for domestic travel/conference, or \$500 for international
  travel/conference. It is expected that there is a matching source (PhD
  Network, department, or advisor). Preference is given for first-time
  awardees for whom it would be their first national conference. Apply
  at least 1 month before conference.
\item
  \href{https://phd.northeastern.edu/network/resources/travelfunding/}{PhD
  Network Funding Application} The PhD Network grants up to \$500 once
  per academic year. It is expected that there is a matching source
  (COS, department, or advisor). Apply at least 2 months before
  conference.\\
\end{itemize}

\hypertarget{design}{%
\chapter{Design}\label{design}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{access-nu-creative-cloud}{%
\section*{\texorpdfstring{\textbf{Access NU Creative
Cloud}}{Access NU Creative Cloud}}\label{access-nu-creative-cloud}}
\addcontentsline{toc}{section}{\textbf{Access NU Creative Cloud}}

We have access to In Design, Illustrator, Photoshop, and Adobe Pro (for
editing pdfs). Here is how to access the
\href{https://adobe.northeastern.edu/}{Adobe Creative Cloud} and make
sure to
\href{https://service.northeastern.edu/ogs?id=kb_article\&sys_id=4128ecdb1b42745069590dc1604bcbb2}{follow
the instructions here} (choose ``active directory'')

\hypertarget{guidance-and-resources}{%
\section*{\texorpdfstring{\textbf{Guidance and
resources}}{Guidance and resources}}\label{guidance-and-resources}}
\addcontentsline{toc}{section}{\textbf{Guidance and resources}}

\href{https://northeastern.zoom.us/rec/share/L4GY5DKi6XCi1nf8_RxkzyGyu6NAojZd4nsNsL1WpxiEfNp5qe_PHd5An9ngPANS.49r97rNHCjo5PNSH}{Aug
31 2021 Lab meeting recording - Data Viz discussion from the NU
specialist} (ask Katie for password)

\begin{itemize}
\tightlist
\item
  \href{https://subjectguides.lib.neu.edu/dataviz/tutorials}{NU library
  page on data visualization resources}
\end{itemize}

\hypertarget{templates}{%
\section*{\texorpdfstring{\textbf{Templates}}{Templates}}\label{templates}}
\addcontentsline{toc}{section}{\textbf{Templates}}

\href{https://github.com/DrK-Lo/lotterhoslabprotocols/raw/gh-pages/ClassicPresentationTheme.key}{A
Keynote color/font template for presentations and posters}

\textbf{A ggtheme for graphs with white backgrounds and dark grey lines}

\begin{verbatim}
ggtheme <- theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
panel.background = element_blank(), panel.border=element_blank(), axis.line = element_line(colour=“grey30”), 
axis.title = element_text(colour=“grey20”), axis.text = (element_text(colour=“grey30”)), 
legend.title = element_text(colour=“grey20”), legend.text = element_text(colour=“grey30”))
\end{verbatim}

\hypertarget{graduate-courses}{%
\chapter{Graduate Courses}\label{graduate-courses}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{bioinformatics}{%
\subsection*{\texorpdfstring{\textbf{Bioinformatics}}{Bioinformatics}}\label{bioinformatics}}
\addcontentsline{toc}{subsection}{\textbf{Bioinformatics}}

I think taking 6308 + 6200 would be the best for new students, and if
you had to pick one, 6308 is probably easier and will give students more
bioinfo exposure. 6200 is much harder, has much more comp sci
theory/programming in it, and for someone who has never seen a fastq or
bam file.. it might be a steeper learning curve.

\hypertarget{binf6200-bioinformatics-programming}{%
\subsubsection*{\texorpdfstring{\textbf{BINF6200: Bioinformatics
Programming}}{BINF6200: Bioinformatics Programming}}\label{binf6200-bioinformatics-programming}}
\addcontentsline{toc}{subsubsection}{\textbf{BINF6200: Bioinformatics
Programming}}

\begin{figure}

{\centering \includegraphics{https://user-images.githubusercontent.com/90652097/154899888-1b33bd9c-eb73-4c15-9597-c6ce8ce5b238.png}

}

\caption{Screen Shot 2022-02-21 at 1 19 30 AM}

\end{figure}

Focuses on the fundamental programming skills required in the
bioinformatics industry. Focuses on Python and R as the main programming
language used. Topics include string operations, file manipulation,
regular expressions, object-oriented programming, data structures,
testing, program design, and implementation. Includes substantial
out-of-classroom assignments.

\hypertarget{binf6308-bioinformatics-computational-methods-i}{%
\subsubsection*{\texorpdfstring{\textbf{BINF6308: Bioinformatics
Computational Methods
I}}{BINF6308: Bioinformatics Computational Methods I}}\label{binf6308-bioinformatics-computational-methods-i}}
\addcontentsline{toc}{subsubsection}{\textbf{BINF6308: Bioinformatics
Computational Methods I}}

\begin{figure}

{\centering \includegraphics{https://user-images.githubusercontent.com/90652097/154899281-621d969b-bc95-4d7a-80dc-5339be17a4ad.png}

}

\caption{Screen Shot 2022-02-21 at 1 13 52 AM}

\end{figure}

Offers the first semester of a two-semester sequence on the use of
computers in bioinformatics research. Offers students an opportunity to
work with current methods and computational algorithms used in
contemporary sequence analysis. Teaches practical skills necessary to
manage and mine the vast biological information being generated and
housed in public databases. Emphasizes the use of Python as the primary
computer language and requires students to learn and understand basic
computer logic and syntax, including an introduction to scalars, arrays,
hashes, decision statements, loops, subroutines, references, and regular
expressions. A focus on fundamental skills, including the command line
interface found in the Linux operating system, is designed to prepare
students for second-semester applications.

\hypertarget{binf6309-bioinformatics-computational-methods-ii}{%
\subsubsection*{\texorpdfstring{\textbf{BINF6309: Bioinformatics
Computational Methods
II}}{BINF6309: Bioinformatics Computational Methods II}}\label{binf6309-bioinformatics-computational-methods-ii}}
\addcontentsline{toc}{subsubsection}{\textbf{BINF6309: Bioinformatics
Computational Methods II}}

\begin{itemize}
\tightlist
\item
  Syllabus:
\end{itemize}

Designed to build upon the core topics covered in BINF 6308, i.e., use
of the computer as a tool for bioinformatics research. Builds upon the
Python language fundamentals covered during the first semester but
requires students to apply these fundamentals to a semester-long
project. The project includes protein family analysis, multiple sequence
analysis, phylogeny, and protein structure analysis. Additionally,
students have an opportunity to learn to build, load, connect, and query
custom MySQL databases, and parse command line flags.

\hypertarget{binf6400-genomics-in-bioinformatics}{%
\subsubsection*{\texorpdfstring{\textbf{BINF6400: Genomics in
Bioinformatics}}{BINF6400: Genomics in Bioinformatics}}\label{binf6400-genomics-in-bioinformatics}}
\addcontentsline{toc}{subsubsection}{\textbf{BINF6400: Genomics in
Bioinformatics}}

\begin{itemize}
\tightlist
\item
  Syllabus:
\end{itemize}

Introduces the field of genomics. With the completion of the Human
Genome Project several years ago, there has been an explosion of genetic
data collected. Focuses on the bioinformatics tools necessary to analyze
large-scale genomic data. Covers topics such as phylogenetic trees,
molecular evolution, gene expression profiling, heterogeneous genomic
data, as well as next-generation sequencing (NGS) data.

\hypertarget{machine-learning}{%
\section*{\texorpdfstring{\textbf{Machine
Learning}}{Machine Learning}}\label{machine-learning}}
\addcontentsline{toc}{section}{\textbf{Machine Learning}}

\hypertarget{ds4400-machine-learning-and-data-mining-i}{%
\subsubsection*{\texorpdfstring{\textbf{DS4400: Machine Learning and
Data Mining
I}}{DS4400: Machine Learning and Data Mining I}}\label{ds4400-machine-learning-and-data-mining-i}}
\addcontentsline{toc}{subsubsection}{\textbf{DS4400: Machine Learning
and Data Mining I}}

\begin{itemize}
\tightlist
\item
  Syllabus: https://www.ccs.neu.edu/home/alina/classes/Spring2021/
\end{itemize}

Machine learning is a fast-pacing and exciting field achieving
human-level performance in tasks such as image classification, speech
recognition. machine translation, precision medicine, and self-driving
cars. Machine learning has already impacted greatly our daily lives and
has the potential to transform the world even more in the near future.
This course will provide a broad introduction to machine learning and
cover the fundamental algorithms for supervised and unsupervised
learning. We will cover topics related to regression, classification,
deep learning, dimensionality reduction, and clustering. The class will
also provide an introduction into adversarial machine learning, an
emerging area that studies the fundamental security issues of machine
learning

\hypertarget{biostatistics}{%
\section*{\texorpdfstring{\textbf{Biostatistics}}{Biostatistics}}\label{biostatistics}}
\addcontentsline{toc}{section}{\textbf{Biostatistics}}

\hypertarget{envr6500-biostatistics}{%
\subsubsection*{\texorpdfstring{\textbf{ENVR6500:
Biostatistics}}{ENVR6500: Biostatistics}}\label{envr6500-biostatistics}}
\addcontentsline{toc}{subsubsection}{\textbf{ENVR6500: Biostatistics}}

\begin{figure}

{\centering \includegraphics{https://user-images.githubusercontent.com/90652097/154899480-d6e17a72-8f9e-4b3f-9021-acd63b098de5.png}

}

\caption{Screen Shot 2022-02-21 at 1 15 30 AM}

\end{figure}

Offers an overview of traditional and modern statistical methods used to
analyze biological data using the free and open-source R programming
environment. Lectures describe core statistical approaches and discuss
their suitability for understanding patterns that arise at different
levels of biological organization, from cellular processes to whole
ecosystems. Supervised lab sessions offer students an opportunity to
develop the R programming skills required to analyze the complex
datasets that often emerge when addressing cutting-edge questions in
biology. Topics include basic probability and sampling theory,
experimental design, null hypothesis significance testing, t-tests and
ANOVA, correlation and regression, likelihood, model selection, and
information theory.

\hypertarget{evolution}{%
\section*{\texorpdfstring{\textbf{Evolution}}{Evolution}}\label{evolution}}
\addcontentsline{toc}{section}{\textbf{Evolution}}

\hypertarget{biol5585-advanced-evolution}{%
\subsubsection*{\texorpdfstring{\textbf{BIOL5585: Advanced
Evolution}}{BIOL5585: Advanced Evolution}}\label{biol5585-advanced-evolution}}
\addcontentsline{toc}{subsubsection}{\textbf{BIOL5585: Advanced
Evolution}}

\begin{figure}

{\centering \includegraphics{https://user-images.githubusercontent.com/90652097/154898463-499bc650-eee4-4d5d-8007-b00c4b6c9b4d.png}

}

\caption{Screen Shot 2022-02-21 at 1 05 51 AM}

\end{figure}

Discusses history of evolutionary theory and lines of evidence. Emphasis
is on mechanisms of speciation. Introduces and discusses current
evolutionary topics.

\hypertarget{gis}{%
\section*{\texorpdfstring{\textbf{GIS}}{GIS}}\label{gis}}
\addcontentsline{toc}{section}{\textbf{GIS}}

\hypertarget{envr5260-geographical-information-systems}{%
\subsubsection*{\texorpdfstring{\textbf{ENVR5260: Geographical
Information
Systems}}{ENVR5260: Geographical Information Systems}}\label{envr5260-geographical-information-systems}}
\addcontentsline{toc}{subsubsection}{\textbf{ENVR5260: Geographical
Information Systems}}

\begin{figure}

{\centering \includegraphics{https://user-images.githubusercontent.com/90652097/154898851-003915ee-c15b-4337-86c0-88d13357d739.png}

}

\caption{Screen Shot 2022-02-21 at 1 08 48 AM}

\end{figure}

This course explores the practical application of GIS to support
geographic inquiry and decision making. It focuses on the technical
knowledge of the common tasks that a GIS analyst faces in applying GIS
to a variety of spatial problems. Students will gain hands-on experience
with a leading commercial GIS software package.Students will have the
opportunity to gain a good understanding of the concepts, principles,
approaches, and techniques of spatial analysis using Geographic
Information Systems.

\begin{itemize}
\item
  Students will have the opportunity to become familiar with the
  everyday challenges facing GIS professionals as they apply GIS
  principles to a variety of spatial problems.
\item
  Students will have the opportunity to gain an understanding of
  real-world GIS applications and how to approach them.
\item
  Students will have the opportunity to explore geographic data, to
  learn how it is stored and managed, and to see how it differs from
  other types of data.
\item
  Students will have the opportunity to become familiar with the basic
  functionality and operation of the ESRI ArcGIS software.
\end{itemize}

\part{Molecular Protocols}

\hypertarget{types-of-ethanol}{%
\chapter{Types of Ethanol}\label{types-of-ethanol}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{denatured-ethanol-vs-non-denatured-ethanol}{%
\section*{\texorpdfstring{\textbf{Denatured ethanol vs non-denatured
ethanol}}{Denatured ethanol vs non-denatured ethanol}}\label{denatured-ethanol-vs-non-denatured-ethanol}}
\addcontentsline{toc}{section}{\textbf{Denatured ethanol vs
non-denatured ethanol}}

The main difference is that denatured ethanol contains additives that
make it unsafe to drink. On the other hand, non-denatured ethanol is
ethanol with no additives or denaturants.

Denatured ethanol is a form of ethanol containing a high amount of
additives and denaturants that makes it poisonous. Denatured ethanol has
bad taste and foul smell that makes it unsafe for drinking. There are
two forms of non-denatured ethanol named 95\% ethanol and absolute
ethanol. These forms are known as ``pure'' forms of ethanol.

\hypertarget{histologicalreagent-ethanol-vs-molecular-grade}{%
\section*{\texorpdfstring{\textbf{Histological/reagent ethanol vs
molecular
grade}}{Histological/reagent ethanol vs molecular grade}}\label{histologicalreagent-ethanol-vs-molecular-grade}}
\addcontentsline{toc}{section}{\textbf{Histological/reagent ethanol vs
molecular grade}}

Histological/reagent ethanol is denatured ethanol. Molecular grade
ethanol is non-denatured.
\href{https://www.differencebetween.com/difference-between-denatured-and-vs-undenatured-ethanol/\#Undenatured\%20Ethanol}{Source}

\hypertarget{use-95-pure-190-proof-non-denatured-ethanol-without-benzene-for-preserving-tissues-for-genomic-dna}{%
\section*{\texorpdfstring{\textbf{Use 95\% Pure (190 proof)
non-denatured Ethanol without Benzene for Preserving tissues for genomic
DNA}}{Use 95\% Pure (190 proof) non-denatured Ethanol without Benzene for Preserving tissues for genomic DNA}}\label{use-95-pure-190-proof-non-denatured-ethanol-without-benzene-for-preserving-tissues-for-genomic-dna}}
\addcontentsline{toc}{section}{\textbf{Use 95\% Pure (190 proof)
non-denatured Ethanol without Benzene for Preserving tissues for genomic
DNA}}

\href{https://link.springer.com/article/10.1007/s13127-010-0012-4}{This
article} has the following information on ethanol for tissue
preservation:

\begin{itemize}
\item
  If possible, the use of non-denaturated ethanol is favoured, but
  traces of denaturating agents (e.g.~isopropanol, acetone, methyl ethyl
  ketone, also gasoline) usually do not significantly alter the
  performance.
\item
  According to plenty of tests, the optimal concentration is about
  95--99\% (King and Porter 2004, who also compared with isopropanol).
\item
  With 70\% ethanol, a high degree of degradation has been observed,
  even after one year.
\item
  Interestingly, absolute ethanol may not be the optimal medium as it is
  most expensive and can contain traces of benzene that seriously affect
  DNA preservation (Ito 1992).

  \begin{itemize}
  \tightlist
  \item
    Side note: KEL tried to find Ito 1992, but the orginal print is not
    available online.
  \end{itemize}
\item
  The ethanol-tissue ratio should exceed 5:1, although some authors
  recommend much higher ratios (e.g.~Martin 1977).
\item
  When ordering, look for ``pure ethanol'' and check the SDS to make
  sure it does not contain additives
\item
  95.6\% ethanol is an azeotrope, which means the vapor state has the
  same ethanol-to-water ratio as the liquid state. This means as it
  evaporates the remaining liquid will still be at 95.6\%
\item
  When adding wet tissues to 95\% ethanol, it can decrease the
  concentration of ethanol. It can be helpful to change the ethanol
  after the first few months.
\item
  More information on other ways to preserve tissues are found
  \href{https://spnhc.biowikifarm.net/wiki/Tissue_Sample_Collection}{in
  this article}
\end{itemize}

\hypertarget{use-70-denatured-ethanol-for-cleaning}{%
\section*{\texorpdfstring{\textbf{Use 70\% Denatured Ethanol for
Cleaning}}{Use 70\% Denatured Ethanol for Cleaning}}\label{use-70-denatured-ethanol-for-cleaning}}
\addcontentsline{toc}{section}{\textbf{Use 70\% Denatured Ethanol for
Cleaning}}

\href{https://bitesizebio.com/13518/which-type-of-ethanol-should-i-use/}{Information
from this article explains the use of ethanol for cleaning}. The main
points are summarized here:

\begin{itemize}
\tightlist
\item
  Although it may seem counter-intuitive, 70\% ethanol seems to be the
  most effective concentration, as 90\% ethanol coagulates the protein,
  and the protein layer that is created effectively renders the cell
  dormant. At a 70\% concentration, however, the ethanol takes longer to
  evaporate and this longer contact time means that it is able to
  penetrate cells more effectively. The water in a 70\% grade is also
  important in denaturing proteins. And 70\% ethanol also tends to be
  cheaper and is less flammable.
\end{itemize}

\hypertarget{there-are-a-couple-of-things-to-watch-out-for-when-diluting-ethanol}{%
\section*{\texorpdfstring{\textbf{There are a couple of things to watch
out for when diluting
ethanol:}}{There are a couple of things to watch out for when diluting ethanol:}}\label{there-are-a-couple-of-things-to-watch-out-for-when-diluting-ethanol}}
\addcontentsline{toc}{section}{\textbf{There are a couple of things to
watch out for when diluting ethanol:}}

\begin{itemize}
\item
  First, diluted ethanol is no longer an azeotrope. At 70\%, ethanol
  will evaporate faster than water, so evaporative loss will result in a
  gradual decrease in ethanol concentration in the remaining solution.
  Don't be surprised when an unsealed beaker of 70\% ethanol `stops
  working'. So don't make more than is necessary at a time, and make
  sure to label the date when it was made (which you should do for
  everything you make anyway).
\item
  Second, ethanol/water mixtures don't have additive volumes. That means
  that if you want to make 1 liter of 70\% ethanol from absolute
  ethanol, you can't do it by mixing 700 ml of ethanol and 300 ml of
  water. The correct method is to measure 700 ml of ethanol and then
  bring it to a volume of 1 liter with water.

  \begin{itemize}
  \tightlist
  \item
    Why? When you mix water and ethanol, the ethanol molecules occupy
    the space between the water molecules. A helpful way to visualize
    this is to think about sand and rocks -- if you were to mix 1 liter
    of each, the final volume would be less than 2 liters because the
    sand would fall between the rocks.
  \end{itemize}
\end{itemize}

\hypertarget{use-100-200-proof-absolute-ethanol-molecular-grade-non-denatured-ethanol-for-molecular-biology-work}{%
\section*{\texorpdfstring{\textbf{Use 100\% (200 Proof, absolute
ethanol, molecular grade) non-denatured Ethanol for Molecular Biology
work}}{Use 100\% (200 Proof, absolute ethanol, molecular grade) non-denatured Ethanol for Molecular Biology work}}\label{use-100-200-proof-absolute-ethanol-molecular-grade-non-denatured-ethanol-for-molecular-biology-work}}
\addcontentsline{toc}{section}{\textbf{Use 100\% (200 Proof, absolute
ethanol, molecular grade) non-denatured Ethanol for Molecular Biology
work}}

\begin{itemize}
\item
  Something about the process of making ethanol makes it hard to get
  from 95\% to 100\%. Sometimes this contains additives to get it up to
  100\%, but they are not denaturing agents.
\item
  Some procedures that are sensitive to the presence of water require
  absolute ethanol.
\item
  A common method to produce ethanol with a higher concentration than
  95\% is to use additives that disrupt the azeotrope composition and
  allow further distillation. For this reason, absolute ethanol
  sometimes contains trace amounts of these additives (such as benzene).
  \href{https://bitesizebio.com/13518/which-type-of-ethanol-should-i-use/}{source}
\end{itemize}

\hypertarget{solution-recipes}{%
\chapter{Solution Recipes}\label{solution-recipes}}

Recipes for solutions and reagents needed for lab protocols

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{label-your-solutions}{%
\section*{\texorpdfstring{\textbf{Label your
solutions}}{Label your solutions}}\label{label-your-solutions}}
\addcontentsline{toc}{section}{\textbf{Label your solutions}}

\begin{itemize}
\item
  Please make sure solutions are labeled with your name/initials, the
  date that you made them
\item
  Please do not use stock solutions for molecular work - pour out
  solutions into aliquots in falcon tubes or glass jars
\item
  Please make sure your aliquots are labeled with the stock solution,
  the date they were made, who made them, and your name on the aliquot
\item
  Do not share aliquots with other people in the lab
\end{itemize}

\hypertarget{m-ammonium-acetate}{%
\subsection*{\texorpdfstring{\textbf{7.5M ammonium
acetate}}{7.5M ammonium acetate}}\label{m-ammonium-acetate}}
\addcontentsline{toc}{subsection}{\textbf{7.5M ammonium acetate}}

\begin{longtable}[]{@{}ll@{}}
\toprule()
\textbf{Ingredient} & \textbf{Amount} \\
\midrule()
\endhead
ammonium acetate (Fisher \#A637500) & 57.81g \\
water & (to a final volume of 100ml) \\
\bottomrule()
\end{longtable}

\begin{itemize}
\item
  If needed, sterilize by filtration in 0.2um filter.
\item
  final pH will be 5.5
\end{itemize}

\hypertarget{bleach-10}{%
\subsection*{\texorpdfstring{\textbf{Bleach
10\%}}{Bleach 10\%}}\label{bleach-10}}
\addcontentsline{toc}{subsection}{\textbf{Bleach 10\%}}

\begin{longtable}[]{@{}ll@{}}
\toprule()
\textbf{Ingredient} & \textbf{Amount} \\
\midrule()
\endhead
Bleach & 10ml \\
DI Water & 90ml \\
\bottomrule()
\end{longtable}

\begin{itemize}
\tightlist
\item
  solution is good for up to 7 days
\end{itemize}

\hypertarget{m-edta-ph---8}{%
\subsection*{\texorpdfstring{\textbf{0.5M EDTA (pH -
8)}}{0.5M EDTA (pH - 8)}}\label{m-edta-ph---8}}
\addcontentsline{toc}{subsection}{\textbf{0.5M EDTA (pH - 8)}}

\begin{longtable}[]{@{}
  >{\raggedright\arraybackslash}p{(\columnwidth - 2\tabcolsep) * \real{0.8286}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 2\tabcolsep) * \real{0.1714}}@{}}
\toprule()
\begin{minipage}[b]{\linewidth}\raggedright
\textbf{Ingredient}
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
\textbf{Amount}
\end{minipage} \\
\midrule()
\endhead
Disodium ethylenediamine tetraacetate (Fisher \#BP120500) & 186.1g \\
Water & 80ml \\
NaOH (Fisher \#BP359-212) & 18g \\
\bottomrule()
\end{longtable}

\begin{itemize}
\item
  Stir vigorously on a magnetic stirrer.
\item
  Adjust the pH to 8.0.
\item
  Sterilize by autoclaving. (Be sure not to screw on the lid or bottle
  will explode)
\end{itemize}

\hypertarget{mm-edta-ph---8-for-storing-extracted-rna}{%
\subsection*{\texorpdfstring{\textbf{1mM EDTA (pH - 8) for storing
extracted
RNA}}{1mM EDTA (pH - 8) for storing extracted RNA}}\label{mm-edta-ph---8-for-storing-extracted-rna}}
\addcontentsline{toc}{subsection}{\textbf{1mM EDTA (pH - 8) for storing
extracted RNA}}

EDTA should help stabilize and protect the RNA, but be sure to check
with downstream library prep first. Use sterile 25mL pipettes to measure
RNA free water and put into a 100mL glassware (autoclaved, wiped clean
with RNAaseZap or RNAase Away, and rinsed with RNAase-free water). Clean
magnetic spinbar with RNAaseZap. Calibrate pH meter and wipe outside
with RNAase zap. Use pasteur pipette to add HCl for pH adjustment, but
wipe outside with RNAase-Zap.

\begin{longtable}[]{@{}
  >{\raggedright\arraybackslash}p{(\columnwidth - 2\tabcolsep) * \real{0.8286}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 2\tabcolsep) * \real{0.1714}}@{}}
\toprule()
\begin{minipage}[b]{\linewidth}\raggedright
\textbf{Ingredient}
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
\textbf{Amount}
\end{minipage} \\
\midrule()
\endhead
Disodium ethylenediamine tetraacetate (Fisher \#BP120500) & 0.372 g \\
Water & 80ml \\
NaOH (Fisher \#BP359-212) & 0.036 g \\
\bottomrule()
\end{longtable}

\begin{itemize}
\tightlist
\item
  Stir vigorously on a magnetic stirrer. Adjust the pH to 8.0. Sterilize
  by autoclaving
\end{itemize}

\hypertarget{ethidium-bromide-10mgml}{%
\subsection*{\texorpdfstring{\textbf{Ethidium Bromide
(10mg/ml)}}{Ethidium Bromide (10mg/ml)}}\label{ethidium-bromide-10mgml}}
\addcontentsline{toc}{subsection}{\textbf{Ethidium Bromide (10mg/ml)}}

Note that we are transitioning to a EtBr-free facility. CAUTION:
Ethidium bromide is a mutagen and is toxic. Wear gloves when working
with ethidium bromide solutions and a mask when weighting the powder.

\begin{longtable}[]{@{}ll@{}}
\toprule()
\textbf{Ingredient} & \textbf{Amount} \\
\midrule()
\endhead
Ethidium bromide (Fisher \#BP1025) & 1g \\
Water & 100ml \\
\bottomrule()
\end{longtable}

\begin{itemize}
\item
  stir on magnetic stirrer for several hours to ensure that the dye has
  dissolved
\item
  light sensitive, wrap the container in aluminum foil or transfer to a
  dark bottle
\item
  store at 4C
\end{itemize}

\hypertarget{ctab}{%
\subsection*{\texorpdfstring{\textbf{CTAB}}{CTAB}}\label{ctab}}
\addcontentsline{toc}{subsection}{\textbf{CTAB}}

\begin{longtable}[]{@{}
  >{\raggedright\arraybackslash}p{(\columnwidth - 2\tabcolsep) * \real{0.7500}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 2\tabcolsep) * \real{0.2500}}@{}}
\toprule()
\begin{minipage}[b]{\linewidth}\raggedright
\textbf{Ingredient}
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
\textbf{Amount}
\end{minipage} \\
\midrule()
\endhead
4M NaCl (Fisher \#S271-1) & 35 mL \\
0.5M EDTA pH 8.0 (Fisher \#BP120500) & 4 mL \\
1M Tris-HCl pH 8.0 & 10 mL \\
CTAB (Cetyl Trimethyl-Ammonium Bromide) (Fisher \#AC22716100) & 2g \\
\bottomrule()
\end{longtable}

\begin{itemize}
\tightlist
\item
  Mix ingredients together in a clean beaker. Stir on hot plate with
  stir bar. Heat gently until CTAB is dissolved. Pour into graduated
  cylinder and bring volume up to 100ml using nanopure H2O. Pour into a
  bottle with a cap. Autoclave before using. Store at room temperature.
  Discard after 1 year.
\end{itemize}

\hypertarget{dntps-10mm}{%
\subsection*{\texorpdfstring{\textbf{dNTPs
(10mM)}}{dNTPs (10mM)}}\label{dntps-10mm}}
\addcontentsline{toc}{subsection}{\textbf{dNTPs (10mM)}}

To make 48 aliquots of 25ul each of 10mM dNTPs

All of the items below come in a Fisher package \#FERR01811

\begin{longtable}[]{@{}lll@{}}
\toprule()
\textbf{Ingredient} & \textbf{Amount for 12X} & \textbf{Amount for
1X} \\
\midrule()
\endhead
dTTP (100mM) & 120 ul & 10 ul \\
dATP (100mM) & 120 ul & 10 ul \\
dGTP (100mM) & 120 ul & 10 ul \\
dCTP (100mM) & 120 ul & 10 ul \\
ddH20 & 720 ul & 60 ul \\
\bottomrule()
\end{longtable}

\hypertarget{m-nacl-500ml}{%
\subsection*{\texorpdfstring{\textbf{5M NaCl
500ml}}{5M NaCl 500ml}}\label{m-nacl-500ml}}
\addcontentsline{toc}{subsection}{\textbf{5M NaCl 500ml}}

\begin{longtable}[]{@{}ll@{}}
\toprule()
\textbf{Ingredient} & \textbf{Amount} \\
\midrule()
\endhead
NaCl (Fisher \#S271-1) & 146.1 g \\
H2O & 350 mL \\
\bottomrule()
\end{longtable}

*Dissolve, then bring up to volume with H2O *Sterilize by autoclaving
(15 minutes)

\hypertarget{naoh-10n}{%
\subsection*{\texorpdfstring{\textbf{NaOH
10N}}{NaOH 10N}}\label{naoh-10n}}
\addcontentsline{toc}{subsection}{\textbf{NaOH 10N}}

\begin{longtable}[]{@{}ll@{}}
\toprule()
\textbf{Ingredient} & \textbf{Amount} \\
\midrule()
\endhead
NaOH (Fisher \#BP359-212) & 40g \\
H20 & 80ml \\
\bottomrule()
\end{longtable}

*continue adding water to 100ml

\hypertarget{tween-20}{%
\subsection*{\texorpdfstring{\textbf{10\%
Tween-20}}{10\% Tween-20}}\label{tween-20}}
\addcontentsline{toc}{subsection}{\textbf{10\% Tween-20}}

\begin{longtable}[]{@{}ll@{}}
\toprule()
\textbf{Ingredient} & \textbf{Amount} \\
\midrule()
\endhead
100\% Tween-20 (Fisher \#97062-332) & 1ml \\
Distilled H20 & 9 ml \\
\bottomrule()
\end{longtable}

\hypertarget{tbe-10x-stock-solution}{%
\subsection*{\texorpdfstring{\textbf{TBE 10X (stock
solution)}}{TBE 10X (stock solution)}}\label{tbe-10x-stock-solution}}
\addcontentsline{toc}{subsection}{\textbf{TBE 10X (stock solution)}}

\begin{longtable}[]{@{}ll@{}}
\toprule()
\textbf{Ingredient} & \textbf{Amount} \\
\midrule()
\endhead
Tris base (Fisher \#BP15210) & 108g \\
Boric acid (Fisher \#A73 1) & 55g *dissolve in Milli-Q water \\
0.5M EDTA (pH 8.0) & 40ml \\
\bottomrule()
\end{longtable}

\begin{itemize}
\item
  increase the final volume to 1 liter
\item
  store at room temperature or at 4C
\item
  might be too concentrated for long term storage, ok if you are
  planning to do a lot of work
\end{itemize}

\hypertarget{tbe-5x-stock-solution}{%
\subsection*{\texorpdfstring{\textbf{TBE 5X (stock
solution)}}{TBE 5X (stock solution)}}\label{tbe-5x-stock-solution}}
\addcontentsline{toc}{subsection}{\textbf{TBE 5X (stock solution)}}

\begin{longtable}[]{@{}ll@{}}
\toprule()
\textbf{Ingredient} & \textbf{Amount} \\
\midrule()
\endhead
Tris base (Fisher \#BP15210) & 54 g \\
Boric acid (Fisher \#A73 1) & 27.5 g *dissolve in Milli-Q water \\
0.5M EDTA (pH 8.0) & 20ml \\
\bottomrule()
\end{longtable}

\begin{itemize}
\item
  increase the final volume to 1 liter
\item
  store at room temperature or at 4C
\item
  (1X TBE = 200ml of 5X TBE + 800ml)
\end{itemize}

\hypertarget{te}{%
\subsection*{\texorpdfstring{\textbf{TE}}{TE}}\label{te}}
\addcontentsline{toc}{subsection}{\textbf{TE}}

A low TE buffer that is 0.1mM EDTA is better for DNA that is destined
for next-generation sequencing, because the EDTA interferes with
enzymatic reactions.

\begin{longtable}[]{@{}ll@{}}
\toprule()
\textbf{Ingredient} & \textbf{Amount} \\
\midrule()
\endhead
Tris-HCl (pH 8.0) X M & ? \\
EDTA (pH 8.0) X mM & ? \\
\bottomrule()
\end{longtable}

\begin{itemize}
\item
  Start with a small volume of nanopure H20, about half of your final
  volume, in a beaker with a stirbar.
\item
  Measure out the amount of Tris-Hcl needed for a 10mM solution
\item
  Add to H20
\item
  Measure out the amount of EDTA needed for a 1mM solution
\item
  Add to H20 and Tris
\item
  While stirring solution, carefully insert calibrated pH probe
\item
  Adjust pH as needed with either HCl or NaOH
\item
  Note: EDTA will not fully dissolve until pH is near 8.0 and will cause
  pH to fluctuate as it dissolves, so be prepared to make many small
  adjustments in either direction ( it is best to work with diluted
  acids and bases).
\item
  Also: if you overshoot 8.0, it is okay to readjust in the other
  direction (even if you wildly overshoot the mark) and you DO NOT need
  to throw the solution out and start again
\item
  After pH is adjusted, transfer solution to graduated cylinder and add
  H20 to bring to final concentration.
\item
  Some people think TE buffer should not be autoclaved, but I usually
  autoclave it. Tris breaks down easily after autoclaving and will
  precipitate out more easily, so you may have to make the solution up
  more frequently. In my opinion, that is better than chancing a
  solution with DNAse.
\item
  Store at room temperature and discard if sediment is forming on the
  bottom of the container or if it starts growing or if it gets cloudy.
\item
  For 100ml: 1M Tris-HCl (pH 8.0) 1ml and 0.5M EDTA (pH 8.0) 200ul
\item
  Bring to a final volume of 100ml with nanopure H20
\end{itemize}

\hypertarget{m-tris}{%
\subsection*{\texorpdfstring{\textbf{1M Tris}}{1M Tris}}\label{m-tris}}
\addcontentsline{toc}{subsection}{\textbf{1M Tris}}

\begin{longtable}[]{@{}ll@{}}
\toprule()
\textbf{Ingredient} & \textbf{Amount} \\
\midrule()
\endhead
Tris-base (Fisher \#BP15210) & 121.1g \\
\bottomrule()
\end{longtable}

\begin{itemize}
\tightlist
\item
  Bring volume up to 1 liter with ddH20
\end{itemize}

\hypertarget{m-tris-hcl}{%
\subsection*{\texorpdfstring{\textbf{1M
Tris-HCL}}{1M Tris-HCL}}\label{m-tris-hcl}}
\addcontentsline{toc}{subsection}{\textbf{1M Tris-HCL}}

\begin{longtable}[]{@{}ll@{}}
\toprule()
\textbf{Ingredient} & \textbf{Amount} \\
\midrule()
\endhead
Tris base (Fisher \#BP15210) & 60.56g \\
ddH20 & 400ml \\
\bottomrule()
\end{longtable}

\begin{itemize}
\tightlist
\item
  Adjust pH to 8.0 using HCL and bring final volume to 500ml\\
\end{itemize}

\hypertarget{determining-barcodes-and-indexes-for-illumina-sequencing}{%
\chapter{Determining Barcodes and Indexes for Illumina
Sequencing}\label{determining-barcodes-and-indexes-for-illumina-sequencing}}

Tips for determining barcodes and indexes for Illumina Sequencing

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

First, you should make a plan for the adapters that you are going to use
for each sample. As you choose adapters, you want to ensure that there
is not low diversity in the indexes or in the first 5-6 bases of every
read. Low diversity can result in high error rates because of an
unbalanced color scheme as explained here by Illumina:

\href{https://www.illumina.com/documents/products/technotes/technote-nextera-rapid-capture-low-plex-pooling-guidelines.pdf}{Illumina
Low Plex Pooling Guidelines}

The term ``nucleotide diversity'' describes the proportion of each
nucleotide (A, T, C, and G) at each position in a sequencing library. A
``balanced'' or ``diverse'' library has equal proportions of A, C, G,
and T nucleotides at each base position in a sequencing library. In
particular, nucleotide diversity is key during the first 4--7 cycles of
Read 1 for all Illumina sequencing systems. The sequencing software uses
images from these early cycles to identify the location of each cluster
in a process called template generation. This results in overclustering.
Overclustering creates image analysis problems, including loss of focus,
poor template generation, and issues with cluster registration. The
increased overall signal brightness of the flow cell makes it difficult
for the sequencer to read and find the appropriate focal plane.

\href{https://support.illumina.com/content/dam/illumina-marketing/documents/products/other/miseq-overclustering-primer-770-2014-038.pdf}{Illumina
Consequences of Overclustering}

Next, and especially if you are doing library prep for the first time, I
suggest that you draw out what is happening to the DNA or RNA at every
step of the process:

\href{https://docs.google.com/spreadsheets/d/1VSqr03MbtS0hFzgnOu3-oHL2fzf4qL5jfEEyDE76bv4/edit\#gid=0}{Example
Genomic Library Prep with p5 inline barcode and index in p7 PCR Primer}

\href{https://docs.google.com/document/d/1xr99sMucP-x2rK6K61vtGHKNahl9UYFwh_oEb0AtHX8/edit}{Example
Library prep with inline barcodes and dual indexes}

For genomic DNA library prep, we are now working with inline barcodes on
the P5 (left) adapter and indexes on the P7 (right) adapter. See above
figure for details (note: KEL thinks there is issues with this system
and this may change). These can give up to 12*12 = 144 unique barcodes.
But, keep in mind that every sample can't have the same barcode (e.g.,
can't have 12 samples with 1 barcode and 12 indexes), and every sample
can't have the same index (e.g., can't have 12 samples with 12 barcodes
and 1 index) because of the low-diversity problem.

Also, note that the indexes will also demultiplex with Illumina's
software at the sequencing facility. But after you receive them, the
reads will need to be further demultiplexed using the inline barcodes
with a program such as STACKS.

\href{https://docs.google.com/spreadsheets/d/1bro9DbuJB9zH8KtIUH3H7FHrlVO2gZYbIsUbx7yax70/edit\#gid=0}{Lotterhos
adapter and primer database}

\href{https://docs.google.com/spreadsheets/d/1Ok9EEkloK6RCCpKsc4tjklF4UygO8SnATWK8HGLgpvY/edit\#gid=0}{Lotterhos
y-adapters with inline barcodes and dual indexes}

\hypertarget{dna-extraction-qiagen}{%
\chapter{DNA Extraction (Qiagen)}\label{dna-extraction-qiagen}}

The protocol for extracting DNA with the Qiagen Tissue Kit

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

This protocol has proven successful for rockfish (Sebastes), cod (Gadus
morhua), and oysters (Crassostrea).

\hypertarget{materials-and-equipment}{%
\section*{\texorpdfstring{\textbf{Materials and
Equipment}}{Materials and Equipment}}\label{materials-and-equipment}}
\addcontentsline{toc}{section}{\textbf{Materials and Equipment}}

\hypertarget{qiagen-extraction-kits}{%
\subsection*{\texorpdfstring{\textbf{Qiagen Extraction
Kits}}{Qiagen Extraction Kits}}\label{qiagen-extraction-kits}}
\addcontentsline{toc}{subsection}{\textbf{Qiagen Extraction Kits}}

\hypertarget{qiagen-blood-and-tissue-extraction-kit---single-sample}{%
\subsubsection*{\texorpdfstring{\textbf{Qiagen Blood and Tissue
Extraction Kit - single
sample}}{Qiagen Blood and Tissue Extraction Kit - single sample}}\label{qiagen-blood-and-tissue-extraction-kit---single-sample}}
\addcontentsline{toc}{subsubsection}{\textbf{Qiagen Blood and Tissue
Extraction Kit - single sample}}

\begin{itemize}
\item
  250 samples - Cat No./ID: 69506 - \textasciitilde700\$
\item
  2.80\$ per sample
\end{itemize}

\hypertarget{qiagen-blood-and-tissue-extraction-kit---96-plate}{%
\subsubsection*{\texorpdfstring{\textbf{Qiagen Blood and Tissue
Extraction Kit - 96
plate}}{Qiagen Blood and Tissue Extraction Kit - 96 plate}}\label{qiagen-blood-and-tissue-extraction-kit---96-plate}}
\addcontentsline{toc}{subsubsection}{\textbf{Qiagen Blood and Tissue
Extraction Kit - 96 plate}}

\begin{itemize}
\item
  4x96 (384 samples) - Cat No./ID: 69506 - \textasciitilde1300\$
\item
  \textasciitilde3.40\$ per sample
\end{itemize}

\hypertarget{additional-reagents-and-consumables}{%
\subsection*{\texorpdfstring{\textbf{Additional Reagents and
Consumables}}{Additional Reagents and Consumables}}\label{additional-reagents-and-consumables}}
\addcontentsline{toc}{subsection}{\textbf{Additional Reagents and
Consumables}}

\begin{itemize}
\item
  Razors (for cutting tissue)
\item
  Flame (sterilization)
\item
  Weigh paper
\item
  Kim wipes
\item
  Pipet Tips (10uL,20uL,200uL,1000uL)
\item
  EtoH (100\% - molecular grade)
\end{itemize}

\hypertarget{equipment}{%
\subsection*{\texorpdfstring{\textbf{Equipment}}{Equipment}}\label{equipment}}
\addcontentsline{toc}{subsection}{\textbf{Equipment}}

\begin{itemize}
\item
  Micro-scale (need to be able to measure down to 20mg)
\item
  Thermo-mixer (It has both single sample and plate attachments)
\item
  Vortexer
\item
  Microcentrifuge (large centrifuge for plate extractions)
\item
  Pipets (10uL,20uL,200uL,1000uL)
\end{itemize}

\hypertarget{protocol-1}{%
\section*{\texorpdfstring{\textbf{Protocol}}{Protocol}}\label{protocol-1}}
\addcontentsline{toc}{section}{\textbf{Protocol}}

\hypertarget{before-starting}{%
\subsection*{\texorpdfstring{\textbf{Before
Starting}}{Before Starting}}\label{before-starting}}
\addcontentsline{toc}{subsection}{\textbf{Before Starting}}

\begin{itemize}
\item
  If any precipitate has formed in either the Buffer ATL or Buffer AL,
  warm solution(s) to 56 degrees C until precipitates have fully
  dissolved.
\item
  Preheat thermo-mixer to 56 degrees C.

  \hypertarget{tissue-lysing}{%
  \subsection*{Tissue Lysing}\label{tissue-lysing}}
  \addcontentsline{toc}{subsection}{Tissue Lysing}

  \begin{enumerate}
  \def\labelenumi{\arabic{enumi}.}
  \item
    Prep surfaces and area around the scale. First clean with bleach
    solution, followed by H2O, then 70\% EtoH.
  \item
    If working with many samples (\textgreater10) get a styrofoam
    container with ice to keep tissue samples cool while prepping for
    extraction.
  \item
    Remove tissues from freezer and let thaw (do not let them sit any
    longer than necessary). Once thawed they can be placed on ice (this
    generally won't cause them to re-freeze).
  \item
    Tare piece of weighing paper on scale.
  \item
    Carefully remove tissue from 2mL tube. Using a sterile razor blade
    to cut a piece of tissue roughly 20mg in weight. Using 100\% EtoH,
    gently rinse tissue and pat dry with clean kim wipe, so that the
    tissue appears nearly dry.
  \item
    Place tissue on weigh paper and measure the weight, if the tissue
    weighs 20mg+/-2mg place it in extraction tube. If not, either remove
    tissue as necessary or add more from sample tissue.
  \end{enumerate}
\end{itemize}

\textbf{Note:} Between samples razors \& forceps should be sterilized by
spraying with EtoH and using a flame to burn off any residual EtoH and
tissue.

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Record weight in lab notebook.
\item
  After tissues have been processed and place into new clean 1.5mL
  tubes, add 180 uL of Buffer ATL and 20 uL of Proteinase K to each
  sample tube (same as standard Qiagen kit).
\item
  Seal tubes, confirm all tube labels, then vortex for 15 seconds.
\item
  Place samples on thermo-mixer set at 56 degrees at 600 rpm.
  \textbf{BEFORE STARTING THE MIXER}: Make sure all tube caps are
  securely fastened. In addition, after 2 minutes on the thermo-mixer
  confirm the tube caps haven't popped open.
\item
  Samples can be left for at least 2 hours or overnight to allow for all
  tissue to be completely lysed.
\end{enumerate}

\hypertarget{extraction}{%
\subsection*{\texorpdfstring{\textbf{Extraction}}{Extraction}}\label{extraction}}
\addcontentsline{toc}{subsection}{\textbf{Extraction}}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Prep surfaces and area around the work bench. First clean with bleach
  solution, followed by H2O, then 70\% EtoH.
\item
  Pre-label top of spin-column tube and 2 additional 1.5 mL tubes (one
  label E1 for elution 1 and E2 for elution 2).
\item
  Add 200 μl Buffer AL. Mix thoroughly by vortexing.
\item
  Add 200 μl ethanol (96--100\%). Mix thoroughly by vortexing.
\end{enumerate}

\textbf{Note:} A working solution of Buffer AL \& ethanol can be made to
be aliquoted into sample tubes.

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Pipet the mixture into a DNeasy Mini spin column placed in a 2 ml
  collection tube. Centrifuge at ≥ 6000 x g (8000 rpm) for 1 min.
  Discard the flow-through and collection tube.
\item
  Place the spin column in a new 2 ml collection tube. Add 500 μl Buffer
  AW1, and centrifuge for 1 min at ≥6000 x g. Discard the flow-through
  and collection tube.
\item
  Place the spin column in a new 2 ml collection tube. Add 500 μl Buffer
  AW2, and centrifuge for 3 min at 20,000 x g (14,000 rpm). Discard the
  flow-through and collection tube.
\end{enumerate}

\textbf{Note:} Spin column membrane should be dry after this step. Spin
for an additional 1 min if not.

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Transfer the spin column to a new 1.5 ml or 2 ml microcentrifuge tube.
\item
  Elute the DNA by adding 200 μl Buffer AE to the center of the spin
  column membrane. Incubate for 1 min at room temperature (15--25°C).
  Centrifuge for 1 min at ≥6000 x g.
\item
  Perform second elution by adding 200 μl Buffer AE to the center of the
  spin column membrane. Incubate for 10 min at room temperature
  (15--25°C). Centrifuge for 1 min at ≥6000 x g.
\item
  Extracted DNA can be place in fridge for a few days, or store at -20
  or -80 for long term storage. \textbf{AVOID} numerous freeze-thaw
  cycles.\\
\end{enumerate}

\hypertarget{dna-extraction-omega}{%
\chapter{DNA Extraction (Omega)}\label{dna-extraction-omega}}

The protocol for extracting DNA with the Omega Tissue DNA Kit

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\emph{Read the entire protocol ahead of time; DNA extraction with this
kit takes two consecutive days in the lab but some preparations and
arrangements need to be done before day 1 and might take more than a day
to be completed, e.g., autoclave materials, order supplies, arrange to
use the large centrifuge, etc.}

\hypertarget{materials-and-equipment-1}{%
\section*{\texorpdfstring{\textbf{Materials and
Equipment}}{Materials and Equipment}}\label{materials-and-equipment-1}}
\addcontentsline{toc}{section}{\textbf{Materials and Equipment}}

\hypertarget{kits}{%
\subsection*{\texorpdfstring{\textbf{Kits}}{Kits}}\label{kits}}
\addcontentsline{toc}{subsection}{\textbf{Kits}}

\hypertarget{omega-e.z.n.a.-tissue-dna-kit---individual-spin-columns-good-for-extracting-fewer-than-8-16-samples-or-low-yield-samples.-cheaper-per-sample-1.4-3.2-per-sample.}{%
\subsubsection*{\texorpdfstring{\textbf{Omega E.Z.N.A. Tissue DNA Kit -
individual spin columns: good for extracting fewer than 8-16 samples or
low yield samples. Cheaper per sample (\textasciitilde\$1.4-\$3.2 per
sample).}}{Omega E.Z.N.A. Tissue DNA Kit - individual spin columns: good for extracting fewer than 8-16 samples or low yield samples. Cheaper per sample (\textasciitilde\$1.4-\$3.2 per sample).}}\label{omega-e.z.n.a.-tissue-dna-kit---individual-spin-columns-good-for-extracting-fewer-than-8-16-samples-or-low-yield-samples.-cheaper-per-sample-1.4-3.2-per-sample.}}
\addcontentsline{toc}{subsubsection}{\textbf{Omega E.Z.N.A. Tissue DNA
Kit - individual spin columns: good for extracting fewer than 8-16
samples or low yield samples. Cheaper per sample
(\textasciitilde\$1.4-\$3.2 per sample).}}

\begin{itemize}
\item
  200 reactions - cat \#: D3396-02
\item
  50 reactions - cat \#: D3396-01
\item
  5 reactions - cat \#: D3396-00
\end{itemize}

\hypertarget{omega-e.z.-96-tissue-dna-kit---96-well-plate-columns-good-for-extracting-a-large-number-of-samples-saves-time-by-using-the-multi-channel-pipette-and-in-number-of-centrifugation-rounds.-more-expensive-per-sample-2.1-2.6-per-sample.}{%
\subsubsection*{\texorpdfstring{\textbf{Omega E.Z. 96 Tissue DNA Kit -
96-well plate columns: good for extracting a large number of samples
(saves time by using the multi-channel pipette and in number of
centrifugation rounds). More expensive per sample
(\textasciitilde\$2.1-\$2.6 per
sample).}}{Omega E.Z. 96 Tissue DNA Kit - 96-well plate columns: good for extracting a large number of samples (saves time by using the multi-channel pipette and in number of centrifugation rounds). More expensive per sample (\textasciitilde\$2.1-\$2.6 per sample).}}\label{omega-e.z.-96-tissue-dna-kit---96-well-plate-columns-good-for-extracting-a-large-number-of-samples-saves-time-by-using-the-multi-channel-pipette-and-in-number-of-centrifugation-rounds.-more-expensive-per-sample-2.1-2.6-per-sample.}}
\addcontentsline{toc}{subsubsection}{\textbf{Omega E.Z. 96 Tissue DNA
Kit - 96-well plate columns: good for extracting a large number of
samples (saves time by using the multi-channel pipette and in number of
centrifugation rounds). More expensive per sample
(\textasciitilde\$2.1-\$2.6 per sample).}}

\begin{itemize}
\item
  4 plates, 384 reactions - cat \#: D1196-01
\item
  1 plate, 96 reactions - cat \#: D1196-00
\end{itemize}

\hypertarget{reagents-and-consumables}{%
\subsection*{\texorpdfstring{\textbf{Reagents and
Consumables}}{Reagents and Consumables}}\label{reagents-and-consumables}}
\addcontentsline{toc}{subsection}{\textbf{Reagents and Consumables}}

Locate/confirm equipment:

\begin{itemize}
\item
  Gloves - wear gloves to handle \textbf{all} extraction materials.
\item
  10\% bleach squirt bottle (for sterilizing the bench top)
\item
  70-90 \% EtOH squirt bottle (for sterilizing the bench top)
\item
  Razor blades (for cutting tissue)
\item
  Forceps (for handling the tissue)
\item
  Lighter (for sterilizing blades and forceps)
\item
  Weigh paper
\item
  Kimwipes
\item
  Calibrated pipettes
\item
  Autoclaved or sterile/filtered (preferred) pipette tips -
  sterile/filtered tips are preferred but check with lab manager before
  you use them.
\item
  Autoclaved 1.5 eppendorf tubes
\item
  Tube racks (if using the individual spin columns)
\item
  Sharps container
\item
  Trash bin
\item
  90\% molecular grade EtOH(for sterilizing blades and forceps)
\item
  100\% molecular grade EtOH in a beaker (for purification if using
  individual spin columns)
\item
  Micro-scale (need to be able to measure between 1 and 20 mg)
\item
  Thermomixer - for the Omega kit, you will use setting \#7: ``TB Buffer
  DNA Digest''. It has both single tube and plate attachments, stored in
  the drawer underneath the counter.
\item
  Vortexer
\item
  Microcentrifuge
\item
  Pipettes (10 uL, 20 uL, 200 uL, 1000uL)
\end{itemize}

\hypertarget{protocol-2}{%
\section*{\texorpdfstring{\textbf{Protocol}}{Protocol}}\label{protocol-2}}
\addcontentsline{toc}{section}{\textbf{Protocol}}

\hypertarget{prepping-for-day-1-of-extraction---check-list}{%
\subsection*{\texorpdfstring{\textbf{Prepping for day 1 of extraction -
check
list:}}{Prepping for day 1 of extraction - check list:}}\label{prepping-for-day-1-of-extraction---check-list}}
\addcontentsline{toc}{subsection}{\textbf{Prepping for day 1 of
extraction - check list:}}

\begin{itemize}
\item
  Calculate how many samples you have in total (including neg and/or pos
  controls, if any), consider time and cost, and decide whether to use
  plates or individual tubes.
\item
  If too many samples to extract in a single round, decide how many
  samples per extraction batch, remember to leave room for controls if
  needed.
\item
  Calculate supplies and volumes needed, and check if you have enough or
  need to order new.
\item
  Autoclave materials not provided in the kit (eppendorf tubes and tips
  (if not using sterile/filtered) and forceps). See autoclave training
  information if this is your first lab work at the MSC!
\item
  When using plates, it can be more difficult to keep track of which
  samples go in which wells, since you can not label wells directly. It
  is advised to draw the layout of the plate and plan sample allocation
  on to the wells ahead of time, and keep that layout near by for
  reference during tissue digestion prep. Also, make sure to keep plates
  in the same orientation (top left corner A-1) at all times.
\end{itemize}

\hypertarget{prepping-for-day-2-of-extraction---check-list}{%
\subsection*{\texorpdfstring{\textbf{Prepping for day 2 of extraction -
check
list:}}{Prepping for day 2 of extraction - check list:}}\label{prepping-for-day-2-of-extraction---check-list}}
\addcontentsline{toc}{subsection}{\textbf{Prepping for day 2 of
extraction - check list:}}

\begin{itemize}
\item
  Reserve plate centrifuge in Hughes lab if using plates.
\item
  Make sure kit purification buffers are diluted as directed in the kit
  manual.
\end{itemize}

\hypertarget{day-1-of-extraction---tissue-digestion}{%
\subsection*{\texorpdfstring{\textbf{Day 1 of extraction - tissue
digestion}}{Day 1 of extraction - tissue digestion}}\label{day-1-of-extraction---tissue-digestion}}
\addcontentsline{toc}{subsection}{\textbf{Day 1 of extraction - tissue
digestion}}

D1-1. Set thermomixer to desired temperature and rpm (typically 60oC and
450 RPM, but may vary if you are tweaking the protocol).

D1-2. Sterilize surfaces and area around the scale - wipe with with 10
\% bleach first, then DI H2O, then 70-90\% EtOH.

D1-3. Remove tissue samples from freezer and let thaw (do not let them
sit any longer than necessary). Take \textasciitilde10 samples at the
time. Once thawed they can be placed on ice/cold beads.

D1-4. Pre-load lysis plate or sterile eppendorf tubes with \textbf{200
uL of TL buffer per well plate or tube} using a serile pipette tip.

D1-5. Tare scale with a piece of weighing paper on it.

D1-6. Dip \textbf{ONLY} the cutting edge of the blade into EtOH in
beaker. Move the beaker aside. Hold the blade \textbf{vertically, away
from your body and away from the EtOH beaker} and flame the blade. When
all EtOH has burned off and the flame is gone, set the sterile blade on
a clean kimwipe on the bench. Dip the tip of the forceps into EtOH
beaker, move the beaker aside. Hold the forceps \textbf{horizontally,
away from your body and from the EtOH beaker} and flame the forceps to
sterilize.

D1-7. Using the sterile forceps, carefully remove tissue from 2mL tube
and set it on the kimwipe. Using the sterile razor blade, cut a piece of
tissue 15-20 mg. Pat dry the piece of tissue with a clean kimwipe, so
that the tissue appears nearly dry.

D1-8. Place tissue on weigh paper and record the weight in lab notebook.
If the tissue weights less than 15mg or more than 20mg, remove or add
more tissue as needed to reach this range.

D1-9. Take the weigh paper out of the scale, set it on a clean kimwipe
and use the sterile forceps and blade to cut up the tissue into smaller
pieces.

D1-10. Transfer the cut up tissue to a well plate or an eppendorf tube,
making sure that all pieces of tissue fall into the buffer.

D1-11. Label the tube with sample ID, or keep track of the plate layout
on your lab notebook, so you know which samples are in which wells.

D1-12. Discard the used blade into the sharps container. Discard all
used kimwipes and wipe the bench again with 70-90\% EtOH.

\begin{quote}
\textbf{Note:} Between samples, a new blade should be used if available
and forceps should be sterilized by diping in EtOH and flaming to burn
off any residual EtOH and tissue.
\end{quote}

D1-13. Repeat steps 5-12 until all samples are in buffer.

D-14. Add 25uL of \emph{OB Protease Solution} (OB Protease and
Proteinase K, provided with the kit, serve the same purpose as digestive
buffers) to each sample tube or well plate. You can reuse the tip if you
are sure that the tip did not touch the liquid or the sides of the tubes
or wells. If the tip does touch any surfaces in the lysis plate or
tubes, replace with a sterile tip.

D-15. Close tubes or seal plate with a silicone mat, confirm all tubes
or plate wells are labelled correctly and vortex the tubes/plate for 15
seconds.

D-16. Place samples on thermomixer. \textbf{Before restarting the
mixer:} Make sure all tube caps are securely fastened/plate is securely
sealed. In addition, after 2 minutes on the thermomixer confirm the tube
caps/silicone mat have not popped open.

\textbf{Note:} Samples can be left for at least 3 hours or overnight to
allow for all tissue to be completely lysed. If possible, pause the
mixer, take plate/tubes out and vortex them for 15 sec every 1/2h to 1h
to help with lysis.

\hypertarget{day-2-of-extraction---dna-purification.}{%
\subsection*{\texorpdfstring{\textbf{Day 2 of extraction - DNA
purification.}}{Day 2 of extraction - DNA purification.}}\label{day-2-of-extraction---dna-purification.}}
\addcontentsline{toc}{subsection}{\textbf{Day 2 of extraction - DNA
purification.}}

D2-1. Sterilize bench top - wipe with with 10\% bleach first, then DI
H2O, then 70-90\% EtOH.

D2-2. Heat up heating block (or check oven temperature in Hughes lab) to
70oC and place the volume needed of elution buffer (or molecular grade
water) in it.

D2-3. Follow the kit manual for plates or individual tubes for DNA
purification with the following modifications, if needed:

\begin{itemize}
\item
  skip the optional step ``add RNAase''
\item
  skip the optional step ``column equilibration''
\item
  when centrifuging the lysis plate and the E-Z 96 DNA plate
  (mini-columns plate), tape them together.
\item
  skip the optional step ``dry samples at 70oC''
\item
  we prefer to use molecular grade water as the elution buffer instead
  of the kit buffer.
\item
  elution volume may vary depending on starting material and extraction
  yield. Minimum volume is 30 uL, to ensure that the entire memembrane
  is soaked with elution buffer.
\item
  the second elution can be eluted into a separate tube, if there are
  concerns about DNA concentration in the first elution.
\item
  Align the tubes in the centrifuge in this manner:
  \includegraphics{https://user-images.githubusercontent.com/98117631/183438550-75c4d11b-8d6a-4b37-8579-f74a95a913f5.png}
\end{itemize}

D2-4. You will likely need to spot-check the concentration of your
extracted DNA, either by Qubit or Picogreen assay. Both are destructive
assays, meaning you will need to use some aliquote (typically 1-2 uL) of
your eluted DNA to measure concentration. Make note of how much you used
from which samples, to keep track of how much is available for
downstream library prep and sequencing.

D2-5. Extracted DNA can be place in fridge for a few days, or store at
-20oC or -80oC for long term storage. Avoid numerous freeze-thaw
cycles.\\

\hypertarget{dna-extraction-spartina}{%
\chapter{DNA Extraction (Spartina)}\label{dna-extraction-spartina}}

A rough outline of the protocol used for extracting DNA from Spartina

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

Originally written by Amber Eule-Nashoba

\hypertarget{extraction-on-spartinaalterniflora-with-qiagen-dnaeasy-plant-mini-kit}{%
\section*{\texorpdfstring{\textbf{Extraction on
\emph{Spartinaalterniflora} with Qiagen DNAeasy Plant Mini
Kit}}{Extraction on Spartinaalterniflora with Qiagen DNAeasy Plant Mini Kit}}\label{extraction-on-spartinaalterniflora-with-qiagen-dnaeasy-plant-mini-kit}}
\addcontentsline{toc}{section}{\textbf{Extraction on
\emph{Spartinaalterniflora} with Qiagen DNAeasy Plant Mini Kit}}

Following collection, fresh leaf tissue samples are stored in silica
gel.

Tissue is homogenized using the TissueLyser II.

Following homogenization, 21 mg of leaf powder is used for each
extraction.

Remainder of extraction follows procedure outlined in Plant Mini Kit
handbook.

\hypertarget{tips-for-pooled-sequences}{%
\section*{\texorpdfstring{\textbf{Tips for Pooled
Sequences:}}{Tips for Pooled Sequences:}}\label{tips-for-pooled-sequences}}
\addcontentsline{toc}{section}{\textbf{Tips for Pooled Sequences:}}

Use an approximately equivalent weight of tissue from each sample. After
homogenizing and prior to weighing out powdered tissue for extraction,
samples can be combined and well mixed or an equal weight can be
collected from each tube of powdered tissue.\\

\hypertarget{rna-extraction}{%
\chapter{RNA Extraction}\label{rna-extraction}}

The protocol for extracting RNA from tissue samples

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\begin{itemize}
\item
  \href{https://drk-lo.github.io/lotterhoslabprotocols/molecprot_rnaextraction/\#general-tips-for-rna-extraction}{General
  tips for RNA extraction}
\item
  \href{https://drk-lo.github.io/lotterhoslabprotocols/molecprot_rnaextraction/\#homogenize-tissue-sample-with-tissueruptor}{Homogenize
  tissue sample with TissueRuptor}
\end{itemize}

\hypertarget{kits-that-we-use}{%
\section*{\texorpdfstring{\textbf{Kits that we
use}}{Kits that we use}}\label{kits-that-we-use}}
\addcontentsline{toc}{section}{\textbf{Kits that we use}}

\begin{itemize}
\item
  \href{https://drk-lo.github.io/lotterhoslabprotocols/molecprot_rnaextraction/\#protocol-for-tri-reagent-rna-isolation}{TRI
  Reagent RNA Isolation}

  \begin{itemize}
  \item
    (Trizol based)
  \item
    Works well with \emph{Crassostrea virginica}
  \end{itemize}
\item
  \href{https://drk-lo.github.io/lotterhoslabprotocols/molecprot_rnaextraction/\#protocol-with-qiagen-rnaeasy-kit}{Qiagen
  RNAeasy Kit}
\item
  \href{https://drk-lo.github.io/lotterhoslabprotocols/molecprot_rnaextraction/\#protocol-for-zymo-quick-dna-rna-miniprep-and-tissueruptor-homogenization}{ZYMO
  Quick DNA/RNA Miniprep}
\end{itemize}

\hypertarget{general-tips-for-rna-extraction}{%
\section*{\texorpdfstring{\textbf{General tips for RNA
extraction}}{General tips for RNA extraction}}\label{general-tips-for-rna-extraction}}
\addcontentsline{toc}{section}{\textbf{General tips for RNA extraction}}

\begin{itemize}
\item
  Work surfaces should be cleaned with 10\% Bleach and then RNAase Away
  to ensure an RNAase free environment.
\item
  RNA extraction is highly dependent on the starting tissue and quality
  of preservation. Only extract RNA from:

  \begin{itemize}
  \item
    Fresh tissue
  \item
    Tissue flash frozen in liquid Nitrogen

    \begin{itemize}
    \item
      Some people recommend freezing tissues in RNAlater, so that when
      defrosting, RNA is still preserved

      \begin{itemize}
      \tightlist
      \item
        We have done this successfully, but have not formally tested it.
      \end{itemize}
    \end{itemize}
  \item
    Tissue preserved in RNAlater

    \begin{itemize}
    \tightlist
    \item
      \textbf{Note}- it is best to put tissue in RNAlater, place in
      refrigerator overnight, and then freeze
    \end{itemize}
  \end{itemize}
\item
  More general RNA tips can be found
  \href{https://www.thermofisher.com/us/en/home/references/ambion-tech-support/nuclease-enzymes/general-articles/working-with-rna.html}{here}
  and
  \href{https://www.thermofisher.com/us/en/home/references/ambion-tech-support/rna-isolation/general-articles/the-basics-rna-isolation.html}{here}
\item
  \textbf{Extractions need to be performed in the chemical hood}
\end{itemize}

\hypertarget{homogenize-tissue-sample-with-tissueruptor}{%
\subsection*{\texorpdfstring{\textbf{Homogenize tissue sample with
TissueRuptor}}{Homogenize tissue sample with TissueRuptor}}\label{homogenize-tissue-sample-with-tissueruptor}}
\addcontentsline{toc}{subsection}{\textbf{Homogenize tissue sample with
TissueRuptor}}

The Tissue Ruptor uses disposable probes that can be sterilized and
re-used.

\textbf{To sterilize probes in the Autoclave:}

\begin{itemize}
\item
  place probes in a large beaker and cover with aluminum and autoclave
  tape
\item
  autoclave at 120°C for 30 min (20 min to reach temperature + 15 min at
  120°C)
\item
  \textbf{Note} TissueRuptor Probes may be cleaned /sterilized up to 5
  times (15 min at 120°C) or until visible signs of wear are seen
\end{itemize}

\textbf{Steps to homogenize:}

\begin{itemize}
\item
  Get each tissue out of the freezer and keep on ice

  \begin{itemize}
  \tightlist
  \item
    do one tissue at a time
  \end{itemize}
\item
  prepare 2 mL vials (with somewhat flat bottoms) each with 600 µl
  \textbf{DNA/RNA Lysis Buffer}
\item
  cut \textasciitilde25 mg of tissue and put in prepared lysis vials
\item
  keep vials on ice
\item
  run TissueRuptor at level 3 for 7 seconds (or until tissue is
  completely lysed)
\item
  return to ice
\item
  clean probe with 3 washes of DI water

  \begin{itemize}
  \item
    you should have 3 separate falcon tubes with DI water
  \item
    run TissueRuptor through each DI wash for 5 seconds
  \item
    remove probe and set aside to be autoclaved prior to re-use
  \item
    secure a new sterilized probe to the TissueRuptor
  \end{itemize}
\end{itemize}

\hypertarget{protocol-for-tri-reagent-rna-isolation}{%
\section*{\texorpdfstring{\textbf{Protocol for TRI Reagent RNA
Isolation}}{Protocol for TRI Reagent RNA Isolation}}\label{protocol-for-tri-reagent-rna-isolation}}
\addcontentsline{toc}{section}{\textbf{Protocol for TRI Reagent RNA
Isolation}}

Adapted from official protocol. Download
\href{http://depts.washington.edu/kellylab/wp-content/uploads/2013/06/Tri-Reagent-Protocol.pdf}{here}

\hypertarget{materials}{%
\subsection*{\texorpdfstring{\textbf{Materials}}{Materials}}\label{materials}}
\addcontentsline{toc}{subsection}{\textbf{Materials}}

\begin{itemize}
\item
  Nuclease-free Water
\item
  Equipment for grinding/homogenizing solid tissue Appropriately sized
  RNase-free centrifuge tubes with secure closures, compatible with
  phenol/chloroform (polypropyl- ene, or polyallomer), and capable of
  withstanding centrifugal forces of 12,000 x g
\item
  Centrifuge capable of 12,000 x g
\item
  1-bromo-3-chloropropane (BCP; recommended, e.g., MRC, Cat \#BP 151)
\item
  100\% isopropanol, ACS grade or better
\item
  100\% ethanol, ACS grade or better
\end{itemize}

\hypertarget{procedure}{%
\subsection*{\texorpdfstring{\textbf{Procedure}}{Procedure}}\label{procedure}}
\addcontentsline{toc}{subsection}{\textbf{Procedure}}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Homogenize tissue samples in 10--20 volumes TRI Reagent solution.

  \begin{itemize}
  \item
    Handling fresh tissue: Immediately after dissection, inactivate
    RNases by any one of the following treatments:

    \begin{itemize}
    \item
      Homogenize in TRI Reagent solution immediately
    \item
      Freeze rapidly in liquid nitrogen (tissue pieces must be small
      enough to freeze in a few seconds)
    \item
      Submerge in a tissue storage buffer RNAlater®
    \end{itemize}
  \item
    Handling frozen tissue: Weigh frozen tissue, break into pieces
    smaller than \textasciitilde50 mg (keeping tissue completely
    frozen), and homogenize directly in TRI Reagent solution. Larger
    pieces of tissue, very hard or fibrous tissues, and tissues with a
    high RNase content must typically be ground to a powder in liquid
    nitrogen for maximum RNA yield.
  \item
    Homogenizing tissue: Homogenize samples in 10--20 volumes TRI
    Reagent solution (e.g., 1 mL TRI Reagent solution per 50--100mg
    tissue), using standard homogenization procedures.

    \begin{itemize}
    \tightlist
    \item
      The sample volume should not exceed 10\% of the volume of TRI
      Reagent solution used for homogenization.
    \end{itemize}
  \end{itemize}
\item
  Incubate the homogenate for 5 min at room temp.

  \begin{itemize}
  \item
    This 5 min room temperature incubation allows nucleoprotein
    complexes to completely dissociate.
  \item
    Homogenized samples can be stored at --70°C for at least one month.
  \end{itemize}
\item
  (Optional) Centrifuge at 12,000 xg for 10 min at 4°C and transfer the
  supernatant to a fresh tube.

  \begin{itemize}
  \tightlist
  \item
    This optional centrifugation is only required to remove insoluable
    material from homogenates that contain high amounts of protein, fat,
    polysaccharide, or extracellular material, such as muscle, fat
    tissue, and tuberous parts of plants. Centrifugation pellets,
    extracellular membranes, polysaccharides, and high molecular weight
    DNA, leaving the RNA in the supernatant. High molecular weight DNA
    can be recovered from the pellet by following the wash and
    solubilization steps of the DNA isolation protocol, available
    \href{https://drk-lo.github.io/lotterhoslabprotocols/molecprot_rnaextraction/www.ambion.com/techlib/append/supp.}{here}
  \end{itemize}
\item
  Add 100 μL BCP per 1 mL of TRI Reagent solution, mix well, and
  incubate at room temp for 5--15 min.

  \begin{itemize}
  \item
    Add 100 μL BCP per 1 mL of TRI Reagent solution used for
    homogenization. Alternatively, 200 μL of chloroform (without isoamyl
    alcohol) can be used in place of BCP.
  \item
    Cap the tubes tightly and shake vigorously for 15sec.
  \item
    Incubate the mixture at room temperature for 5--15 min.
  \end{itemize}
\item
  Centrifuge at 12,000 x g for 10--15 min at 4°C, then transfer the
  aqueous phase to a fresh tube.

  \begin{itemize}
  \item
    \textbf{Centrifuge at 12,000 x g for 10--15 min at 4°C.}
  \item
    Transfer the aqueous phase (colorless top layer) to a fresh tube.
  \item
    RNA remains exclusively in the aqueous phase whereas DNA and protein
    are in the interphase and organic phase.
  \item
    The interphase and lower, red, organic phase can be stored at 4°C
    for subsequent isolation of DNA and protein. A protocol is posted
    \href{https://drk-lo.github.io/lotterhoslabprotocols/molecprot_rnaextraction/www.ambion.com/techlib/append/supp}{here}
  \end{itemize}
\item
  Add 500 μL of isopropanol per 1 mL of TRI Reagent solution, vortex for
  5--10 sec, and incubate at room temp for 5--10 min.

  \begin{itemize}
  \item
    Add 500 μL of isopropanol per 1 mL of TRI Reagent solution used for
    sample homogenization.
  \item
    Vortex at moderate speed for 5--10 sec.
  \item
    Incubate the samples at room temp for 5--10 min.
  \item
    \textbf{Note} For oysters, we use a modified protocol:

    \begin{itemize}
    \item
      Transfer the aqueous phase from step 5 to a fresh tube. For each 1
      mL of TRI Reagent solution used for the homogenization, add 250 μL
      of isopropanol and 250 μL of a high salt precipitation solution
      (e.g.~0.8 M sodium citrate and 1.2 M NaCl).
    \item
      Mix well, store for 5--10 min at room temperature, and centrifuge
      at 12,000 x g for 8 min at 4--25°C.
    \item
      Wash the resulting RNA pellet as described in steps 8 and 9 of the
      procedure.
    \end{itemize}
  \end{itemize}
\item
  Centrifuge at 12,000 x g for 8 min at 4--25°C, and discard the
  supernatant.

  \begin{itemize}
  \item
    Centrifuge at 12,000 x g for 8 min at 4--25°C.
  \item
    Carefully remove the supernatant without disturbing the pellet.

    \begin{itemize}
    \tightlist
    \item
      Precipitated RNA forms a gel-like or white pellet on the side and
      bottom of the tube.
    \end{itemize}
  \end{itemize}
\item
  Add 1 mL of 75\% ethanol per 1 mL of TRI Reagent solution.

  \begin{itemize}
  \tightlist
  \item
    Add 1 mL of 75\% ethanol per 1 mL TRI Reagent solution used for
    sample homogenization to each sample to wash the RNA pellets.
  \end{itemize}
\item
  Centrifuge at 7,500 x g for 5 min, remove the ethanol, and briefly air
  dry the RNA pellet.

  \begin{itemize}
  \item
    Centrifuge at 7,500 x g for 5 min at 4--25°C.

    \begin{itemize}
    \tightlist
    \item
      If the precipitated RNA floats or does not form a compact pellet,
      repeat the centrifugation at 12,000 x g for 5 min to consolidate
      the pellet at the bottom of the tube.
    \end{itemize}
  \item
    Remove the ethanol wash without disturbing the pellet.
  \item
    Remove all residual ethanol by centrifuging again briefly and
    removing the ethanol that collects with a fine tip pipette. Complete
    removal of ethanol is necessary for the RNA to perform well in
    downstream applications.
  \item
    Air dry the RNA pellet for 3--5min.

    \begin{itemize}
    \tightlist
    \item
      \textbf{Do not completely dry the RNA pellet as this will greatly
      decrease its solubility. Do not dry RNA by vacuum centrifugation.}
      10.Dissolve RNA in the buffer of your choice.
    \end{itemize}
  \item
    Dissolve RNA in THE RNA Storage Solution (P/N AM7000, AM7001),
    Nuclease-free Water, or your choice of buffer‡ by passing the
    solution a few times through a pipette tip or by vigorous vortexing
  \item
    The resuspension volume is determined by the size of the RNA pellet.
    3--5 mm pellets typically require 300--500 μL. If necessary,
    increase the resuspension volume or incubate at 55--60°C to
    completely dissolve the pellet.
  \item
    Store at 4°C for immediate analysis. For long-term storage, store at
    --70°C or colder.
  \end{itemize}
\end{enumerate}

\hypertarget{protocol-with-qiagen-rnaeasy-kit}{%
\section*{\texorpdfstring{\textbf{Protocol with Qiagen RNAeasy
Kit}}{Protocol with Qiagen RNAeasy Kit}}\label{protocol-with-qiagen-rnaeasy-kit}}
\addcontentsline{toc}{section}{\textbf{Protocol with Qiagen RNAeasy
Kit}}

\hypertarget{reagents-and-supplies}{%
\subsection*{\texorpdfstring{\textbf{Reagents and
supplies}}{Reagents and supplies}}\label{reagents-and-supplies}}
\addcontentsline{toc}{subsection}{\textbf{Reagents and supplies}}

\begin{itemize}
\item
  RNAeasy kit purification of total RNA from small amounts of starting
  material
\item
  Molecular grade EtOH
\end{itemize}

\hypertarget{equipment-1}{%
\subsection*{\texorpdfstring{\textbf{Equipment}}{Equipment}}\label{equipment-1}}
\addcontentsline{toc}{subsection}{\textbf{Equipment}}

\begin{itemize}
\item
  Tissue homogenizer with replacable probes (Jon can you add specs of
  ours here)
\item
  Micro centrifuge \textgreater{} 8,000 rpm
\end{itemize}

\hypertarget{notes-before-starting}{%
\subsection*{\texorpdfstring{\textbf{Notes before
starting}}{Notes before starting}}\label{notes-before-starting}}
\addcontentsline{toc}{subsection}{\textbf{Notes before starting}}

\begin{itemize}
\item
  If purifying RNA from cell lines rich in RNases, or tissue, add either
  10 μl β-mercaptoethanol (β-ME), or 20 μl 2 M dithiothreitol (DTT)*, to
  1 ml Buffer RLT. Buffer RLT with β-ME or DTT can be stored at room
  temperature for up to 1 month.
\item
  Add 4 volumes of ethanol (96--100\%) to Buffer RPE for a working
  solution.
\item
  Remove RNAlater®-stabilized tissue from the reagent using forceps.

  \hypertarget{procedure-1}{%
  \subsection*{Procedure}\label{procedure-1}}
  \addcontentsline{toc}{subsection}{Procedure}

  \begin{enumerate}
  \def\labelenumi{\arabic{enumi}.}
  \item
    Do not use more than 30 mg tissue. Disrupt the tissue and homogenize
    the lysate in the appropriate volume of Buffer RLT (350 μl for less
    than 20 mg; 650 μl for less than 30 mg). Centrifuge the lysate for 3
    min at maximum speed. Carefully remove the supernatant by pipetting,
    and use it in step 2.
  \item
    Add 1 volume of 70\% ethanol to the lysate, and mix well by
    pipetting. Do not centrifuge. Proceed immediately to step 3.
  \item
    Transfer up to 700 μl of the sample, including any precipitate, to
    an RNeasy Mini spin column placed in a 2 ml collection tube
    (supplied). Close the lid, and centrifuge for 15 s at ≥8000 x g.
    Discard the flow-through.
  \item
    Add 700 μl Buffer RW1 to the RNeasy spin column. Close the lid, and
    centrifuge for 15 s at ≥8000 x g. Discard the flow-through.
  \item
    Add 500 μl Buffer RPE to the RNeasy spin column. Close the lid, and
    centrifuge for 15 s at ≥8000 x g. Discard the flow-through.
  \item
    Add 500 μl Buffer RPE to the RNeasy spin column. Close the lid, and
    centrifuge for 2 min at ≥8000 x g.
  \item
    \textbf{Optional:} Place the RNeasy spin column in a new 2 ml
    collection tube (supplied). Centrifuge at full speed for 1 min to
    dry the membrane.
  \item
    Place the RNeasy spin column in a new 1.5 ml collection tube
    (supplied). Add 30--50 μl RNase-free water directly to the spin
    column membrane. Close the lid, and centrifuge for 1 min at ≥8000 x
    g to elute the RNA.
  \item
    If the expected RNA yield is \textgreater30 μg, repeat step 7 using
    another 30--50 μl of RNase-free water, or using the eluate from step
    7 (if high RNA concentration is required). Reuse the collection tube
    from step 7.
  \end{enumerate}
\end{itemize}

\hypertarget{protocol-for-zymo-quick-dna-rna-miniprep-and-tissueruptor-homogenization}{%
\section*{\texorpdfstring{\textbf{Protocol for ZYMO Quick DNA RNA
Miniprep and TissueRuptor
homogenization}}{Protocol for ZYMO Quick DNA RNA Miniprep and TissueRuptor homogenization}}\label{protocol-for-zymo-quick-dna-rna-miniprep-and-tissueruptor-homogenization}}
\addcontentsline{toc}{section}{\textbf{Protocol for ZYMO Quick DNA RNA
Miniprep and TissueRuptor homogenization}}

\href{https://files.zymoresearch.com/protocols/_d7001_quick-dna-rna_miniprep_kit.pdf}{Protocol}

\hypertarget{materials-1}{%
\subsection*{\texorpdfstring{\textbf{Materials}}{Materials}}\label{materials-1}}
\addcontentsline{toc}{subsection}{\textbf{Materials}}

\begin{itemize}
\item
  ZYMO Quick-DNA/RNA Miniprep
\item
  TissueRuptor
\item
  TissueRuptor Disposable Probes
\item
  Ethanol (99\%)
\end{itemize}

\hypertarget{equipment-2}{%
\subsection*{\texorpdfstring{\textbf{Equipment}}{Equipment}}\label{equipment-2}}
\addcontentsline{toc}{subsection}{\textbf{Equipment}}

\begin{itemize}
\tightlist
\item
  Centrifuge for 1.5 mL spin columns
\end{itemize}

\hypertarget{prep}{%
\section*{\texorpdfstring{\textbf{Prep}}{Prep}}\label{prep}}
\addcontentsline{toc}{section}{\textbf{Prep}}

\begin{itemize}
\item
  Sterilize TissueRuptor probes in the Autoclave:

  \begin{itemize}
  \item
    place probes in a large beaker and cover with aluminum and autoclave
    tape
  \item
    autoclave at 120°C for 35 min (20 min to reach temperature + 15 min
    at 120°C).
  \item
    \textbf{Note} \emph{TissueRuptor Probes may be cleaned /sterilized
    up to 5 times (15 min at 120°C) or until visible signs of wear are
    seen}
  \end{itemize}
\item
  Wipe of bench with 10\% Bleach and RNAse Away
\item
  Leave UV hood on for 1 hour to sterilize the working area
\end{itemize}

\hypertarget{procedure-2}{%
\subsection*{\texorpdfstring{\textbf{Procedure}}{Procedure}}\label{procedure-2}}
\addcontentsline{toc}{subsection}{\textbf{Procedure}}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Homogenize- TissueRuptor

  \begin{itemize}
  \item
    Get each tissue out of the freezer and keep on ice

    \begin{itemize}
    \tightlist
    \item
      do one tissue at a time
    \end{itemize}
  \item
    prepare 2 mL vials (with somewhat flat bottoms) each with 600 µl
    \textbf{DNA/RNA Lysis Buffer}
  \item
    cut \textasciitilde25 mg of tissue and put in prepared lysis vials
  \item
    keep vials on ice
  \item
    run TissueRuptor at level 3 for 7 seconds (or until tissue is
    completely lysed)
  \item
    return to ice
  \item
    clean probe with 3 washes of DI water

    \begin{itemize}
    \item
      you should have 3 separate falcon tubes with DI water
    \item
      run TissueRuptor through each DI wash for 5 seconds
    \item
      remove probe and set aside to be autoclaved prior to re-use
    \item
      secure a new sterilized probe to the TissueRuptor
    \end{itemize}
  \item
    repeat steps with the next tissue
  \item
    centrifuge samples at 14`,000 rfc x 30 seconds (may want to increase
    this)
  \end{itemize}
\item
  Transfer the sample lysed in DNA/RNA Lysis Buffer into a
  \textbf{SpinAway™ Filter (yellow)} in a Collection Tube

  \begin{itemize}
  \item
    centrifuge 16,000 rfc x 30 sec
  \item
    Save the flow-through for RNA purification and the filter for DNA
    purification!
  \end{itemize}
\item
  DNA/RNA Purification

  \begin{itemize}
  \item
    \textbf{\emph{3a: DNA Purification}} (DNA is in the filter)

    \begin{itemize}
    \tightlist
    \item
      Transfer the Spin-Away Filter (yellow) into a new Collection Tube
    \end{itemize}
  \item
    \textbf{\emph{3b: RNA Purification}}(RNA is in the flow-through)

    \begin{itemize}
    \item
      Add 600 μL volume ethanol (95-100\%) to the flow-through (1:1 vol)
      and mix well.
    \item
      Transfer the sample into a Zymo-Spin™ IICR Column in a Collection
      Tube

      \begin{itemize}
      \tightlist
      \item
        max 600 μL at a time
      \end{itemize}
    \item
      centrifuge 16,000 rfc x 30 sec
    \item
      Discard the flow-through
    \end{itemize}
  \end{itemize}
\item
  Add 400 µl \textbf{DNA/RNA Prep Buffer} to the column

  \begin{itemize}
  \item
    centrifuge 16,000 rfc x 30 sec
  \item
    discard the flow-through
  \end{itemize}
\item
  Add 700 µl \textbf{DNA/RNA Wash Buffer} to the column

  \begin{itemize}
  \item
    centrifuge 16,000 rfc x 30 sec
  \item
    Discard the flow-through.
  \end{itemize}
\item
  Add 400 µl \textbf{DNA/RNA Wash Buffer}

  \begin{itemize}
  \item
    centrifuge 16,000 rfc x 2 min to ensure complete removal of the wash
    buffer
  \item
    Then carefully, transfer the column into a nuclease-free tube (not
    provided).
  \end{itemize}
\item
  Elute DNA/RNA

  \begin{itemize}
  \item
    \textbf{\emph{7a: DNA}}

    \begin{itemize}
    \item
      add 50 µl DNase/RNase-Free Water

      \begin{itemize}
      \tightlist
      \item
        or 100 µl if you need it less concentrated
      \end{itemize}
    \item
      let stand 2-5 min
    \item
      centrifuge 16,000 rfc x 30 sec
    \end{itemize}
  \item
    \textbf{\emph{7b: RNA}}

    \begin{itemize}
    \item
      add 30 µl DNase/RNase-Free Water

      \begin{itemize}
      \tightlist
      \item
        or 50 µl if you need it less concentrated
      \end{itemize}
    \item
      centrifuge 16,000 rfc x 30 sec
    \end{itemize}
  \end{itemize}
\end{enumerate}

\hypertarget{gel-electrophoresis}{%
\chapter{Gel Electrophoresis}\label{gel-electrophoresis}}

The protocol for running a gel

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

We are Ethidium-Bromide Free!

\hypertarget{materials-and-equipment-2}{%
\section*{\texorpdfstring{\textbf{Materials and
Equipment}}{Materials and Equipment}}\label{materials-and-equipment-2}}
\addcontentsline{toc}{section}{\textbf{Materials and Equipment}}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Agarose
\item
  1X TBE
\item
  GelRed® 10,000X stock reagent (left-top drawer in the room with gel
  docks).
\item
  Erlenmeyer flask (should be in the room with gel docks).

  \hypertarget{protocol-3}{%
  \section*{Protocol}\label{protocol-3}}
  \addcontentsline{toc}{section}{Protocol}
\item
  Tare a weigh boat on the scale.
\item
  Add 1.4 g of agarose to the weigh boat.
\item
  Pour 70 mL of 1x TBE buffer into a graduated cylinder.
\item
  Pour 35-40 mL of the TBE buffer into an Erlenmeyer flask, then add the
  agarose and swirl.
\item
  Microwave solution for \textasciitilde45 seconds, or until it begins
  to boil.
\item
  Add more TBE buffer, but leave 10-20 mL in the graduated cylinder.
\item
  Microwave solution again until it begins to boil. Repeat until agarose
  is completely dissolved.
\item
  Add remaining TBE buffer (to cool solution).
\item
  Dilute the GelRed® 10,000X stock reagent into the molten agarose gel
  solution at 1:10,000 and mix thoroughly.

  \begin{itemize}
  \tightlist
  \item
    (e.g., 5 µL of the GelRed 10,000X stock reagent added to 50 mL of
    the gel solution)
  \end{itemize}
\end{enumerate}

\hypertarget{running-the-gel}{%
\section*{\texorpdfstring{\textbf{Running the
Gel}}{Running the Gel}}\label{running-the-gel}}
\addcontentsline{toc}{section}{\textbf{Running the Gel}}

\begin{itemize}
\item
  This source detailes a full protocol for running a gel:
  https://www.addgene.org/protocols/gel-electrophoresis/
\item
  Source from Remy for Gel protocol (note that this is for an Ethidium
  Bromide system, and we are Ethidium-Bromide free at the MSC):
  \href{https://github.com/DrK-Lo/lotterhoslabprotocols/files/8584679/Gel.Electrophoresis.protocol_lab.interns.pdf}{Gel
  Electrophoresis protocol\_lab interns.pdf}\\
\end{itemize}

\hypertarget{illumina-library-prep}{%
\chapter{Illumina Library Prep}\label{illumina-library-prep}}

The protocol for doing a library prep using Illumina kit

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\emph{Note: This protocol is unfinished}

\hypertarget{background}{%
\section*{\texorpdfstring{\textbf{Background}}{Background}}\label{background}}
\addcontentsline{toc}{section}{\textbf{Background}}

Before you start this, you should
\href{https://bmford11.github.io/lotterhoslabprotocols/barcodesnindexes/}{determine
the barcodes and index scheme for your library.}

Then, you should draw an outline of what is happening at each step in
this protocol:

\href{https://docs.google.com/document/d/1xr99sMucP-x2rK6K61vtGHKNahl9UYFwh_oEb0AtHX8/edit}{Example
Genomic Library Prep with inline barcodes and dual indexes}

This protocol is based on the KAPA Hyper Prep Kit
\href{https://rochesequencingstore.com/wp-content/uploads/2017/10/KAPA-HyperPrep-Kit_KR0961-\%E2\%80\%93-v7.19.pdf}{protocol}
with added notes from both Jon Puritz and Sara Schaal. We performed the
protocol using half reactions of everything compared to the original
protocol.

\hypertarget{kits-used}{%
\section*{\texorpdfstring{\textbf{Kits
used}}{Kits used}}\label{kits-used}}
\addcontentsline{toc}{section}{\textbf{Kits used}}

\begin{itemize}
\tightlist
\item
  KAPA Hyper Prep Kit (Kit Code: KK8505)
\end{itemize}

\hypertarget{materials-needed}{%
\section*{\texorpdfstring{\textbf{Materials
needed}}{Materials needed}}\label{materials-needed}}
\addcontentsline{toc}{section}{\textbf{Materials needed}}

\begin{itemize}
\item
  500 ng of DNA in 25 uL of RNAse free water that is sheared to the size
  appropriate for your library
\item
  Primers at 15 uM (aliquots of i5 and i7 custom primers if using)
\item
  Adapters at 15 uM (aliquots of annealed custom adapters if using)
\item
  KAPA beads (or other beads)
\end{itemize}

\hypertarget{equipment-needed}{%
\section*{\texorpdfstring{\textbf{Equipment
needed}}{Equipment needed}}\label{equipment-needed}}
\addcontentsline{toc}{section}{\textbf{Equipment needed}}

\begin{itemize}
\item
  Thermo Mixer set at 20C and 400 rpm (this can be done in the
  thermocycler as well if you don't have a thermo Mixer)
\item
  Thermocycler - preprogrammed to have all thermocycler steps already
  set up: End Repair and A-Tailing and PCR.
\item
  Magnetic Bead Rack
\end{itemize}

\hypertarget{tips-exposables}{%
\section*{\texorpdfstring{\textbf{Tips \&
Exposables}}{Tips \& Exposables}}\label{tips-exposables}}
\addcontentsline{toc}{section}{\textbf{Tips \& Exposables}}

\begin{itemize}
\item
  filtered 10 uL
\item
  filtered 20-200 uL
\item
  Strip tubes or individual tubes depending on how many libraries you
  are prepping
\end{itemize}

\hypertarget{protocol-4}{%
\section*{\texorpdfstring{\textbf{Protocol}}{Protocol}}\label{protocol-4}}
\addcontentsline{toc}{section}{\textbf{Protocol}}

\hypertarget{end-repair-a-tailing}{%
\subsection*{\texorpdfstring{\textbf{End Repair \&
A-tailing}}{End Repair \& A-tailing}}\label{end-repair-a-tailing}}
\addcontentsline{toc}{subsection}{\textbf{End Repair \& A-tailing}}

1.1 Assemble each end repair and A-tailing reaction in a tube or well of
a PCR plate as follows:

\begin{longtable}[]{@{}ll@{}}
\toprule()
\textbf{Component} & \textbf{Volume} \\
\midrule()
\endhead
Fragmented Double Stranded DNA & 25 uL \\
End Repair \& A-tailing Buffer & 3.5 uL \\
End Repair \& A-tailing Enzyme & 1.5 uL \\
Total Volume & 30 uL \\
\bottomrule()
\end{longtable}

Depending on how many libraries you are prepping, you can make a master
mix of the buffer and enzyme to add to each sample well. If you do this,
be sure to add about 10\% volume of each component due to pipetting
error.

1.2 Vortex gently and spin down briefly. Return the plate/tube(s) to
ice. Proceed immediately to the next step. 1.3 Incubate libraries in
thermocycler set to the following protocol:

\begin{longtable}[]{@{}lll@{}}
\toprule()
\textbf{Step} & \textbf{Temp} & \textbf{Time} \\
\midrule()
\endhead
End Repair \& A-Tailing & 20 C & 30 min \\
'\,' & 65 C & 30 min \\
HOLD & 4 C & * \\
\bottomrule()
\end{longtable}

1.4 Proceed immediately to Adapter Ligation

\hypertarget{adapter-ligation}{%
\subsection*{\texorpdfstring{\textbf{Adapter
Ligation}}{Adapter Ligation}}\label{adapter-ligation}}
\addcontentsline{toc}{subsection}{\textbf{Adapter Ligation}}

2.1 In the same plate/tube(s) in which end repair and A-tailing was
performed, assemble each adapter ligation reaction as follows:

\begin{longtable}[]{@{}ll@{}}
\toprule()
\textbf{Component} & \textbf{Volume} \\
\midrule()
\endhead
End Repair \& A-Tailing Reaction Product & 30 uL \\
Adapter stock (concentration as required) & 2.5 uL \\
PCR-grade water & 2.5 uL \\
Ligation Buffer & 15 uL \\
DNA Ligase & 5 uL \\
Total Volume & 55 uL \\
\bottomrule()
\end{longtable}

Again, at this step you can make a master mix of the buffer, Ligase
enzyme and water. If you do this, be sure to add about 10\% volume of
each component due to pipetting error.

2.2 Mix thoroughly by pipetting up and down 15 times and centrifuge
briefly. 2.3 Incubate at 20C for 60 minutes on the ThermoMixer with the
rotations set at 400 rpm. 2.4 After incubation, proceed immediately to
the next step

\hypertarget{post-ligation-cleanup}{%
\subsection*{\texorpdfstring{\textbf{Post-Ligation
Cleanup}}{Post-Ligation Cleanup}}\label{post-ligation-cleanup}}
\addcontentsline{toc}{subsection}{\textbf{Post-Ligation Cleanup}}

3.1 In the same plate/tube(s), perform a 0.8X bead based cleanup by
combining the following:

\begin{longtable}[]{@{}ll@{}}
\toprule()
\textbf{Component} & \textbf{Volume} \\
\midrule()
\endhead
Adapter ligation reaction product & 55 uL \\
KAPA Pure Beads & 44 uL \\
Total Volume & 99 uL \\
\bottomrule()
\end{longtable}

3.2 Mix thoroughly by pipetting up and down 15 times. 3.3 Incubate the
plate/tube(s) at room temperature for 10 min to bind DNA to the beads.

\hypertarget{pcr}{%
\subsection*{\texorpdfstring{\textbf{PCR}}{PCR}}\label{pcr}}
\addcontentsline{toc}{subsection}{\textbf{PCR}}

\hypertarget{post-pcr-cleanup}{%
\subsection*{\texorpdfstring{\textbf{Post-PCR
Cleanup}}{Post-PCR Cleanup}}\label{post-pcr-cleanup}}
\addcontentsline{toc}{subsection}{\textbf{Post-PCR Cleanup}}

\hypertarget{spot-check-libraries-on-a-tapestation-before-qpcr}{%
\subsection*{\texorpdfstring{\textbf{Spot check libraries on a
tapestation before
qPCR}}{Spot check libraries on a tapestation before qPCR}}\label{spot-check-libraries-on-a-tapestation-before-qpcr}}
\addcontentsline{toc}{subsection}{\textbf{Spot check libraries on a
tapestation before qPCR}}

\hypertarget{library-quantification}{%
\chapter{Library Quantification}\label{library-quantification}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{qpcr-protocol---library-quantification}{%
\section*{\texorpdfstring{\textbf{qPCR Protocol - Library
Quantification}}{qPCR Protocol - Library Quantification}}\label{qpcr-protocol---library-quantification}}
\addcontentsline{toc}{section}{\textbf{qPCR Protocol - Library
Quantification}}

The qPCR machine is located in the Bowen Lab. Please coordinate with
their lab before using the machine. This protocol is used to quantify
the amount of each of your libraries that will be sequenced by the
sequencer. Quantifying by other means will tell you how much dsDNA you
have in your library, but the qPCR will tell you how much library was
successfully amplified and how much will be read by the sequencer. You
should run each sample in triplicate on the qPCR to minimize error in
these measurements especially when pooling samples for sequencing.

\hypertarget{what-you-need-before-you-start}{%
\section*{\texorpdfstring{\textbf{What you need before you
start}}{What you need before you start}}\label{what-you-need-before-you-start}}
\addcontentsline{toc}{section}{\textbf{What you need before you start}}

\begin{itemize}
\item
  Kit: KAPA Library Quantification Kit (Universal) (Cat \#: KK4923)
\item
  Specific strip tubes \& caps: Agilent AriaMx 96 Tube Strips (Cat \#: )
  \& Agilent Optical Cap, 8x Strip (Cat \#: )
\item
  Make a stock of 10 mM Tris-HCl + 0.05\% Tween-20 (optional but
  recommended for the dilution of library DNA, as it leads to improved
  accuracy by reducing the adherence of DNA to plastic surfaces)
\item
  Copy of KAPA Protocol
  \href{https://rochesequencingstore.com/wp-content/uploads/2017/10/KAPA-Lib-Quant-ILMN_9.17-IfU_1.pdf}{Found
  here}
\item
  Ensure qPCR reagents are thawed and properly mixed. If the kit is new,
  the 2X KAPA SYBR-FAST qPCR Master Mix (5mL) should be combined with 1
  mL 10X Illumina Primer Premix and then aliquoted out. These should be
  labeled with the date it was mixed because once mixed the reagent is
  only good for (\emph{60 days double check that timeframe}).
\item
  1.5 mL microcentrifuge tube per library
\item
  Previously quantified libraries and their concentrations
\item
  Turn on qPCR machine (needs some time to warm up so turn on before you
  start setting up your reactions)
\end{itemize}

\hypertarget{step-1-make-library-dilutions}{%
\section*{\texorpdfstring{\textbf{Step 1: Make library
dilutions}}{Step 1: Make library dilutions}}\label{step-1-make-library-dilutions}}
\addcontentsline{toc}{section}{\textbf{Step 1: Make library dilutions}}

\begin{itemize}
\tightlist
\item
  Libraries need to be diluted to be read on the QPCR machine. All
  dilutions should be done in 10 mM Tris-HCl + 0.05\% Tween-20. First
  dilute sample to a 1:1000 concentration in a 1.5mL microcentrifuge
  tube (Table 1).
\end{itemize}

\hypertarget{table-1}{%
\subsubsection*{\texorpdfstring{\textbf{Table
1:}}{Table 1:}}\label{table-1}}
\addcontentsline{toc}{subsubsection}{\textbf{Table 1:}}

\textbar{} Reagent \textbar{} Amount \textbar{}
\textbar---------------------------------\textbar---------\textbar{}
\textbar{} 10 mM Tris-HCl + 0.05\% Tween-20 \textbar{} 999 uL \textbar{}
\textbar{} Library dsDNA \textbar{} 1 uL \textbar{} \textbar{} Total
\textbar{} 1000 uL \textbar{}

\begin{itemize}
\tightlist
\item
  Next dilute your sample libraries to ensure they fall within the range
  of the standards provided by the KAPA kit (Table 2). Typically for a
  library in the range of 1-200 ng/uL you need a 1:18000 dilution to be
  read on the QPCR machine. However, you can use the calculation below
  to determine what you need to dilute your undiluted library down to:
\end{itemize}

\hypertarget{calculation-1---determining-library-concentration-in-pm}{%
\subsubsection*{\texorpdfstring{\textbf{Calculation 1 - determining
library concentration in
pM:}}{Calculation 1 - determining library concentration in pM:}}\label{calculation-1---determining-library-concentration-in-pm}}
\addcontentsline{toc}{subsubsection}{\textbf{Calculation 1 - determining
library concentration in pM:}}

\begin{itemize}
\item
  Calc 1: (Undiluted Library Conc. (ng/uL)/ (617.9 g/mol * fragment size
  of library)) * 10\^{}6 = undiluted library in nM
\item
  Calc 2: undiluted library in nM * 1000 = undiluted library in pM
\item
  Calc 3: dilute final library to be in the range of the standards in
  Table 2
\end{itemize}

\hypertarget{table-2}{%
\subsubsection*{\texorpdfstring{\textbf{Table
2:}}{Table 2:}}\label{table-2}}
\addcontentsline{toc}{subsubsection}{\textbf{Table 2:}}

\textbar{} Standard Name \textbar{} dsDNA concentration (pM) \textbar{}
\textbar---------------\textbar--------------------------\textbar{}
\textbar{} Standard 1 \textbar{} 20 \textbar{} \textbar{} Standard 2
\textbar{} 2 \textbar{} \textbar{} Standard 3 \textbar{} 0.2 \textbar{}
\textbar{} Standard 4 \textbar{} 0.02 \textbar{} \textbar{} Standard 5
\textbar{} 0.002 \textbar{} \textbar{} Standard 6 \textbar{} 0.0002
\textbar{}

\hypertarget{step-2-qpcr-setup}{%
\section*{\texorpdfstring{\textbf{Step 2: qPCR
setup}}{Step 2: qPCR setup}}\label{step-2-qpcr-setup}}
\addcontentsline{toc}{section}{\textbf{Step 2: qPCR setup}}

\begin{itemize}
\item
  qPCR reactions for the six standards and each library dilution should
  be set up in triplicate. Load each well of the qPCR strip tubes as
  indicated below in Table 3, for a total reaction volume of 20 uL.
  Below is an example set up when running 8 libraries in triplicate
  (Table 4).
\item
  Note: you can not label the individual tubes. Make sure you pay
  attention to your strip tube orientation and order. Tip: Always put
  the ``A'' side of strip tude on top and label each strip tube tap with
  the numbers 1 - \# of strip tubes. Have a key in your notebook to what
  each of those strips contain. When you are doing a lot of these it is
  easy to mix up what was in what tube.
\item
  Note: mix contents of each reaction \emph{very} well by pipetting up
  and down \textasciitilde20 times.
\end{itemize}

\hypertarget{table-3-qpcr-reaction-setup}{%
\subsubsection*{\texorpdfstring{\textbf{Table 3: qPCR reaction
setup}}{Table 3: qPCR reaction setup}}\label{table-3-qpcr-reaction-setup}}
\addcontentsline{toc}{subsubsection}{\textbf{Table 3: qPCR reaction
setup}}

\textbar{} Reagent \textbar{} Amount \textbar{}
\textbar---------------------------------------------------\textbar--------\textbar{}
\textbar{} KAPA SYBR FAST qPCR Master Mix containing Primers \textbar{}
12 uL \textbar{} \textbar{} PCR-grade water \textbar{} 4 uL \textbar{}
\textbar{} Diluted dsDNA library dsDNA Standard (1-6) \textbar{} 4 uL
\textbar{} \textbar{} Total \textbar{} 20 uL \textbar{}

\hypertarget{table-4-example-set-up-for-6-samples-in-triplicate}{%
\subsubsection*{\texorpdfstring{\textbf{Table 4: Example set up for 6
samples in
triplicate}}{Table 4: Example set up for 6 samples in triplicate}}\label{table-4-example-set-up-for-6-samples-in-triplicate}}
\addcontentsline{toc}{subsubsection}{\textbf{Table 4: Example set up for
6 samples in triplicate}}

\textbar{} \textbar{} 1 \textbar{} 2 \textbar{} 3 \textbar{} 4
\textbar{} 5 \textbar{} 6 \textbar{}
\textbar---\textbar------\textbar------\textbar------\textbar-------\textbar-------\textbar-------\textbar{}
\textbar{} A \textbar{} Std1 \textbar{} Std1 \textbar{} Std1 \textbar{}
Samp1 \textbar{} Samp1 \textbar{} Samp1 \textbar{} \textbar{} B
\textbar{} Std2 \textbar{} Std2 \textbar{} Std2 \textbar{} Samp2
\textbar{} Samp2 \textbar{} Samp2 \textbar{} \textbar{} C \textbar{}
Std3 \textbar{} Std3 \textbar{} Std3 \textbar{} Samp3 \textbar{} Samp3
\textbar{} Samp3 \textbar{} \textbar{} D \textbar{} Std4 \textbar{} Std4
\textbar{} Std4 \textbar{} Samp4 \textbar{} Samp4 \textbar{} Samp4
\textbar{} \textbar{} E \textbar{} Std5 \textbar{} Std5 \textbar{} Std5
\textbar{} Samp5 \textbar{} Samp5 \textbar{} Samp5 \textbar{} \textbar{}
F \textbar{} Std6 \textbar{} Std6 \textbar{} Std6 \textbar{} Samp6
\textbar{} Samp6 \textbar{} Samp6 \textbar{} \textbar{} G \textbar{} Blk
\textbar{} Blk \textbar{} Blk \textbar{} Samp7 \textbar{} Samp7
\textbar{} Samp7 \textbar{} \textbar{} H \textbar{} \textbar{}
\textbar{} \textbar{} Samp8 \textbar{} Samp8 \textbar{} Samp8 \textbar{}

\begin{itemize}
\item
  Ensure that the qPCR plate is sealed.
\item
  Collect all components in the bottom of the well by brief
  centrifucation.
\end{itemize}

\hypertarget{step-3-using-the-qpcr-machine}{%
\section*{\texorpdfstring{\textbf{Step 3: Using the qPCR
machine}}{Step 3: Using the qPCR machine}}\label{step-3-using-the-qpcr-machine}}
\addcontentsline{toc}{section}{\textbf{Step 3: Using the qPCR machine}}

\begin{itemize}
\item
  Take final reaction strip tubes to the Bowen lab for use on the qPCR
  machine.
\item
  Turn on the power button in the bottom left hand corner. This will
  take a few minutes because the machine runs self diagnostics when
  booting up.
\item
  On the home screen, click ``Quantitative PCR Fluoresence Probe''
  \includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/qPCRmachine_step1.jpg}
\item
  The first screen that will pop up is your plate set up. Here click the
  top left corner of the plate to select all wells and in the bottom
  right corner change the dye from ``FAM'' to ``SYBR''. This updates all
  wells to have the correct dye information for your run.
  \includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/qPCRmachine_step1.jpg}
\item
  Next for each well make sure to label as either a ``Standard'' well,
  ``NTC'' (no template control) well, or ``Unknown'' well. For example,
  to do this for standards, highlight all cells that are locations of
  standards and move to the top right corner dropdown menu called ``Well
  Types'' and set it to standard. You can click and drag to highlight
  multiple locations in a row or just click on all the cells
  individually. You can always deselect by clicking that cell again or
  deselect everything using the top left corner square.
  \includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/qPCRmachine_step3.jpg}
\item
  Next click on the left tab that says ``Thermal Profile'' and set the
  specifications for your run (Table 5). You do this by clicking any
  value (e.g., time or temperature) and manually editing the values as
  shown in the image below.
  \includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/qPCRmachine_step4.jpg}
\end{itemize}

\hypertarget{table-5}{%
\subsubsection*{\texorpdfstring{\textbf{Table
5:}}{Table 5:}}\label{table-5}}
\addcontentsline{toc}{subsubsection}{\textbf{Table 5:}}

\textbar{} Step \textbar{} Temperature \textbar{} Time \textbar{} Cycles
\textbar{}
\textbar--------------------------------------\textbar-------------\textbar--------\textbar--------\textbar{}
\textbar{} Initial activation/denaturation \textbar{} 95C \textbar{} 5
min \textbar{} 1 \textbar{} \textbar{} Denaturation \textbar{} 95C
\textbar{} 30 sec \textbar{} 35 \textbar{} \textbar{}
Annealing/extension/data acquisition \textbar{} 60C \textbar{} 45 sec
\textbar{} * \textbar{}

The denature and annealing steps are both run together for 35 cycles.
Markdown tables are frustrating and you can't merge cells of a table.
See the image at the bottom of this section for clarification. Once the
specifications are set properly, place the tubes in the qPCR machine and
click ``Run Experiment''.
\includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/qPCRmachine_step6.jpg}

\includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/qPCRmachine_step5.jpg}

\hypertarget{step-4-analyzing-output-data}{%
\section*{\texorpdfstring{\textbf{Step 4: Analyzing output
data}}{Step 4: Analyzing output data}}\label{step-4-analyzing-output-data}}
\addcontentsline{toc}{section}{\textbf{Step 4: Analyzing output data}}

\begin{itemize}
\item
  To analyze the output you have to use the AriaMx program located on
  the Bowen laptop that is typically plugged into the tapestation in the
  Shared Molecular Space.
\item
  First use the red flash drive (pictured below) to get your data off of
  the qPCR machine. Only this flash drive works (not sure why?).
  \includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/qPCRmachine_step7.jpg}
\item
  Then shut down the qPCR machine and take the flash drive to the
  tapestation computer. Here you should open the Agilent Aria program to
  see your data.
  \includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/qPCRmachine_step8.jpg}
\item
  Once the program opens a similar table to what you saw on the qPCR
  machine will open. Standard cells will be a redish pink color, sample
  cells blue and no template control green. For the standard cells you
  need to manually enter each value into the program. This is done by
  highlighting one cell and moving to the properties tab on the right
  side panel. Then in the standard quantities section place the value of
  that standard in the ``Amount'' field and change the Units field to
  picograms. These values will correspond to the order you put your
  ladder in and are the values in Table 2. Double check all inputs once
  you complete all your standard wells then you are ready to view your
  results.
  \includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/qPCRmachine_step9.jpg}
\item
  Highlight all cells then move to the ``Graphical Displays'' to inspect
  your amplification plots and where your samples fall on your standard
  curve. You can do a visual inspection here to ensure your samples fall
  within your standards, but you need to export your Cq values to
  determine actually concentrations of your samples.
  \includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/qPCRmachine_step10.jpg}
\item
  To export your Cq values make sure all samples are highlighted in the
  right hand column. Right click and select ``export to excel''. Save
  your excel file to a hard drive and take it to your personal computer.
  \includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/qPCRmachine_step11.jpg}\\
\end{itemize}

\part{Molecular Equipment}

\hypertarget{numsc---shared-lab-space}{%
\chapter{NUMSC - Shared Lab Space}\label{numsc---shared-lab-space}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

Introductory information on the shared lab space @ NU Marine Science
Center

\url{https://numsc-corefacility.github.io/sharedLabSpace/}

\hypertarget{take3-dna-quantification}{%
\chapter{Take3 DNA Quantification}\label{take3-dna-quantification}}

The protocol for quantifying DNA/RNA samples using the Take3

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{what-is-the-take3}{%
\section*{\texorpdfstring{\textbf{What is the
Take3?}}{What is the Take3?}}\label{what-is-the-take3}}
\addcontentsline{toc}{section}{\textbf{What is the Take3?}}

The Take3 is like a nanodrop for DNA quantification using
spectrophotometry on the plate reader. For more accurate quantification
similar to Qubit, considering using the PicoGreen assay.

\hypertarget{take3-run}{%
\section*{\texorpdfstring{\textbf{Take3
Run}}{Take3 Run}}\label{take3-run}}
\addcontentsline{toc}{section}{\textbf{Take3 Run}}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Select File -\textgreater{} New Task
\item
  When prompted by Task Manager, select Read Now in left pane.
\item
  Select Nucleic Acid Quantification (or Protein A280 if measuring
  proteins) under Take3 Application at the bottom of prompt
\item
  Information in top three drop down menus should be autopopulated with
  plate serial number, well type, and sample type.
\item
  Change sample type if desired
\item
  Chose Scan 240-300 nm with 2nm step if desired
\item
  Click Read. Reader reads the blanks and takes an average, then tray
  opens.
\item
  Click Approve if reading blanks was successful (low \%CV value).
\item
  You can now specify sample wells. Samples must be in rectangular
  blocks for reader.
\item
  Click Read again to read samples. Reader reads samples and once
  finished ejects tray.
\item
  Export Data.
\item
  Blot liquid off with Kimwipe. Store with proper packaging.
\item
  Replace carrier tray with priming plate.\\
\end{enumerate}

\hypertarget{picogreen-assay}{%
\chapter{PicoGreen Assay}\label{picogreen-assay}}

The protocol for checking quantification of DNA using the PicoGreen
Assay on the Plate Reader

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{background-1}{%
\section*{\texorpdfstring{\textbf{Background}}{Background}}\label{background-1}}
\addcontentsline{toc}{section}{\textbf{Background}}

PicoGreen dsDNA Quantitation Reagent is an ultra-sensitive fluorescent
nucleic acid stain for quantitating double-stranded DNA (dsDNA) in
molecular biological procedures such as cDNA synthesis for library
production and DNA fragment purification for subcloning, as well as
diagnostic applications, such as quantitating DNA amplification products
and primer extension assays. The most commonly used technique for
measuring nucleic acid concentration is the determination of absorbance
at 260 nm (A260).

\hypertarget{equipment-reagents}{%
\section*{\texorpdfstring{\textbf{Equipment \&
Reagents}}{Equipment \& Reagents}}\label{equipment-reagents}}
\addcontentsline{toc}{section}{\textbf{Equipment \& Reagents}}

\begin{itemize}
\item
  flat bottom black Costar 96-well plate (Fisher Cat \#: 07-200-627)
  (FIND OUT IF AUTOCLAVE SAFE)
\item
  two - 15 mL Falcon tubes
\item
  two - 0.2 mL snap cap tubes for ladder dilutions
\item
  samples diluted to appropriate concentration to be read on the plate
  reader
\item
  20X TE Buffer (comes with PicoGreen Kit)
\item
  PicoGreen Dye (comes with PicoGreen Kit)
\item
  DNA Ladder 100 ug/mL (comes with PicoGreen Kit)
\item
  Synergy HTX plate reader in shared molecular space
\end{itemize}

\hypertarget{important-notes}{%
\section*{\texorpdfstring{\textbf{Important
Notes}}{Important Notes}}\label{important-notes}}
\addcontentsline{toc}{section}{\textbf{Important Notes}}

\begin{itemize}
\item
  Plate reader can read samples between 25 pg/mL - 1,000 ng/mL (or
  0.000025 - 1 ng/uL). To be conservative we make the highest
  concentration for our ladder 500 ng/mL.
\item
  Samples should be diluted to be read within this range. For example,
  typical quantifications for Atlantic cod DNA extractions using a spin
  column plate is between 10 - 200 ng/uL. Therefore, you should do a
  1:10 dilution because the reaction does a 1:100 dilution in the
  reaction itself. This brings the final DNA concentration to 10 - 200
  ng/mL which is well within the range of the assay.
\item
  PicoGreen dye is a solid when stored at 4C. Allow the dye to come to
  room temperature before use. Take out the kit and leave on the counter
  before starting protocol to ensure it becomes a liquid. You can
  pipette TE and samples while it is coming to room temp.
\item
  The first column of your plate needs to be saved for the ladder
  standards. Therefore if you are using 6 standards as recommended in
  this protocol, the most samples you could run on a plate is 90.
\end{itemize}

\hypertarget{protocol-5}{%
\section*{\texorpdfstring{\textbf{Protocol}}{Protocol}}\label{protocol-5}}
\addcontentsline{toc}{section}{\textbf{Protocol}}

\begin{itemize}
\item
  Dilute your 20X TE to 1X TE by pipetting 9.5 mL of Molecular Grade H2O
  and 0.5 mL of 20X TE into a 15mL falcon tube. This solution is stable
  long term so you can make more than you will need and store for later
  use. If you will not be using frequently, you can also calculate
  exactly how much you need and just make enough for that use.
\item
  Create your ladder standards using Table 1. The stock ladder is at 100
  ug/mL. So for example, to make Dilution 1 at 2 ug/mL, combine 1 uL of
  stock ladder with 49 uL of 1X TE.
\item
  Pipette 49 uL of 1X TE in all sample wells and the appropriate volume
  of 1X TE into each standard well according to Table 2. For example for
  standard 1 pipette 25 uL of 1X TE in the appropriate well.
\item
  Add 1 uL of sample to each of the sample wells and the appropriate
  amount of each ladder to the standard wells according to Table 2. For
  example, add 25 uL of Ladder Dilution 1 to standard 1 well.
\item
  At this point your PicoGreen should be at room temperature. Create
  your Diluted PicoGreen and use Equation 1 to determine how much stock
  PicoGreen dye you need for the number of samples/standards you are
  running. Then use that value to calculate the amount of 1X TE this
  should be added to for a 200 times dilution using Equation 2.
\item
  Pipette diluted PicoGreen to each well with each well getting 50 uL of
  the 200 times diluted PicoGreen. When adding the diluted PicoGreen
  pipette up and down at least 20 times to mix sample, TE and PicoGreen
  together.
\item
  Use a plate cover and briefly spin down the plate to get all liquid to
  the bottom of the well.
\item
  Run plate on a Plate Reader using the Gen5 software. First turn on the
  machine to warm up (takes \textasciitilde2-3 minutes)
\item
  Open Gen5 software and on the opening window make sure you are on the
  ``Run Now'' option on the left side column and click on the PicoGreen
  Assay full plate file.
\item
  This will open another window with a full plate already loaded just
  click ``cancel''
\item
  Click ``Plate Layout'' on the top bar which will open a new window to
  edit with your standard curve and sample wells
\item
  Clear plate by highlighting ``EMPTY'' on the left panel and click and
  drag across whole plate
\item
  Then right click on the ``STD'' and click ``edit'' to change your
  standard curve values to what you set up for your plate when you are
  done editing click okay
\item
  Highlight the first standard curve value in the left panel and click
  on the appropriate well where that value is located. You can click and
  drag down the column if they are all in a row.
\item
  Highlight the first sample (SP1) from the left panel and click on the
  well with your first sample well and drag across the plate until all
  wells you want readings for are filled. ** You do not need to put
  water in empty wells can just leave empty **
\item
  When this is done click okay and the machine will have a bulb warm up
  period after which it will ask you to load your plate. You do not need
  to close the drawer it will close automatically after you load the
  plate and click ``okay''.
\item
  The run should take \textasciitilde2-3 minutes, then click on the
  ``Statistics'' tab and output that spreadsheet to an excel file by
  copy and pasting.
\item
  Finally use XXX spreadsheet to convert absorbance values to associated
  concentrations.
\end{itemize}

\hypertarget{equation-1}{%
\subsection*{\texorpdfstring{\textbf{Equation
1}}{Equation 1}}\label{equation-1}}
\addcontentsline{toc}{subsection}{\textbf{Equation 1}}

(1.1 * (50 uL diluted PicoGreen * (\#samples + \#standards))/200 = vol
of Stock PicoGreen

\hypertarget{equation-2}{%
\subsection*{\texorpdfstring{\textbf{Equation
2}}{Equation 2}}\label{equation-2}}
\addcontentsline{toc}{subsection}{\textbf{Equation 2}}

199 * vol of Stock PicoGreen = vol of 1x TE

\hypertarget{table-1-1}{%
\subsection*{\texorpdfstring{\textbf{TABLE
1}}{TABLE 1}}\label{table-1-1}}
\addcontentsline{toc}{subsection}{\textbf{TABLE 1}}

\begin{longtable}[]{@{}
  >{\raggedright\arraybackslash}p{(\columnwidth - 8\tabcolsep) * \real{0.1667}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 8\tabcolsep) * \real{0.2647}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 8\tabcolsep) * \real{0.1667}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 8\tabcolsep) * \real{0.2059}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 8\tabcolsep) * \real{0.1961}}@{}}
\toprule()
\begin{minipage}[b]{\linewidth}\raggedright
\textbf{Ladder Type}
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
\textbf{Concentration (ug/mL)}
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
\textbf{Ladder Type}
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
\textbf{Ladder Vol (uL)}
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
\textbf{1X TE Vol (uL)}
\end{minipage} \\
\midrule()
\endhead
D1 & 2 & Stock & 1 & 49 \\
D2 & 0.2 & D1 & 1 & 9 \\
\bottomrule()
\end{longtable}

\hypertarget{table-2-1}{%
\subsection*{\texorpdfstring{\textbf{TABLE
2}}{TABLE 2}}\label{table-2-1}}
\addcontentsline{toc}{subsection}{\textbf{TABLE 2}}

\begin{longtable}[]{@{}
  >{\raggedright\arraybackslash}p{(\columnwidth - 10\tabcolsep) * \real{0.1136}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 10\tabcolsep) * \real{0.2045}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 10\tabcolsep) * \real{0.1288}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 10\tabcolsep) * \real{0.1591}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 10\tabcolsep) * \real{0.1515}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 10\tabcolsep) * \real{0.2424}}@{}}
\toprule()
\begin{minipage}[b]{\linewidth}\raggedright
\textbf{Standards}
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
\textbf{Concentration (ng/mL)}
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
\textbf{Ladder Type}
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
\textbf{Ladder Vol (uL)}
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
\textbf{1X TE Vol (uL)}
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
\textbf{Diluted PicoGreen Vol (uL)}
\end{minipage} \\
\midrule()
\endhead
1 & 500 & D1 & 25 & 25 & 50 \\
2 & 250 & D1 & 12.5 & 37.5 & 50 \\
3 & 100 & D1 & 5 & 45 & 50 \\
4 & 10 & D2 & 5 & 45 & 50 \\
5 & 4 & D2 & 2 & 48 & 50 \\
blank & 0 & - & - & 50 & 50 \\
\bottomrule()
\end{longtable}

\hypertarget{table-3}{%
\subsection*{\texorpdfstring{\textbf{TABLE 3}}{TABLE 3}}\label{table-3}}
\addcontentsline{toc}{subsection}{\textbf{TABLE 3}}

\begin{longtable}[]{@{}ll@{}}
\toprule()
\textbf{Reagent} & \textbf{Volume (uL)} \\
\midrule()
\endhead
1X TE & 49 \\
sample & 1 \\
dil PG & 50 \\
\bottomrule()
\end{longtable}

\hypertarget{magnetic-bead-wash}{%
\chapter{Magnetic Bead Wash}\label{magnetic-bead-wash}}

The protocol for doing a magnetic bead wash

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{overview-1}{%
\section*{\texorpdfstring{\textbf{Overview}}{Overview}}\label{overview-1}}
\addcontentsline{toc}{section}{\textbf{Overview}}

Magnetic beads are used for purification and size selection of single or
double-stranded DNA. size selection may be performed at the following
points during the overall library construction process:

\begin{itemize}
\item
  before or after DNA fragmentation;
\item
  after individual enzymatic reactions;
\item
  after the first post-ligation cleanup; or
\item
  after library amplification.
\end{itemize}

We use
\href{https://www.kapabiosystems.com/document/kapa-pure-beads-tds/?dl=1}{Kapa
Pure Beads} for magnetic bead washes. We find that they work as well as
the Ampure Beads and are MUCH cheaper.

Before using, be sure to read the
\href{https://www.kapabiosystems.com/document/kapa-pure-beads-tds/?dl=1}{PDF}
from Kapa.

\hypertarget{table-of-contents}{%
\section*{\texorpdfstring{\textbf{Table of
Contents}}{Table of Contents}}\label{table-of-contents}}
\addcontentsline{toc}{section}{\textbf{Table of Contents}}

\hypertarget{protocols}{%
\subsection*{\texorpdfstring{\textbf{Protocols}}{Protocols}}\label{protocols}}
\addcontentsline{toc}{subsection}{\textbf{Protocols}}

\href{https://drk-lo.github.io/lotterhoslabprotocols/molecprot_magbeadwash/\#clean-up-genomic-dna}{Clean
up genomic DNA}

\href{https://drk-lo.github.io/lotterhoslabprotocols/molecprot_magbeadwash/\#cleanup-of-fragmented-dna}{Cleanup
of fragmented DNA}

\href{https://drk-lo.github.io/lotterhoslabprotocols/molecprot_magbeadwash/\#size-selection}{Size
Selection}

\hypertarget{troubleshooting}{%
\subsection*{\texorpdfstring{\textbf{Troubleshooting}}{Troubleshooting}}\label{troubleshooting}}
\addcontentsline{toc}{subsection}{\textbf{Troubleshooting}}

\href{https://drk-lo.github.io/lotterhoslabprotocols/molecprot_magbeadwash/\#factors-that-impact-overall-performance}{Factors
that Impact Overall Performance}

\hypertarget{protocols-1}{%
\section*{\texorpdfstring{\textbf{Protocols}}{Protocols}}\label{protocols-1}}
\addcontentsline{toc}{section}{\textbf{Protocols}}

See the
\href{https://www.kapabiosystems.com/document/kapa-pure-beads-tds/?dl=1}{PDF}
for the complete protocol. Tips:

\hypertarget{cleanup-genomic-dna}{%
\subsection*{\texorpdfstring{\textbf{Cleanup genomic
DNA}}{Cleanup genomic DNA}}\label{cleanup-genomic-dna}}
\addcontentsline{toc}{subsection}{\textbf{Cleanup genomic DNA}}

1.1 Ensure that KAPA Pure Beads has been equilibrated to room
temperature and that the beads are fully resuspended before proceeding.

1.2 Add 300 μL of KAPA Pure Beads (3:1 ratio for gDNA) to the 100 μL
genomic DNA sample.

1.3 Mix thoroughly by vortexing and/or pipetting up and down multiple
times.

1.4 Incubate the plate/tube(s) at room temperature for 10 min to bind
the DNA to the beads.

1.5 Place the plate/tube(s) on a magnet to capture the beads. Incubate
until the liquid is clear.

1.6 Carefully remove and discard the supernatant.

1.7 Keeping the plate/tube(s) on the magnet, add 200 μL of 80\% ethanol.

1.8 Incubate the plate/tube(s) on the magnet at room temperature for ≥30
sec.

1.9 Carefully remove and discard the ethanol.

1.10 Keeping the plate/tube(s) on the magnet, add 200 μL of 80\%
ethanol.

1.11 Incubate the plate/tube(s) on the magnet at room temperature for
≥30 sec.

1.12 Carefully remove and discard the ethanol. Try to remove all
residual ethanol without disturbing the beads.

1.13 Dry the beads at room temperature for 3 -- 5 min, or until all of
the ethanol has evaporated. Caution: over-drying the beads may result in
reduced yield.

1.14 Remove the plate/tube(s) from the magnet.

1.15 Resuspend the beads in an appropriate volume of pre-heated elution
buffer at 37°C and/or perform the elution incubation (step 1.16) in a
thermocycler or heating block set to 37°C. The appropriate elution
buffer may be either 10 mM Tris-HCl, (pH 8.0 -- 8.5) or PCR-grade water,
depending on the downstream application.

1.16 Incubate the plate/tube(s) for 10 min to elute the DNA off the
beads.

1.17 Place the plate/tube(s) on a magnet to capture the beads. Incubate
until the liquid is clear.

1.18 Transfer the clear supernatant to a new plate/ tube(s). Proceed
with your downstream application, or store DNA at 4ºC for 1 -- 2 weeks,
or at -20ºC.

\hypertarget{notes}{%
\subsection*{\texorpdfstring{\textbf{Notes}}{Notes}}\label{notes}}
\addcontentsline{toc}{subsection}{\textbf{Notes}}

\begin{itemize}
\item
  3X KAPA Pure Beads to-sample volumetric ratio recommended
\item
  The detailed protocol below is an example of a 3X cleanup of genomic
  DNA in 100 µL. Please pay special attention to steps 1.15 and 1.16
  (elution of DNA off beads). To ensure optimal recovery, these steps
  may be performed at an elevated temperature: 37°C for 10 min. Elution
  buffer may be pre-heated for this step and/or the elution performed in
  a thermocycler or heating block.
\item
  As indicated above, a heated elution is recommended to ensure optimal
  recovery of high-quality genomic DNA. This is not a requirement for
  the cleanup of other DNA types (e.g.~fragmented DNA).
\item
  For maximum recovery of the original DNA sample during cleanup, buffer
  exchange, and/or concentration of genomic DNA, fragmented DNA for NGS
  library construction, PCR or qPCR products---regardless of quality---a
  KAPA Pure Beads-to-sample volumetric ratio of 3X is recommended.
  Expected recoveries are 50 -- 90\%, depending upon sample quality and
  quantity. If a lower ratio is used, it may result in lower yields and
  exclusion of smaller fragments.
\end{itemize}

\hypertarget{cleanup-of-fragmented-dna}{%
\section*{\texorpdfstring{\textbf{Cleanup of fragmented
DNA}}{Cleanup of fragmented DNA}}\label{cleanup-of-fragmented-dna}}
\addcontentsline{toc}{section}{\textbf{Cleanup of fragmented DNA}}

2.1 Ensure that KAPA Pure Beads has been equilibrated to room
temperature and that the beads are fully resuspended before proceeding.

2.2 Add 80 μL of KAPA Pure Beads to the 100 μL fragmented DNA sample.

2.3 Mix thoroughly by vortexing and/or pipetting up and down multiple
times.

2.4 Incubate the plate/tube(s) at room temperature for 5 -- 15 min to
bind the DNA to the beads.

2.5 Place the plate/tube(s) on a magnet to capture the beads. Incubate
until the liquid is clear.

2.6 Carefully remove and discard the supernatant.

2.7 Keeping the plate/tube(s) on the magnet, add 200 μL of 80\% ethanol.

2.8 Incubate the plate/tube(s) on the magnet at room temperature for ≥30
sec.

2.9 Carefully remove and discard the ethanol.

2.10 Keeping the plate/tube(s) on the magnet, add 200 μL of 80\%
ethanol.

2.11 Incubate the plate/tube(s) on the magnet at room temperature for
≥30 sec.

2.12 Carefully remove and discard the ethanol. Try to remove all
residual ethanol without disturbing the beads.

2.13 Dry the beads at room temperature for 3 -- 5 min, or until all of
the ethanol has evaporated. Caution: over-drying the beads may result in
reduced yield.

2.14 Remove the plate/tube(s) from the magnet.

2.15 Resuspend the beads in an appropriate volume of elution buffer (10
mM Tris-HCl, pH 8.0 -- 8.5) or PCR-grade water, depending on the
downstream application.

2.16 Incubate the plate/tube(s) at room temperature for 2 min to elute
the DNA off the beads. The elution time may be extended up to 10 min if
necessary to improve DNA recovery.

2.17 Place the plate/tube(s) on a magnet to capture the beads. Incubate
until the liquid is clear.

2.18 Transfer the clear supernatant to a new plate/ tube(s). Proceed
with your downstream application, or store DNA at 4ºC for 1 -- 2 weeks,
or at -20ºC.

\hypertarget{notes-1}{%
\subsection*{\texorpdfstring{\textbf{Notes}}{Notes}}\label{notes-1}}
\addcontentsline{toc}{subsection}{\textbf{Notes}}

\begin{itemize}
\tightlist
\item
  0.8X cleanup of a 100 µL fragmented DNA sample recommended (will
  retain fragments greater than 300 bp)
\end{itemize}

\begin{longtable}[]{@{}cc@{}}
\toprule()
\textbf{Fragments to be retained} & \textbf{Recommended Beads-to-sample
ratio} \\
\midrule()
\endhead
≥ kb & 0.5X \\
≥ 450 bp & 0.6X \\
≥ 350 bp & 0.7X \\
≥ 300 bp & 0.8X \\
≥ 250 bp & 0.9X \\
≥ 150 bp & 1.5X \\
≥ 100 bp & 2.2X -3X \\
\bottomrule()
\end{longtable}

\hypertarget{size-selection}{%
\section*{\texorpdfstring{\textbf{Size
Selection}}{Size Selection}}\label{size-selection}}
\addcontentsline{toc}{section}{\textbf{Size Selection}}

3.1 Ensure that KAPA Pure Beads has been equilibrated to room
temperature and that the beads are fully resuspended before proceeding.

3.2 Perform the first size cut (0.6X, to exclude large library
fragments) by adding 30 μL of KAPA Pure Beads to 50 μL of
adapter-ligated library (0.6 x 50 μL = 30 μL).

3.3 Mix thoroughly by vortexing and/or pipetting up and down multiple
times.

3.4 Incubate the plate/tube(s) at room temperature for 5 -- 15 min to
bind large library molecules (\textgreater450 bp) to the beads.

3.5 Place the plate/tube(s) on a magnet to capture the beads. Incubate
until the liquid is clear.

3.6 Carefully transfer the supernatant containing the smaller library
molecules (\textless450 bp) to a new plate/ tube(s). It is critical that
no beads are transferred with the supernatant.

3.7 Discard the plate/tube(s) containing the beads to which library
fragments larger than \textasciitilde450 bp are bound.

3.8 Perform the second size cut (0.8X) by adding 10 μL of KAPA Pure
Beads to the supernatant. This volume is calculated relative to the
original sample volume of 50 μL, e.g., (0.8 -- 0.6) x 50 μL = 10 μL.

3.9 Mix thoroughly by vortexing and/or pipetting up and down multiple
times.

3.10 Incubate the plate/tube(s) at room temperature for 5 -- 15 min to
bind library molecules \textgreater250 bp to the beads.

3.11 Place the plate/tube(s) on a magnet to capture the beads. Incubate
until the liquid is clear. Carefully remove and discard the supernatant
which contains library molecules smaller than

\begin{quote}
250 bp.
\end{quote}

3.12 Keeping the plate/tube(s) on the magnet, add 200 μL of 80\%
ethanol.

3.13 Incubate the plate/tube(s) on the magnet at room temperature for
≥30 sec.

3.14 Carefully remove and discard the ethanol. 3.15 Keeping the
plate/tube(s) on the magnet, add 200 μL of 80\% ethanol.

3.16 Incubate the plate/tube(s) on the magnet at room temperature for
≥30 sec.

3.17 Carefully remove and discard the ethanol. Try to remove all
residual ethanol without disturbing the beads.

3.18 Dry the beads at room temperature for 3 -- 5 min, or until all of
the ethanol has evaporated. ⚠️ \textbf{Caution: over-drying the beads
may result in reduced yield.}

3.19 Remove the plate/tube(s) from the magnet.

3.20 Thoroughly resuspend the beads in the required volume of elution
buffer (10 mM Tris-HCl, pH 8.0 -- 8.5), or PCR-grade water, depending on
the downstream application.

3.21 Incubate the plate/tube(s) at room temperature for 2 min to elute
DNA off the beads. The elution time may be extended up to 10 min if
necessary to improve DNA recovery.

3.22 Place the plate/tube(s) on a magnet to capture the beads. Incubate
until the liquid is clear.

3.23 Transfer the clear supernatant with size-selected DNA to a new
plate/tube(s). Proceed with your downstream application, or store DNA at
4ºC for 1 -- 2 weeks, or at -20ºC.

\hypertarget{notes-2}{%
\subsection*{\texorpdfstring{\textbf{Notes}}{Notes}}\label{notes-2}}
\addcontentsline{toc}{subsection}{\textbf{Notes}}

Bead-based size selection or ``double-sided size selection'' consists of
a ``first'' and ``second cut'', represented as ratios of beads-to-sample
volumes.

\hypertarget{factors-that-impact-overall-performance}{%
\section*{\texorpdfstring{\textbf{Factors that Impact Overall
Performance}}{Factors that Impact Overall Performance}}\label{factors-that-impact-overall-performance}}
\addcontentsline{toc}{section}{\textbf{Factors that Impact Overall
Performance}}

\begin{itemize}
\item
  Store at 2°C to 8°C in a constant-temperature refrigerator; do not
  store at -20°C. \textbf{Always ensure that KAPA} \textbf{Pure Beads is
  fully equilibrated to room temperature} \textbf{before use.}
\item
  For optimal performance, always ensure that the KAPA Pure Beads is
  equilibrated to room temperature, that the beads are fully resuspended
  before use, and that the DNA and bead solutions are thoroughly mixed
  prior to the binding incubation.
\item
  The volumetric ratio of KAPA Pure Beads to sample is the critical
  factor in determining the size distribution of DNA fragments retained
  by the beads. The volume (ratio) may be modified/optimized based upon
  the specific application and/or point in the library construction
  workflow where a cleanup or size selection is employed.
\item
  The volume in which cleanups and size selection are performed may
  impact performance. DNA samples may be diluted in 10 mM Tris-HCl (pH
  8.0 -- 8.5) or PCR-grade water to increase the working volume prior to
  the addition of KAPA Pure Beads. This is recommended for starting
  volumes \textless50 µL, DNA preparations that are viscous, and/ or
  contain a high concentration of salts or PEG.
\item
  It is important to prepare fresh 80\% ethanol prior to use, since
  ethanol absorbs atmospheric water.
\item
  The volume of 80\% ethanol used for bead washes may be adjusted to
  accommodate different plate/tube volumes and/or limited pipetting
  capacity, but it is important that the beads are entirely submerged
  during the wash steps. Where possible, use a wash volume that is equal
  to the volume of sample plus KAPA Pure Beads.
\item
  It is important to remove all ethanol before proceeding with
  subsequent reactions. However, over-drying of beads may make them
  difficult to resuspend, and may result in a dramatic loss of DNA.
  Drying of beads for 3 -- 5 min at room temperature should be
  sufficient. Drying of beads at 37ºC is not recommended.
\item
  Elution time and temperature may impact DNA recovery, particularly for
  high-molecular weight genomic DNA.
\item
  Purified DNA in elution buffer should be stable at 4°C for 1 -- 2
  weeks, or for at least 1~month at -20°C. The long-term stability of
  library DNA at -20°C depends on a number of factors, including library
  concentration. Always use low DNA-binding tubes for long-term storage
  and avoid excessive freezing and thawing.
\item
  To increase the recovery of DNA at a specific stage in an NGS library
  construction protocol, the volume of KAPA Pure Beads used in the
  cleanup may be increased, but this will most likely result in the
  retention of smaller DNA fragments.
\end{itemize}

\hypertarget{upper-size-limit-modification}{%
\subsection*{\texorpdfstring{\textbf{Upper size limit \textbar{}
Modification}}{Upper size limit \textbar{} Modification}}\label{upper-size-limit-modification}}
\addcontentsline{toc}{subsection}{\textbf{Upper size limit \textbar{}
Modification}}

\begin{itemize}
\item
  Increase upper size limit: Decrease the ratio of the first cut
\item
  Decrease upper size limit: Increase the ratio of the first cut
\end{itemize}

\hypertarget{lower-size-limit-modification}{%
\subsection*{\texorpdfstring{\textbf{Lower size limit \textbar{}
Modification}}{Lower size limit \textbar{} Modification}}\label{lower-size-limit-modification}}
\addcontentsline{toc}{subsection}{\textbf{Lower size limit \textbar{}
Modification}}

\begin{itemize}
\item
  Increase lower size limit: Decrease the ratio of the second cut
\item
  Decrease lower size limit: Increase the ratio of the second cut
\item
  The second size cut should be performed with at least 0.2~volumes of
  KAPA Pure Beads. Please note the volume of the KAPA Pure Beads needed
  for the second cut is calculated relative to the volume of the sample
  at the start of the size selection procedure, not the volume of the
  DNA-containing supernatant transferred after the first cut. DNA
  recovery is dramatically reduced if the difference between the first
  and second cuts is less than \textasciitilde0.2~ volumes. To increase
  the amount of DNA recovered, more than 0.2~volumes of KAPA Pure Beads
  may be used for the second cut, but note that this may result in the
  recovery of smaller library fragments and a broader size distribution.
\end{itemize}

\hypertarget{tapestation-library-check}{%
\chapter{TapeStation Library Check}\label{tapestation-library-check}}

The protocol for checking libraries on the TapeStation

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{background-2}{%
\section*{\texorpdfstring{\textbf{Background}}{Background}}\label{background-2}}
\addcontentsline{toc}{section}{\textbf{Background}}

The Agilent TapeStation provides an accurate, efficient quantification
and quality assessment of RNA, DNA and protein samples. RIN numbers are
given to RNA samples and quantification can be done on each individual
band in your sample. Output comes as both a gel image and a size
distribution electropharogram. Can handle strip tubes or plates.

\hypertarget{equipment-reagents-1}{%
\section*{\texorpdfstring{\textbf{Equipment \&
Reagents}}{Equipment \& Reagents}}\label{equipment-reagents-1}}
\addcontentsline{toc}{section}{\textbf{Equipment \& Reagents}}

\begin{itemize}
\item
  2200 TapeStation (Agilent Cat \# G2965AA)
\item
  Loading Tips (1pk = 384 tips, Agilent Cat \# 5067-5153) ** must use
  their tips
\item
  Optical Tube Strips (Agilent Cat \# 401428) ** can use other tubes
  with exact same dimensions
\item
  Strip caps (Agilent Cat \# 401425) ** can use other caps not needed
  for actual machine run
\item
  Kits:

  \begin{enumerate}
  \def\labelenumi{\arabic{enumi}.}
  \tightlist
  \item
    \textbf{D1000}
  \end{enumerate}

  \begin{itemize}
  \item
    ScreenTape (Agilent Cat \# 5067-5582)
  \item
    Reagents (Agilent Cat \# 5067-5583)

    \begin{enumerate}
    \def\labelenumi{\arabic{enumi}.}
    \tightlist
    \item
      \textbf{High Sensitivity D1000}
    \end{enumerate}
  \item
    ScreenTape (Agilent Cat \# 5067-5584)
  \item
    Reagents (Agilent Cat \# 5067-5585)

    \begin{enumerate}
    \def\labelenumi{\arabic{enumi}.}
    \tightlist
    \item
      \textbf{D5000}
    \end{enumerate}
  \item
    ScreenTape (Agilent Cat \# 5067-5588)
  \item
    Reagents (Agilent Cat \# 5067-5589)

    \begin{enumerate}
    \def\labelenumi{\arabic{enumi}.}
    \tightlist
    \item
      \textbf{High Sensitivity D5000}
    \end{enumerate}
  \item
    ScreenTape (Agilent Cat \# 5067-5592)
  \item
    Reagents (Agilent Cat \# 5067-5593)
  \end{itemize}
\end{itemize}

\textbf{Store all tapes in their box with the label facing towards you
(Fig 1 below)}

\includegraphics{https://docs.google.com/drawings/d/1t0-f8TKZFptM4MrDHA_w4AtsOcy_YRR7boeD1aCk2mg/pub?w=1429\&h=495.pdf}

\hypertarget{important-notes-read-before-using}{%
\section*{\texorpdfstring{\textbf{Important Notes READ BEFORE
USING:}}{Important Notes READ BEFORE USING:}}\label{important-notes-read-before-using}}
\addcontentsline{toc}{section}{\textbf{Important Notes READ BEFORE
USING:}}

\begin{itemize}
\item
  RNA Reagent kits and tapes will be GREEN and DNA Reagent kits and
  tapes will be BLUE double check before you get started
\item
  Make sure you use the correct tape and buffers for the size fragments
  that you are looking at:
\item
  Pairs only EVEN number of samples because the two probes move
  together. If you have 3 samples fill a 4th with buffer
\item
  Tapes can only do up to 16 samples at a time (2 columns of a 96 well
  plate or 2 strip tubes)
\item
  If you are doing a 96 well plate you need to replace the Tape after
  the 16 samples are complete. Computer program will notify you when
  this is okay to do.
\item
  \textbf{NEVER OPEN TAPESTATION WHILE RUNNING SAMPLES} unless you want
  to start over! This will throw out all your data because it completes
  calculations after it runs through every sample.
\item
  Always wear gloves when handling the tapes, oils on your hands can
  impact the accuracy of the reads
\end{itemize}

\hypertarget{things-to-do-before-starting}{%
\section*{\texorpdfstring{\textbf{Things to do before
starting}}{Things to do before starting}}\label{things-to-do-before-starting}}
\addcontentsline{toc}{section}{\textbf{Things to do before starting}}

\begin{itemize}
\item
  Set reagents including tapes, buffer, and ladder on counter for 30
  minutes to bring to room temperature
\item
  Label strip tubes or plate for your samples
\item
  Login to machine using the sticker log in information on the computer
  next to tapestation
\end{itemize}

\hypertarget{protocol-6}{%
\section*{\texorpdfstring{\textbf{Protocol}}{Protocol}}\label{protocol-6}}
\addcontentsline{toc}{section}{\textbf{Protocol}}

\begin{itemize}
\item
  Load tips into machine with multi-channel pipette; fill tips
  completely because tapestation does not use tips in order. Can remove
  unused tips and put back in storage box for later use.

  \begin{itemize}
  \tightlist
  \item
    If you are doing a plate, you will need to refill tips after each
    set of 16 samples completes. The program will alert you when it is
    time to refill them.
  \end{itemize}
\item
  After tape has come to room temperature, flick out any air bubbles
  from the lanes

  \begin{itemize}
  \tightlist
  \item
    Usually only happens if the tape has had some lanes used before
  \end{itemize}
\item
  Insert tape in the appropriate slot
\end{itemize}

\includegraphics{https://docs.google.com/drawings/d/1ajGI6Yz1gpO-WCYh8056qCoPAlh_n4mYoNaOQg73TOc/pub?w=465\&h=336.pdf}

\begin{itemize}
\item
  Tapes have a barcode that needs to be read by the machine
\item
  If in the wrong orientation, the computer will spit back an error
\item
  Save the wrapping if you aren't using all the lanes to place tape back
  in for storage after run
\item
  Highlight where your samples are and name them via the spreadsheet on
  the right

  \begin{itemize}
  \tightlist
  \item
    CSV file with names can be uploaded
  \end{itemize}
\end{itemize}

\includegraphics{https://docs.google.com/drawings/d/1joVrT3Hetfx3kbLPvPmTcUZlsE-SD004qC5AmqHDxq8/pub?w=239\&h=283.pdf}

\textbf{Table 1} : Each kit with the appropriate amount of sample buffer
with either sample or ladder that should be mixed in each well

\begin{longtable}[]{@{}
  >{\raggedright\arraybackslash}p{(\columnwidth - 4\tabcolsep) * \real{0.3544}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 4\tabcolsep) * \real{0.3038}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 4\tabcolsep) * \real{0.3418}}@{}}
\toprule()
\begin{minipage}[b]{\linewidth}\raggedright
\textbf{Kit}
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
\textbf{Sample Buffer (uL)}
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
\textbf{Sample or Ladder (uL)}
\end{minipage} \\
\midrule()
\endhead
\textbf{D1000} & 3 & 1 \\
\textbf{High Sensitivity D1000} & 2 & 2 \\
\textbf{D5000} & 10 & 1 \\
\textbf{High Sensitivity D5000} & 2 & 2 \\
\textbf{Genomic DNA} & 10 & 1 \\
\textbf{RNA} & 5 & 1 \\
\textbf{High Sensitivity RNA} & 1 & 2 \\
\bottomrule()
\end{longtable}

\begin{itemize}
\item
  Vortex and spin down reagents that you will be using for run
\item
  Add the appropriate amount of buffer and sample to each tube (Table 1)
  with one tube designated for the ladder

  \begin{itemize}
  \tightlist
  \item
    Only one well needs to be used for the ladder and it doesn't matter
    which well you use just make sure to label properly in the
    spreadsheet
  \end{itemize}
\item
  If the number of samples + 1 ladder is an odd number, you need to add
  an additional well with just buffer in the appropriate amount for the
  kit (e.g.~D1000 4 uL of buffer). The machine only runs samples in
  twos.
\item
  Spin for 1 min on fancy vortexer with caps on
\item
  Remove caps
\item
  Place in machine in the orientation that you filled the spreadsheet
  out as
\item
  Double check tips are in
\item
  Click Start
\item
  Once complete File Open your file
\item
  You can view the results as both a gel image and an electropherogram
\item
  Click the electropherogram button and then scale to sample

  \begin{itemize}
  \tightlist
  \item
    All concentrations are outputted as a table below the
    electropherogram
  \end{itemize}
\item
  Set upper and lower markers by right clicking on peaks
\item
  To zoom into a peak, highlight the region on the electropharogram
\item
  You can also compare results between different files by clicking the
  comparison tab
\item
  Open the files that you wish to compare

  \begin{itemize}
  \tightlist
  \item
    NOTE you can only compare those DNA to DNA or RNA to RNA AND they
    must be from the same kit
  \end{itemize}
\item
  You can make a report of all of your data by clicking File Make a
  report

  \begin{itemize}
  \tightlist
  \item
    Check off the items you want in the report and change the order you
    want them to appear by using the arrows
  \end{itemize}
\end{itemize}

\hypertarget{molecular-protocols-for-using-the-biorad-qpcr-machine}{%
\chapter{Molecular Protocols for Using the BioRAD qPCR
Machine}\label{molecular-protocols-for-using-the-biorad-qpcr-machine}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

This protocols page does not exist yet.

\hypertarget{outdated-molecular-protocols}{%
\chapter{Outdated Molecular
Protocols}\label{outdated-molecular-protocols}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

Outdated molecular protocols

\begin{itemize}
\item
  \href{https://drk-lo.github.io/lotterhoslabprotocols/molecprot_ddradlibprep/}{ddRAD
  Library Prep}
\item
  \href{https://drk-lo.github.io/lotterhoslabprotocols/molecprot_bsddradlibprep/}{bs
  RAD Library Prep}
\item
  \href{https://drk-lo.github.io/lotterhoslabprotocols/molecprot_splatlibraryprep/}{SPLAT
  Library Prep}
\item
  \href{https://drk-lo.github.io/lotterhoslabprotocols/molecprot_rnaseqlibraryprep/}{RNA
  seq}\\
\end{itemize}

\part{Field and Sampling Protocols}

\hypertarget{collecting-samples-for-dna}{%
\chapter{Collecting Samples for DNA}\label{collecting-samples-for-dna}}

Protocols and tips for collecting samples for DNA extraction

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{overview-2}{%
\section*{\texorpdfstring{\textbf{Overview}}{Overview}}\label{overview-2}}
\addcontentsline{toc}{section}{\textbf{Overview}}

With next generation sequencing, it is important to consider all
possible sources of contamination when collecting samples. It is also
important to consider downstream analysis that may be affected by the
method of preservation for collected samples. For example, best results
for DNA extraction will come from flash-frozen DNA. For whole genome
sequencing, DNA methylation and bisulfite sequencing, flash-frozen DNA
is also the preferred collection method, but DNA stored in ethanol or
OGLFix can also be used, as long as samples are not stored for a long
time. It is very important that we all make the upmost effort to ensure
that the samples are collected using sterile materials and appropriate
technique, and that samples are well preserved, with the method of
preservation well documented. Care and attention in sample collection
for DNA extraction will ensure that time and resources are spent on good
quality samples and are the first steps to obtaining a robust and
reliable dataset.

\hypertarget{tips}{%
\section*{\texorpdfstring{\textbf{Tips}}{Tips}}\label{tips}}
\addcontentsline{toc}{section}{\textbf{Tips}}

\begin{itemize}
\item
  \textbf{Prepare ahead of time:} All sample vials used to store tissue
  for DNA extraction should be sterilized by autoclaving ahead of time.
  Screw-cap cryovials are preferred (Fisher, cat \#10-500-26). Be sure
  to get trained (online course and in-person training by OGL personnel)
  before using the autoclave. Make a list of all other materials needed
  to collect samples and autoclave what is autoclavable ahead of time.
  Remember to gather boxes, gloves and other PPE.
\item
  \textbf{Smart label and metadata}: In general, it is best practice to
  have tubes organized in a box, with at least 2 labels on each tube (in
  case one falls off). We have a barcode label printer (with
  freezer-proof labels) that can be used. Label the vials with species
  name or initials, and sample ID. \textbf{Make sure to have unique
  sample IDs for each sample for your entire project}. Label boxes with
  species name, project name, your initials, sampling date and location
  ahead of time. Make a spreadsheet listing the sample IDs and columns
  to record all other info and data about the study site, individuals
  and samples (lat/long, size and weight of animals, experimental
  treatment, etc).
\item
  \textbf{Avoid cross-contamination}: If possible, use a sterilized
  surface to handle or collect tissue or samples. Sterilize surface with
  10\% bleach, UV light and/or 70\% ethanol, or use sterile plastic
  weigh boats. Always wear gloves. For anything that comes in contact
  with the animals and samples, use sterile disposable materials, one
  per individual, whenever possible. If using blades, use sterile blades
  or wipe each blade and all other materials (forceps, spatulas, etc)
  with 70\% ethanol \textbf{between individuals} to avoid
  cross-contamination. If available, flame the blades to burn off the
  excess ethanol and further sterilize them. Be sure to get all bits of
  tissue off the blade and into the sample vial. Remember to dispose of
  sharps in a sharps container.
\item
  \textbf{Amount of tissue sample:} Collect an appropriate amount of
  tissue, so there will be enough sample to repeat or troubleshoot
  extraction if needed, and to archive the remaining tissue for future
  studies, to use as positive controls, etc. \textbf{Typically, that
  means 1 to 1.5 inch piece of tissue.} Use your best judgment to make
  sure that the entire tissue sample is submersed in fixative liquid (if
  that is your method of preservation). A sample that is too small may
  not yield enough DNA to proceed with sequencing and a sample that is
  too big for the vial may not be preserved well as the amount of
  preservative liquid in the vial may not permeate the whole tissue (see
  photo below).
\item
  \textbf{Storage:} Samples can be stored temporarily in liquid N2 but
  should be stored short term in the -20oC freezer and long term in the
  -80oC freezer. Minimize the amount of time that you keep the -80oC
  freezer door open and do not let the temperature rise above -70oC
  (this will set off the freezer alarms). \textbf{Be mindful of the
  location you select to store your boxes in the -80oC freezer - we have
  an inventory that needs to be checked ahead of time and filled out to
  keep track of where samples are}. Using the inventory helps minimizing
  the time that de freezer door is open when storing or retrieving
  boxes.
\end{itemize}

\hypertarget{flash-freezing-in-liquid-n2}{%
\section*{\texorpdfstring{\textbf{Flash freezing in liquid
N2}}{Flash freezing in liquid N2}}\label{flash-freezing-in-liquid-n2}}
\addcontentsline{toc}{section}{\textbf{Flash freezing in liquid N2}}

Be sure to get trained by OGL personnel beforehand if using the liquid
N2 dispenser at NUMSC. Place tissue into labelled cryovials. Make sure
to use liquid N2 and -80oC freezer proof labels so they do not fall off.
Use protective equipment to handle liquid N2: cryogenic gloves,
closed-toe shoes, lab coat and goggles. Dispense liquid N2 into the
dewar flask. Place a sample vials on a long-handle strainer and slowly
submerse the strainer into liquid N2 in the dewar, releasing the
cryovials into the liquid. If flash-freezing samples in the lab, wait
1-2 min and retreive the vials using the strainer. Make sure to retreive
all vials from the dewar. Place cryovials in a storage box and transfer
the box to the -80oC freezer. These cryovials are used for
flash-freezing and temporary storage-only in liquid N2. If
flash-freezing away from the lab, make sure to transfer vials to storage
boxes and into the -80oC freezer as soon as possible.

\hypertarget{ethanol}{%
\section*{\texorpdfstring{\textbf{Ethanol}}{Ethanol}}\label{ethanol}}
\addcontentsline{toc}{section}{\textbf{Ethanol}}

Use regular grade 95\% ethanol. Calculate how much ethanol will be
needed for all samples you are collecting and pour that volume plus
little extra from the stock ethanol container (stored into the
flammables cabinet) into a beaker or another container of appropriate
volume. You may need to adjust the concentration of ethanol, e.g., from
100\% to 95\% - use ultrapure water (MiliQ) or molecular grade water to
do this dilution. Pre-fill screw-cap vials with 1.5-2mL of 95\% ethanol
using a pipette. \textbf{Make sure to sumberse the whole tissue into
ethanol so the liquid can permeate the whole tissue} (see photo below).
If samples preserved in ethanol are being shipped via FedEx, UPS, etc,
you might need a Special or Hazardous Shipping form, label and to follow
specific packaging instructions. Contact lab manager before shipping
containers with ethanol.

\hypertarget{oglfix}{%
\section*{\texorpdfstring{\textbf{OGLFix}}{OGLFix}}\label{oglfix}}
\addcontentsline{toc}{section}{\textbf{OGLFix}}

OGLFix can be used instead of ethanol to preserve tissue for DNA
extraction. Follow the same instructions described above for ethanol (no
dilution needed). OGLFix can be shipped without hazardous designation.
As for ethanol preservation, \textbf{it is critical the the entire
tissue is submersed in fixative.} Refer to the OGLFix sampling protocol
for detailed instructions.

\begin{figure}

{\centering \includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/Samplingprotocolimmerse.png}

}

\caption{Sampling protocol immerse}

\end{figure}

\begin{figure}

{\centering \includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/Samplingprotocolsize.png}

}

\caption{Samplingprotocolsize}

\end{figure}

\hypertarget{collecting-samples-for-rna}{%
\chapter{Collecting Samples for RNA}\label{collecting-samples-for-rna}}

The protocol for collecting samples for RNA extraction

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{general-tips}{%
\section*{\texorpdfstring{\textbf{General
tips}}{General tips}}\label{general-tips}}
\addcontentsline{toc}{section}{\textbf{General tips}}

With next generation sequencing, it is important to consider all sources
of possible contamination when collecting samples. It is also important
to consider the study question. For example, best results for RNA
extraction will come from flash-frozen tissue or tissue stored in
RNA-later. It is important to consider the source tissue (because of
tissue-specific expression) when designing an experiment with RNA.

A good resource is the \href{http://rnaseq.uoregon.edu/}{RNA-seqlopedia}
and our RNA-seq page.

\begin{itemize}
\item
  RNAases will ``eat'' DNA and they are everywhere. Use a sterilized
  surface. Sterilize surface with 10\% bleach or UV light, then use
  RNAase Away to clean the surface.
\item
  RNA degrades in as little as 30 seconds, so tissues should be
  harvested quickly.
\item
  Use sterile blades and switch the blade between samples.
\item
  All centrifuge tubes used to store tissues should be sterilized and
  prepared before going out. In general, it is best practice to have
  tubes organized in a box, with at least 2 labels on each tube (in case
  one falls off). We have a special barcode label printer (with
  freezer-proof labels) that can be used.
\end{itemize}

\hypertarget{materials-2}{%
\section*{Materials}\label{materials-2}}
\addcontentsline{toc}{section}{Materials}

\begin{itemize}
\item
  Scalpels with replaceable, sterile blades
\item
  Autoclaved glass pyrex dish for cutting sample on and a flame to torch
  it; alternatively can use weigh boats that are RNAase/DNAase free
\item
  Autoclaved tweezers, probes, etc. Ideally use tweezers without the
  ridges, between which tissue gets stuck.
\item
  Ethanol in an autoclaved beaker for rinsing dissecting tools
\item
  A small flame for sterilizing dissecting tools between samples
\item
  Prepared, Autoclaved and labeled (with EtOH-proof pen) microcentrifuge
  tube(s)
\item
  RNA-ase away to wipe off surfaces and blades between samples
\item
  Nitex gloves
\end{itemize}

\hypertarget{tips-for-flash-freezing-in-liquid-n2}{%
\section*{\texorpdfstring{\textbf{Tips for flash freezing in liquid
N2}}{Tips for flash freezing in liquid N2}}\label{tips-for-flash-freezing-in-liquid-n2}}
\addcontentsline{toc}{section}{\textbf{Tips for flash freezing in liquid
N2}}

We use Fisher cryovials (\#10-500-26), although note that they are
really not designed to put into liquid N2. They work well for short-term
storage in liquid N2 and then transfer to the -80C.

\hypertarget{tips-for-rnalater}{%
\section*{\texorpdfstring{\textbf{Tips for
RNAlater}}{Tips for RNAlater}}\label{tips-for-rnalater}}
\addcontentsline{toc}{section}{\textbf{Tips for RNAlater}}

We have successfully placed tissue in RNAlater and flash frozen, with
good results for RNA extraction.

If the tissue cannot be flash frozen in RNAlater, than it is important
to:

(1) Get the tissue:RNAlater ratio correct following manufacturer's
directions: Trim the tissue to be less than 0.5 cm in at least one
dimension and simply submerge it in 5 volumes of RNAlater solution
(e.g., a 0.5 g sample requires about 2.5 mL of RNAlater solution).

(2) Keep the tissue at the correct temperature following the
manufacturer's direction for the specified amount of time before
transferring to -80C. This allows the RNAlater to permeate the tissues.
Do not freeze samples in RNAlater® Solution immediately;
\href{https://www.thermofisher.com/content/dam/LifeTech/migration/en/filelibrary/nucleic-acid-purification-analysis/pdfs.par.18819.file.dat/bp-7020.pdf}{READ
THE INSTRUCTIONS}. Typically we store at 4°C overnight (to allow the
solution to thoroughly penetrate the tissue), remove supernatant, then
move to --20°C or --80°C for long-term storage. Although, we have had
good luck flash-freezing tissues in RNAlater.

Samples can be stored in RNAlater at 4°C for one month, at 25°C for one
week, or at --20°C indefinitely. Archive tissues treated with RNAlater
solution at --20°C.

\begin{itemize}
\item
  If any precipitation of RNAlater® Solution is seen, heat it to 37°C
  and agitate to redissolve it.
\item
  If the crystals do not go into solution at 37°C, loosen the cap and
  heat the solution at a higher temperature, up to 65°C, for
  \textasciitilde30 minutes, mixing periodically. Once the crystals have
  dissolved, store the solution in smaller aliquots in case crystals
  re-form after cooling. Warming the solution in this way does not
  affect performance of RNAlater Solution.
\end{itemize}

\hypertarget{water-sampling}{%
\chapter{Water Sampling}\label{water-sampling}}

The protocol for collecting water samples for the BioOce project

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

The purpose of taking water samples is to measure local environmental
variability relative to where oysters were sampled from The protocol
below is based on sampling from \textbf{three sites}, sampled
approximately once per month

\hypertarget{materials-3}{%
\section*{\texorpdfstring{\textbf{Materials}}{Materials}}\label{materials-3}}
\addcontentsline{toc}{section}{\textbf{Materials}}

\begin{itemize}
\item
  3 50 mL Falcon Tubes
\item
  3 glass bottles from the Ries Lab
\item
  A styrofoam cooler with ice
\item
  Digital thermometers or PreSens
\item
  Field notebook
\end{itemize}

\hypertarget{prep-1}{%
\section*{\texorpdfstring{\textbf{Prep}}{Prep}}\label{prep-1}}
\addcontentsline{toc}{section}{\textbf{Prep}}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Grab three glass bottles from the Ries Lab. You'll need card access
  for this. Grab the large bottles unless there aren't any clean
\item
  For the three glass bottles, you need glass stoppers with grease
  applied to them. Stoppers are in the plastic bottles
\item
  Gently wipe grease off glass stoppers with a paper towel, and put a
  conservative amount of new grease on them (the grease is expensive; it
  can be found in drawer to right of sink). Place into bottles and
  rotate
\item
  Make sure there is no resistance when rotating stoppers in bottles, if
  there is, add more grease
\item
  Label them with a sharpie or one of the EtOH proof pens (typically
  LOCATION and DATE go on each bottle: ex, IpswichLIV 2017-01-25)
\item
  Store all tubes on ice. Ice is in gel room
\end{enumerate}

\hypertarget{sampling}{%
\section*{\texorpdfstring{\textbf{Sampling}}{Sampling}}\label{sampling}}
\addcontentsline{toc}{section}{\textbf{Sampling}}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Start at Rowley SI site
\item
  Go into shallow water, about mid-calf in depth
\item
  Let temperature gauges sit freely in water while taking water samples
\item
  Rinse bottle three times with salt water, then gently fill the bottle
  by making opening parallel to water surface \textbf{Minimize the
  creation of bubbles when you are filling the glass bottles}
\item
  As bottle begins to fill, slowly submerge lower end to continue
  filling bottle \textbf{Leave about 1cm between the glass stopper and
  the the top of the water}
\item
  Repeat with falcon tube
\item
  Store bottles on ice between sites and when traveling back to the MSC
\end{enumerate}

\textbf{Repeat at each site}

\hypertarget{upon-returning-to-msc}{%
\section*{\texorpdfstring{\textbf{Upon returning to
MSC}}{Upon returning to MSC}}\label{upon-returning-to-msc}}
\addcontentsline{toc}{section}{\textbf{Upon returning to MSC}}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Poison glass bottle samples with mecuric chloride (This is located in
  the Ries lab fume hood on the left, for a small glass bottle it takes
  100ul or 200ul for larger bottles of the solution in the falcon tubes)
\item
  Place bottles in the Ries lab freezer with other samples \textbf{Be
  careful not to get mecuric chloride on the outside of the glass
  bottles or caps} (since we handle them without gloves when we
  eventually run them on the VINDTA) Therefore, \textbf{take out
  stoppers, use mecuric chloride, dispose of gloves, and restopper}
\end{enumerate}

\hypertarget{conductivity-loggers}{%
\chapter{Conductivity Loggers}\label{conductivity-loggers}}

The protocol for calibrating/initializing, deploying, and offloading
Data for conductivity loggers

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{initializing-loggers}{%
\section*{\texorpdfstring{\textbf{Initializing
Loggers}}{Initializing Loggers}}\label{initializing-loggers}}
\addcontentsline{toc}{section}{\textbf{Initializing Loggers}}

We use the
\href{http://www.onsetcomp.com/products/data-loggers/u24-002-c}{HOBO
Salt Water Conductivity/Salinity Data Logger} to measure local changes
in salinity, conductivity, and temperature where we sample oysters.

To initialize the logger:

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  First, make sure you have the
  \href{http://www.onsetcomp.com/products/communications/u-dtw-1}{Hobo
  Waterproof Shuttle} and appropriate coupler (Blue COUPLER2-C for
  U22/U24/U26/U20L)
\item
  Make sure you have HOBOWare downloaded to your computer
\item
  Attach conductivity logger to the shuttle with the coupler. Align the
  arrows between the logger and shuttle coupler for proper connection
\item
  Use miniUSB to connect the logger to the computer. Unscrew white cap
  on shuttle to access the cord port
\item
  Open HoboWare
\item
  In the bottom left bar of the program you will see that the software
  recognizes your device (`Dev: HOBO Waterproof Shuttle
  U-DTW-1\ldots{}'). It will 1 device connected in lower right bar.
  \textbf{This does not mean the shuttle has been properly started and
  connected}
\item
  To properly start and connect the logger press the coupler lever
\end{enumerate}

\includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/couplerlever.jpg}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  This starts the logger and you should see a `Installing device drive
  software' message on your screen
\item
  When the software has been installed, it should then say `USB Input
  Device'
\item
  The device name in the lower left bar of the HOBOWare program should
  now be correct
\item
  To get some basic information about the logger, go to `Device' in the
  menu bar, then choose `Status' This gives basic information about the
  logger such as the device name, serial number, battery state, etc. **
  You should use a Dixon standard (saltwater standard) to get an initial
  Conductivity reading form this Status pane before initializing**
\item
  To launch the logger, go to `Device' -\textgreater{} `Launch'
\item
  Change the name to something meaningful like
  `20180615\_Barnstable\_conductivity'
\item
  Choose the logging interval and when to start logging (when the logger
  will be deployed) \textbf{THIS IS VERY IMPORTANT: When you choose the
  logging interval, check the `Logs until' time in the pane above it to
  see how long it will take before the storage runs out. If your
  interval is too small, the logger will fill up before you can offload
  the data and continue measuring}
\item
  Click `Start' and logger will launch
\item
  Unplug from the computer
\item
  \textbf{Place rubber cap that comes with the logger on cylindrical end
  to prevent scratches}
\end{enumerate}

\hypertarget{deploying-loggers}{%
\section*{\texorpdfstring{\textbf{Deploying
loggers}}{Deploying loggers}}\label{deploying-loggers}}
\addcontentsline{toc}{section}{\textbf{Deploying loggers}}

\hypertarget{materials-youll-need-before-deploying}{%
\subsection*{\texorpdfstring{\textbf{Materials you'll need before
deploying}}{Materials you'll need before deploying}}\label{materials-youll-need-before-deploying}}
\addcontentsline{toc}{subsection}{\textbf{Materials you'll need before
deploying}}

\begin{itemize}
\item
  Cinderblock
\item
  Metal tubing straps to attach PVC fixed loggers to
\item
  Zipties
\item
  PVC pipe large and long enough to fit logger, ``swiss-cheesed' with
  holes \textbf{There must be enough holes for regular flow of water
  through the PVC pipe. The PVC pipe prevents large rocks and debris
  from damaging loggers}
\item
  Rope and buoy
\item
  Glass bottles and 50 mL Falcon tubes. See
  \href{https://bmford11.github.io/lotterhoslabprotocols/watersampling/}{this
  page} for water sampling protocol
\end{itemize}

\hypertarget{before-leaving-for-the-field}{%
\subsection*{\texorpdfstring{\textbf{Before leaving for the
field}}{Before leaving for the field}}\label{before-leaving-for-the-field}}
\addcontentsline{toc}{subsection}{\textbf{Before leaving for the field}}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Attach metal straps to cinder block
\item
  Attach PVC pipe to the metal rings that are fastened to the
  cinderblock
\item
  Label cinderblock, PVC pipe, buoy, and loggers with contact
  information (Katie's)
\item
  Make sure you've contacted local shellfish constable to make sure they
  can keep an eye on the loggers
\end{enumerate}

\hypertarget{deploying}{%
\subsection*{\texorpdfstring{\textbf{Deploying}}{Deploying}}\label{deploying}}
\addcontentsline{toc}{subsection}{\textbf{Deploying}}

Loggers should be deployed at low tide, so that we know they'll be
constantly exposed to water while logging

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Before deploying, attach logger(s) to PVC using many zipties
\item
  Wade to waist-deep water at site of deployment
\item
  Deploy logger
\item
  Record time of deployment. If loggers start logging before they are
  deployed they give strange readings. It is good to know when the
  logger was actually deployed, so we know when logging was actually
  initiated
\item
  Record location using GPS
\item
  Take water samples
\end{enumerate}

\hypertarget{last-steps}{%
\subsection*{\texorpdfstring{\textbf{Last
Steps}}{Last Steps}}\label{last-steps}}
\addcontentsline{toc}{subsection}{\textbf{Last Steps}}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Add the geographic location, deployment time, and additional notes
  about site to the log file (should be a google doc- ask Katie)
\item
  Make sure field sample in glass bottle has been poisoned with Mecuric
  Chloride (Ries Lab) and place in freezer
\end{enumerate}

\hypertarget{offloading-data}{%
\section*{\texorpdfstring{\textbf{Offloading
Data}}{Offloading Data}}\label{offloading-data}}
\addcontentsline{toc}{section}{\textbf{Offloading Data}}

Data from loggers should be off-loaded about every two weeks

\hypertarget{materials-youll-need-before-going-into-the-field-for-off-loading-data}{%
\subsection*{\texorpdfstring{\textbf{Materials you'll need before going
into the field for off-loading
data}}{Materials you'll need before going into the field for off-loading data}}\label{materials-youll-need-before-going-into-the-field-for-off-loading-data}}
\addcontentsline{toc}{subsection}{\textbf{Materials you'll need before
going into the field for off-loading data}}

\begin{itemize}
\item
  Glass bottle and 50mL Falcon tube fore water samples
\item
  HOBO Waterproof shuttle and correct coupler
\item
  Kimwipes
\item
  Wire cutters (to clip zipties)
\item
  Zipties- to reattached logger
\end{itemize}

\hypertarget{transferring-data-to-the-shuttle}{%
\subsection*{\texorpdfstring{\textbf{Transferring Data to the
Shuttle}}{Transferring Data to the Shuttle}}\label{transferring-data-to-the-shuttle}}
\addcontentsline{toc}{subsection}{\textbf{Transferring Data to the
Shuttle}}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Retrieve logger and \textbf{record the time you collected logger}
\item
  Detach logger from the cinderblock and make sure it's is completely
  dry before hooking up to shuttle and coupler
\item
  Remove rubber cap
\item
  Connect logger to shuttle with coupler \textbf{Make sure the shuttle
  cap to the USB port is tightly sealed when in the field to prevent any
  water damage}
\item
  Press coupler level, the yellow `Transfer' light should start blinking
  (if it doesn't, you probably didn't press the lever hard enough)
\item
  Once the transfer has completed the `OK' light will start blinking,
  which means it is ready for relaunch
\item
  The green light will blink for 15 minutes before relaunching.
  \textbf{If you want to relaunch immediately, press the coupler lever}
\item
  The green `OK' light should turn off, and the logger has been
  succesfully relaunched
\item
  Place the rubber cap back on the logger and reattach to the PVC pipe
\item
  Deploy
\item
  Record time of deployment
\item
  Take water samples \textbf{THIS IS IMPORTANT, as this data will be
  used to correct deviations for salinity data}
\end{enumerate}

\textbf{You do not do a calibration step before relaunching the salinity
loggers like you do for the pH loggers}

\hypertarget{offloading-the-data}{%
\subsection*{\texorpdfstring{\textbf{Offloading the
Data}}{Offloading the Data}}\label{offloading-the-data}}
\addcontentsline{toc}{subsection}{\textbf{Offloading the Data}}

**You need the HoboWare Pro software to properly offload data from the
shuttle. You can download the base version then get the license number
from the Helmuth Lab (Francis) which you enter by selecting `Manage
License Key' under Help

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Connect shuttle to computer with miniUSB cord. You \textbf{DO NOT}
  need to press coupler lever for offloading data from shuttle
\item
  Go to Device-\textgreater{} Manage Shuttle
\item
  A pane will come up with the device details for the shuttle. Any data
  that has not been offloaded will blink red. Select the data you want
  to offload by clicking the checkmark next to the file and click
  `Offload Checked'
\item
  Save by choosing a folder, \textbf{change filename} (especially if
  offloading from the same logger as it defaults to logger name), and
  click `Save Checked'
\item
  Once saved, the pane will return to the Files on Shuttle list
\item
  Click close and disconnect shuttle from computer
\end{enumerate}

\hypertarget{notes-3}{%
\subsection*{\texorpdfstring{\textbf{Notes}}{Notes}}\label{notes-3}}
\addcontentsline{toc}{subsection}{\textbf{Notes}}

To save space on the shuttle you can click `Delete Checked'. The shuttle
has 63 banks, but I'm not sure the size of these banks, so banks should
be deleted after they have been saved and backed-up so the memory does
not max out when you go to transfer data again

\hypertarget{ph-loggers}{%
\chapter{pH Loggers}\label{ph-loggers}}

The protocol for calibrating/initializing, deploying, and offloading
Data from the HOBO pH loggers

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

The pH loggers require the iPad with the special HOBO software

\hypertarget{initializing-loggers-1}{%
\section*{\texorpdfstring{\textbf{Initializing
Loggers}}{Initializing Loggers}}\label{initializing-loggers-1}}
\addcontentsline{toc}{section}{\textbf{Initializing Loggers}}

\hypertarget{calibrating-the-logger}{%
\subsection*{\texorpdfstring{\textbf{Calibrating the
Logger}}{Calibrating the Logger}}\label{calibrating-the-logger}}
\addcontentsline{toc}{subsection}{\textbf{Calibrating the Logger}}

To initialize the logger:

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  First, make sure you have the IPad the is used for the pH loggers
\item
  If the logger isn't on, turn the logger on
\item
  Under `IN RANGE' the logger should appear. Select the correct logger
\item
  If the logger has an expired calibration, you will be promoted to
  calibrate the logger. If it isn't expired, you should still calibrate
  before deploying by selecting `pH Calibration'
\item
  This will open a window with a very easy-to-follow guide on how to
  calibrate the logger. We typically do a three-point calibration, so
  choose `Three-Point' \textbf{Follow the important tips at the bottom
  of the page. Namely, make sure the copper tip has been removed before
  calibrating and make sure the temperature logger is fully submerged
  when calibrating}
\end{enumerate}

\textbf{Tip: Have the logger in the 7 pH buffer before clicking `Start
Calibration' and before clikcing `Next Buffer' for the 4 and 10 pH
buffers}

If you put it in the wrong buffer or click `Next Buffer' before putting
the logger in, you should be able to place the logger in the buffer and
it will properly stabilize

\hypertarget{configuring-and-launching-logger}{%
\subsection*{\texorpdfstring{\textbf{Configuring and Launching
Logger}}{Configuring and Launching Logger}}\label{configuring-and-launching-logger}}
\addcontentsline{toc}{subsection}{\textbf{Configuring and Launching
Logger}}

After the logger has been calibrated you can configure it and launch it.
To do so:

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Choose `Configure' under the `Actions' list
\item
  Some things I would change: Name (to something meaningful like
  20180615\_Barnstable\_pH), Logging Interval (typically 30 min), Start
  Logging (to the date and time it will be deployed, or `Now' if
  relaunching), Stop Logging (make sure it's on `When memory fills'),
  and make sure bluetooth is always on.
\item
  Click `Start' and the logger should launch at the date and time you
  set.
\end{enumerate}

\hypertarget{deploying-loggers-1}{%
\section*{\texorpdfstring{\textbf{Deploying
loggers}}{Deploying loggers}}\label{deploying-loggers-1}}
\addcontentsline{toc}{section}{\textbf{Deploying loggers}}

\hypertarget{materials-youll-need-before-deploying-1}{%
\subsection*{\texorpdfstring{\textbf{Materials you'll need before
deploying}}{Materials you'll need before deploying}}\label{materials-youll-need-before-deploying-1}}
\addcontentsline{toc}{subsection}{\textbf{Materials you'll need before
deploying}}

\begin{itemize}
\item
  Cinderblock
\item
  Metal tubing straps to attach PVC fixed loggers to
\item
  Zipties
\item
  PVC pipe large and long enough to fit logger, ``swiss-cheesed' with
  holes \textbf{There must be enough holes for regular flow of water
  through the PVC pipe. The PVC pipe prevents large rocks and debris
  from damaging loggers}
\item
  Rope and buoy
\item
  Glass bottles and 50 mL Falcon tubes. See
  \href{https://bmford11.github.io/lotterhoslabprotocols/watersampling/}{this
  page} for water sampling protocol
\end{itemize}

\hypertarget{before-leaving-for-the-field-1}{%
\subsection*{\texorpdfstring{\textbf{Before leaving for the
field}}{Before leaving for the field}}\label{before-leaving-for-the-field-1}}
\addcontentsline{toc}{subsection}{\textbf{Before leaving for the field}}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Attach metal straps to cinder block
\item
  Attach PVC pipe to the metal rings that are fastened to the
  cinderblock
\item
  Label cinderblock, PVC pipe, buoy, and loggers with contact
  information (Katie's)
\item
  Make sure you've contacted local shellfish constable to make sure they
  can keep an eye on the logger
\end{enumerate}

\hypertarget{deploying-1}{%
\subsection*{\texorpdfstring{\textbf{Deploying}}{Deploying}}\label{deploying-1}}
\addcontentsline{toc}{subsection}{\textbf{Deploying}}

Loggers should be deployed at low tide, so that we know they'll be
constantly exposed to water while logging

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Before deploying, attach logger(s) to PVC using many zipties
  \textbf{IMPORTANT: MAKE SURE THE COPPER GUARD HAS BEEN PLACED ON
  SENSOR BEFORE DEPLOYING}
\item
  Wade to waist-deep water at site of deployment
\item
  Deploy logger
\item
  Record time of deployment. If loggers start logging before they are
  deployed they give strange readings. It is good to know when the
  logger was actually deployed, so we know when logging was actually
  initiated
\item
  Record location using GPS
\item
  Take water samples
\end{enumerate}

\hypertarget{last-steps-1}{%
\subsection*{\texorpdfstring{\textbf{Last
Steps}}{Last Steps}}\label{last-steps-1}}
\addcontentsline{toc}{subsection}{\textbf{Last Steps}}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Add the geographic location, deployment time, and additional notes
  about site to the log file (should be a google doc- ask Katie)
\item
  Make sure field sample in glass bottle has been poisoned with Mecuric
  Chloride (Ries Lab) and place in freezer
\end{enumerate}

\hypertarget{offloading-data-1}{%
\section*{\texorpdfstring{\textbf{Offloading
Data}}{Offloading Data}}\label{offloading-data-1}}
\addcontentsline{toc}{section}{\textbf{Offloading Data}}

Data from loggers should be off-loaded about every two weeks

\hypertarget{materials-youll-need-before-going-into-the-field-for-off-loading-data-1}{%
\subsection*{\texorpdfstring{\textbf{Materials you'll need before going
into the field for off-loading
data}}{Materials you'll need before going into the field for off-loading data}}\label{materials-youll-need-before-going-into-the-field-for-off-loading-data-1}}
\addcontentsline{toc}{subsection}{\textbf{Materials you'll need before
going into the field for off-loading data}}

\begin{itemize}
\item
  Glass bottle and 50mL Falcon tube fore water samples
\item
  IPad with HOBO software
\item
  Kimwipes
\item
  Wire cutters (to clip zipties)
\item
  Zipties- to reattached logger
\end{itemize}

\hypertarget{transferring-data-to-the-ipad}{%
\subsection*{\texorpdfstring{\textbf{Transferring Data to the
iPad}}{Transferring Data to the iPad}}\label{transferring-data-to-the-ipad}}
\addcontentsline{toc}{subsection}{\textbf{Transferring Data to the
iPad}}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  Retrieve logger and \textbf{record the time you collected logger}
\end{enumerate}

\emph{pH protocols is not finished}

\hypertarget{managing-seatables}{%
\chapter{Managing Seatables}\label{managing-seatables}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

If you are using the seatables, please follow the following:

\hypertarget{daily-checklist}{%
\section*{\texorpdfstring{\textbf{Daily
Checklist}}{Daily Checklist}}\label{daily-checklist}}
\addcontentsline{toc}{section}{\textbf{Daily Checklist}}

\href{https://docs.google.com/spreadsheets/d/1oQnhrPk-GJYOvP48skjc4ehimgx4G3M1Ne0vrJx371c/edit\#gid=0}{Link
to Daily Checklist Table}

\begin{itemize}
\item
  Make sure freshwater is turned off at the end of the day. (This is a
  high pressure line and would flood the bunker if burst).
\item
  Check to make sure any heat sources in the tables are on top of
  silicone pads.
\item
  Check all plugs are working and the appropriate plugs are on emergency
  backup (red).
\item
  Make sure anything that could scratch the tables (oyster shells,
  ceiling tile) are on silicone pads.
\item
  Influent - check flow rate and position of all hoses. Pressure
  fluctuates with MSC-wide system use from day to day. Check that hoses
  are secure at the base.
\item
  Effluent float valve - Make sure the float valve is working. It should
  turn on when it is lifted up, and turn off when it falls down. If the
  float valve does not turn on, the sump tank will overflow and flood
  the bunker. If the float valve does not fall all the way down it will
  not turn off, which will burn out the pump. If the sump pump burns
  out, the sump tank will overflow and flood the bunker.
\item
  Effluent UV lights - If effluent is being treated, check that both the
  UV lights are working and the water is flowing in the correct
  direction.
\item
  Sea Table Drains - Make sure drains are clear of debris.
\item
  Effluent - write down the pressure on the valve before the blue filter
  and the valve after the blue filter.
\item
  Influent filtered water - write down the pressure on the valve before
  and after the filters on the incoming water.
\item
  Housekeeping - Make sure all lines are labeled (e.g.~unfiltered
  seawater, filtered seawater, municipal freshwater, etc.), all lab
  equipment is labeled ``Lotterhos'', and all equipment and tools are
  put out of sight (things tend to walk off).
\item
  Check for dead animals.
\end{itemize}

\hypertarget{weekly-checklist}{%
\section*{\texorpdfstring{\textbf{Weekly
Checklist}}{Weekly Checklist}}\label{weekly-checklist}}
\addcontentsline{toc}{section}{\textbf{Weekly Checklist}}

\href{https://docs.google.com/spreadsheets/d/1oQnhrPk-GJYOvP48skjc4ehimgx4G3M1Ne0vrJx371c/edit\#gid=1794014838}{Link
to Weekly Checklist Table}

\begin{itemize}
\item
  Effluent - Clean the sump in a bucket of freshwater. First, lift up
  the float valve and drain most of the salt water from the sump tank
  down to normal stop level. Fill a ``E'' bucket (be careful, some
  buckets have had chemicals and not for placement in sump) with
  freshwater. Unplug the sump pump, and carefully place it in the
  bucket. Plug it back in, and use the float valve to turn it on and
  hold it up until the pump runs through the bucket of freshwater. Then
  unplug the pump again, place it back in the tank, and plug it in. Test
  the float valve again as described in ``Daily checklist: Effluent
  float valve.''
\item
  Effluent - inspect sump pump hose for cracks, especially around where
  the hose clamps are. Make sure the hose clamps are tight.
\end{itemize}

\hypertarget{monthly-checklist}{%
\section*{\texorpdfstring{\textbf{Monthly
Checklist}}{Monthly Checklist}}\label{monthly-checklist}}
\addcontentsline{toc}{section}{\textbf{Monthly Checklist}}

\href{https://docs.google.com/spreadsheets/d/1oQnhrPk-GJYOvP48skjc4ehimgx4G3M1Ne0vrJx371c/edit\#gid=626032260}{Link
to Monthly Table}

\begin{itemize}
\item
  Use the shop vac to clean out the debris at the bottom of the sump.
\item
  Check for leaks along the main lines to the sea tables.
\item
  Check for leaks along the the freshwater line.
\item
  Check for leaks along the bottom pipe that runs below the sea tables
  to the sump pump
\item
  Check for leaks when the sump pump is running between the sump and the
  drainage pipe.
\item
  All motors (fans, pumps, etc), solenoids, and relays are inductive in
  nature. Do not plug these into power strips, plug them directly into
  the red receptacles. Check monthly.
\item
  Check the plugs for corrosion. Scrub with green abrasive pad.

  \begin{itemize}
  \tightlist
  \item
    \includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/UVpicts1.JPG}
  \end{itemize}
\end{itemize}

\hypertarget{unfiltered-seawater}{%
\subsection*{\texorpdfstring{\textbf{Unfiltered
Seawater}}{Unfiltered Seawater}}\label{unfiltered-seawater}}
\addcontentsline{toc}{subsection}{\textbf{Unfiltered Seawater}}

\begin{itemize}
\tightlist
\item
  Unfiltered seawater has a central connection to a PVC pipe that runs
  across all the tables.
\end{itemize}

\hypertarget{filtered-seawater}{%
\subsection*{\texorpdfstring{\textbf{Filtered
Seawater}}{Filtered Seawater}}\label{filtered-seawater}}
\addcontentsline{toc}{subsection}{\textbf{Filtered Seawater}}

\begin{itemize}
\tightlist
\item
  Filtered seawater is connected on the left side of the tables and
  currently only runs to the first table.
\end{itemize}

\hypertarget{changing-filters-on-influent}{%
\subsection*{\texorpdfstring{\textbf{Changing Filters on
Influent}}{Changing Filters on Influent}}\label{changing-filters-on-influent}}
\addcontentsline{toc}{subsection}{\textbf{Changing Filters on Influent}}

\begin{itemize}
\tightlist
\item
  TO DO.
\end{itemize}

\hypertarget{when-to-treat-effluent}{%
\section*{\texorpdfstring{\textbf{When to treat
effluent}}{When to treat effluent}}\label{when-to-treat-effluent}}
\addcontentsline{toc}{section}{\textbf{When to treat effluent}}

\begin{itemize}
\item
  See MSC Best Practices Document. Usually we treat effluent for any
  animals not collected from Nahant or the Massachusetts Bay area.
\item
  Make sure that the valve to the filters is open and the other valve is
  closed.
\item
  Make sure that both UV lights are turned ON.
\end{itemize}

\hypertarget{changing-filters-on-effluent}{%
\subsection*{\texorpdfstring{\textbf{Changing Filters on
effluent}}{Changing Filters on effluent}}\label{changing-filters-on-effluent}}
\addcontentsline{toc}{subsection}{\textbf{Changing Filters on effluent}}

\begin{itemize}
\tightlist
\item
  Changing the filter in the blue housing. Turn off the flow to the
  seatables, unplug the pump and UV lights. Close the valves. Open the
  housing and remove the filter. There is a replacement filter behind
  the blue housing. Grease the O-ring on the housing with silicon grease
  and change the filter. Set all the valves correctly, plug everything
  back in and turn everything back on. Check to make sure float valve is
  working, check for leaks in pipes and around filter housing.
\end{itemize}

\hypertarget{untreated-effluent}{%
\subsection*{\texorpdfstring{\textbf{Untreated
effluent}}{Untreated effluent}}\label{untreated-effluent}}
\addcontentsline{toc}{subsection}{\textbf{Untreated effluent}}

\begin{itemize}
\item
  Make sure that the valve to the filters is closed and the other valve
  is open.
\item
  Make sure the UV lights are unplugged.
\end{itemize}

\hypertarget{changing-the-uv-lights}{%
\section*{\texorpdfstring{\textbf{Changing the UV
lights}}{Changing the UV lights}}\label{changing-the-uv-lights}}
\addcontentsline{toc}{section}{\textbf{Changing the UV lights}}

Read the instructions!

\includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/UVpicts2.JPG}
\includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/UVpicts3.JPG}

\hypertarget{signs-there-is-a-problem}{%
\subsection*{\texorpdfstring{\textbf{Signs there is a
problem}}{Signs there is a problem}}\label{signs-there-is-a-problem}}
\addcontentsline{toc}{subsection}{\textbf{Signs there is a problem}}

If you see water in the quartz sleeve, this means it was not put
together correctly and it leaked.

\includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/UVpicts4.JPG}

Also look for signs that the bulb is burning out by the black residue
inside the quartz tube:

\includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/UVpicts10.JPG}

If the light is blue does that mean the UV light is working? Read the
instructions to find the answer to this question. The answer will
surprise you.

\hypertarget{changing-the-bulb}{%
\subsection*{\texorpdfstring{\textbf{Changing the
Bulb}}{Changing the Bulb}}\label{changing-the-bulb}}
\addcontentsline{toc}{subsection}{\textbf{Changing the Bulb}}

Make sure when you change the bulb, to follow the instructions and use
silicone grease. It can be found in a plastic box with a white top
labeled ``hot glue, silicone grease, etc.''

\includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/UVpicts6.JPG}

On the gasket (which should be changed once a year), make sure to put
some silicone grease on the inside and smoosh it around with your
finger:

\includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/UVpicts8.JPG}

Make sure to follow these instructions as you put it back together:

\includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/UVpicts9.JPG}

This is wrong:

\includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/UVpicts14.JPG}

Make sure when you put it back together, screw in the coupling tight
before attaching it to the housing:

\includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/UVpicts15.JPG}

It's hard to see, but check that the gasket is securely in place:

\includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/UVpicts16.JPG}

When you put the wires in, make sure not to let them get twisted by
carefully stuffing them inside,

\includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/UVpicts17.JPG}

then holding the top wire taught (don't let it twist) while you screw
the top on:

\includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/UVpicts18.JPG}

\hypertarget{annual-maintenance-for-uv-lights}{%
\subsection*{\texorpdfstring{\textbf{Annual maintenance for UV
lights}}{Annual maintenance for UV lights}}\label{annual-maintenance-for-uv-lights}}
\addcontentsline{toc}{subsection}{\textbf{Annual maintenance for UV
lights}}

\begin{itemize}
\item
  Replace bulbs. If the light is blue does that mean the UV light is
  working? Read the instructions to find the answer to this question.
  The answer will surprise you.
\item
  Replace gasket. (\#5 in instructions)
\item
  Replace O-ring. (\#8 in instructions)
\item
  Clean the glass tubing per instructions
\end{itemize}

\hypertarget{lab-kayak-care}{%
\chapter{Lab Kayak Care}\label{lab-kayak-care}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{kayak-checkout-list}{%
\section*{\texorpdfstring{\textbf{Kayak checkout
list}}{Kayak checkout list}}\label{kayak-checkout-list}}
\addcontentsline{toc}{section}{\textbf{Kayak checkout list}}

\begin{itemize}
\item
  Inflatable Columbia Kayak in backpack/bag
\item
  Quick-Release Fin with pin
\item
  2 Footrest
\item
  2 Splash-Guard inserts
\item
  Repair kit - THIS IS IN THE LAB ON THE SHELF IN THE BROWN ACCORDION
  FOLDER WITH THE KAYAK MANUAL
\item
  4 PVC Rod Holder Sleeves
\item
  4 Backstraps
\item
  1 throw rope
\item
  Emergency chemical sticks
\item
  2 life jackets
\item
  2 kayak paddles
\item
  emergency whistles
\item
  2 seats
\item
  pump with hose (I THINK THIS IS CURRENTLY BROKEN)
\end{itemize}

\hypertarget{kayak-predeparture-checks}{%
\section*{Kayak predeparture checks}\label{kayak-predeparture-checks}}
\addcontentsline{toc}{section}{Kayak predeparture checks}

\begin{itemize}
\item
  Read the
  \href{https://drk-lo.github.io/lotterhoslabprotocols/img/kayak_Aquaglide_Manual_2017.pdf}{Instructions}.
  You should be able to answer these questions. Which parts of the kayak
  do you inflate first? What position should the Boston valves be in
  when you inflate them? How do you deflate?
\item
  Make sure all the zippers are completely closed, with the zipper
  tucked under the fabric.
\end{itemize}

\includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/kayak_zipper2.JPG}

\begin{itemize}
\tightlist
\item
  Make sure the drain in the bow is closed tight!
\end{itemize}

\includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/kayak_drain_close2.JPG}

\begin{itemize}
\tightlist
\item
  The kayak may be muddy or sandy when you are done. Do your best to
  keep it clean, but be very careful NOT to get sand inside the zippers
  or inside the valves. After you open the valves to deflate the kayak,
  \emph{be sure to close them before you fold the kayak into the bag.}
  This will help keep sand out of the inside of the kayak.
\end{itemize}

\hypertarget{kayak-care-and-cleaning}{%
\section*{\texorpdfstring{\textbf{Kayak care and
cleaning}}{Kayak care and cleaning}}\label{kayak-care-and-cleaning}}
\addcontentsline{toc}{section}{\textbf{Kayak care and cleaning}}

\begin{itemize}
\item
  Inflate
\item
  Open the drain in the bow:
\end{itemize}

\includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/kayak_drain_close.JPG}

\begin{itemize}
\tightlist
\item
  Prop up the kayak for rinsing with the drain at the bottom:
\end{itemize}

\includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/kayak_drain.JPG}

\begin{itemize}
\item
  NEVER open the zippers!
\item
  Make sure kayak is completely dry before putting it up.
\end{itemize}

\hypertarget{oyster-sampling}{%
\chapter{Oyster Sampling}\label{oyster-sampling}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

This protocols page does not exist yet.

\hypertarget{field-safety-protocol}{%
\chapter{Field Safety Protocol}\label{field-safety-protocol}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\begin{itemize}
\item
  \href{https://drk-lo.github.io/lotterhoslabprotocols/field_fieldsafety_protocol/\#planning}{Planning}
\item
  \href{https://drk-lo.github.io/lotterhoslabprotocols/field_fieldsafety_protocol/\#training}{Training}
\item
  \href{https://drk-lo.github.io/lotterhoslabprotocols/field_fieldsafety_protocol/\#emergency-procedures}{Incident
  Response \& Reporting}
\end{itemize}

\hypertarget{planning}{%
\section*{\texorpdfstring{\textbf{Planning}}{Planning}}\label{planning}}
\addcontentsline{toc}{section}{\textbf{Planning}}

\begin{itemize}
\item
  Will you be travelling to another state?
\item
  Do you anticipate having phone signal
\item
  What are the local emergency numbers
\item
  Is anyone on the team first aid/ CPR certified?
\end{itemize}

\hypertarget{identify-appropriate-equipment-gear-and-first-aid-supplies}{%
\subsection*{\texorpdfstring{\textbf{Identify appropriate equipment,
gear and first aid
supplies}}{Identify appropriate equipment, gear and first aid supplies}}\label{identify-appropriate-equipment-gear-and-first-aid-supplies}}
\addcontentsline{toc}{subsection}{\textbf{Identify appropriate
equipment, gear and first aid supplies}}

\begin{itemize}
\item
  First Aid Kits
\item
  Jumper cables in vehicles
\item
  Charged cell phone; satellite phone if you anticipate patchy or no
  service; extra battery or charger
\item
  Map, compass, GPS
\item
  extra food/snacks
\item
  Hats, sunscreen, sunglasses
\item
  Appropriate footwear and clothing, layers
\item
  Flashlight or headlamp
\item
  Signal/mirror, whistle
\item
  Knife or multi-tool
\end{itemize}

\hypertarget{training}{%
\section*{\texorpdfstring{\textbf{Training}}{Training}}\label{training}}
\addcontentsline{toc}{section}{\textbf{Training}}

\begin{itemize}
\item
  Title IX training
\item
  First Aid
\item
  CPR
\item
  \href{https://nupd.northeastern.edu/our-services/safety-training/}{NU
  Self Defense Class}
\item
  https://greaterbostonsafetytraining.org/
\end{itemize}

\hypertarget{emergency-procedures}{%
\subsection*{\texorpdfstring{\textbf{Emergency
procedures}}{Emergency procedures}}\label{emergency-procedures}}
\addcontentsline{toc}{subsection}{\textbf{Emergency procedures}}

In case of an emergency while in the field:

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Assess the situation (do not put yourself in danger)
\item
  Contact the emergency officials if needed to be prepared
\item
  Address the medical emergency to the extent possible (e.g., first aid,
  control bleeding,CPR)

  \begin{itemize}
  \item
    \href{https://en.wikipedia.org/wiki/ABC_(medicine)}{ABC} (Airway,
    Breathing, Circulation)
  \item
    if you suspect a spinal injury do not move the victim unless they
    are in imminent danger.
  \item
    If bleeding, apply pressure to control bleeding
  \end{itemize}
\item
  Transport victim to nearest emergency facility
\end{enumerate}

\hypertarget{emergency-contact-numbers}{%
\subsubsection*{\texorpdfstring{\textbf{Emergency contact
numbers}}{Emergency contact numbers}}\label{emergency-contact-numbers}}
\addcontentsline{toc}{subsubsection}{\textbf{Emergency contact numbers}}

\textbar\textbar{} Phone number\textbar{}
\textbar---\textbar---\textbar{}
\textbar\textbar{}\textbf{911}\textbar{} \textbar Katie Lotterhos
(PI)\textbar\textbar{} \textbar Remy Gatins \textbar(805) 704
8877\textbar{} \textbar Madeline Eppley\textbar\textbar{}
\textbar{}\href{https://provost.northeastern.edu/international-travel/health-safety/emergency-assistance/}{Northeastern
Travel Support Network} (for International Emergencies)\textbar+1.857.
214.5332\textbar{} \textbar Title IX Northeastern
University\textbar\textbar{} \textbar Title IX \textbar\textbar{}

\hypertarget{other-resources}{%
\section*{\texorpdfstring{\textbf{Other
Resources}}{Other Resources}}\label{other-resources}}
\addcontentsline{toc}{section}{\textbf{Other Resources}}

\begin{itemize}
\tightlist
\item
  \href{https://www.ucop.edu/safety-and-loss-prevention/_files/field-research-safety/uc-field-research-safety-manual.pdf}{UC
  Field Research Safety}
\end{itemize}

\part{Coding Protocols}

\hypertarget{learning-to-code}{%
\chapter{Learning to Code}\label{learning-to-code}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{options-for-downloading-programs}{%
\section{Options for downloading
programs}\label{options-for-downloading-programs}}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  You can download and install on your computer
\item
  You can use NU's
  \href{https://drk-lo.github.io/lotterhoslabprotocols/code_LearningToCode/ood.discovery.neu.edu}{Open
  On Demand} for R Studio, command line, and python, and more!

  \begin{itemize}
  \item
    To use OOD, you have to have an NU email, and have Discovery access
    (NU's HPC cluster).

    \begin{itemize}
    \tightlist
    \item
      \href{https://drk-lo.github.io/lotterhoslabprotocols/discovery_nonNEUpersonnel/}{Instructions
      to get Discovery access for non-NU people}
    \end{itemize}
  \item
    Once you have Discovery Access, you can follow
    \href{https://docs.google.com/document/d/1sJAnQXE8-r_4ssr5VfVv0oZKFrrrRFxspdxx2kNdaLI/edit\#heading=h.61692qeq0wnf}{these
    intructions}, but instead of the \texttt{course} dropdown menu, use
    the \texttt{Interactive\ Apps} dropdown to find the program you need
  \end{itemize}
\end{enumerate}

\hypertarget{learning-r}{%
\section*{\texorpdfstring{\textbf{Learning
R}}{Learning R}}\label{learning-r}}
\addcontentsline{toc}{section}{\textbf{Learning R}}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  \href{https://docs.google.com/document/d/1-qm-gVOIb-k61PxlQvav_GZXgmZP4t7q1fjs4BN_Ov8/edit\#heading=h.jl4d8sgkpjrk}{Follow
  the instructions for downloading R and R studio in this syllabus (if
  you scroll down)}
\item
  Start with this
  \href{https://rafalab.github.io/dsbook/getting-started.html}{great
  introduction to R and R studio} and data science with R. Start with
  the first 3 chapters.
\item
  After you get a handle on R, start using \texttt{swirl}. It's an
  interactive learning environment
  \href{https://swirlstats.com/students.html}{Visit this site for an
  overview of swirl}
  \href{https://github.com/swirldev/swirl_courses\#swirl-courses}{Then
  visit this site for an overview of courses}
\end{enumerate}

I recommend starting with the ``R programming E'' course

\begin{verbatim}
library(swirl)
install_course("R programming E")
swirl()
\end{verbatim}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  After you finish the above steps, it's time to review what you know
  with your advisor and discuss the best way to build your skills based
  on your goals. Next steps may include:

  \begin{itemize}
  \item
    \href{https://swirlstats.com/scn/title.html}{Other swirl courses}
  \item
    \href{https://rafalab.github.io/dsbook/getting-started.html}{More
    advanced chapters in R for data science}
  \item
    \href{https://carpentries.org/}{Software carpentry and Data
    Carpentry courses} Taking a course is great, but the course material
    is public and it's easy to learn from
  \end{itemize}
\end{enumerate}

\hypertarget{learning-the-command-line}{%
\chapter*{\texorpdfstring{\textbf{Learning the command
line}}{Learning the command line}}\label{learning-the-command-line}}
\addcontentsline{toc}{chapter}{\textbf{Learning the command line}}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  The first hurdle might be what kind of computer you have. If you have
  a Mac, use the Terminal. If you have a PC, google ``use the command
  line on a PC''. The command line functions (and how well they work)
  might vary a bit among platforms.
\item
  Get started with Software Carpentry's
  \href{https://swcarpentry.github.io/shell-novice/}{The Unix Shell}
\item
  I also like the book \texttt{Practical\ Computing\ for\ Biologists} (a
  book you have to buy)
\end{enumerate}

\hypertarget{learning-python}{%
\section*{\texorpdfstring{\textbf{Learning
Python}}{Learning Python}}\label{learning-python}}
\addcontentsline{toc}{section}{\textbf{Learning Python}}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  On a mac, I recommend downloading
  \href{https://www.anaconda.com/}{Anaconda}. I don't know about running
  python on a PC.
\item
  This free python textbook is the best
  https://python-textbok.readthedocs.io/en/1.0/Python\_Basics.html
\item
  This is a great resource for learning python, but the courses are not
  free:
  (https://pythonforbiologists.com/){[}https://pythonforbiologists.com/{]}

  \begin{itemize}
  \item
    In the menu, under training courses, there is a lot of options
  \item
    I bought the book ``Introduction to Python for Biologists'' and went
    through it myself
  \end{itemize}
\item
  Here are some
  \href{https://wiki.python.org/moin/BeginnersGuide/Programmers}{resources
  I have yet to vet}
\end{enumerate}

\hypertarget{learning-bioinformatics}{%
\section*{\texorpdfstring{\textbf{Learning
Bioinformatics}}{Learning Bioinformatics}}\label{learning-bioinformatics}}
\addcontentsline{toc}{section}{\textbf{Learning Bioinformatics}}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  Vince Buffalo's \href{https://vincebuffalo.com/book/}{Bioinformatics
  Data Skills}
\end{enumerate}

\hypertarget{learning-machine-learning}{%
\section*{\texorpdfstring{\textbf{Learning Machine
Learning}}{Learning Machine Learning}}\label{learning-machine-learning}}
\addcontentsline{toc}{section}{\textbf{Learning Machine Learning}}

https://christophm.github.io/interpretable-ml-book/

https://www.fast.ai/

https://course.fast.ai/start\_colab

\hypertarget{learning-slim}{%
\section*{\texorpdfstring{\textbf{Learning
SLiM}}{Learning SLiM}}\label{learning-slim}}
\addcontentsline{toc}{section}{\textbf{Learning SLiM}}

http://benhaller.com/workshops/workshops.html

\hypertarget{advanced-concepts}{%
\section*{\texorpdfstring{\textbf{Advanced
concepts}}{Advanced concepts}}\label{advanced-concepts}}
\addcontentsline{toc}{section}{\textbf{Advanced concepts}}

\href{https://distill.pub/}{Interactive vignettes for machine learning
and stats}

\href{https://gephi.org/}{Making beautiful networks}

\hypertarget{making-maps-in-r}{%
\section*{\texorpdfstring{\textbf{Making Maps in
R}}{Making Maps in R}}\label{making-maps-in-r}}
\addcontentsline{toc}{section}{\textbf{Making Maps in R}}

\href{https://datacarpentry.org/organization-geospatial/}{Data Carpentry
Geospatial Materials} Introduction to raster and vector data

\href{https://mikkovihtakari.github.io/ggOceanMaps/articles/ggOceanMaps.html\#advanced-use-1}{GG
ocean maps} beautiful ocean maps.

\begin{itemize}
\tightlist
\item
  For shallow bathymetries, the default is not great. Scroll down to the
  advanced data and see ``bathymetries'' and how to set the depths you
  want. You save the raster file, then you have to do the same thing for
  your land data.
\end{itemize}

\href{https://rpubs.com/kvistj/mapping}{Basic intro to maps in R}

\href{https://www.molecularecologist.com/2016/07/01/making-maps-in-r-volume-2-ggplots/}{GG
plot maps - molecular ecologist}

\hypertarget{other}{%
\section*{\texorpdfstring{\textbf{Other}}{Other}}\label{other}}
\addcontentsline{toc}{section}{\textbf{Other}}

https://github.com/thais-neu/Thais-learning-code

\hypertarget{bioinformatics-resources}{%
\chapter{Bioinformatics Resources}\label{bioinformatics-resources}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{data-skills}{%
\section*{\texorpdfstring{\textbf{Data
Skills}}{Data Skills}}\label{data-skills}}
\addcontentsline{toc}{section}{\textbf{Data Skills}}

Vince Buffalo's \emph{Bioinformatic Data Skills} is a great book to
build skills. Katie has a set of great readings for learning, contact
her for more info.

\hypertarget{rad-seq}{%
\section*{\texorpdfstring{\textbf{RAD seq}}{RAD seq}}\label{rad-seq}}
\addcontentsline{toc}{section}{\textbf{RAD seq}}

https://radcamp.github.io/NYC2019/RADCamp-PartII-Day1-AM.html

https://github.com/dereneaton/ipyrad/blob/master/newdocs/assembly\_guidelines.rst

http://www.popgen.dk/software/index.php/PCAngsd handle genotype
likelihoods

\hypertarget{vcf-files-and-population-structure}{%
\section*{\texorpdfstring{\textbf{VCF files and population
structure}}{VCF files and population structure}}\label{vcf-files-and-population-structure}}
\addcontentsline{toc}{section}{\textbf{VCF files and population
structure}}

What is a VCF? https://samtools.github.io/hts-specs/VCFv4.2.pdf

Katie also has some lectures on VCF files.

\hypertarget{filter-vcf-table-and-produce-a-pca-plot}{%
\subsection*{\texorpdfstring{\textbf{Filter VCF table and produce a PCA
plot}}{Filter VCF table and produce a PCA plot}}\label{filter-vcf-table-and-produce-a-pca-plot}}
\addcontentsline{toc}{subsection}{\textbf{Filter VCF table and produce a
PCA plot}}

When getting started, it's best to filter for relatedness and linkage
disequilibrium, so that you have a quasi-independent set of individuals
and SNPs.

Check related individuals with vcf using vcftools (--relatedness)

Filter the VCF file. Use VCF tools to do this:
http://vcftools.sourceforge.net/

\begin{itemize}
\item
  for minor alleles \textless{} 0.05 (if you have a lot of SNPs, or 0.01
  if you don't have a lot of SNPs)
\item
  for missing data at SNPs (keep SNPs in at least X\% of individuals in
  all dataset)
\item
  for missing data at population level (keeps SNPs with calls in X\% or
  higher of individuals per population)
\item
  for individuals with low SNP count (keep individuals with higher than
  X\% of SNPs called)
\item
  Depending on the context, X\% might be 80\% or might be 100\%
\end{itemize}

To use the bigsnpr R package, you will have to change your data from the
VCF format to a genotype matrix with 0/1/2 counts of the alternate
allele. \href{https://github.com/laurabenestan/RDA_outlier}{Transform
from vcf to (plink to) raw}. See the first steps in this tutorial.

The presence of LD can bias principal components. This
\href{https://academic.oup.com/bioinformatics/article/36/16/4449/5838185?login=true}{paper}
explains the problem. Read about
\href{https://privefl.github.io/bigsnpr/articles/pruning-vs-clumping.html}{filtering
for LD with pruning and clumping}

\href{https://privefl.github.io/bigsnpr/articles/how-to-PCA.html}{Filter
the raw data using snp\_autoSVD (it's at the end of the tutorial)} This
tutorial also teaches you how to do a PCA.

\hypertarget{structure-plot}{%
\subsection*{\texorpdfstring{\textbf{Structure
plot}}{Structure plot}}\label{structure-plot}}
\addcontentsline{toc}{subsection}{\textbf{Structure plot}}

For Structure use LEA program sNMF package vcf input in R.
(http://membres-timc.imag.fr/Olivier.Francois/LEA/tutorial.htm)

\hypertarget{run-an-rda}{%
\subsection*{\texorpdfstring{\textbf{Run an
RDA}}{Run an RDA}}\label{run-an-rda}}
\addcontentsline{toc}{subsection}{\textbf{Run an RDA}}

Before you run an RDA and do some of the things in these tutorials, talk
to Dr.~L about your plan. Outlier SNP analysis with RDA have a lot of
false positives, but the individual scores are very useful.

https://popgen.nescent.org/2018-03-27\_RDA\_GEA.html

https://github.com/laurabenestan/RDA\_outlier

Katie also has some RDA code on simulations and oyster data.

\hypertarget{run-a-gwas-or-gea}{%
\subsection*{\texorpdfstring{\textbf{Run a GWAS or
GEA}}{Run a GWAS or GEA}}\label{run-a-gwas-or-gea}}
\addcontentsline{toc}{subsection}{\textbf{Run a GWAS or GEA}}

Genotype-phenotype association: LFMM2.
\href{https://academic.oup.com/mbe/article/36/4/852/5290100}{Read the
paper lfmm2} and
\href{https://onlinelibrary.wiley.com/doi/10.1111/1755-0998.13366}{LEA3}

\href{https://rdrr.io/bioc/LEA/man/lfmm2.html}{LFMM2 functions}

\href{http://www.bioconductor.org/packages/3.3/bioc/html/LEA.html}{LEA
installation} Check with Katie this is correct before running

\hypertarget{other-resources-1}{%
\subsection*{\texorpdfstring{\textbf{Other
resources}}{Other resources}}\label{other-resources-1}}
\addcontentsline{toc}{subsection}{\textbf{Other resources}}

\href{https://marineomics.github.io/index.html}{Marine Omics working
group}

\hypertarget{version-control-with-github}{%
\chapter{Version Control with
GitHub}\label{version-control-with-github}}

The protocol for keeping track of different versions using GitHub

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{what-is-version-control}{%
\section*{\texorpdfstring{\textbf{What is version
control?}}{What is version control?}}\label{what-is-version-control}}
\addcontentsline{toc}{section}{\textbf{What is version control?}}

In running bioinformatics pipelines and analyzing data, a user often
makes hundreds of changes to his/her code and ends up with multiple
versions of the same file with slight differences. Using version control
allows you to keep track of different versions as you progress through
your work. \href{https://github.com/}{GitHub} is a great platform to
track the changes in your work, share with collaborators, and have a
backup online.

\textbf{Read more about using git in the terminal in Vince Buffalo's
textbook}

\hypertarget{how-it-is-used-in-the-lab}{%
\section*{\texorpdfstring{\textbf{How it is used in the
lab\ldots{}}}{How it is used in the lab\ldots{}}}\label{how-it-is-used-in-the-lab}}
\addcontentsline{toc}{section}{\textbf{How it is used in the
lab\ldots{}}}

Most major projects in the lab will have their own git repository. If
you are starting a project that will have a lot of coding involved, you
should discuss with Katie about starting a new repository and how you
will manage data, code, and directories. The easiest way to utilize
GitHub is to do the following:

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Make sure you have GitHub Desktop downloaded (you can use the
  Terminal, but this is much more user-friendly, especially for
  beginners)
\item
  Log in to your GitHub account (or create one if you don't have one)
\item
  Find the repository on GitHub that was created by Katie
\item
  On the main repository page you should see a green button towards the
  right of the screen that says `Clone or download'. Click this button
\item
  You can either copy the https and use \texttt{git\ clone\ https...} in
  the directory that you want the git repository, or you can choose
  `Open in Desktop' which opens up the GitHub Desktop app, and allows
  you to choose which directory it will be cloned to.
\item
  You can then make changes on your local directory
\item
  After you have made sufficient changes that you want to save and
  share, you need to commit these changes
\item
  GitHub Desktop will keep track of these changes, which will appear in
  the left column of the app (make sure you've specified the right
  directory in the top navigation bar)
\item
  To commit these changes, select the changes you want to commit, then
  type a message in the summary box and click `Commit to
  \textless repository\textgreater{}'
\end{enumerate}

\textbf{It's good practice to commit groups of changes separately
depending on what you changed. For example, if you edited three text
documents and added an image to the directory, you could commit the text
documents with the summary `edited text documents' then separately
commit the image with the summary `added image'. This makes it so mass
changes aren't all assigned the same summary message}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Then in the upper right corner, on the black navigation bar, it should
  give you the option to push these commits to the main repository on
  GitHub. Do so by clicking on the box.
\item
  If changes have been made on the online GitHub repository, you will
  need to `pull' these before pushing your commits (essentially, making
  sure you have the most up-to-date versions of files before making
  changes). You do this by choosing `pull' in the upper right hand box,
  and then you can `push' your changes.
\end{enumerate}

\hypertarget{general-directory-organization}{%
\chapter{General Directory
Organization}\label{general-directory-organization}}

The protocol for maintaining consistency when creating directories

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

For each new project you'll want to have a dedicated folder to contain
all major data files and code script related to that project. Even if
you are re-using scripts that are already located in another project's
folder, consider copying that code over into your new projects' folder,
instead of reading it from another location.

The guiding principle should be that you want to be able to run your
whole project's code (barring any software you have to download) using
the scripts that are located within that projects' folder. That way, if
you ever have to move your project files to another computer, you won't
be missing any scripts on which you're dependent.

So once you've made a new project folder, for example by using the
\texttt{mkdir} command in the terminal

\begin{verbatim}
> mkdir NewProject
\end{verbatim}

you'll want to navigate into that folder (\texttt{cd\ NewProject}) and
create subdirectories to house specific file types.

Two good startig folders to have are a \texttt{src} folder and a
\texttt{results} folder. An \texttt{src} folder (derived from
``source''), should house all the core scripts necessary for your
project. So all novel R scripts, Python code, Perl pipelines you've
dredged up, \emph{etc}. would be housed here. The \texttt{results}
folder would house all output files generated by scripts in
\texttt{src}.

Depending on the project, you'll want to have a \texttt{data} folder to
keep the raw data sets that you yourself did not generate with the
scripts in \texttt{src}.

Finally, you'll want to create a simple text README file that explicitly
states what subdirectories you have in your project folder, and what
they should contain. Ideally you should have a README file in each
directory that states what files it contains.

A README file written in
\href{https://www.markdownguide.org/cheat-sheet}{markdown} could look
like this:

\begin{verbatim}
# Files for New Project by Student McGee and PI ForFree, Year Month(s)

Short description of New Project

## Description of each folder or file and the data contained within:
  src - What scripts are here?
  results - What kind of result files are here?
  data - What sort of data?
\end{verbatim}

Once you've set up your project folder, it should look something like
this: \includegraphics{https://drk-lo.github.io/img/eg_structure.png}

Quick notes: simfiles folder results folder src folder

\hypertarget{basic-unix-commands}{%
\chapter{Basic Unix Commands}\label{basic-unix-commands}}

A resources for useful lines when coding

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{table-of-contents-1}{%
\section*{\texorpdfstring{\textbf{Table of
Contents}}{Table of Contents}}\label{table-of-contents-1}}
\addcontentsline{toc}{section}{\textbf{Table of Contents}}

1)
\href{https://drk-lo.github.io/lotterhoslabprotocols/code_unixcommands/\#cd}{\texttt{cd}
- Changing Directories}\\
2)
\href{https://drk-lo.github.io/lotterhoslabprotocols/code_unixcommands/\#find}{\texttt{find}
- Finding files}\\
3)
\href{https://drk-lo.github.io/lotterhoslabprotocols/code_unixcommands/\#ls}{\texttt{ls}
- Listing files}\\
4)
\href{https://drk-lo.github.io/lotterhoslabprotocols/code_unixcommands/\#sed}{\texttt{sed}
- Replacing texts in a file name}\\
5)
\href{https://drk-lo.github.io/lotterhoslabprotocols/code_unixcommands/\#chmod}{\texttt{chmod}-
Changing file or folder permissions}\\
6)
\href{https://drk-lo.github.io/lotterhoslabprotocols/code_unixcommands/\#wc}{\texttt{wc}
- Count words or lines or a file}\\
7)
\href{https://drk-lo.github.io/lotterhoslabprotocols/code_unixcommands/\#awk}{\texttt{awk}
- Printing specific lines or columns}

\hypertarget{change-directory---cd}{%
\section*{\texorpdfstring{\textbf{Change directory -
\texttt{cd}}}{Change directory - cd}}\label{change-directory---cd}}
\addcontentsline{toc}{section}{\textbf{Change directory - \texttt{cd}}}

\textbf{Description} - Command allows you to change your directory.

Simple directory change:

\begin{verbatim}
cd /path/to/desired/directory
\end{verbatim}

Additional ways to change your current directory:

\begin{verbatim}
# One level up in the path
cd ..
# Can be done more than one to move mutiple levels
cd ../../..
# Provided with no path information it will change directory to the base directory
cd
\end{verbatim}

\textbf{Changing directory with click and drag method}

\begin{figure}

{\centering \includegraphics{https://drk-lo.github.io/img/quickdirectory.gif}

}

\caption{Quick Directory Access}

\end{figure}

\hypertarget{find-target-files---find}{%
\section*{\texorpdfstring{\textbf{Find target files -
\texttt{find}}}{Find target files - find}}\label{find-target-files---find}}
\addcontentsline{toc}{section}{\textbf{Find target files -
\texttt{find}}}

\textbf{Description} - The \texttt{find} command in UNIX is a command
line utility for walking a file hierarchy. It can be used to find files
and directories and perform subsequent operations on them. It supports
searching by file, folder, name, creation date, modification date, owner
and permissions. By using the \texttt{-\ exec} other UNIX commands can
be executed on files or folders found. (from
https://shapeshed.com/unix-find/)

\begin{itemize}
\item
  Additional Information

  \begin{itemize}
  \item
    \href{https://shapeshed.com/unix-find/}{Extended Tutorial}
  \item
    \href{https://linux.die.net/man/1/find}{\texttt{find} manual}
  \end{itemize}
\end{itemize}

\textbf{Simple Find}

\begin{verbatim}
find ./directory/path/ -name FileExample.txt
\end{verbatim}

This will with find file \texttt{./directory/path/FileExample.txt}

\textbf{Remove files created before a certain date with find}

\begin{verbatim}
find /home/br.ford/br.ford_remote/slim/results/ -ctime +14 -print -exec rm {} \;
\end{verbatim}

This finds all files in the \texttt{.../results/} directory that are
older than 14 days and removes them

You can also be more specific. For example, if you want to remove only
txt files older than two days:

\begin{verbatim}
find /home/br.ford/br.ford_remote/slim/results/*.txt -ctime +2 -print -exec mv {} finished_results \;
\end{verbatim}

\hypertarget{listing-files-and-directories---ls}{%
\section*{\texorpdfstring{\textbf{Listing files and directories -
\texttt{ls}}}{Listing files and directories - ls}}\label{listing-files-and-directories---ls}}
\addcontentsline{toc}{section}{\textbf{Listing files and directories -
\texttt{ls}}}

\textbf{List just subdirectories in the current directory}

\begin{verbatim}
ls -d */
\end{verbatim}

\hypertarget{replacing-characters-in-a-txt-file--sed}{%
\section*{\texorpdfstring{\textbf{Replacing characters in a txt file
-\texttt{sed}}}{Replacing characters in a txt file -sed}}\label{replacing-characters-in-a-txt-file--sed}}
\addcontentsline{toc}{section}{\textbf{Replacing characters in a txt
file -\texttt{sed}}}

\begin{verbatim}
sed -i '' 's/["]//g' test.txt
\end{verbatim}

The -i argument says to use in-place editing (edit and return file
without making another copy), the empty quotes identifies extension of
backup file (none, unless you want a backup file), the statement in the
quotes specifies: to substitute (s/) quotations ({[}``{]}) with nothing
(//) globally (g') in test.txt

\hypertarget{changing-permission---chmod}{%
\section*{\texorpdfstring{\textbf{Changing Permission -
\texttt{chmod}}}{Changing Permission - chmod}}\label{changing-permission---chmod}}
\addcontentsline{toc}{section}{\textbf{Changing Permission -
\texttt{chmod}}}

\textbf{Want to change permissions to a directory and all
subdirectories, but are not sure which \texttt{chmod} arguments to use
to do so recursively?}

Use mac GUI:
\includegraphics{https://drk-lo.github.io/img/changepermissions.gif}

\hypertarget{counting-words-or-lines---wc}{%
\section*{\texorpdfstring{\textbf{Counting words or lines -
\texttt{wc}}}{Counting words or lines - wc}}\label{counting-words-or-lines---wc}}
\addcontentsline{toc}{section}{\textbf{Counting words or lines -
\texttt{wc}}}

\begin{verbatim}
wc -l myfile.txt
\end{verbatim}

\hypertarget{selecting-a-specific-lines-or-columns-from-a-files---awk}{%
\section*{\texorpdfstring{\textbf{Selecting a specific line(s) or
column(s) from a file(s) -
\texttt{awk}}}{Selecting a specific line(s) or column(s) from a file(s) - awk}}\label{selecting-a-specific-lines-or-columns-from-a-files---awk}}
\addcontentsline{toc}{section}{\textbf{Selecting a specific line(s) or
column(s) from a file(s) - \texttt{awk}}}

\texttt{awk\ \textquotesingle{}FNR\ ==\ 11\ \{print\ FILENAME,\ \$0\}\textquotesingle{}\ NAMEOFFILE.txt}

\begin{itemize}
\item
  \texttt{FNR} : Line number you want to print
\item
  \texttt{print\ FILENAME} : built in \texttt{awk} variable that prints
  the file name (not needed but nice if you are printing lines from
  multiple files into a single log file).
\item
  \texttt{\$0} : Tells \texttt{awk} to print the whole line.
\item
  \texttt{NAMEOFFILE.txt} : The specific file you want to print
\end{itemize}

\textbf{Example}

Print out the 11th line from a file.

Command

\begin{verbatim}
downey-wall.a@comp5[20190719_fastqc_trim_10bp_Cvirginica_MBD]# awk 'FNR == 11 {print FILENAME $0}' 17005*R1*.txt 
\end{verbatim}

Output

\begin{verbatim}
17005_DNAm_R1.fastq.gz_trimming_report.txtAdapter sequence: 'CTGTCTCTTATA' (Nextera Transposase sequence; auto-detected)
\end{verbatim}

\textbf{Code Extensions}

Multiple single lines with OR (\texttt{\textbar{}\textbar{}} operation)

\begin{verbatim}
awk 'FNR==5 || FNR==9 {$0}' NAMEOFFILE.txt
\end{verbatim}

A range of lines with AND (\texttt{\&\&} operation)

\begin{verbatim}
awk 'FNR >= 5 && FNR <= 10 {$0}' NAMEOFFILE.txt
\end{verbatim}

Print from multiple files

\begin{verbatim}
awk 'FNR==11 {$0}' *.txt
\end{verbatim}

\textbf{Skipping lines}

\begin{verbatim}
awk 'NR % 2 {print} !(NR % 2) && /pattern/ {print}' FILENAME.txt
\end{verbatim}

\textbf{Counting the columns}

\begin{verbatim}
awk -F'\t' '{print NF; exit}' myfile.txt
\end{verbatim}

-F'\textbackslash t' tells awk that your columns are tab delimited, you
can switch this for another character depending on your dataset. For
example, if the data is comma separated just use -F',' instead. To print
the column count for every line instead of just the first one, leave out
`exit'.

\hypertarget{still-editting-below-this-point-possibly-moving-to-new-file-since-this-is-not-a-basic-unix-command}{%
\subsubsection*{\texorpdfstring{\textbf{Still editting below this point
(possibly moving to new file since this is not a basic unix
command)}}{Still editting below this point (possibly moving to new file since this is not a basic unix command)}}\label{still-editting-below-this-point-possibly-moving-to-new-file-since-this-is-not-a-basic-unix-command}}
\addcontentsline{toc}{subsubsection}{\textbf{Still editting below this
point (possibly moving to new file since this is not a basic unix
command)}}

\textbf{Want to rename lots of files at once?}

Say you have a directory with the following images:

pic1.png

pic2.png

pic3.png

\ldots{}

pic27.png

You decide that they should have a more descriptive name, so you want to
change them to `oyster\_pic1.png', `oyster\_pic2.png',
`oyster\_pic3.png'\ldots.

You don't have to move them individually, just use the bash command
`rename' with a regular expression!

\begin{verbatim}
$ rename 's/pic/oyster_pic/g' *
\end{verbatim}

\begin{itemize}
\item
  `s' at the begining tells the command to \textbf{s}ubstitute a string
  with another string
\item
  /pic/ is the string you want to replace
\item
  /oyster\_pic/ is the string to replace it with

  \begin{figure}

  {\centering 

  }

  \caption{Close document tab}

  \end{figure}
\item
  `g' is the \textbf{g}lobal flag -- this means replace every instance
  of the 1st string with the 2nd. Ex, pic1pic.png would become
  oyster1oyster.png. This may or may not be desirable depending on what
  you want to do.
\item
  * after the regex just tells rename what to rename - in this case
  everything in the directory
\end{itemize}

The rename command uses perl regular expressions - you can do a lot with
these, google perl regex if you want to learn more.

\textbf{On MacOS you may have to install rename before you can do this:
\texttt{\$\ brew\ install\ rename}}

\hypertarget{regular-expressions}{%
\chapter{Regular Expressions}\label{regular-expressions}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

This protocols page does not exist yet.

\hypertarget{setting-alias-or-path-variable}{%
\chapter{Setting Alias or PATH
variable}\label{setting-alias-or-path-variable}}

Basics of altering the `\textasciitilde/.bashrc` or
`\textasciitilde/.bash\_profile` file

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{reason}{%
\section*{\texorpdfstring{\textbf{Reason}}{Reason}}\label{reason}}
\addcontentsline{toc}{section}{\textbf{Reason}}

Often times you may need to run a command from a downloaded piece of
software, but would like to avoid the inconvience of including the
entire path as part of the command. This can be avoided by updating you
\texttt{\textasciitilde{}/.bashrc} or
\texttt{\textasciitilde{}/.bash\_profile} with either an alias for a
particular pathway/file or by amending you \$PATH variable to include
the specific folder in which your command is located. The later may be
particularly useful if the software contains multiple commands you need
to call.

\hypertarget{disclaimer}{%
\section*{\texorpdfstring{\textbf{Disclaimer}}{Disclaimer}}\label{disclaimer}}
\addcontentsline{toc}{section}{\textbf{Disclaimer}}

\begin{itemize}
\tightlist
\item
  This tutorial is intended for those that are using bash shell, so if
  you are working with another shell (i.e.~zsl) these specific files we
  are modifying will be different.
\end{itemize}

\hypertarget{creating-an-alias}{%
\section*{\texorpdfstring{\textbf{Creating an
Alias}}{Creating an Alias}}\label{creating-an-alias}}
\addcontentsline{toc}{section}{\textbf{Creating an Alias}}

Open your \texttt{bashrc} (or \texttt{bash\_profile}) file

\begin{verbatim}
nano ./bash_profile
\end{verbatim}

Add an alias

\begin{verbatim}
alias new_command="/absolute/path/to/actual/command.sh"
\end{verbatim}

Source your bash file (or restart terminal) to make the changes active

\begin{verbatim}
source(~/.bash_profile)
\end{verbatim}

\hypertarget{updating-your-path-variable}{%
\section*{\texorpdfstring{\textbf{Updating your \texttt{\$PATH}
variable}}{Updating your \$PATH variable}}\label{updating-your-path-variable}}
\addcontentsline{toc}{section}{\textbf{Updating your \texttt{\$PATH}
variable}}

Open your \texttt{bashrc} (or \texttt{bash\_profile}) file

\begin{verbatim}
nano ./bash_profile
\end{verbatim}

Add to existing items of \texttt{\$PATH}

\begin{verbatim}
PATH=$PATH:/path/to/folderWith/Commands
\end{verbatim}

\begin{itemize}
\tightlist
\item
  Generally good idea to add this line after where you see other
  \texttt{\$PATH} variable entries, but before the line that exports the
  entries (i.e.~\texttt{export\ PATH})
\end{itemize}

Source your bash file (or restart terminal) to make the changes active

\begin{verbatim}
source(~/.bash_profile)
\end{verbatim}

\hypertarget{example-of-a-bash_profile-file}{%
\section*{\texorpdfstring{\textbf{Example of a \texttt{bash\_profile}
file}}{Example of a bash\_profile file}}\label{example-of-a-bash_profile-file}}
\addcontentsline{toc}{section}{\textbf{Example of a
\texttt{bash\_profile} file}}

\begin{verbatim}
# .bash_profile

# Get the aliases and functions
if [ -f ~/.bashrc ]; then
        . ~/.bashrc
fi

# User specific environment and startup programs

PATH=/shared_lab/scripts:/shared_lab/scripts/bin:$PATH:$HOME/.local/bin
PATH=$PATH:/shared_lab/20180226_RNAseq_2017OAExp/RNA/scripts/salmon_scripts/
PATH=$PATH:/shared_lab/20180226_RNAseq_2017OAExp/RNA/scripts/software/RSEM/RSEM-1.3.1
PATH=$PATH:/shared_lab/20180226_RNAseq_2017OAExp/RNA/scripts/software/gffread
PATH=$PATH:$HOME/downey-wall.a_remote/scripts
PATH=$PATH:/shared_lab/20180226_RNAseq_2017OAExp/RNA/scripts/software/gffcompare
PATH=$PATH:/shared_lab/20180226_RNAseq_2017OAExp/RNA/scripts/STAR_scripts
PATH=$PATH:/home/downey-wall.a/software/hisat2-2.1.0
PATH=$PATH:/home/downey-wall.a/software/Bismark-0.22.1
PATH=$PATH:/home/downey-wall.a/software/TrimGalore-0.6.0
PATH=$PATH:/shared_lab/20180226_RNAseq_2017OAExp/DNAm/scripts
PATH=$PATH:/usr/local/programs/R/3.6.0/bin
PATH=$PATH:/home/downey-wall.a/software/bowtie2-2.3.5.1-linux-x86_64
PATH=$PATH:/home/downey-wall.a/downey-wall.a_remote/software
export PATH

# Aliases
alias python='/home/downey-wall.a/python_3.5.2/bin/python3.5'
\end{verbatim}

\hypertarget{setting-seeds-for-reproducibility}{%
\chapter{Setting Seeds for
Reproducibility}\label{setting-seeds-for-reproducibility}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

Any time you call a random varible or random sample (e.g.~\texttt{rnorm}
or \texttt{sample} in R), your code will not be reproductible unless you
set a seed. sets the starting number used to generate a sequence of
random numbers -- it ensures that you get the same result if you start
with that same seed each time you run the same process.

The function \texttt{set.seed} can make your code reproducible, but you
have to call it on the line before you ask R to do anything random.

Try:

\begin{verbatim}
set.seed(98234)
rnorm(100)
set.seed(98234)
rnorm(100)
\end{verbatim}

And compare it to:

\begin{verbatim}
set.seed(98234)
rnorm(100)
rnorm(100)
\end{verbatim}

If you are running a large number of replicate simulations, it's
important to make them reproducible. A good practice is to think about
what parameters you are going to model and make a giant table. Here is
an example for a simulation with three replicates of two different sets
of parameters:

\begin{longtable}[]{@{}llll@{}}
\toprule()
\textbf{Seed} & \textbf{Rep} & \textbf{Param 1} & \textbf{Param 2} \\
\midrule()
\endhead
98234 & 1 & 0.5 & 1.2 \\
98235 & 2 & 0.5 & 1.2 \\
98236 & 3 & 0.5 & 1.2 \\
98237 & 1 & 0.9 & 1.5 \\
98238 & 2 & 0.9 & 1.5 \\
98239 & 3 & 0.9 & 1.5 \\
\bottomrule()
\end{longtable}

As you loop through this table, you can use the Seed as the
\texttt{mySeed} for your simulations. However, there are bad and good
ways to take the seed from this table and incorporate it into your
simulations.

\hypertarget{bad-practice}{%
\subsection*{\texorpdfstring{\textbf{Bad
Practice:}}{Bad Practice:}}\label{bad-practice}}
\addcontentsline{toc}{subsection}{\textbf{Bad Practice:}}

\begin{verbatim}
set.seed(mySeed)
a<-rnorm(100)
set.seed(mySeed)
b<-rnorm(100)
set.seed(mySeed)
c<-rnorm(100)
\end{verbatim}

Try this in R. \emph{What values do you get for a, b, and c?}

\hypertarget{good-practice}{%
\subsection*{\texorpdfstring{\textbf{Good
practice:}}{Good practice:}}\label{good-practice}}
\addcontentsline{toc}{subsection}{\textbf{Good practice:}}

\begin{verbatim}
set.seed(mySeed)
sim_seeds <- round(runif(100)*10000000) 
  # may need more than 100 if there are more than 
  # 100 places in your sim where you set seeds
set.seed(sim_seeds[1])
a<-rnorm(100)
set.seed(sim_seeds[2])
b<-rnorm(100)
set.seed(sim_seeds[3])
c<-rnorm(100)
\end{verbatim}

\hypertarget{transferring-files-md5}{%
\chapter{Transferring Files md5}\label{transferring-files-md5}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

We recommend using GLOBUS (add link) to tranfer files. However, if you
use another file tranfer mechanism, use md5sums to make sure your files
transferred successfully.

\hypertarget{how-to-ensure-the-files-you-are-transferring-are-transferred-successfully}{%
\section*{\texorpdfstring{\textbf{How to ensure the files you are
transferring are transferred
successfully}}{How to ensure the files you are transferring are transferred successfully}}\label{how-to-ensure-the-files-you-are-transferring-are-transferred-successfully}}
\addcontentsline{toc}{section}{\textbf{How to ensure the files you are
transferring are transferred successfully}}

\hypertarget{step-1-make-a-list-of-md5sums-of-your-files}{%
\subsection*{\texorpdfstring{\textbf{Step 1: Make a list of md5sums of
your
files}}{Step 1: Make a list of md5sums of your files}}\label{step-1-make-a-list-of-md5sums-of-your-files}}
\addcontentsline{toc}{subsection}{\textbf{Step 1: Make a list of md5sums
of your files}}

If md5 sums provided by the sequencing facility in a file:
\texttt{md5\_sum.txt} (on mac is md5, on unix md5sum)

If not: \texttt{md5sum\ (list\ files)\ \textgreater{}\ md5sum.txt}

\hypertarget{step-2-transfer-files-to-new-location-including-the-md5sum.txt}{%
\subsection*{\texorpdfstring{\textbf{Step 2: Transfer files to new
location (including the
md5sum.txt)}}{Step 2: Transfer files to new location (including the md5sum.txt)}}\label{step-2-transfer-files-to-new-location-including-the-md5sum.txt}}
\addcontentsline{toc}{subsection}{\textbf{Step 2: Transfer files to new
location (including the md5sum.txt)}}

change directory to that location

Then, create an md5sum of the transferred files:

\texttt{md5sum\ (list\ files\ in\ new\ location)\ \textgreater{}\ md5sum\_new.txt}

\hypertarget{step-3-compare-md5sum-from-before-the-transfer-to-md5sum-after-the-transfer}{%
\subsection*{\texorpdfstring{\textbf{Step 3: Compare md5sum from before
the transfer to md5sum after the
transfer}}{Step 3: Compare md5sum from before the transfer to md5sum after the transfer}}\label{step-3-compare-md5sum-from-before-the-transfer-to-md5sum-after-the-transfer}}
\addcontentsline{toc}{subsection}{\textbf{Step 3: Compare md5sum from
before the transfer to md5sum after the transfer}}

Compare sorted files FILE1 and FILE2 line by line:

\texttt{comm\ -3\ md5sum.txt\ md5sum\_new.txt}

With no options, produce three-column output. Column one contains lines
unique to FILE1, column two contains lines unique to FILE2, and column
three contains lines common to both files. -1 suppress lines unique to
FILE1 -2 suppress lines unique to FILE2 -3 suppress lines that appear in
both files

If the last command produces no output, the two files are identical.

\hypertarget{troubleshooting-cellprofiler-installation}{%
\chapter{Troubleshooting CellProfiler
Installation}\label{troubleshooting-cellprofiler-installation}}

Steps to take if errors occur during installation

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

When initially trying to use CellProfiler, I had some issues during the
installation process. This was primarily due to the fact that
CellProfiler only uses Python 2.7 (They are developing a new version
that will be built on Python 3) and is fairly sensitive to versions of
libraries it needs. Also, I was having some issues with permissions as
well. I will walk through these issues here.

\hypertarget{pay-attention-to-pip-version}{%
\section*{\texorpdfstring{\textbf{Pay attention to pip
version}}{Pay attention to pip version}}\label{pay-attention-to-pip-version}}
\addcontentsline{toc}{section}{\textbf{Pay attention to pip version}}

One subtle mistake that person could make during the installation
process is not using the correct version of pip. Given that there are
separate versions of pip for Python 2.7 and for Python 3, you could just
be mistakenly trying to install CellProfiler with the wrong version of
pip. I say this because the command install CellProfiler is this:

\begin{verbatim}
user@comp5:~/Cellprofiler$ pip install -e .
\end{verbatim}

Seems pretty straightforward but that assumption could lead to some
trouble. The \texttt{pip} command could be specifying either pip for
python 2.7 or some version of python3. There should be a \texttt{pip2.7}
command to specify which of the two versions you are trying to use. Use
this command. If you do not do this, you will likely get an error saying
that you do not have the correct version of Python and you will be
confused.

This also applies to prerequisite libraries you are installing as well.

\hypertarget{permission-errors}{%
\section*{\texorpdfstring{\textbf{Permission
Errors}}{Permission Errors}}\label{permission-errors}}
\addcontentsline{toc}{section}{\textbf{Permission Errors}}

Another issue that you might run into when installing is permission
errors. Here is what I got when I ran the command for installation
suggested by CellProfiler documentation:

\begin{verbatim}
user@comp5:~/Cellprofiler$ pip2.7 install -e .

ERROR: Could not install packages due to an EnvironmentError: [Errno 13] Permission denied: '/usr/bin/cellprofiler'
Consider using the `--user` option or check the permissions.
\end{verbatim}

You can try to run it with the \texttt{-\/-user} flag as suggested but
this is what I got when I did that:

\begin{verbatim}
user@comp5:~/Cellprofiler$ pip2.7 install -e . --user

ERROR: Could not install packages due to an EnvironmentError: [Errno 13] Permission denied: '/usr/bin/cellprofiler'
Check the permissions.
\end{verbatim}

The only way that I was able to get by these permission errors was to
install it using sudo:

\begin{verbatim}
user@comp5:~/Cellprofiler$ sudo pip2.7 install -e .
\end{verbatim}

\hypertarget{libraries-having-been-installed-incorrectly-or-in-an-incompatible-manner}{%
\section*{\texorpdfstring{\textbf{Libraries having been installed
incorrectly or in an incompatible
manner}}{Libraries having been installed incorrectly or in an incompatible manner}}\label{libraries-having-been-installed-incorrectly-or-in-an-incompatible-manner}}
\addcontentsline{toc}{section}{\textbf{Libraries having been installed
incorrectly or in an incompatible manner}}

Now this is where a installation complication lies. Proper installation
of certain libraries and packages is incredibly important on RedHat
systems and it really matters here. If you do not so this you will get
an error that looks like this:

\begin{verbatim}
Notice: /Stage[main]/Neutron::Db::Sync/Exec[neutron-db-sync]/returns: ImportError: cannot import name UnrewindableBodyError Error: /Stage[main]/Neutron::Db::Sync/Exec[neutron-db-sync]: Failed to call refresh: neutron-db-manage upgrade heads returned 1 instead of one of [0] Error: /Stage[main]/Neutron::Db::Sync/Exec[neutron-db-sync]: neutron-db-manage upgrade heads returned 1 instead of one of [0] Notice: /Stage[main]/Neutron::Deps/Anchor[neutron::service::begin]: Triggered 'refresh' from 1 events Error: Failed to apply catalog: Execution of '/usr/bin/nova-manage cell_v2 list_cells --verbose' returned 1: Traceback (most recent call last): File "/usr/bin/nova-manage", line 6, in <module> from nova.cmd.manage import main File "/usr/lib/python2.7/site-packages/nova/cmd/manage.py", line 77, in <module> from nova.api.ec2 import ec2utils File "/usr/lib/python2.7/site-packages/nova/api/ec2/ec2utils.py", line 25, in <module> from nova import cache_utils File "/usr/lib/python2.7/site-packages/nova/cache_utils.py", line 22, in <module> import nova.conf File "/usr/lib/python2.7/site-packages/nova/conf/__init__.py", line 27, in <module> from nova.conf import cinder File "/usr/lib/python2.7/site-packages/nova/conf/cinder.py", line 16, in <module> from keystoneauth1 import loading as ks_loading File "/usr/lib/python2.7/site-packages/keystoneauth1/loading/__init__.py", line 13, in <module> from keystoneauth1.loading import adapter File "/usr/lib/python2.7/site-packages/keystoneauth1/loading/adapter.py", line 13, in <module> from keystoneauth1 import adapter File "/usr/lib/python2.7/site-packages/keystoneauth1/adapter.py", line 18, in <module> from keystoneauth1 import session File "/usr/lib/python2.7/site-packages/keystoneauth1/session.py", line 26, in <module> import requests File "/usr/lib/python2.7/site-packages/requests/__init__.py", line 43, in <module> import urllib3 File "/usr/lib/python2.7/site-packages/urllib3/__init__.py", line 8, in <module> from .connectionpool import ( File "/usr/lib/python2.7/site-packages/urllib3/connectionpool.py", line 35, in <module> from .connection import ( File "/usr/lib/python2.7/site-packages/urllib3/connection.py", line 44, in <module> from .util.ssl_ import ( File "/usr/lib/python2.7/site-packages/urllib3/util/__init__.py", line 4, in <module> from .request import make_headers File "/usr/lib/python2.7/site-packages/urllib3/util/request.py", line 5, in <module> from ..exceptions import UnrewindableBodyError ImportError: cannot import name UnrewindableBodyError
\end{verbatim}

If you have \texttt{urllib3} installed via pip, and \texttt{requests}
installed via yum you will have this issue, even if you have the correct
packages installed. The same will apply if you have \texttt{urllib3}
installed via yum, and \texttt{requests} installed via pip. Here's what
I did to fix the issue (\textbf{Again, Make sure that you are installing
for the correct version of Python}):

\begin{verbatim}
sudo pip2.7 uninstall requests

sudo pip2.7 uninstall urllib3

sudo yum remove python-urllib3

sudo yum remove python-requests

# confirm that all those libraries have been removed

sudo yum install python-urllib3

sudo yum install python-requests
\end{verbatim}

Just be aware that this will only work for systems that are running
Fedora, Redhat, or CentOS.

\hypertarget{accessing-and-using-defiance-cluster}{%
\chapter{Accessing and Using Defiance
Cluster}\label{accessing-and-using-defiance-cluster}}

Notes on how to access and use the defiance cluster

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{note-we-no-longer-use-this-resource.}{%
\section{\texorpdfstring{\textbf{Note we no longer use this
resource.}}{Note we no longer use this resource.}}\label{note-we-no-longer-use-this-resource.}}

To use the defiance cluster you need to contact Stefan (graduate student
in Vollmer lab; S.Kaluziak@northeastern.edu) and request access

You'll need to send Stefan your public key. Public keys can be created
following the steps here:
\url{https://confluence.atlassian.com/bitbucketserver/creating-ssh-keys-776639788.html}
And is described in
\includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/RSAKeys.pdf}

Ask Stefan to give you access to \texttt{/shared\_lab} directory

Stefan has pre-installed a lot of packages on compliance.
\href{https://drk-lo.github.io/installedpackages.md}{Click here} for a
complete list, including version numbers.

If you require additional programs while using cluster, you'll need to
ask Stefan to install it. E-mail him the software and version, and he
should have it installed within 24 hours.

\hypertarget{description-of-defiance}{%
\section{\texorpdfstring{\textbf{Description of
defiance}}{Description of defiance}}\label{description-of-defiance}}

Defiance is set up with a main node (where you log in) and compute nodes
Not all files/folder map perfectly across the main node and compute
nodes Each user has their own remote repository that is private

\hypertarget{accessing-cluster}{%
\section{\texorpdfstring{\textbf{Accessing
Cluster}}{Accessing Cluster}}\label{accessing-cluster}}

\textbf{After Gaining Access from Stefan} Log onto cluster by opening
Terminal and typing: \texttt{ssh\ username@defiance.neu.edu}
(e.g.~\texttt{ssh\ br.ford@defiance.neu.edu}) You'll then log onto a
computing node. The basic outline of computing nodes is below:

\texttt{comp5} (Lotterhos) \texttt{comp6} \texttt{comp7} (Bowen)
\texttt{comp8}

Log onto comp5 by typing: \texttt{ssh\ comp5}

\textbf{You can only access the \texttt{/shared\_lab} directory by
logging on to one of the computing nodes (comp5)}

To transfer files from your computer to a defiance remote directory:
First, go to the folder on your computer with the file in it. Then
transfer the file(s) into a remote folder
\texttt{my\_laptop\textbackslash{}\$\ scp\ ./my\_file\ br.ford@defiance.neu.edu:/home/br.ford/br.ford\_remote/my\_folder}

Or to download to your personal computer from defiance, go to the folder
on your computer where you want the file, then:
\texttt{my\_laptop\textbackslash{}\$\ scp\ br.ford@defiance.neu.edu:/home/br.ford/br.ford\_remote/my\_folder/my\_file\ .}

To download the file to a different directory than the one you are
currently in, simply replace the `.' at the end of the command with the
path to that directory.

\texttt{/shared\_lab} cannot be accessed with \texttt{scp}, so you'll
need to move (\texttt{mv}) things to your remote repository before
downloading to your personal computer

\hypertarget{checking-and-managing-resource-usage-on-defiance}{%
\section{\texorpdfstring{\textbf{Checking and managing resource usage on
defiance}}{Checking and managing resource usage on defiance}}\label{checking-and-managing-resource-usage-on-defiance}}

Defiance is a shared resource, so it is important to monitor the
resources being used in order to avoid overloading the server. Before
beginning any resource-intensive analysis, you should log into comp5 and
enter the command \texttt{htop} to bring up the following interface:

\begin{figure}

{\centering \includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/img/htop_screenshot.png}

}

\caption{htop example}

\end{figure}

One way to immediately get a sense of whether too many resources are
being used is to look at the load average, circled in red in this image.
These 3 numbers represent the CPU load over the past minute, 5 minutes,
and 15 minutes (from left to right). The higher the load average and the
longer it has been high, the more concerned you should be. A spike in
the load average for the past minute may be ok, but a sustained high
load for the past 15 minutes means some processes need to be killed
before the server crashes. How high is too high? A common rule of thumb
is 0.70 x total number of CPUs. Since we have 72 CPUs on comp5, if you
see the \textbf{load above or approaching 50.0 you should wait for other
jobs to finish} before starting any new ones.

There is also a lot more information htop gives you that you can take
into account. The top portion of the screen shows a real-time
representation of how much load is on each of the 72 CPUs. In this image
you can see that 2.0\% of the total capacity of CPU number 38 is being
used and none of the others are in use. Directly below the CPU grid is a
bar showing the memory usage. 2.92 of 252 GB of memory is being used in
this image. The bottom half of the htop display shows a list of
processes currently running, the user they belong to, and information
about the resources they are using.

If you want to check on your jobs only, you can use \texttt{htop\ -u}
followed by your username (ex: \texttt{htop\ -u\ freeman.k}) to show
only jobs started by your username. This can be useful to check how long
your jobs have been running, to make sure they still are running and
haven't been terminated, or to see how much of the server's capacity you
are using.

If you would like to run a resource-intensive job without slowing down
other jobs on the server, you can use the command \texttt{nice} when you
start the job.
\texttt{nice\ -n\ 19\ \textless{}your\_program\textgreater{}} will start
your program with a ``niceness'' value of 19, meaning it will defer to
all other processes when competing for CPU resources. The default
niceness value is 0, 19 gives the lowest priority, and -20 gives the
highest priority. Only the root user can run processes with negative
niceness values.

\hypertarget{pseudo-parallelize-your-code-on-the-cluster}{%
\chapter{Pseudo-Parallelize Your Code on the
Cluster}\label{pseudo-parallelize-your-code-on-the-cluster}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

Defiance has 72 (check) cores per comp node. Generally we like to leave
a few available, so can parallelize code across up to 68 cores at once.

For genomics, we're often doing tests for many loci. Depending on the
test, it may be faster to run each locus in serial on a core, and spread
out different parts of the data across several cores (as opposed to
running each locus in serial across all cores in parallel).

To accomplish this, you will need a bash script and an R script.
Examples are below.

The bash script takes in the total number of tests you want to do
(number of loci or number of rows in a data frame, for instance), and
then calculates how many should be run in serial on each core. It then
runs through a loop. In each iteration of the loop (the total number of
iterations should be equal to the number of cores), the bash script
calls the R code and sends information about which rows in the dataframe
are to be run in the R code. Each iteration of the loop calls R in the
background on a different core.

Within the R code, another loop would be necessary to conduct the test
for each row in the dataframe (not shown).

\hypertarget{bash-script-runscripts.sh}{%
\section*{\texorpdfstring{\textbf{Bash script
runScripts.sh}}{Bash script runScripts.sh}}\label{bash-script-runscripts.sh}}
\addcontentsline{toc}{section}{\textbf{Bash script runScripts.sh}}

Save this script as \texttt{runRscripts.sh}

\begin{verbatim}
#!/bin/bash
set -e
set -u
set -o pipefail


## Use these two lines of code if you have a conda environmnent you'd like to activate
## and list what's in that environment
# source activate msprime-env
# conda list

mypath="/XXX" # set your path

cd $mypath

nrows=280000 # number of loci or rows to be tested. Improve this code by
            # actually calculating it from the relevant dataframe with `wc`
nlocipercore=$((nrows/67)) # the 67 comes from the fact we want to pseudoparallel over 68 cores
echo nlocipercore $nlocipercore
start=$(seq 1 $nlocipercore $nrows)
echo $start

for i in $start
do
    echo -e "\n\n"
    end=$((i+nlocipercore-1))
    
    if [ $end -gt $nrows ]
    then
        end=$nrows
    fi
    
    ## Use this code if you need to name an outfile here
    ## You may also name it inside your R script
    #outfile=${i}"_Invers_out.txt"
    #outpath=${mypath}"/results/"
    #outpathfile=${outpath}${outfile}
    
    ### Call R script
    echo "Running R scripts for lines" echo 'start row' $i 'end row' $end
    
    Rscript --vanilla ../src/MyScript.R ${i} $end > ${i}"_R.out" 2> ${i}"_R.error" & echo $!
    
    # the above line, using the $! at the end, will run the script in the background on a single core
    # in the next iteration, it will run the script in the background on another core for a different set of rows
    # do not wait until the last background process is finished
    
done
\end{verbatim}

\hypertarget{corresponding-myscript.r}{%
\section*{\texorpdfstring{\textbf{Corresponding
MyScript.R}}{Corresponding MyScript.R}}\label{corresponding-myscript.r}}
\addcontentsline{toc}{section}{\textbf{Corresponding MyScript.R}}

MyScript.R should be located in the \texttt{src/MyScript.R} working
directory. If it is located in a different place, than that needs to be
specified in the bash script above.

\begin{verbatim}
# install libraries

args = commandArgs(trailingOnly=TRUE)

# test if there is at least one argument: if not, return an error
if (length(args)==0) {
  stop("At least one argument must be supplied (input file).n", call.=FALSE)
} 

start_row = args[1]
end_row = args[2]

print(c("start_row", "end_row"))
# Here you would call a loop or apply() function through these indexes of your data and output the results
\end{verbatim}

Once you have these two scripts set up and working on a subset of data,
you can just run the bash script like this

\begin{verbatim}
chmod +x runRscripts.sh
./runRscripts.sh
\end{verbatim}

\hypertarget{github-figures}{%
\chapter{GitHub Figures}\label{github-figures}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{integrating-table-of-contents-and-equations}{%
\section{Integrating table of contents and
equations}\label{integrating-table-of-contents-and-equations}}

Here we discuss adding a table of contents and integrated equation into
your \texttt{github\_document} markdown. These can be nice features when
trying to make an organized markdown file for a github repository.
\textbf{Note}, I believe the table of contents syntax will work for an
\texttt{html\_document}, but the equations solution is specific to the
\texttt{github\_document} file markdown type.

\hypertarget{table-of-contents-2}{%
\subsection*{\texorpdfstring{\textbf{Table of
Contents}}{Table of Contents}}\label{table-of-contents-2}}
\addcontentsline{toc}{subsection}{\textbf{Table of Contents}}

Code: \texttt{toc:\ True}

Optionally, you can specify how many levels you want in your table of
contents using - \texttt{toc\_depth:\ 3}

\hypertarget{equations}{%
\subsection*{\texorpdfstring{\textbf{Equations}}{Equations}}\label{equations}}
\addcontentsline{toc}{subsection}{\textbf{Equations}}

Code: \texttt{pandoc\_args:\ "-\/-webtex"}

\hypertarget{simple-example-header}{%
\subsection*{\texorpdfstring{\textbf{Simple example
header}}{Simple example header}}\label{simple-example-header}}
\addcontentsline{toc}{subsection}{\textbf{Simple example header}}

\begin{verbatim}
---
title: "Github_markdown_template"
output: 
  github_document:
    toc: true
    toc_depth: 3
    pandoc_args: "--webtex"
---
\end{verbatim}

\hypertarget{template-file}{%
\subsection*{\texorpdfstring{\textbf{Template
File}}{Template File}}\label{template-file}}
\addcontentsline{toc}{subsection}{\textbf{Template File}}

\href{https://github.com/DrK-Lo/lotterhoslabprotocols/blob/gh-pages/_data/github_markdown_template.md}{Markdown}
or
\href{https://github.com/DrK-Lo/lotterhoslabprotocols/blob/gh-pages/_data/github_markdown_template.Rmd}{Rmd}

\part{Discovery}

\hypertarget{connecting-to-the-discovery-cluster}{%
\chapter{Connecting to the Discovery
Cluster}\label{connecting-to-the-discovery-cluster}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

Intro to Discovery (how to set up and run SLURM jobs ) tutorial video:
https://rc.northeastern.edu/support/training/

\begin{itemize}
\item
  Discovery account. Apply by opening a Service-Now ticket with an
  access request at
  https://rc-docs.northeastern.edu/en/latest/welcome/gettinghelp.html\#submit-a-ticket
\item
  Login by typing e.g.~\texttt{ssh\ lotterhos@login.discovery.neu.edu},
  but use your own username
\item
  Read
  \href{https://cpb-us-w2.wpmucdn.com/express.northeastern.edu/dist/1/43/files/2019/08/GettingStartedGuide-1.pdf}{NU's
  Getting started on Discovery} (this link is outdated, need to find new
  location)
\item
  Check out \href{https://rc-docs.northeastern.edu/en/latest/}{NU
  Discovery research computing docs}
\item
  Need help? Create a ticket automatically by emailing
  rchelp@northeastern.edu

  \begin{itemize}
  \tightlist
  \item
    RC bookings
    https://outlook.office365.com/owa/calendar/ResearchComputing2@northeastern.onmicrosoft.com/bookings/
  \end{itemize}
\end{itemize}

\hypertarget{software-on-discovery-for-lotterhos-lab}{%
\section*{\texorpdfstring{\textbf{Software on Discovery for Lotterhos
Lab}}{Software on Discovery for Lotterhos Lab}}\label{software-on-discovery-for-lotterhos-lab}}
\addcontentsline{toc}{section}{\textbf{Software on Discovery for
Lotterhos Lab}}

\hypertarget{conda-environments}{%
\subsection*{\texorpdfstring{\textbf{Conda
environments}}{Conda environments}}\label{conda-environments}}
\addcontentsline{toc}{subsection}{\textbf{Conda environments}}

We use conda environments to load software on Discovery.

See this site for more information on how to load environments and
update the \texttt{yml} files.

If you create a \texttt{yml} for your analysis, be sure to talk with
Dr.~L about sharing it with the group.

\href{https://github.com/neu-rc-admin/lotterhos_group}{Shared Lotterhos
Group Environments with Research Computing}

\begin{verbatim}
    lotterhos_R.yml
    lotterhos_RADseq.yml
    lotterhos_RNAseq.yml
    lotterhos_aligners.yml
    lotterhos_gatk.yml
    lotterhos_methylation.yml
    lotterhos_popgen.yml
    lotterhos_utils.yml
    lotterhos_variantCallers.yml
\end{verbatim}

Use \texttt{conda\ deactivate} to change conda environments

\hypertarget{information-about-outdated-modules}{%
\subsection*{\texorpdfstring{\textbf{information about outdated
modules}}{information about outdated modules}}\label{information-about-outdated-modules}}
\addcontentsline{toc}{subsection}{\textbf{information about outdated
modules}}

When we first started using Discovery, we had modules set up for our
research

\begin{itemize}
\item
  We have a software module on Discovery for the lab's research
\item
  There are some programs loaded in modules, and others loaded in conda
  environments. If you are working in a conda environment, it won't
  detect programs loaded in the module outside that environment
\item
  One conda environment is \texttt{lotterhos-py38}, which is running
  python 3.8. Note that many bioinformatics programs are not updated for
  this environment. Slim 3.3.2 is in this environment.
\item
  Another conda environment is \texttt{ddocent2}, which is running
  python 3.7 and all of the ddocent dependencies

\begin{verbatim}
module show lotterhos/2020-08-24
module load lotterhos/2020-08-24
source activate ddocent2
conda list
\end{verbatim}

  Use the \texttt{module\ avail} command to show a list of the most
  currently available software on Discovery.
\end{itemize}

sratoolkit/2.10.8 was added as a seperate module that was then added to
your module, while ddocent-2.7.8 was added to the conda environment
``lotterhos-py38'' within the anaconda3/L2020-03 module

Old module: lotterhos/2019-11-15, lotterhos/2020-03-18,
lotterhos/2020-07-21

\begin{itemize}
\tightlist
\item
  In the future we hope to work on Docker containers for loading R
  packages and developing containers for specific projects. A container
  allows you to create the same environment that you were using when you
  analyzed the data.
\end{itemize}

\hypertarget{packrat-environments-for-r}{%
\subsection*{\texorpdfstring{\textbf{PackRat environments for
R}}{PackRat environments for R}}\label{packrat-environments-for-r}}
\addcontentsline{toc}{subsection}{\textbf{PackRat environments for R}}

PackRat environments can be great for using on OOD or Discovery

\href{https://rc-docs.northeastern.edu/en/latest/software/software.html}{Create
PackRat environments for R}

\hypertarget{using-lotterhos-nodes-on-discovery}{%
\section*{\texorpdfstring{\textbf{Using Lotterhos nodes on
Discovery}}{Using Lotterhos nodes on Discovery}}\label{using-lotterhos-nodes-on-discovery}}
\addcontentsline{toc}{section}{\textbf{Using Lotterhos nodes on
Discovery}}

There is a lotterhos unix group, which you need to request Dr.~L to add
you to. It takes a day for the change to propagate so you should be able
to see you are part of the `lotterhos' group when typing
\texttt{groups\ \$USER} in the terminal. Once you confirm you are part
of the unix group and have access to the nodes, you can submit regular
jobs using the \texttt{-\/-partition\ lotterhos}

There are 2 nodes each with 36 cores, so we can run parallel programs
with up to 36 cores, and serial on up to 72 cores. Our nodes are d3037
and d3038. Please communicate with Dr.~L about your project and the best
way to use resources, so that everyone in the lab has access if they
need it.

To get info on the nodes type \texttt{sinfo\ -p\ lotterhos}, which gives
the node IDs, and \texttt{lscpu\ \#\#\#\#\#} to get info on the CPUs for
that node.

\texttt{scontrol\ show\ partition\ lotterhos} shows what we can do.
There are no limits on job submissions, the default time is 7 days with
a max of 30 days. When you type the code, you can see
MaxTime=30-00:00:00

\texttt{squeue\ -p\ lotterhos} shows what is current running. You
definitely want to check this before you try to \texttt{srun} (see
below) or submit jobs!

There is 5GB of memory on each core, with 36 cores/node and 72 cores
total across both. So it should be possible to submit an array with 140
jobs at a time, each with 2.5 GB memory, but I haven't tried it yet.

If you are on campus, or offcampus using NU's VPN, you can use this
webpage to view resources: https://xdmod.rc.northeastern.edu/

\hypertarget{troubleshooting-and-trying-code}{%
\section*{\texorpdfstring{\textbf{Troubleshooting and trying
code}}{Troubleshooting and trying code}}\label{troubleshooting-and-trying-code}}
\addcontentsline{toc}{section}{\textbf{Troubleshooting and trying code}}

To develop code and troubleshoot, there is a \texttt{debug} partition.
You can also use the \texttt{lotterhos} partition. Use the \texttt{srun}
command with the \texttt{-\/-pty} option will output the results of your
code on the screen

To run on the lotterhos partition:

\texttt{srun\ -p\ lotterhos\ -N\ 1\ -\/-pty\ /bin/bash}

OR For 20 minutes on the debug partition:

\texttt{srun\ -p\ debug\ -N\ 1\ -\/-pty\ /bin/bash}

\href{https://rc-docs.northeastern.edu/en/latest/hardware/partitions.html\#introduction}{See
this page for information about default times on different partitions}

\hypertarget{overview-presentation}{%
\section*{\texorpdfstring{\textbf{Overview
presentation}}{Overview presentation}}\label{overview-presentation}}
\addcontentsline{toc}{section}{\textbf{Overview presentation}}

Here is a presentation from Research Computing on terminology and
optimizing workflows in the lab:
\href{https://drk-lo.github.io/lotterhoslabprotocols/img/Optimization_Lotterhos_labmeeting_Sekar-V2.pdf}{Optimization\_Lotterhos\_labmeeting\_Sekar-V2.pdf}

\hypertarget{open-ondemand}{%
\chapter{Open OnDemand}\label{open-ondemand}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\begin{itemize}
\item
  Check on the Open On Demand - R studio and Jupyter notebooks! Open
  OnDemand
  \href{https://drk-lo.github.io/lotterhoslabprotocols/discovery_1a-OOD/ood.discovery.neu.edu}{ood.discovery.neu.edu}

  \begin{itemize}
  \item
    You also open a JupyterHub interface (jupyterhub.discovery.neu.edu)
  \item
    Integrated Genome Viewer!
  \end{itemize}
\item
  To view your home directory on Discovery with OOD, you have to log in
  to Discovery at the command line / Terminal first
\item
  Previous workshop materials are available at
  \texttt{/shared/rc/training}. For example, the latest one research
  computing did focused on introduction to Slurm and Discovery as well
  as how to set up python environments.

  \begin{itemize}
  \item
    You can copy it to your home directory by typing the following
    command (at the command line):
    \texttt{cp\ -r\ /shared/rc/training/training\_111219\ \textasciitilde{}}
  \item
    You can then view it via Open OnDemand
  \end{itemize}
\item
  Although OOD allows file transfer, this is not recommended because
  there are no automatic checks for file corruption. Use Globus instead.
\end{itemize}

\hypertarget{using-our-environments-in-jupyter-lab-notebooks}{%
\section*{\texorpdfstring{\textbf{Using our environments in Jupyter Lab
Notebooks}}{Using our environments in Jupyter Lab Notebooks}}\label{using-our-environments-in-jupyter-lab-notebooks}}
\addcontentsline{toc}{section}{\textbf{Using our environments in Jupyter
Lab Notebooks}}

\begin{itemize}
\tightlist
\item
  we do have a way to load a custom conda env on your jupyter notebook
  when opening jupyterlab through OOD. The only additional step needed
  for this is to install jupyterlab onto the env you wish to load using:
  ``` conda activate
\end{itemize}

conda install jupyterlab ```

and then go to

OOD \textgreater{} Interactive apps \textgreater{} JupyterLab Notebook
{[}Custom Anaconda Environment{]}

{[}fill in all the fields and provide the env name where you installed
jupyterlab under the field ``Custom Anaconda Environment (provide name
only)''{]}

\href{https://rc-docs.northeastern.edu/en/latest/using-ood/interactiveapps.html\#working-with-jupyter-notebook-custom-anaconda-environment}{You
can find some instructions on this here as well in RC Docs}

\hypertarget{globus-file-transfer}{%
\chapter{Globus File Transfer}\label{globus-file-transfer}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\begin{itemize}
\item
  Globus is a system for transferring files. The pros:

  \begin{itemize}
  \item
    It's fast
  \item
    If your connection gets interrupted, it automatically picks up where
    it left off
  \item
    It automatically checks that your files were not corrupted during
    the transfer
  \item
    Globus is already integrated with Discovery and our lab files.
  \item
    It makes transferring files among your laptop, your collaborator(s),
    and Discovery easy.
  \item
    A \textbf{Globus endpoint} is a folder on a computer that you want
    to transfer files to or from
  \item
    A \textbf{Globus Collection} is a folder on a computer that contains
    a collection of files. You may create several endpoints within a
    collection to share with specific people.
  \end{itemize}
\end{itemize}

\hypertarget{getting-started-with-globus-on-discovery}{%
\section*{\texorpdfstring{\textbf{Getting started with Globus on
Discovery}}{Getting started with Globus on Discovery}}\label{getting-started-with-globus-on-discovery}}
\addcontentsline{toc}{section}{\textbf{Getting started with Globus on
Discovery}}

\begin{itemize}
\item
  Go to
  \href{https://drk-lo.github.io/lotterhoslabprotocols/discovery_2-globus/globus.org}{globus.org}
\item
  Click on ``Login'', from the dropdown list choose ``Northeastern
  University'', login with your NU credentials
\item
  Click on ``File Manager'' and the Search Icon. In the Search Box type
  ``Northeastern''. Choose ``northeastern\#discovery''. This is your
  home directory on Discovery. Click on the Bookmark Icon to add this to
  your bookmarks.
\item
  Dr.~Lotterhos will share through Globus the specific folders that you
  will have access to on the Discovery Lab Storage
\end{itemize}

\hypertarget{getting-started-with-globus-on-your-laptop}{%
\section*{\texorpdfstring{\textbf{Getting started with Globus on your
laptop}}{Getting started with Globus on your laptop}}\label{getting-started-with-globus-on-your-laptop}}
\addcontentsline{toc}{section}{\textbf{Getting started with Globus on
your laptop}}

\begin{itemize}
\item
  Download \href{https://www.globus.org/globus-connect-personal}{Globus
  Connect Personal} for your laptop and follow the instructions to setup
  and configure your laptop as an endpoint
\item
  Make sure to save your setup key, you'll need it again to open the
  Globus App on your computer
\item
  If you get error messages, try restarting your Terminal
\end{itemize}

\hypertarget{transferring-files-in-globus}{%
\section*{\texorpdfstring{\textbf{Transferring files in
Globus}}{Transferring files in Globus}}\label{transferring-files-in-globus}}
\addcontentsline{toc}{section}{\textbf{Transferring files in Globus}}

\begin{itemize}
\item
  Click on ``File Manager''
\item
  In the upper right, click on the Double-Panel Icon. This will open two
  panels - choose the collections that you would like to transfer files
  between, and drag and drop!
\item
  can view other's scratch directories and copy files, but not edit
\item
  other users can view /research/lotterhos and edit/delete files
\end{itemize}

\hypertarget{sharing-endpoints}{%
\section*{\texorpdfstring{\textbf{Sharing
endpoints}}{Sharing endpoints}}\label{sharing-endpoints}}
\addcontentsline{toc}{section}{\textbf{Sharing endpoints}}

\begin{itemize}
\item
  A \textbf{Globus endpoint} is a folder on a computer that you want to
  transfer files to or from. In Globus, click on the folder
  (``Collection'') that you would like to allow file transfer to or
  from.
\item
  To let someone download files in that folder, give them access to
  ``read'' the files
\item
  To let someone upload files to that folder, give them access to ``read
  and write'' to the folder

  \begin{itemize}
  \tightlist
  \item
    Note that if you allow the person you are sharing with to write to
    the folder, this gives the person you are sharing with the
    opportunity to delete or replace any files within that folder, so
    proceed with caution. I recommend creating a specific
    ``sharedwith\_X'' folder for having someone transfer files to you.
  \end{itemize}
\item
  Steps

  \begin{itemize}
  \item
    In Globus, click on the folder (``Collection'') that you would like
    to allow file transfer to or from.
  \item
    Click on ``Share'', ``Add a Shared Endpoint''
  \item
    To view your shared endpoints, click on ``endpoints'', ``shareable
    by you''

    \begin{itemize}
    \item
      click on the endpoint and the ``sharing'' tab
    \item
      click on ``Add Permissions - share with'' and enter the email
      associated with the globus account you want to share with
    \end{itemize}
  \end{itemize}
\end{itemize}

Question for research computing - can we link Globus with Google drive?
https://docs.globus.org/how-to/access-google-drive/

\hypertarget{lotterhos-lab-vs.-personal-storage-on-discovery}{%
\chapter{Lotterhos Lab vs.~Personal Storage on
Discovery}\label{lotterhos-lab-vs.-personal-storage-on-discovery}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\begin{itemize}
\item
  We have \textbf{active} storage (10 TB) that should be used for
  `active' projects, and \textbf{archival} storage for storing raw files
  and outputs from projects. Please be smart about how you use archival
  storage and work with Dr.~Lotterhos to make sure only relevant files
  are being archived.
\item
  In addition each user has a home directory and a scratch directory.
\end{itemize}

\hypertarget{working-with-github}{%
\section*{\texorpdfstring{\textbf{Working with
github:}}{Working with github:}}\label{working-with-github}}
\addcontentsline{toc}{section}{\textbf{Working with github:}}

It is good practice to create a repo on Github and then clone it to
Discovery to start a project. You can follow instructions in Buffalo's
Bioinformatics book to work with git on linux.

In addition, you may want to collaborate on Discovery through your git
repo.

example: When person A started working on his git branch in Discovery,
he cloned that branch to Discovery (e.g in the folder and repo
/work/lotterhos/MVP-NonClinalAF). To give person B access,

\begin{verbatim}
1. Person-B Go to the terminal and copy your public key file content.
cat ~/.ssh/id_rsa.pub
(If you don't have a key pair generated already, generate a new one using command "ssh-keygen" )
2. Go to the github website and login with Person-B credentials.
3. In the upper-right corner of the github page, click Person-B profile photo, then click Settings.
4. Click New SSH key or Add SSH key.
5. Paste your key into the "Key" field.
6. Click Add SSH key.
7. Test the connection by running the following command in the terminal on person B Discovery:
ssh -T git@github.com 
\end{verbatim}

Get message: ``Hi ModelValidationProgram/MVP-NonClinalAF! You've
successfully authenticated, but GitHub does not provide shell access.''

Then, Person-B navigated to the folder on Discovery

\begin{verbatim}
cd /work/lotterhos/MVP-NonClinalAF
\end{verbatim}

This seemed to work

\hypertarget{your-home-directory-on-discovery.}{%
\subsection*{\texorpdfstring{\textbf{Your home directory on
Discovery**.}}{Your home directory on Discovery**.}}\label{your-home-directory-on-discovery.}}
\addcontentsline{toc}{subsection}{\textbf{Your home directory on
Discovery**.}}

\begin{itemize}
\item
  Path: \texttt{/\textasciitilde{}/} or \texttt{/home/username}
\item
  Note that only you have access to these files.
\end{itemize}

\hypertarget{your-scratch-directory-on-discovery.-use-this-for-intermediate-outputs-that-are-not-backed-up.}{%
\subsection*{\texorpdfstring{\textbf{Your scratch directory on
Discovery**. Use this for intermediate outputs that are not backed
up.}}{Your scratch directory on Discovery**. Use this for intermediate outputs that are not backed up.}}\label{your-scratch-directory-on-discovery.-use-this-for-intermediate-outputs-that-are-not-backed-up.}}
\addcontentsline{toc}{subsection}{\textbf{Your scratch directory on
Discovery**. Use this for intermediate outputs that are not backed up.}}

\begin{itemize}
\item
  Path: \texttt{/scratch/username}
\item
  Note that only you have access to these files.
\end{itemize}

\hypertarget{lotterhos-shared-storage-on-discovery}{%
\subsection*{\texorpdfstring{\textbf{Lotterhos shared storage on
Discovery}}{Lotterhos shared storage on Discovery}}\label{lotterhos-shared-storage-on-discovery}}
\addcontentsline{toc}{subsection}{\textbf{Lotterhos shared storage on
Discovery}}

\begin{itemize}
\item
  /work/lotterhos/
\item
  Katie will create a folder for each lab project in this directory. For
  each folder, Katie will submit a ticket to research computing to ask
  them to create a group for the project with members who will have
  access to that folder.
\end{itemize}

Example:

\begin{verbatim}
mkdir BlackSeaBass                  # make the folder
chgrp -R lotterhos-bsb 2020_NOAA_BlackSeaBass_ddRADb    # change the group on the directory. I forget what the -R does
chmod -R 775 2020_NOAA_BlackSeaBass_ddRADb            # give group permissions to the directory
ls -la 2020_NOAA_BlackSeaBass_ddRADb # check permissions
\end{verbatim}

\begin{itemize}
\item
  Groups
\item
  \texttt{lotterhos} - all Lotterhos Lab members
\item
  \texttt{lotterhos-bsb} - access to Black Sea Bass project
\item
  \texttt{lotterhos-cod}
\item
  Groups are created by research computing
\end{itemize}

Find the groups of a user: \texttt{groups\ tbittar} will give the groups
for user tbittar

Find the users in a group: \texttt{users\ lotterhos-bsb\ lotterhos} will
give the users for group \texttt{lotterhos-bsb} and group
\texttt{lotterhos}

\texttt{usermod\ -a\ -G\ groupname\ username} I forget what this does

Sometimes I have still had trouble making the code work on existing
directories, but it can work if you copy the directory to a new one

\hypertarget{github-enterprise}{%
\subsection*{\texorpdfstring{\textbf{GitHub
Enterprise}}{GitHub Enterprise}}\label{github-enterprise}}
\addcontentsline{toc}{subsection}{\textbf{GitHub Enterprise}}

\begin{itemize}
\tightlist
\item
  To do: Set up a time to talk to Raphael
\end{itemize}

\hypertarget{archival-lotterhos-lab-storage-25-tb-for-lab.-northeastern-nese-collection.}{%
\subsection*{\texorpdfstring{\textbf{Archival Lotterhos Lab storage**
(25 TB for lab). Northeastern NESE
collection.}}{Archival Lotterhos Lab storage** (25 TB for lab). Northeastern NESE collection.}}\label{archival-lotterhos-lab-storage-25-tb-for-lab.-northeastern-nese-collection.}}
\addcontentsline{toc}{subsection}{\textbf{Archival Lotterhos Lab
storage** (25 TB for lab). Northeastern NESE collection.}}

\begin{itemize}
\item
  Path: \texttt{/lotterhosarchive/}
\item
  TO DO: FIGURE OUT WHAT TO DO ABOUT SARA's COD BACKUPS ON SMB/LOTTERHOS
  - ASK JAMES
\item
  Within this storage space we will have a shared folder for each
  project, that only the people on that project will have access to.
\end{itemize}

\hypertarget{active-lotterhos-lab-storage-more-information-tb-for-lab.}{%
\subsection*{\texorpdfstring{\textbf{Active Lotterhos Lab storage: more
information** (? TB for
lab).}}{Active Lotterhos Lab storage: more information** (? TB for lab).}}\label{active-lotterhos-lab-storage-more-information-tb-for-lab.}}
\addcontentsline{toc}{subsection}{\textbf{Active Lotterhos Lab storage:
more information** (? TB for lab).}}

\begin{itemize}
\item
  smb://nunet.neu.edu/rc-shares/lotterhos/
\item
  Ask Katie to share a Globus endpoint with you. The endpoint with
  include the words ``smb Volumes/lotterhos''
\item
  Within this storage space we have a shared folder for each project,
  that only the people on that project have access to.
\item
  For this endpoint to work, it has to be mapped through Katie's laptop
  or the iMac in the lab. \emph{Katie's laptop from home: sign into NU
  VPN, make sure Globus Connect Personal App is running, plug in
  ethernet cord}
\item
  Because accessing this server is done through a computer, it makes
  transferring files specific to the computer that the server is mapped
  to. Typically, you will ask Katie to share the Globus endpoint for
  this server that is set up through the lab iMac. As long as the server
  is mapped to the computer (typically lab iMac) and that connection is
  not interrupted, you should have no problems.
\item
  From a laptop or over wifi, transfers of large files are very slow.
  This is why we map to the server through the iMac. The point of all
  the steps we outline below is so we can use Globus to transfer files
  from the smb drive to Discovery.
\item
  To connect to the smb server, follow these directions to map to the
  server on a computer
  \href{https://drk-lo.github.io/lotterhoslabprotocols/discovery_3-storage/accessing_shared_storage_2020.pdf}{this
  document} (typically Katie only)
\item
  Setting up a Globus endpoint for the folder to share (typically Katie
  only)
\item
  \emph{Intitial step if Globus personal endpoint for the computer is
  not set up}: Download and install the Globus Connect Personal App. In
  Globus web app, click on ``Add a personal endpoint'' and enter the key
  in the Desktop app.
\item
  \emph{Intial setup if working from a home laptop}: Use GlobalProtect
  to establish a VPN connection (vpn.northeastern.edu) (if you are
  having trouble establishing the VPN connection on a Mac, check your
  Security and Permissions settings and give the app permissions, see
  image). Then you should be able to map to the server.
\item
  \includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/discovery_3-storage/globalprotectgetittowork.png}
\item
  Step 1: In the Globus Desktop App, then go to
  Preferences-\textgreater{} Access and click the ``+'' sign. On a Mac,
  at the top center click on the bar, click on the name of the computer,
  and you should see the mapped server there as a volume
  \texttt{Volumes/lotterhos}. Click on it and add it to the list.
\item
  \includegraphics{https://drk-lo.github.io/lotterhoslabprotocols/discovery_3-storage/howtofindvolumeinglobusconnect.png}
\item
  Step 2: In the Globus Web App, go to the endpoint for the computer.
  Then type \texttt{Volumes/lotterhos/} in the path. Save it as a
  bookmark.
\item
  Step 3: Before you can conduct file transfers from a personal
  endpoint, you need to have a ``Globus Plus Account''. Email James
  Macalino at NU research computing to get added. You will then receive
  an email.
\end{itemize}

\hypertarget{workflow-parallelization-approaches}{%
\chapter{Workflow Parallelization
Approaches}\label{workflow-parallelization-approaches}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

Workflow parallelization approaches:

\emph{a. One the shared resources}, limits on each partition are
determined according to the following table:
https://rc-docs.northeastern.edu/en/latest/hardware/partitions.html\#introduction.
If your jobs have exceeded those limits, this may have been the cause of
job terminations.

\emph{b. On the ``lotterhos'' partition}, you do not have any
restrictions on number of submitted jobs, or requested cpus. To submit
to the lotterhos partition, simply set the SLURM flag for this
partition:

For an interactive job (example):
\texttt{srun\ -p\ lotterhos\ -\/-nodes=1\ -\/-cpus-per-task=1\ -\/-pty\ -\/-time=24:00:00\ /bin/bash}

This example will allocate a single node with 1 CPU (and --ntasks=1 by
default, but alternatively, you can also use this instead:

\texttt{srun\ -p\ lotterhos\ -\/-nodes=1\ -\/-ntasks\ =1\ -\/-pty\ -\/-time=24:00:00\ /bin/bash}
for 24 hours, and flag \texttt{-\/-pty} will set all output to be
printed to the screen. \texttt{/bin/bash} is the command to be run for
this job, so any bash command after this will be executed.

If you run this command, you can type: \texttt{lscpu} to get the
hardware info (such as core count).

For a batch (background) job, adding the following to your bash script
is equivalent:

\begin{verbatim}
#SBATCH -p lotterhos
#SBATCH --nodes=1
#SBATCH --cpus-per-task=1
#SBATCH --time=24:00:00
\end{verbatim}

If your shell script looks the same, and the only thing changing is a
particular parameter, for example:

\begin{verbatim}
#SBATCH -p lotterhos
#SBATCH --nodes=1
#SBATCH --cpus-per-task=1
#SBATCH --time=24:00:00
program input-X output-X
\end{verbatim}

Then you can also take advantage of the job-array structure to submit
multiple jobs at a time. Each SLURM job that is defined as array will
have several bash variables associated with this that are defined once
the job is submitted. The main important variable that can be useful is:
\texttt{\$SLURM\_ARRAY\_TASK\_ID} For example, if you create a SLURM
array job with 10 jobs in the array, then the values
\texttt{\$SLURM\_ARRAY\_TASK\_ID} will get are: 1,\ldots,10 (each job in
the array will be assigned with a number 1-10, and can be distinguished
this way). Here, \%10 means all 10 jobs are set to run together. Setting
\%1 means they will run one after another.

\begin{verbatim}
#SBATCH -p lotterhos
#SBATCH --nodes=1
#SBATCH --cpus-per-task=1
#SBATCH --time=24:00:00
#SBATCH --array=1-10:%10
program input-$SLURM_ARRAY_TASK_ID output-$SLURM_ARRAY_TASK_ID
\end{verbatim}

This setup will run the program on a specific input file based on the
\$SLURM\_ARRAY\_TASK\_ID ~value (input-1,input-2,\ldots,input-10) and
will output the results to output file named based on the value of
\$SLURM\_ARRAY\_TASK\_ID ~(output-1,output-2,\ldots,output-10).

There is a hard limit of an array size of 1000 that research computing
cannot increase.

In addition, this kind of code will not work:
\texttt{\#SBATCH\ —array=1001-1951\%68} because the array index goes
over 1000 (Even though the number of jobs in the array is less than
1000). So if you have more than 1000 simulations, I don't recommend the
array approach.

\hypertarget{parallelizing-your-r-script}{%
\section*{\texorpdfstring{\textbf{Parallelizing your R
script}}{Parallelizing your R script}}\label{parallelizing-your-r-script}}
\addcontentsline{toc}{section}{\textbf{Parallelizing your R script}}

\emph{c.~Parallelizing your R script:} You can also leverage the entire
36 cores on each on the lotterhos nodes in a single job by using a
threading parallelization scheme. Specifically, you can use the parallel
processing R libraries such as ``foreach'' to increase the efficiency of
your loops. You can then parallelize specific chunks of your code
(mainly, loops) by importing the parallel libraries. For example, if
your r code is this:

\begin{verbatim}
#import packages
library(foreach)
library(doParallel)
#number of iterations
iters<-1e4
#setup parallel backend to use 8 processors
cl<-makeCluster(36)
registerDoParallel(cl)
#start time
strt<-Sys.time()
#loop
ls<-foreach(icount(iters)) %dopar% {
    to.ls<-rnorm(1e6)
    to.ls<-summary(to.ls)
    to.ls
    }
print(Sys.time()-strt)
stopCluster(cl)
\end{verbatim}

source:
https://beckmw.wordpress.com/2014/01/21/a-brief-foray-into-parallel-processing-with-r/

Then, you can set up an array job that looks like this:

\begin{verbatim}
#SBATCH -p lotterhos
#SBATCH --nodes=1
#SBATCH --cpus-per-task=36
#SBATCH --time=24:00:00
#SBATCH --array=1-10:%10
r-parallel-program  input-$SLURM_ARRAY_TASK_ID  output-$SLURM_ARRAY_TASK_ID
\end{verbatim}

\hypertarget{optimizing}{%
\chapter{Optimizing}\label{optimizing}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

Here is a presentation from Research Computing on terminology and
optimizing workflows in the lab:
\href{https://drk-lo.github.io/lotterhoslabprotocols/img/Optimization_Lotterhos_labmeeting_Sekar-V2.pdf}{Optimization\_Lotterhos\_labmeeting\_Sekar-V2.pdf}

To view your recent jobs, use
\texttt{acct\ -S\ 2020-08-01\ -u\ lotterhos} will list all jobs since
that date for user lotterhos

To learn about how much memory a job used, type \texttt{seff\ jobID} -
will give memory for the job that was requested and how much was used.

\emph{Check job efficiency and performance:}

\emph{a. R code in part c from Workflow parralelization approaches shows
an example of how you can capture the run time of a particular job}, or
a parallel segment by printing out the time difference it takes to
calculate a particular part. You can also modify this to calculate the
total time it took to run your code. To find the best performance, I
would recommend to benchmark the calculations on (for example)
1,24,8,16,32 cores and see if at some point it levels off in terms of
run time, this may mean that there would be no benefit to request more
cores beyond that. Research Computing can discuss performance testing,
just schedule another consultation time.

\emph{b. You can also look at the efficiency and memory usage of your
jobs by looking at the completed jobs from your slurm history}, by
typing: \texttt{acct\ -S\ 2020-08-10\ -u\ albecker} Then, selecting a
particular job (might be beneficial to have a small script that runs
through all jobs, since you had over 100,000 jobs this month, I can help
with that too if needed) and using the seff command to extract the
efficiency. For example:

\begin{verbatim}
$ seff 13291706
Job ID: 13291706
Cluster: discovery
User/Group: albecker/users
State: COMPLETED (exit code 0)
Cores: 1
CPU Utilized: 00:01:58
CPU Efficiency: 99.16% of 00:01:59 core-walltime
Job Wall-clock time: 00:01:59
Memory Utilized: 284.50 MB
Memory Efficiency: 0.56% of 50.00 GB
\end{verbatim}

Here, you can see that the CPU efficiency was very high at 99\%, but you
ended up using less than 1\% of the memory you requested (284.50 MB out
of 50.00 GB). This information is useful to better allocate and utilize
resources in the future.

\emph{c.~It is also recommended to perform short testing prior to submit
long jobs to the cluster.} You can use the ``debug'', ``express'' or the
``lotterhos'' partition and you can run the job interactively. For
example, this will give you resources on the debug partition (up to 20
minutes).
\texttt{srun\ -p\ debug\ -\/-nodes=1\ -\/-ntasks=16\ -\/-pty\ /bin/bash}

\hypertarget{make-sure-to-optimize-on-the-paritition-that-you-will-use}{%
\section*{\texorpdfstring{\textbf{Make sure to optimize on the
paritition that you will
use}}{Make sure to optimize on the paritition that you will use}}\label{make-sure-to-optimize-on-the-paritition-that-you-will-use}}
\addcontentsline{toc}{section}{\textbf{Make sure to optimize on the
paritition that you will use}}

For example, the \texttt{lotterhos} partition runs a lot faster than the
\texttt{short} partition

If you optimize on the \texttt{lotterhos} partition, then your jobs will
run out of time on the \texttt{short} partition

\hypertarget{optimization-and-arrays}{%
\section*{\texorpdfstring{\textbf{Optimization and
arrays}}{Optimization and arrays}}\label{optimization-and-arrays}}
\addcontentsline{toc}{section}{\textbf{Optimization and arrays}}

Here is an example header for an array script:

\begin{verbatim}
#SBATCH --job-name=SlimRun
#SBATCH --mail-user=k.lotterhos@northeastern.edu
#SBATCH --mail-type=FAIL
#SBATCH --partition=lotterhos
#SBATCH --mem=4G
#SBATCH --nodes=1
#SBATCH --CPUs-per-task=2
#SBATCH --array=50-151%70
#SBATCH --output=/work/lotterhos/MVP-NonClinalAF/slurm_log/SlimRun20210826_%j.out
#SBATCH --error=/work/lotterhos/MVP-NonClinalAF/slurm_log/SlimRun20210826_%j.err

source ~/miniconda3/bin/activate MVP_env
\end{verbatim}

The Lotterhos partition has 72 Cores, divided across two nodes. First,
you need to know more about the how the programs that your script will
call work. Can thay utilize multiple cores (threads)? Or do they only
use one core?

This is an example header from a submission script for an array that
reads in rows 50-150 from a file that has the parameters that I want to
run.

And explanation of the following terms:

\texttt{\#SBATCH\ -\/-array=50-151\%70}

\begin{itemize}
\item
  This will run 101 array tasks, for parameters 50-151 in the filename,
  and submit 70 jobs at a time.
\item
  If each task in array requires 1 CPU, you could also run up to 72 jobs
  at at time with \texttt{50-151\%72}, to maximize the resources in the
  lotterhos partition. But, it is good to leave a couple of cores open
  for \texttt{srun} tasks.
\item
  If each task requires 2 CPUS, set \texttt{-\/-CPUs-per-task=2} (look
  it up) and set \texttt{-\/-array=50-151\%36} to maximize the resources
  in the lotterhos partition.
\item
  IF programs can use multiple threads, do benchmarking to determine CPU
  per task, 1-2 jobs at a time, and do \texttt{seff} on those tasks that
  completed (see below). Test 1, 2, 4, 8, 16 CPUs. Then set that in the
  line.
\end{itemize}

\texttt{\#SBATCH\ -\/-nodes=1}

\begin{itemize}
\item
  Set this to 1 unless we know the program can communicate between the
  nodes.
\item
  If you set this to 1 and you set \texttt{50-151\%72}, it will use both
  nodes on lotterhos, but only one CPU per node. If you set
  \texttt{\#SBATCH\ -\/-nodes=2} and \texttt{-\/-array=50-151\%72}, it
  will only submit up to 36 jobs at a time because it will set aside 1
  CPU from each node, and only one CPU will be used while the other one
  will remain idle if the program does not communicate between the
  nodes.
\end{itemize}

\texttt{\#SBATCH\ -\/-mem=4GB}

\begin{itemize}
\item
  This is the amount of memory needed for each task.
\item
  If I set \texttt{\#SBATCH\ -\/-array=50-151\%70}, and each task needs
  3GB memory (see below), it's best to add a add buffer in case one task
  requires more memory.
\item
  If your submission script doesn't specify anthing, it will use the
  default amount which I would have to look up. For some array tasks,
  the default might work well, but it might not work well every time.
  The default memory is limited, so it's always recommended to set it
  higher so you don't run out.
\end{itemize}

\hypertarget{seff-20714872_151-for-an-array}{%
\subsection*{\texorpdfstring{\textbf{\texttt{seff\ 20714872\_151} for an
array}}{seff 20714872\_151 for an array}}\label{seff-20714872_151-for-an-array}}
\addcontentsline{toc}{subsection}{\textbf{\texttt{seff\ 20714872\_151}
for an array}}

\begin{verbatim}
seff 20714872
Job ID: 20714872
Array Job ID: 20714872_151
Cluster: discovery
User/Group: lotterhos/users
State: COMPLETED (exit code 0)
Nodes: 2
Cores per node: 1
CPU Utilized: 00:13:00
CPU Efficiency: 48.51% of 00:26:48 core-walltime
Job Wall-clock time: 00:13:24
Memory Utilized: 768.23 MB
Memory Efficiency: 19.21% of 3.91 GB
\end{verbatim}

This report is for the specified array task (in this case index 151 in
the array)

This array task required 0.1921 x 3.91 \textasciitilde{} 0.75 GB memory.

This task used 2 cores (1 from each node), but the actual programs being
used didn't commuicate between the nodes. So it was not a well formatted
array.

It only used half of the CPU, but the array will run each task on an
individual CPU. So there is not much we can do about that.

\hypertarget{example-scripts}{%
\chapter{Example Scripts}\label{example-scripts}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

This protocols page does not exist yet.

\hypertarget{what-are-vs-code-and-pycharm}{%
\chapter{What are VS Code and
PyCharm?}\label{what-are-vs-code-and-pycharm}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\href{https://code.visualstudio.com/}{VS Code} and
\href{https://www.jetbrains.com/pycharm/}{Pycharm} are both Integrated
Developement Environments, or IDEs. You've probably worked with an IDE,
RStudio is an IDE. Ultimately the purpose of an IDE is to make coding
easier. Some of the most basic features of IDEs are syntax highlighting
and autocompletion capabilities, but they can also do a lot more than
that - like displaying hover-over metadata (eg displaying a function's
documentation when hovering over code), jumping to the part of the file
where a function is defined/used, showing where an error will occur,
among others. While PyCharm is specifically built for python (but see
other IDEs from \href{https://www.jetbrains.com/}{jet brains}), VS Code
supports hundreds of coding languages.

Both PyCharm and VS Code have free professional versions for students
and academics - see eg here for
\href{https://www.jetbrains.com/community/education/\#students}{PyCharm}.
I won't say much more about the purposes and cool features of these
IDEs, for that it would be good to google, use the intro/how-to
resources from each program's page, or better yet open one up and
explore.

The purpose of this post is to show how to set up VS Code and PyCharm to
connect to NEU's Discovery cluster.

\begin{itemize}
\item
  First download and install the program by visiting the landing pages
  linked above.
\item
  Make sure you can ssh to discovery - in terminal
  \texttt{ssh\ user@discovery.neu.edu}

  \begin{itemize}
  \item
    TODO - link to example of ssh config
  \item
    to set up passwordless ssh so you don't have to enter a password
    each time, see \href{http://www.linuxproblem.org/art_9.html}{this
    link}
  \end{itemize}
\end{itemize}

\hypertarget{vs-code}{%
\section*{\texorpdfstring{\textbf{VS Code}}{VS Code}}\label{vs-code}}
\addcontentsline{toc}{section}{\textbf{VS Code}}

\begin{itemize}
\item
  Install the
  \href{https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.vscode-remote-extensionpack}{Remote
  Development Extensionpack}
  \includegraphics{https://user-images.githubusercontent.com/8821725/159176412-c43f26ad-855a-46d8-918a-31bf41156bea.png}
\item
  Once the Remote Development Extension is installed, open the Command
  Palette (⇧⌘P) and type ``\textbf{Remote-SSH: Connect to
  Host\ldots{}}'' and you should see something like this
  \includegraphics{https://user-images.githubusercontent.com/8821725/159713469-dcb1aebc-915e-495f-89b1-d3ca29ff7fd0.png}
\item
  After a moment, VS Code will connect to the SSH server and set itself
  up. VS Code will keep you up-to-date using a progress notification.
  After you are connected, you'll be in an empty window. You can always
  refer to the Status bar to see which host you are connected to.\\
  \includegraphics{https://user-images.githubusercontent.com/8821725/159713970-a6471acf-e7db-4b43-a87f-316ee8feb9a5.png}
\item
  You should now be able to choose the folder you wish to open\\
  \includegraphics{https://user-images.githubusercontent.com/8821725/159714594-fa30ac5e-3666-492a-abb9-52de33fa2879.png}
\item
  After opening the folder destination you can begin to create new files
  or open that already exist
  \includegraphics{https://user-images.githubusercontent.com/8821725/159715126-e0cc754b-ddb3-43ea-98a1-a1fa6a665f16.png}
\end{itemize}

\hypertarget{pycharm}{%
\section*{\texorpdfstring{\textbf{PyCharm}}{PyCharm}}\label{pycharm}}
\addcontentsline{toc}{section}{\textbf{PyCharm}}

\begin{itemize}
\item
  After setting up a project first add the Remote Host tool window to
  the
  sidebar\includegraphics{https://user-images.githubusercontent.com/8821725/159718453-3750eb37-6127-4c53-9e00-ffec62ba4a22.png}
\item
  If PyCharm opens the Remote Host tool window like a browser tab, you
  can drag it to the sidebar. Afterwards, click on it. Then click on the
  \texttt{...} next to the drop down menu. In the Deployment window,
  click the \texttt{+} button to add a new SFTP connection. This will
  then prompt you to give a server name, let's use ``discovery''.\\
  \includegraphics{https://user-images.githubusercontent.com/8821725/159719312-ca86411a-1d7d-4131-95a3-fa894d6eee94.png}
\item
  Next we need to set up the SSH Configuration - click the \texttt{...}
  next to the SSH Configuration drop-down menu. Fill in your
  credentials. I have passwordless ssh already configured (how-to link
  above) so I have selected to authenticate using my RSA key. Make sure
  to test the connection.\\
  \includegraphics{https://user-images.githubusercontent.com/8821725/159720760-74429eab-6aa7-4a5e-bbb9-eb3672879f72.png}
\item
  Close the SSH Configuration window to return to the Deployment window.
  Test the connection in the Deployment window. After a successful
  connection, close the Deployment window and select `discovery' from
  the Remote Host drop-down menu. You should now be able to browse files
  on the remote
  server.\includegraphics{https://user-images.githubusercontent.com/8821725/159721461-d1106690-d6f9-452d-983e-02913f4b625c.png}
\end{itemize}

\hfill\break

\hypertarget{instructions-for-discovery-for-non-northeastern-personnel}{%
\chapter{Instructions for Discovery for Non-Northeastern
Personnel}\label{instructions-for-discovery-for-non-northeastern-personnel}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

1) Request a Northeastern sponsored
\href{https://research.northeastern.edu/nu-res/request-nu-sponsored-account/}{account}

2) Download Duo Two Factor Authentication

3) In order to login into Duo with your new Northeastern Account, call
\href{https://its.northeastern.edu/support/}{Northeastern IT} to set up
duo access with them (needed for non-northeastern people)

4) submit a second
\href{https://rc-docs.northeastern.edu/en/latest/get_started/get_access.html}{form}
to get a sponsored Discovery account in addition to a sponsored NEU ID

\hypertarget{parallelization-with-array}{%
\chapter{Parallelization with Array}\label{parallelization-with-array}}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{overview-3}{%
\section*{\texorpdfstring{\textbf{Overview}}{Overview}}\label{overview-3}}
\addcontentsline{toc}{section}{\textbf{Overview}}

Brief review of using the \texttt{\#SBATCH\ -\/-array} slurm argument to
restrict the number of jobs running in parallel on Discovery. Arrays
seem to be most effective when you have a large number of jobs that all
require the same number of resources and the user wants to limit the
amount of the computing resource being used at a particular time. This
guide follows the same template format from the lotterhos wiki, which
reads through and leverages a reference file that contains a number of
file names corresponding to files you wish to process. For example, this
cod contain a list of raw sequence files (e.g., \texttt{fq.gz}) that you
with to perform trimming and qc on.

\hypertarget{breaking-down-the---array-argument}{%
\subsection*{\texorpdfstring{\textbf{Breaking down the
\texttt{-\/-array}
argument}}{Breaking down the -\/-array argument}}\label{breaking-down-the---array-argument}}
\addcontentsline{toc}{subsection}{\textbf{Breaking down the
\texttt{-\/-array} argument}}

The \texttt{-\/-array} argument can take a single integer,\texttt{1}, or
a range, \texttt{1-32}, of integers that the slurm script will iterate
through. Following the range of integers you specify the number of these
values you wish to process simultaneously using the \texttt{\%}
operator. For example, if you wish to iterate through the range of
integers in batches of 4 you could code:
\texttt{\#SBATCH\ -\/-array=1-32\%4}

\textbf{Other options}

\begin{verbatim}
# A job array with array tasks numbered 1, 2, 5, 19, 27.
#SBATCH --array=1,2,5,19,27

# A job array with array tasks numbered 1, 3, 5 and 7.
#SBATCH --array=1-7:2
\end{verbatim}

\hypertarget{using-the---array-argument-in-your-script}{%
\subsection*{\texorpdfstring{\textbf{Using the \texttt{-\/-array}
argument in your
script}}{Using the -\/-array argument in your script}}\label{using-the---array-argument-in-your-script}}
\addcontentsline{toc}{subsection}{\textbf{Using the \texttt{-\/-array}
argument in your script}}

Including the above example (\texttt{\#SBATCH\ -\/-array=1-32\%4}) in
your slurm script will effectively iterate through your slurm script 32
times, by running 4 simultaneous instances (or jobs) of the script at a
time on Discovery. However, running the same script with identical
parameters (or files) may not be particularly useful. Instead we
probably want to run the same script but apply it to a different set of
data or use different parameters. At the very least we want to be able
to create unique outputs for each iteration!

Conviently, the \texttt{-\/-array} argument creats a unix variable -
\texttt{SLURM\_ARRAY\_TASK\_ID}, which in the case above references the
specific integer that corresponds to the particular instances of the
script. For example, the first iteration of your slurm script will have
\texttt{SLURM\_ARRAY\_TASK\_ID=1}.

This could be leveraged in multiple ways, but one option is using this
index to extract specific lines from a reference script. The data on
these lines could be anything, but having file names is a particular
useful approach if you want to apply your script to multiple files.

\textbf{Short illustration}

\begin{verbatim}
VARNAME=`sed "${SLURM_ARRAY_TASK_ID}q;d" path/to/reference/reference.txt`

file1 = path/to/data/${VARNAME}

echo '${file1}'
\end{verbatim}

Example of reference file:

\begin{verbatim}
datafile1.csv
datafile2.csv
datafile3.csv
\end{verbatim}

Example output in \texttt{.out} file

\begin{verbatim}
datafile1.csv
datafile2.csv
datafile3.csv
\end{verbatim}

\hypertarget{calculations}{%
\subsection*{\texorpdfstring{\textbf{Calculations}}{Calculations}}\label{calculations}}
\addcontentsline{toc}{subsection}{\textbf{Calculations}}

The lotterhos partition has 2 nodes, 36 cores per node, and 5GB per
core. A job that only uses a single cpu should not take up more than 5GB
of memory.

Here is an example of an array that runs 72 jobs at a time, each job
runs on one core. In the remaining script, not shown, is code that reads
the input parameters from a file that has 1000 lines (the first line is
the header, and each line after that has parameters). Hence, the array
starts on line 2 and goes to line 1000. Unless the program is able to
cross-talk among nodes (most do not have the architecture to do that),
then nodes should always be set to 1. However, this script will run
arrays across both lotterhos nodes (the nodes=1 just indicates that each
task will use one node).

\begin{verbatim}
#SBATCH --partition=lotterhos
#SBATCH --mem=5GB
#SBATCH --nodes=1
#SBATCH --cpus-per-task=1
#SBATCH --array=2-1000%70
\end{verbatim}

1000 is the upper limit that can be requested by the ``array'' command.
It's also good to keep the total number of jobs submitted at a time
capped at 70, so that a couple of cores are left open for \texttt{srun}

Below is an example of a job that calls a program that can use multiple
cpus. If 32 cpus are requested, then 32 cpus * 5GB/cpu = 160GB can be
also requested for memory. Unless the program is able to cross-talk
among nodes (most do not have the architecture to do that), then nodes
should always be set to 1. y Since there are only 2 nodes, each with 36
cpus, then we can only run 2 jobs at a time and stay within the bounds
of the architecture. Hence the\texttt{array=2-1000\%2} argument

\begin{verbatim}
#SBATCH --partition=lotterhos
#SBATCH --mem=160GB
#SBATCH --nodes=1
#SBATCH --cpus-per-task=32
#SBATCH --array=2-1000%2
\end{verbatim}

\hypertarget{pratical-example}{%
\subsection*{\texorpdfstring{\textbf{Pratical
example}}{Pratical example}}\label{pratical-example}}
\addcontentsline{toc}{subsection}{\textbf{Pratical example}}

Script performs bismark mapping, \texttt{bismark} on list of files based
on the file names in a reference text file, \texttt{sample\_files.txt}.
Based on the \texttt{-\/-array} argument, I am peforming the mapping on
32 samples, processing 4 samples at a time.

\begin{verbatim}
    #!/bin/bash

    #SBATCH --job-name=DNAMethylation_mapping
    #SBATCH --mem=100Gb
    #SBATCH --mail-user=downey-wall.a@northeastern.edu
    #SBATCH --mail-type=FAIL
    #SBATCH --partition=lotterhos
    #SBATCH --nodes=1
    #SBATCH --cpus-per-task=32
    #SBATCH --array=1-32%2
    #SBATCH --output=/work/lotterhos/2018OALarvae_DNAm/slurm_log/DNAm_mapping_%j.out
    #SBATCH --error=/work/lotterhos/2018OALarvae_DNAm/slurm_log/DNAm_mapping_%j.err

    ## Store name of individual sample
    VARNAME=`sed "${SLURM_ARRAY_TASK_ID}q;d" 2018OALarvae_DNAm/RAW/sample_files.txt`

    ## Code for running bismark (DNA methylation sequence mapper)

    i=$(ls 2018OALarvae_DNAm/trimmed_files/${VARNAME}_R1*fq.gz | rev | cut -d'/' -f 1 | rev) 
    j=$(ls 2018OALarvae_DNAm/trimmed_files/${VARNAME}_R2*fq.gz | rev | cut -d'/' -f 1 | rev)

    # Run Bismark
    bismark --non_directional \
    --parallel 32 \
    --genome oyster_references/haplo_tig_genome \
    --score_min L,0,-0.4 \
    --bowtie2 \
    -1 2018OALarvae_DNAm/trimmed_files/${i} \
    -2 2018OALarvae_DNAm/trimmed_files/${j} \
    -o 2018OALarvae_DNAm/mapped_files
\end{verbatim}



\end{document}
