<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Lotterhos Lab Protocols – molecprot_splatlibraryprep</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


</head>

<body class="nav-sidebar floating nav-fixed">


<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">Lotterhos Lab Protocols</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link active" href="./index.html" aria-current="page">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./about.html">
 <span class="menu-text">About</span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title"></h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
      <a href="./index.html" class="sidebar-logo-link">
      <img src="./img/Lotterhos_logo2.png" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false">Welcome</a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_welcome.html" class="sidebar-item-text sidebar-link">Welcome</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_expectations.html" class="sidebar-item-text sidebar-link">Expectations</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_GettingStarted.html" class="sidebar-item-text sidebar-link">Getting Started in the Lab</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./code_quarto.html" class="sidebar-item-text sidebar-link">Editing These Protocols with Quarto</a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false">General</a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_ordering.html" class="sidebar-item-text sidebar-link">Ordering Supplies and Equipment</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_fedex_ship.html" class="sidebar-item-text sidebar-link">FedEx Account</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_Safety.html" class="sidebar-item-text sidebar-link">General Safety Protocols</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_ShippingHazMat.html" class="sidebar-item-text sidebar-link">Shipping of Hazardous Materials</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_purchasing.html" class="sidebar-item-text sidebar-link">Purchasing</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_receiving.html" class="sidebar-item-text sidebar-link">Receiving Supplies and Equipment</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_reimbursement.html" class="sidebar-item-text sidebar-link">Reimbursement</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_creating_barcoded_labels.html" class="sidebar-item-text sidebar-link">Creating Barcoded Labels</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_chemical_inventory.html" class="sidebar-item-text sidebar-link">Managing Chemical Inventory</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_waste_clearing.html" class="sidebar-item-text sidebar-link">Clearing Lab Waste</a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false">General Equipment</a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_autoclave.html" class="sidebar-item-text sidebar-link">Autoclave (OGL)</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_MVI-scope-camera.html" class="sidebar-item-text sidebar-link">Scope Camera MVI Nikon SMZ 1000</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_freezer.html" class="sidebar-item-text sidebar-link">Freezer Alarm System</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_OT2-Pipetting-Robot.html" class="sidebar-item-text sidebar-link">Opentrons OT-2 Pipetting Robot</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_shared_equipment.html" class="sidebar-item-text sidebar-link">NUMSC - Shared Lab Space</a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="false">Molecular Equipment</a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="false">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_take3dnaquantification.html" class="sidebar-item-text sidebar-link">Take3 DNA Quantification</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_picoGreenAssay.html" class="sidebar-item-text sidebar-link">PicoGreen Assay</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_magbeadwash.html" class="sidebar-item-text sidebar-link">Magnetic Bead Wash</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_tapestationlibrarycheck.html" class="sidebar-item-text sidebar-link">TapeStation Library Check</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_qPCRandBioRAD.html" class="sidebar-item-text sidebar-link">Molecular Protocols for Using the BioRAD qPCR Machine</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_outdated.html" class="sidebar-item-text sidebar-link">Outdated Molecular Protocols</a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="false">Professional Development</a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="false">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./profdev.html" class="sidebar-item-text sidebar-link">Professional Development Resources</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./profdev_ReducingCognitiveLoad.html" class="sidebar-item-text sidebar-link">Reducing Cognitive Load</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./profdev_How-to-publish-a-paper.html" class="sidebar-item-text sidebar-link">How to Publish a Paper</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./profdev_design.html" class="sidebar-item-text sidebar-link">Design</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./profdev_courses.html" class="sidebar-item-text sidebar-link">Graduate Courses</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./profdev_ColdBrew.html" class="sidebar-item-text sidebar-link">Chett’s Cold Brew</a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="false">Molecular Protocols</a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="false">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-6" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_etoh.html" class="sidebar-item-text sidebar-link">Types of Ethanol</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_barcodesnindexes.html" class="sidebar-item-text sidebar-link">Determining Barcodes and Indexes for Illumina Sequencing</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_dnaextractionomega.html" class="sidebar-item-text sidebar-link">DNA Extraction (Omega)</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_dnaextractionqiagen.html" class="sidebar-item-text sidebar-link">DNA Extraction (Qiagen)</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_dnaextractionqiagenplate.html" class="sidebar-item-text sidebar-link">Plate DNA Extraction (Qiagen)</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_dnaextractionspartina.html" class="sidebar-item-text sidebar-link">DNA Extraction (Spartina)</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_gelelectrophoresis.html" class="sidebar-item-text sidebar-link">Gel Electrophoresis</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_dnadrying.html" class="sidebar-item-text sidebar-link">molecprot_dnadrying.html</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_illuminalibraryprep.html" class="sidebar-item-text sidebar-link">Illumina Library Prep</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_qPCR-Libraryquant.html" class="sidebar-item-text sidebar-link">Library Quantification</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_rnaextraction.html" class="sidebar-item-text sidebar-link">RNA Extraction</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_solutionrecipes.html" class="sidebar-item-text sidebar-link">Solution Recipes</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_gonad_buffer.html" class="sidebar-item-text sidebar-link">Changing Gonad Buffer</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_Acid_Washes.html" class="sidebar-item-text sidebar-link">Cleaning the S-Blocks for reuse</a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" aria-expanded="false">Field and Sampling Protocols</a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" aria-expanded="false">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-7" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./field_collectsampdna.html" class="sidebar-item-text sidebar-link">Collecting Samples for DNA</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./field_collectsamprna.html" class="sidebar-item-text sidebar-link">Collecting Samples for RNA</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./field_conductivityloggers.html" class="sidebar-item-text sidebar-link">Conductivity Loggers</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./field_fieldsafety_protocol.html" class="sidebar-item-text sidebar-link">Field Safety Protocol</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./field_oystersampling.html" class="sidebar-item-text sidebar-link">Oyster Sampling</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./field_kayak.html" class="sidebar-item-text sidebar-link">Lab Kayak Care</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./field_phloggers.html" class="sidebar-item-text sidebar-link">pH Loggers</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./field_Seatables.html" class="sidebar-item-text sidebar-link">Managing Seatables</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./field_watersampling.html" class="sidebar-item-text sidebar-link">Water Sampling</a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" aria-expanded="false">Coding Protocols</a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" aria-expanded="false">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-8" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./code_bioinformatics.html" class="sidebar-item-text sidebar-link">Bioinformatics Resources</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./code_checksums.html" class="sidebar-item-text sidebar-link">Transferring Files md5</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./code_defiance.html" class="sidebar-item-text sidebar-link">Accessing and Using Defiance Cluster</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./code_directoryorganization.html" class="sidebar-item-text sidebar-link">General Directory Organization</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./code_githhubMarkdown.html" class="sidebar-item-text sidebar-link">GitHub Figures</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./code_LearningToCode.html" class="sidebar-item-text sidebar-link">Learning to Code</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./code_pathAndAliasMod.html" class="sidebar-item-text sidebar-link">Setting Alias or PATH variable</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./code_pseudoparallel.html" class="sidebar-item-text sidebar-link">Pseudo-Parallelize Your Code on the Cluster</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./code_setseeds.html" class="sidebar-item-text sidebar-link">Setting Seeds for Reproducibility</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./code_TroubleshootingCellprofiler.html" class="sidebar-item-text sidebar-link">Troubleshooting CellProfiler Installation</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./code_unixcommands.html" class="sidebar-item-text sidebar-link">Basic Unix Commands</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./code_versioncontrol.html" class="sidebar-item-text sidebar-link">Version Control with GitHub</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./code_quarto.html" class="sidebar-item-text sidebar-link">Editing These Protocols with Quarto</a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-9" aria-expanded="false">Discovery</a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-9" aria-expanded="false">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-9" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./discovery_1-gettingstarted.html" class="sidebar-item-text sidebar-link">Connecting to the Discovery Cluster</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./discovery_1a-OOD.html" class="sidebar-item-text sidebar-link">Open OnDemand</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./discovery_2-globus.html" class="sidebar-item-text sidebar-link">Globus File Transfer</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./discovery_3-storage.html" class="sidebar-item-text sidebar-link">Lotterhos Lab vs.&nbsp;Personal Storage on Discovery</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./discovery_4-examples.html" class="sidebar-item-text sidebar-link">Example Scripts</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./discovery_5-parallelization.html" class="sidebar-item-text sidebar-link">Workflow Parallelization Approaches</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./discovery_6-optimization.html" class="sidebar-item-text sidebar-link">Optimizing</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./discovery_slurmArrays.html" class="sidebar-item-text sidebar-link">Parallelization with Array</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./discovery_nonNEUpersonnel.html" class="sidebar-item-text sidebar-link">Instructions for Discovery for Non-Northeastern Personnel</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./discovery_integrated-development-environment.html" class="sidebar-item-text sidebar-link">What are VS Code and PyCharm?</a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-10" aria-expanded="false">Bioinformagics</a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-10" aria-expanded="false">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-10" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./bioinformagics_1_install_iterm.html" class="sidebar-item-text sidebar-link">install iTerm terminal app for MacOS</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./bioinformagics_2-3_advanced_sshing.html" class="sidebar-item-text sidebar-link">Advanced SSH-ing</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./bioinformagics_4_anaconda.html" class="sidebar-item-text sidebar-link">Installing Anaconda and Anaconda Environments</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./bioinformagics_6_jupyter.html" class="sidebar-item-text sidebar-link">jupyter Notebooks</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./bioinformagics_7_terminal_colors.html" class="sidebar-item-text sidebar-link">How to color code your terminal</a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-11" aria-expanded="false">Outreach and Teaching</a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-11" aria-expanded="false">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-11" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./outreach_oysterworkshop.html" class="sidebar-item-text sidebar-link">All About Oysters: Outreach Workshop</a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#splat-library-prep" id="toc-splat-library-prep" class="nav-link active" data-scroll-target="#splat-library-prep">SPLAT Library Prep</a></li>
  <li><a href="#background" id="toc-background" class="nav-link" data-scroll-target="#background">Background</a></li>
  <li><a href="#materials-and-equipment" id="toc-materials-and-equipment" class="nav-link" data-scroll-target="#materials-and-equipment">Materials and Equipment</a>
  <ul class="collapse">
  <li><a href="#dna-extractions" id="toc-dna-extractions" class="nav-link" data-scroll-target="#dna-extractions">1. DNA Extractions</a></li>
  <li><a href="#rad-digestion" id="toc-rad-digestion" class="nav-link" data-scroll-target="#rad-digestion">2. RAD digestion</a></li>
  <li><a href="#bisulfite-conversion" id="toc-bisulfite-conversion" class="nav-link" data-scroll-target="#bisulfite-conversion">3. BiSulfite Conversion</a></li>
  <li><a href="#splat-adapter-ligation" id="toc-splat-adapter-ligation" class="nav-link" data-scroll-target="#splat-adapter-ligation">4. SPLAT Adapter ligation</a></li>
  <li><a href="#bisulfite-pcr" id="toc-bisulfite-pcr" class="nav-link" data-scroll-target="#bisulfite-pcr">5. BiSulfite PCR</a></li>
  <li><a href="#size-selection" id="toc-size-selection" class="nav-link" data-scroll-target="#size-selection">6. Size Selection</a></li>
  <li><a href="#quality-control-steps" id="toc-quality-control-steps" class="nav-link" data-scroll-target="#quality-control-steps">6. Quality Control Steps</a></li>
  </ul></li>
  <li><a href="#overview" id="toc-overview" class="nav-link" data-scroll-target="#overview">Overview</a>
  <ul class="collapse">
  <li><a href="#dna-extractions-1" id="toc-dna-extractions-1" class="nav-link" data-scroll-target="#dna-extractions-1">1. DNA Extractions</a></li>
  <li><a href="#rad-digestion-1" id="toc-rad-digestion-1" class="nav-link" data-scroll-target="#rad-digestion-1">2. RAD digestion</a></li>
  <li><a href="#bisulfite-conversion-1" id="toc-bisulfite-conversion-1" class="nav-link" data-scroll-target="#bisulfite-conversion-1">3. BiSulfite Conversion</a></li>
  <li><a href="#splat-adapters" id="toc-splat-adapters" class="nav-link" data-scroll-target="#splat-adapters">4. SPLAT Adapters</a></li>
  <li><a href="#bisulfite-pcr-1" id="toc-bisulfite-pcr-1" class="nav-link" data-scroll-target="#bisulfite-pcr-1">4. Bisulfite PCR</a></li>
  <li><a href="#pooling" id="toc-pooling" class="nav-link" data-scroll-target="#pooling">5. Pooling</a></li>
  <li><a href="#quality-control-steps-1" id="toc-quality-control-steps-1" class="nav-link" data-scroll-target="#quality-control-steps-1">6. Quality Control Steps</a></li>
  </ul></li>
  <li><a href="#protocol" id="toc-protocol" class="nav-link" data-scroll-target="#protocol">Protocol</a>
  <ul class="collapse">
  <li><a href="#dna-extractions-2" id="toc-dna-extractions-2" class="nav-link" data-scroll-target="#dna-extractions-2">1. DNA Extractions</a></li>
  <li><a href="#rad-digestion-2" id="toc-rad-digestion-2" class="nav-link" data-scroll-target="#rad-digestion-2">2. RAD digestion</a></li>
  <li><a href="#bisulfite-conversion-2" id="toc-bisulfite-conversion-2" class="nav-link" data-scroll-target="#bisulfite-conversion-2">3. BiSulfite Conversion</a>
  <ul class="collapse">
  <li><a href="#acceptable-sample-types-and-recommended-input-amounts" id="toc-acceptable-sample-types-and-recommended-input-amounts" class="nav-link" data-scroll-target="#acceptable-sample-types-and-recommended-input-amounts">Acceptable sample types and recommended input amounts</a></li>
  <li><a href="#prepare-the-reagents-for-new-kits" id="toc-prepare-the-reagents-for-new-kits" class="nav-link" data-scroll-target="#prepare-the-reagents-for-new-kits">Prepare the reagents (for new kits)</a></li>
  <li><a href="#bisulfite-conversion-3" id="toc-bisulfite-conversion-3" class="nav-link" data-scroll-target="#bisulfite-conversion-3">Bisulfite conversion</a></li>
  <li><a href="#assess-the-yield-and-quality-of-the-bisulfite-converted-dna" id="toc-assess-the-yield-and-quality-of-the-bisulfite-converted-dna" class="nav-link" data-scroll-target="#assess-the-yield-and-quality-of-the-bisulfite-converted-dna">Assess the yield and quality of the bisulfite-converted DNA</a></li>
  </ul></li>
  <li><a href="#pnk-treatment-and-adapter-ligation" id="toc-pnk-treatment-and-adapter-ligation" class="nav-link" data-scroll-target="#pnk-treatment-and-adapter-ligation">4. PNK Treatment and Adapter Ligation</a>
  <ul class="collapse">
  <li><a href="#polynucleotide-kinase-pnk-treatment" id="toc-polynucleotide-kinase-pnk-treatment" class="nav-link" data-scroll-target="#polynucleotide-kinase-pnk-treatment">Polynucleotide kinase (PNK) treatment</a></li>
  <li><a href="#adapter-ligation-1" id="toc-adapter-ligation-1" class="nav-link" data-scroll-target="#adapter-ligation-1">Adapter Ligation 1</a></li>
  </ul></li>
  <li><a href="#kapa-hifi-hotstart-uracil-pcr" id="toc-kapa-hifi-hotstart-uracil-pcr" class="nav-link" data-scroll-target="#kapa-hifi-hotstart-uracil-pcr">5. KAPA HiFi Hotstart Uracil+ PCR</a>
  <ul class="collapse">
  <li><a href="#prepare-the-pcr-master-mix" id="toc-prepare-the-pcr-master-mix" class="nav-link" data-scroll-target="#prepare-the-pcr-master-mix">1. Prepare the PCR master mix</a></li>
  <li><a href="#set-up-individual-reactions" id="toc-set-up-individual-reactions" class="nav-link" data-scroll-target="#set-up-individual-reactions">2. Set up individual reactions</a></li>
  <li><a href="#run-the-pcr" id="toc-run-the-pcr" class="nav-link" data-scroll-target="#run-the-pcr">3. Run the PCR</a></li>
  <li><a href="#notes-and-troubleshooting" id="toc-notes-and-troubleshooting" class="nav-link" data-scroll-target="#notes-and-troubleshooting">4. Notes and troubleshooting</a></li>
  </ul></li>
  <li><a href="#pooling-1" id="toc-pooling-1" class="nav-link" data-scroll-target="#pooling-1">6. Pooling</a></li>
  <li><a href="#quality-control-steps-2" id="toc-quality-control-steps-2" class="nav-link" data-scroll-target="#quality-control-steps-2">7. Quality Control Steps</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">



<section id="splat-library-prep" class="level1">
<h1>SPLAT Library Prep</h1>
<p><a href="https://github.com/DrK-Lo/lotterhoslabprotocols/edit/main/molecprot_splatlibraryprep.qmd">Edit this page</a></p>
<p>The protocol for doing a library prep using SPLAT.</p>
<hr>
<p><em>Note: this protocol is outdated. Not sure if it works</em></p>
</section>
<section id="background" class="level1">
<h1>Background</h1>
</section>
<section id="materials-and-equipment" class="level1">
<h1>Materials and Equipment</h1>
<section id="dna-extractions" class="level2">
<h2 class="anchored" data-anchor-id="dna-extractions">1. DNA Extractions</h2>
<ul>
<li>For oyster extractions see: <a href="https://github.com/DrK-Lo/lab-wiki/wiki/3a-DNA-extraction-Omega">DNA Extraction - Omega</a></li>
</ul>
</section>
<section id="rad-digestion" class="level2">
<h2 class="anchored" data-anchor-id="rad-digestion">2. RAD digestion</h2>
<ul>
<li>See <a href="https://github.com/DrK-Lo/lab-wiki/wiki/5---ddRAD-library-prep#materials-and-equipment">ddRADseq Material and Methods</a></li>
</ul>
</section>
<section id="bisulfite-conversion" class="level2">
<h2 class="anchored" data-anchor-id="bisulfite-conversion">3. BiSulfite Conversion</h2>
<p>(based on 96 well plate conversion)</p>
<p><strong>Cells-to-CpG BiSulfite Conversion Kit (2x96 = 550$)</strong></p>
<table class="table">
<thead>
<tr class="header">
<th>Item</th>
<th>Amount</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Binding Buffer</td>
<td>1 x 192-rxn bottle</td>
</tr>
<tr class="even">
<td>Binding Plates</td>
<td>2 plates</td>
</tr>
<tr class="odd">
<td>Conversion Buffer</td>
<td>10 mL</td>
</tr>
<tr class="even">
<td>Conversion Reagent</td>
<td>2 x 96-rxn bottles</td>
</tr>
<tr class="odd">
<td>Denaturation Reagent</td>
<td>1.mL</td>
</tr>
<tr class="even">
<td>Desulfonation Reagent</td>
<td>1 x 192-rxn bottle</td>
</tr>
<tr class="odd">
<td>Elution Buffer</td>
<td>20 mL</td>
</tr>
<tr class="even">
<td>Elution Plates</td>
<td>2 plates</td>
</tr>
<tr class="odd">
<td>Lysis Enhancer</td>
<td>2 x 0.5 mL</td>
</tr>
<tr class="even">
<td>Wash Buffer</td>
<td>2 x 96-rxn bottles</td>
</tr>
</tbody>
</table>
<p><strong>Other Materials</strong></p>
<table class="table">
<thead>
<tr class="header">
<th>Item</th>
<th># Needed</th>
<th>Cost</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>MicroAmp Clear Adhesive Film</td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>MicroAmp Optical Film Compression Pad</td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>Nuclease-free Water (not DEPC-treated) (1x500mL)</td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Reaction Plates</td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>Ethanol, molecular grade, &gt;99.5% or 200 proof</td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Isopropyl alcohol, ACS, reagent grade, &gt;99.5%</td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>Microcentrifuge tubes (Size)</td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Pipette tips, nuclease-free (Sizes)</td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<p><strong>Equipment</strong></p>
<table class="table">
<thead>
<tr class="header">
<th>Items</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ThermoMixer</td>
</tr>
<tr class="even">
<td>Microcentrifuge</td>
</tr>
<tr class="odd">
<td>Pipettors (sizes)</td>
</tr>
<tr class="even">
<td>Vortex</td>
</tr>
<tr class="odd">
<td>Water bath 60<sup>o</sup>C</td>
</tr>
</tbody>
</table>
</section>
<section id="splat-adapter-ligation" class="level2">
<h2 class="anchored" data-anchor-id="splat-adapter-ligation">4. SPLAT Adapter ligation</h2>
</section>
<section id="bisulfite-pcr" class="level2">
<h2 class="anchored" data-anchor-id="bisulfite-pcr">5. BiSulfite PCR</h2>
<p><strong>KAPA HiFi HotStart Uracil+ ReadyMix</strong></p>
<table class="table">
<colgroup>
<col style="width: 16%">
<col style="width: 38%">
<col style="width: 44%">
</colgroup>
<thead>
<tr class="header">
<th>Item</th>
<th>Amount Total</th>
<th>Amount per Kit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>2X KAPA HiFi HotStart Uracil+ ReadyMix</td>
<td>1.25mL (50x50uL) or 6.25mL (250x50uL)</td>
<td>12.5 uL per single reaction</td>
</tr>
</tbody>
</table>
<p><strong>Other Materials</strong></p>
<table class="table">
<thead>
<tr class="header">
<th>Item</th>
<th># Needed</th>
<th>Cost</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>MicroAmp Clear Adhesive Film</td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>MicroAmp Optical Film Compression Pad</td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>Nuclease-free Water (not DEPC-treated) (1x500mL)</td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Ethanol, molecular grade, &gt;99.5% or 200 proof</td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>Microcentrifuge tubes (Size)</td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Pipette tips, nuclease-free (Sizes)</td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<p><strong>Equipment</strong></p>
<table class="table">
<thead>
<tr class="header">
<th>Items</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Thermocycler</td>
</tr>
<tr class="even">
<td>Microcentrifuge</td>
</tr>
<tr class="odd">
<td>Pipettors (sizes)</td>
</tr>
<tr class="even">
<td>Vortex</td>
</tr>
</tbody>
</table>
</section>
<section id="size-selection" class="level2">
<h2 class="anchored" data-anchor-id="size-selection">6. Size Selection</h2>
<p>Depending on whether the pippin needs dsDNA (i.e.&nbsp;post-PCR amplicons). This step could occur after adapter ligation after samples have been pooled by barcode.</p>
<table class="table">
<thead>
<tr class="header">
<th>Items</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Pippin (Sage Science)</td>
</tr>
</tbody>
</table>
</section>
<section id="quality-control-steps" class="level2">
<h2 class="anchored" data-anchor-id="quality-control-steps">6. Quality Control Steps</h2>
</section>
</section>
<section id="overview" class="level1">
<h1>Overview</h1>
<p>SPLAT method uses splinted adapters that ligate to ssDNA created during the bisulfite conversion process. These adapters bind to the ssDNA using a generalized or specific hexamer (i.e.&nbsp;NNNNNN) overhang. Using a ddRAD protocol, gDNA is digested into small dsDNA fragments with predictable overhangs that can be targeted using specifically designed hexamer overhangs within either/both of the adapters. <strong>Care should be taken in designing overhangs that compliment the specific restriction enzyme design.</strong> See figure below for a complete workflow of the SPLAT protocol. <img src="https://docs.google.com/drawings/d/1TDOZWNagpkqOpV0hqWujv4ZQvZBTPbTKrbEqg5W1U5c/pub?w=1440&amp;h=1080.png" class="img-fluid" alt="SPLAT4"></p>
<section id="dna-extractions-1" class="level2">
<h2 class="anchored" data-anchor-id="dna-extractions-1">1. DNA Extractions</h2>
<ul>
<li>See <a href="https://github.com/DrK-Lo/lab-wiki/wiki/3-DNA-extraction">3 DNA Extraction - Overview</a></li>
</ul>
</section>
<section id="rad-digestion-1" class="level2">
<h2 class="anchored" data-anchor-id="rad-digestion-1">2. RAD digestion</h2>
<ul>
<li>See <a href="https://github.com/DrK-Lo/lab-wiki/wiki/5---ddRAD-library-prep#overview">ddRADseq - Overview</a></li>
</ul>
</section>
<section id="bisulfite-conversion-1" class="level2">
<h2 class="anchored" data-anchor-id="bisulfite-conversion-1">3. BiSulfite Conversion</h2>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://tools.thermofisher.com/content/sfs/gallery/high/bisulfite1.jpg" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">BiSulfite Conversion</figcaption><p></p>
</figure>
</div>
</section>
<section id="splat-adapters" class="level2">
<h2 class="anchored" data-anchor-id="splat-adapters">4. SPLAT Adapters</h2>
<p>SPLinted Adapters are largely dsDNA adapters with a hexamer overhang that will bind with a ssDNA created during bisulfite conversion. The adapter container the appropriate sequence necessary for successful downstream PCR amplification. The hexamer overhang in the original protocol (Raines et al.&nbsp;2016) was a randomly generated hexamer (NNNNNN) that would bind with complimentary 6 base ends (both the 3’ and 5’ end of the ssDNA). This works, but is likely highly inefficient. With a ddRAD protocol, known 3’ and 5’ end sequences are produced. We take advantage of this by designing hexamers that will specifically target these sequences. However, for this to work special care is needed</p>
</section>
<section id="bisulfite-pcr-1" class="level2">
<h2 class="anchored" data-anchor-id="bisulfite-pcr-1">4. Bisulfite PCR</h2>
<p>KAPA HiFi HotStart is a novel B-family DNA polymerase, engineered to have increased affinity for DNA, without the need for accessory proteins or DNA-binding domains. The intrinsic high processivity of the enzyme results in significant improvement in yield, speed and sensitivity when compared with wild-type B-family DNA polymerases and polymerase blends. When used for the amplification of next-generation sequencing (NGS) libraries, KAPA HiFi HotStart DNA Polymerase exhibits high yields with minimal amplification bias and provides extremely uniform sequence coverage.</p>
<p>The read-ahead function of proofreading DNA polymerases detects pro-mutagenic uracil residues in the template strand and prevents further strand extension, thereby reducing or completely inhibiting PCR amplification. In KAPA HiFi HotStart Uracil+ DNA Polymerase, this uracilbinding pocket is inactivated to enable the amplification of uracil-containing DNA. The enzyme shows the same high yield, low GC-bias and coverage uniformity as the unmodified KAPA HiFi HotStart DNA Polymerase, making it particularly advantageous for applications employing bisulfite DNA conversion, which typically produces low concentrations of AT-rich DNA.</p>
<p>KAPA HiFi HotStart Uracil+ DNA Polymerase has 5’g3’ polymerase and 3’g5’ exonuclease (proofreading) activity, but no 5’g3’ exonuclease activity. The strong 3’g5’ exonuclease activity results in superior accuracy during DNA amplification. A proprietary antibody inactivates the polymerase until the first cycle of denaturation, minimizing spurious amplification products that may result from nonspecific priming events during reaction setup and initiation, and increasing overall reaction efficiency. KAPA HiFi HotStart Uracil+ ReadyMix (2X) is a ready-touse cocktail containing all components required for PCR, except primers and template. The ReadyMix contains 0.2 mM of each dNTP (dATP, dCTP, dGTP, dTTP), and does not contain dUTP.</p>
</section>
<section id="pooling" class="level2">
<h2 class="anchored" data-anchor-id="pooling">5. Pooling</h2>
<ul>
<li>See <a href="https://github.com/DrK-Lo/lab-wiki/wiki/5---ddRAD-library-prep#overview">ddRADseq - Overview</a></li>
</ul>
</section>
<section id="quality-control-steps-1" class="level2">
<h2 class="anchored" data-anchor-id="quality-control-steps-1">6. Quality Control Steps</h2>
</section>
</section>
<section id="protocol" class="level1">
<h1>Protocol</h1>
<section id="dna-extractions-2" class="level2">
<h2 class="anchored" data-anchor-id="dna-extractions-2">1. DNA Extractions</h2>
<p>for oyster extractions see <a href="https://github.com/DrK-Lo/lab-wiki/wiki/3a-DNA-extraction-Omega#protocol">3a DNA Extraction Omega - Protocol</a></p>
</section>
<section id="rad-digestion-2" class="level2">
<h2 class="anchored" data-anchor-id="rad-digestion-2">2. RAD digestion</h2>
<p>see <a href="https://github.com/DrK-Lo/lab-wiki/wiki/5---ddRAD-library-prep">‘5 ddRAD seq’</a> for details on ddRAD digest.</p>
</section>
<section id="bisulfite-conversion-2" class="level2">
<h2 class="anchored" data-anchor-id="bisulfite-conversion-2">3. BiSulfite Conversion</h2>
<p>Link to <a href="http://www.zymoresearch.com/downloads/dl/file/id/490/d5030i.pdf">Zymo EZ DNA Methylation-lightning bisulfite conversion kit.</a></p>
<section id="acceptable-sample-types-and-recommended-input-amounts" class="level3">
<h3 class="anchored" data-anchor-id="acceptable-sample-types-and-recommended-input-amounts">Acceptable sample types and recommended input amounts</h3>
<table class="table">
<colgroup>
<col style="width: 14%">
<col style="width: 28%">
<col style="width: 28%">
<col style="width: 28%">
</colgroup>
<thead>
<tr class="header">
<th>Input type</th>
<th>Optimal input amount</th>
<th>Minimum input amount</th>
<th>Maximum input amount</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Purified gDNA</td>
<td>200-500ng</td>
<td>100pg</td>
<td>2ug</td>
</tr>
</tbody>
</table>
</section>
<section id="prepare-the-reagents-for-new-kits" class="level3">
<h3 class="anchored" data-anchor-id="prepare-the-reagents-for-new-kits">Prepare the reagents (for new kits)</h3>
<p>Before you use a new kit, prepare the Binding Buffer, Wash Buffer, and Desulfonation Reagent.</p>
<p>Add 24 mL ethanol (&gt;99.5% or 200 proof) to 6 mL bottle of M-Wash Buffer concentrate, then swirl the bottles to mix. <em>Note: This amount will depend on the kit. Make sure you add the correct amount of EtoH for the kit you are using!</em></p>
<p>After you prepare the reagents, you will have sufficient reagents for 192 reactions (two 96-well plates).</p>
</section>
<section id="bisulfite-conversion-3" class="level3">
<h3 class="anchored" data-anchor-id="bisulfite-conversion-3">Bisulfite conversion</h3>
<ol type="1">
<li><p>Add 130 μl of Lightning Conversion Reagent to 20 μl of a DNA sample in a PCR tube. Mix, then centrifuge briefly to ensure there are no droplets in the cap or sides of the tube.</p>
<p><strong>Note:</strong> If the volumne of DNA is less than 20uL, compensate with water.</p></li>
<li><p>Place the PCR tube in a thermal cycler and perform the following steps:</p>
<table class="table">
<thead>
<tr class="header">
<th>Temp</th>
<th>Time</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>98<sup>o</sup>C</td>
<td>8 minutes</td>
</tr>
<tr class="even">
<td>54<sup>o</sup>C</td>
<td>60 minutes</td>
</tr>
<tr class="odd">
<td>4<sup>o</sup>C</td>
<td>up to 20 hours</td>
</tr>
</tbody>
</table></li>
</ol>
<p><strong>STOPPING POINT:</strong> If you cannot proceed with desalting and desulfonation immediately, you can store the converted sample at -20<sup>o</sup>C for up to 2 hours. Thaw the samples at room temp, then proceed with desalting and desulfonation.</p>
<p><img src="http://rs870.pbsrc.com/albums/ab261/rainbowquery/STOP.gif~c200" width="30"></p>
<ol start="3" type="1">
<li><p>Add 600 μl of M-Binding Buffer to a Zymo-Spin™ IC Column and place the column into a provided Collection Tube.</p></li>
<li><p>Load the sample (from Step 2) into the Zymo-Spin™ IC Column containing the M-Binding Buffer. Close the cap and mix by inverting the column several times.</p></li>
<li><p>Centrifuge at full speed (&gt; 10,000 x g) for 30 seconds. Discard the flow-through.</p></li>
<li><p>Add 100 μl of M-Wash Buffer to the column. Centrifuge at full speed for 30 seconds.</p></li>
<li><p>Add 200 μl of L-Desulphonation Buffer to the column and let stand at room temperature (20-30°C) for 15-20 minutes. After the incubation, centrifuge at full speed for 30 seconds.</p></li>
<li><p>Add 200 μl of M-Wash Buffer to the column. Centrifuge at full speed for 30 seconds. Repeat this wash step.</p></li>
<li><p>Place the column into a 1.5 ml microcentrifuge tube and add 10 μl of M-Elution Buffer directly to the column matrix. Centrifuge for 30 seconds at full speed to elute the DNA.</p></li>
</ol>
<p>The DNA is ready for immediate analysis or can be stored at or below -20°C for later use. For long term storage, store at or below -70°C. We recommend using 1-4 μl of eluted DNA for each PCR, however, up to 10 μl can be used if necessary. The elution volume can be &gt; 10 μl depending on the requirements of your experiments, but small elution volumes will yield higher DNA concentrations.</p>
</section>
<section id="assess-the-yield-and-quality-of-the-bisulfite-converted-dna" class="level3">
<h3 class="anchored" data-anchor-id="assess-the-yield-and-quality-of-the-bisulfite-converted-dna">Assess the yield and quality of the bisulfite-converted DNA</h3>
</section>
</section>
<section id="pnk-treatment-and-adapter-ligation" class="level2">
<h2 class="anchored" data-anchor-id="pnk-treatment-and-adapter-ligation">4. PNK Treatment and Adapter Ligation</h2>
<section id="polynucleotide-kinase-pnk-treatment" class="level3">
<h3 class="anchored" data-anchor-id="polynucleotide-kinase-pnk-treatment">Polynucleotide kinase (PNK) treatment</h3>
<p>In pcr tube, combine:</p>
<table class="table">
<thead>
<tr class="header">
<th>Component</th>
<th>Amount</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>bs-treated DNA</td>
<td>10uL</td>
</tr>
<tr class="even">
<td>T4 ligase buffer (30U/uL; Thermo Fisher)</td>
<td>1uL</td>
</tr>
<tr class="odd">
<td>Nuclease free H20</td>
<td>3.5uL</td>
</tr>
<tr class="even">
<td>PNK (10U/uL;Thermo Fisher Scientific)</td>
<td>0.5uL</td>
</tr>
</tbody>
</table>
<p>Heat in thermocycler for 15min at 37C, then heated for 4 min at 95C (w/ heated lid). Afterwards cool the mixture using an ice bath.</p>
</section>
<section id="adapter-ligation-1" class="level3">
<h3 class="anchored" data-anchor-id="adapter-ligation-1">Adapter Ligation 1</h3>
<p>In pcr tube, combine:</p>
</section>
</section>
<section id="kapa-hifi-hotstart-uracil-pcr" class="level2">
<h2 class="anchored" data-anchor-id="kapa-hifi-hotstart-uracil-pcr">5. KAPA HiFi Hotstart Uracil+ PCR</h2>
<p><strong>IMPORTANT!</strong> The KAPA HiFi HotStart Uracil+ ReadyMix contains an engineered B-family (proofreading) DNA polymerase and uniquely-formulated buffers, and requires specialized reaction conditions. If these conditions are not adhered to, reaction failure is likely. Refer to Important Parameters for more information.</p>
<section id="prepare-the-pcr-master-mix" class="level3">
<h3 class="anchored" data-anchor-id="prepare-the-pcr-master-mix">1. Prepare the PCR master mix</h3>
<p>1.1 KAPA HiFi HotStart Uracil+ reactions MUST be set up on ice since the high proofreading activity of the enzyme will result in rapid primer degradation at room temperature.</p>
<p>1.2 Ensure that all reagents are properly thawed and mixed.</p>
<p>1.3 Prepare a PCR master mix containing the appropriate volume of all reaction components common to all or a subset of reactions to be performed.</p>
<p>1.4 Calculate the required volumes of each component based on the following table:</p>
<table class="table">
<thead>
<tr class="header">
<th>Component</th>
<th>25 uL reaction</th>
<th>Final conc.</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PCR-grade water</td>
<td>Up to 25 uL</td>
<td>N/A</td>
</tr>
<tr class="even">
<td>2X KAPA HiFi Hotstart Uracil+ ReadyMix</td>
<td>12.5 uL</td>
<td>1X</td>
</tr>
<tr class="odd">
<td>10 uM Forward Primer</td>
<td>0.75 uL</td>
<td>0.3 uM</td>
</tr>
<tr class="even">
<td>10 uM Reverse Primer</td>
<td>0.75 uL</td>
<td>0.3 uM</td>
</tr>
<tr class="odd">
<td>Template DNA</td>
<td>As required</td>
<td>As required</td>
</tr>
</tbody>
</table>
</section>
<section id="set-up-individual-reactions" class="level3">
<h3 class="anchored" data-anchor-id="set-up-individual-reactions">2. Set up individual reactions</h3>
<p>2.1 Transfer the appropriate volumes of PCR master mix, template and primer to individual PCR tubes or wells of a PCR plate. 2.2 Cap or seal individual reactions, mix and centrifuge briefly.</p>
</section>
<section id="run-the-pcr" class="level3">
<h3 class="anchored" data-anchor-id="run-the-pcr">3. Run the PCR</h3>
<p>3.1 Perform PCR with the following cycling protocol:</p>
<table class="table">
<thead>
<tr class="header">
<th>Step</th>
<th>Temp.</th>
<th>Duration</th>
<th>Cycles</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Initial denaturation</td>
<td>95<sup>o</sup>C</td>
<td>3 min</td>
<td>1</td>
</tr>
<tr class="even">
<td>Denaturation</td>
<td>98<sup>o</sup>C</td>
<td>20 sec</td>
<td>15-36</td>
</tr>
<tr class="odd">
<td>Annealing</td>
<td>60<sup>o</sup>C-75<sup>o</sup>C</td>
<td>15 sec</td>
<td>15-35</td>
</tr>
<tr class="even">
<td>Extension</td>
<td>72<sup>o</sup>C</td>
<td>15-60 sec/kb</td>
<td>15-35</td>
</tr>
<tr class="odd">
<td>Final Extension</td>
<td>72<sup>o</sup>C</td>
<td>1 min/kb</td>
<td>1</td>
</tr>
<tr class="even">
<td>Hold</td>
<td>4<sup>o</sup>C</td>
<td>Infinite</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="notes-and-troubleshooting" class="level3">
<h3 class="anchored" data-anchor-id="notes-and-troubleshooting">4. Notes and troubleshooting</h3>
<section id="amplification-of-bisulfite-converted-dna" class="level5">
<h5 class="anchored" data-anchor-id="amplification-of-bisulfite-converted-dna">Amplification of bisulfite-converted DNA</h5>
<p>Amplification of bisulfite-treated DNA can be problematic due to DNA damage arising from the harsh conditions required for near-complete conversion of unmethylated cytosines. Conversion conditions should be optimized to minimize the extent of DNA damage while ensuring sufficient conversion efficiency. Because bisulfite treatment converts cytosines to uracils, which are subsequently substituted with thymines during PCR, bisulfite-converted DNA is typically much more AT-rich than the original source DNA. This should be considered carefully when designing PCR primers, and cycling parameters such as annealing temperature may require special attention.</p>
</section>
<section id="mgcl2-concentration" class="level5">
<h5 class="anchored" data-anchor-id="mgcl2-concentration">MgCl2 concentration</h5>
<p>KAPA HiFi HotStart Uracil+ ReadyMix contains a final (1X) MgCl2 concentration of 2.5 mM, which is sufficient for most applications. Applications which are likely to require higher MgCl2 concentrations include long PCR (&gt;10 kb) and AT-rich PCR, as well as amplification using primers with a low GC content (&lt;40%). Note that bisulfiteconverted DNA is typically very AT-rich, as are the primers used for amplification of these templates; this application may therefore require additional MgCl2.</p>
</section>
<section id="prevention-of-amplicon-contamination-with-udg" class="level5">
<h5 class="anchored" data-anchor-id="prevention-of-amplicon-contamination-with-udg">Prevention of amplicon contamination with UDG</h5>
<p>For effective removal of carryover contamination, dUTP must be added routinely to PCR reactions with KAPA HiFi HotStart Uracil+ ReadyMix. Typically, dUTP is added to a concentration of 0.2 mM, but short AT-poor amplicons may require up to 0.3 mM dUTP for effective removal using UDG. For amplicons &gt;600 bp, a lower dUTP concentration (≤ 0.1 mM) may be required for efficient amplification. Low amplification efficiencies may result from incomplete UDG inactivation, in which case a longer initial denaturation may be required to inactivate the UDG prior to cycling. Refer to the manufacturer’s recommendations for optimal UDG concentration, and incubation/inactivation conditions.</p>
</section>
<section id="denaturation" class="level5">
<h5 class="anchored" data-anchor-id="denaturation">Denaturation</h5>
<p>Due to the high salt concentration of the KAPA HiFi HotStart Uracil+ ReadyMix, it is important to use appropriate denaturation parameters. An initial denaturation time of 2–5 min at 95°C is recommended to ensure that complex template DNA is fully denatured before the first primer annealing step. Use 5 min for complex, genomic DNA and/ or GC-rich targets, and at least 45 sec for less complex templates such as purified viral or plasmid DNA.</p>
</section>
<section id="annealing-temperature" class="level5">
<h5 class="anchored" data-anchor-id="annealing-temperature">Annealing temperature</h5>
<p>Due to the high salt concentration of the KAPA HiFi HotStart Uracil+ ReadyMix, the optimal annealing temperature for a given primer set is usually different when compared with other PCR buffer systems. When using the kit with a specific primer pair for the first time, determine the optimal annealing temperature with annealing temperature gradient PCR. We recommend a gradient from 60–72°C, although some assays may require even higher annealing temperatures. For assays with optimal annealing temperatures of 68°C or higher, 2-step cycling may be performed at the optimal annealing temperature. Optimal annealing temperatures below 60°C are typically rare, but may be necessary when using primers with a high AT content, as is often the case with amplification of bisulfite-converted DNA. If a gradient PCR is not feasible, use an annealing temperature of 60°C as a first approach, and adjust the annealing temperature based on the results obtained: • If a low yield of only the specific product is obtained, lower the annealing temperature in 1–2°C increments. • If nonspecific products are formed in addition to the specific product, increase the annealing temperature in 1–2°C increments. • If no product is formed (specific or nonspecific), reduce the annealing temperature by 5°C. MgCl2 concentration may have to be increased. If only nonspecific products are formed (in a ladder-like pattern), increase the annealing temperature by 5°C.</p>
</section>
<section id="amplicon-length" class="level5">
<h5 class="anchored" data-anchor-id="amplicon-length">Amplicon length</h5>
<p>KAPA HiFi HotStart Uracil+ ReadyMix can amplify targets up to 18 kb in length from simple purified template such as plasmid DNA, and up to 15 kb in length from high quality (i.e.&nbsp;not bisulfite-treated), complex genomic DNA. For efficient amplification of fragments ≥10 kb, higher template concentrations, and optimization of the Mg2+ concentration, may be required.</p>
</section>
<section id="primer-and-template-dna-quality" class="level5">
<h5 class="anchored" data-anchor-id="primer-and-template-dna-quality">Primer and template DNA quality</h5>
<p>Primers should be designed to minimize the possibility of primer-dimer formation, self-priming, or nonspecific priming. Primer pairs should have similar theoretical melting temperatures (Tm), and should have a GC content of 40–60%, although this may not be feasible for bisulfite-converted template DNA. Primers with a GC content 60% may require higher denaturation temperatures or longer denaturation times, while primers with a GC content 40% may require lower annealing temperatures and increased MgCl2 concentrations. Template DNA quality has a significant impact on the success of PCR amplification. Degraded, damaged, or sheared template DNA is usually problematic. While KAPA HiFi HotStart Uracil+ ReadyMix tolerates uracil, deamination of dCMP to dUMP in the DNA template will generate G/C to A/T mutations during amplification. NOTE: Always dilute and store primers and DNA in a buffered solution (e.g.&nbsp;10 mM Tris-HCl, pH 8.0 – 8.5) instead of PCR-grade water, and minimize freeze-thaw cycles to limit degradation and maintain primer quality.</p>
</section>
<section id="cycle-number" class="level5">
<h5 class="anchored" data-anchor-id="cycle-number">Cycle number</h5>
<p>Excessive library amplification should be avoided to minimize the following adverse effects: • increased duplicate reads • uneven coverage depth and sequence dropout • chimeric library inserts • nucleotide substitutions • heteroduplex formation To minimize over-amplification and associated unwanted artifacts, the number of amplification cycles should be optimized to ensure a sufficient amount of amplified library for the next step in the workflow (capture or sequencing), plus the amount needed for library QC and/or archiving. Depending on the sequencing application and degree of multiplexing, 100 ng – 1.5 μg of amplified library is typically required. The number of cycles typically required will vary, depending on input amount and quality. Size selection of libraries at any part in the library construction process results in significant loss of material and as a result, 2–4 additional cycles are required for workflows which include a sizeselection step prior to library amplification.</p>
</section>
<section id="primer-depletion-and-library-over-amplification" class="level5">
<h5 class="anchored" data-anchor-id="primer-depletion-and-library-over-amplification">Primer depletion and library over-amplification</h5>
<p>In library amplification reactions (set up according to the recommended protocol), primers are typically depleted before dNTPs. When DNA synthesis can no longer take place due to primer depletion, subsequent rounds of DNA denaturation and annealing result in the separation of complementary DNA strands, followed by imperfect annealing to non-complementary partners. This presumably results in the formation of so-called “daisychains” or tangled knots, comprising large assemblies of improperly annealed, partially double-stranded, heteroduplex DNA. These species migrate slower and are observed as secondary, higher molecular weight peaks during the electrophoretic analysis of amplified libraries. However, they are typically comprised of library molecules of the desired length, which are separated during denaturation prior to target enrichment (capture) or cluster amplification. Since these heteroduplexes contain significant portions of single-stranded DNA, over-amplification leads to the under-quantification of library molecules with assays employing dsDNA-binding dyes. qPCR-based library quantification methods, such as that employed by the KAPA Library Quantification Kit, quantify DNA by denaturation and amplification, thereby providing a more accurate measurement of the amount of adapter-ligated molecules—even if the library was over-amplified.</p>
</section>
</section>
</section>
<section id="pooling-1" class="level2">
<h2 class="anchored" data-anchor-id="pooling-1">6. Pooling</h2>
</section>
<section id="quality-control-steps-2" class="level2">
<h2 class="anchored" data-anchor-id="quality-control-steps-2">7. Quality Control Steps</h2>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>