<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Lotterhos Lab Protocols – molecprot_ddradlibprep</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


</head>

<body class="nav-sidebar floating nav-fixed">


<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">Lotterhos Lab Protocols</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link active" href="./index.html" aria-current="page">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./about.html">
 <span class="menu-text">About</span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title"></h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
      <a href="./index.html" class="sidebar-logo-link">
      <img src="./img/Lotterhos_logo2.png" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false">Welcome</a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_welcome.html" class="sidebar-item-text sidebar-link">Welcome</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_expectations.html" class="sidebar-item-text sidebar-link">Expectations</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_GettingStarted.html" class="sidebar-item-text sidebar-link">Getting Started in the Lab</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./code_quarto.html" class="sidebar-item-text sidebar-link">Editing These Protocols with Quarto</a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false">General</a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_ordering.html" class="sidebar-item-text sidebar-link">Ordering Supplies and Equipment</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_fedex_ship.html" class="sidebar-item-text sidebar-link">FedEx Account</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_Safety.html" class="sidebar-item-text sidebar-link">General Safety Protocols</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_ShippingHazMat.html" class="sidebar-item-text sidebar-link">Shipping of Hazardous Materials</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_purchasing.html" class="sidebar-item-text sidebar-link">Purchasing</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_receiving.html" class="sidebar-item-text sidebar-link">Receiving Supplies and Equipment</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_reimbursement.html" class="sidebar-item-text sidebar-link">Reimbursement</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_creating_barcoded_labels.html" class="sidebar-item-text sidebar-link">Creating Barcoded Labels</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_chemical_inventory.html" class="sidebar-item-text sidebar-link">Managing Chemical Inventory</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_waste_clearing.html" class="sidebar-item-text sidebar-link">Clearing Lab Waste</a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false">General Equipment</a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_autoclave.html" class="sidebar-item-text sidebar-link">Autoclave (OGL)</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_MVI-scope-camera.html" class="sidebar-item-text sidebar-link">Scope Camera MVI Nikon SMZ 1000</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_freezer.html" class="sidebar-item-text sidebar-link">Freezer Alarm System</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_OT2-Pipetting-Robot.html" class="sidebar-item-text sidebar-link">Opentrons OT-2 Pipetting Robot</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general_shared_equipment.html" class="sidebar-item-text sidebar-link">NUMSC - Shared Lab Space</a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="false">Molecular Equipment</a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="false">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_take3dnaquantification.html" class="sidebar-item-text sidebar-link">Take3 DNA Quantification</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_picoGreenAssay.html" class="sidebar-item-text sidebar-link">PicoGreen Assay</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_magbeadwash.html" class="sidebar-item-text sidebar-link">Magnetic Bead Wash</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_tapestationlibrarycheck.html" class="sidebar-item-text sidebar-link">TapeStation Library Check</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_qPCRandBioRAD.html" class="sidebar-item-text sidebar-link">Molecular Protocols for Using the BioRAD qPCR Machine</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_outdated.html" class="sidebar-item-text sidebar-link">Outdated Molecular Protocols</a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="false">Professional Development</a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="false">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./profdev.html" class="sidebar-item-text sidebar-link">Professional Development Resources</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./profdev_ReducingCognitiveLoad.html" class="sidebar-item-text sidebar-link">Reducing Cognitive Load</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./profdev_How-to-publish-a-paper.html" class="sidebar-item-text sidebar-link">How to Publish a Paper</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./profdev_design.html" class="sidebar-item-text sidebar-link">Design</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./profdev_courses.html" class="sidebar-item-text sidebar-link">Graduate Courses</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./profdev_ColdBrew.html" class="sidebar-item-text sidebar-link">Chett’s Cold Brew</a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="false">Molecular Protocols</a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="false">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-6" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_etoh.html" class="sidebar-item-text sidebar-link">Types of Ethanol</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_barcodesnindexes.html" class="sidebar-item-text sidebar-link">Determining Barcodes and Indexes for Illumina Sequencing</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_dnaextractionomega.html" class="sidebar-item-text sidebar-link">DNA Extraction (Omega)</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_dnaextractionqiagen.html" class="sidebar-item-text sidebar-link">DNA Extraction (Qiagen)</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_dnaextractionqiagenplate.html" class="sidebar-item-text sidebar-link">Plate DNA Extraction (Qiagen)</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_dnaextractionspartina.html" class="sidebar-item-text sidebar-link">DNA Extraction (Spartina)</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_gelelectrophoresis.html" class="sidebar-item-text sidebar-link">Gel Electrophoresis</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_illuminalibraryprep.html" class="sidebar-item-text sidebar-link">Illumina Library Prep</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_qPCR-Libraryquant.html" class="sidebar-item-text sidebar-link">Library Quantification</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_rnaextraction.html" class="sidebar-item-text sidebar-link">RNA Extraction</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_solutionrecipes.html" class="sidebar-item-text sidebar-link">Solution Recipes</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecprot_gonad_buffer.html" class="sidebar-item-text sidebar-link">Changing Gonad Buffer</a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" aria-expanded="false">Field and Sampling Protocols</a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" aria-expanded="false">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-7" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./field_collectsampdna.html" class="sidebar-item-text sidebar-link">Collecting Samples for DNA</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./field_collectsamprna.html" class="sidebar-item-text sidebar-link">Collecting Samples for RNA</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./field_conductivityloggers.html" class="sidebar-item-text sidebar-link">Conductivity Loggers</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./field_fieldsafety_protocol.html" class="sidebar-item-text sidebar-link">Field Safety Protocol</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./field_oystersampling.html" class="sidebar-item-text sidebar-link">Oyster Sampling</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./field_kayak.html" class="sidebar-item-text sidebar-link">Lab Kayak Care</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./field_phloggers.html" class="sidebar-item-text sidebar-link">pH Loggers</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./field_Seatables.html" class="sidebar-item-text sidebar-link">Managing Seatables</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./field_watersampling.html" class="sidebar-item-text sidebar-link">Water Sampling</a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" aria-expanded="false">Coding Protocols</a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" aria-expanded="false">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-8" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./code_bioinformatics.html" class="sidebar-item-text sidebar-link">Bioinformatics Resources</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./code_checksums.html" class="sidebar-item-text sidebar-link">Transferring Files md5</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./code_defiance.html" class="sidebar-item-text sidebar-link">Accessing and Using Defiance Cluster</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./code_directoryorganization.html" class="sidebar-item-text sidebar-link">General Directory Organization</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./code_githhubMarkdown.html" class="sidebar-item-text sidebar-link">GitHub Figures</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./code_LearningToCode.html" class="sidebar-item-text sidebar-link">Learning to Code</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./code_pathAndAliasMod.html" class="sidebar-item-text sidebar-link">Setting Alias or PATH variable</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./code_pseudoparallel.html" class="sidebar-item-text sidebar-link">Pseudo-Parallelize Your Code on the Cluster</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./code_setseeds.html" class="sidebar-item-text sidebar-link">Setting Seeds for Reproducibility</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./code_TroubleshootingCellprofiler.html" class="sidebar-item-text sidebar-link">Troubleshooting CellProfiler Installation</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./code_unixcommands.html" class="sidebar-item-text sidebar-link">Basic Unix Commands</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./code_versioncontrol.html" class="sidebar-item-text sidebar-link">Version Control with GitHub</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./code_quarto.html" class="sidebar-item-text sidebar-link">Editing These Protocols with Quarto</a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-9" aria-expanded="false">Discovery</a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-9" aria-expanded="false">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-9" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./discovery_1-gettingstarted.html" class="sidebar-item-text sidebar-link">Connecting to the Discovery Cluster</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./discovery_1a-OOD.html" class="sidebar-item-text sidebar-link">Open OnDemand</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./discovery_2-globus.html" class="sidebar-item-text sidebar-link">Globus File Transfer</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./discovery_3-storage.html" class="sidebar-item-text sidebar-link">Lotterhos Lab vs.&nbsp;Personal Storage on Discovery</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./discovery_4-examples.html" class="sidebar-item-text sidebar-link">Example Scripts</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./discovery_5-parallelization.html" class="sidebar-item-text sidebar-link">Workflow Parallelization Approaches</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./discovery_6-optimization.html" class="sidebar-item-text sidebar-link">Optimizing</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./discovery_slurmArrays.html" class="sidebar-item-text sidebar-link">Parallelization with Array</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./discovery_nonNEUpersonnel.html" class="sidebar-item-text sidebar-link">Instructions for Discovery for Non-Northeastern Personnel</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./discovery_integrated-development-environment.html" class="sidebar-item-text sidebar-link">What are VS Code and PyCharm?</a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-10" aria-expanded="false">Bioinformagics</a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-10" aria-expanded="false">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-10" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./bioinformagics_1_install_iterm.html" class="sidebar-item-text sidebar-link">install iTerm terminal app for MacOS</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./bioinformagics_2-3_advanced_sshing.html" class="sidebar-item-text sidebar-link">Advanced SSH-ing</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./bioinformagics_4_anaconda.html" class="sidebar-item-text sidebar-link">Installing Anaconda and Anaconda Environments</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./bioinformagics_6_jupyter.html" class="sidebar-item-text sidebar-link">jupyter Notebooks</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./bioinformagics_7_terminal_colors.html" class="sidebar-item-text sidebar-link">How to color code your terminal</a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#ddrda-library-prep" id="toc-ddrda-library-prep" class="nav-link active" data-scroll-target="#ddrda-library-prep">ddRDA Library Prep</a></li>
  <li><a href="#background" id="toc-background" class="nav-link" data-scroll-target="#background">Background</a></li>
  <li><a href="#materials-and-equipment" id="toc-materials-and-equipment" class="nav-link" data-scroll-target="#materials-and-equipment">Materials and Equipment</a>
  <ul class="collapse">
  <li><a href="#reagents-and-consumables" id="toc-reagents-and-consumables" class="nav-link" data-scroll-target="#reagents-and-consumables">Reagents and Consumables</a>
  <ul class="collapse">
  <li><a href="#equipment" id="toc-equipment" class="nav-link" data-scroll-target="#equipment">Equipment</a></li>
  </ul></li>
  <li><a href="#dna-quantification" id="toc-dna-quantification" class="nav-link" data-scroll-target="#dna-quantification">DNA quantification</a>
  <ul class="collapse">
  <li><a href="#dna-quantification-1" id="toc-dna-quantification-1" class="nav-link" data-scroll-target="#dna-quantification-1">DNA quantification</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#overview" id="toc-overview" class="nav-link" data-scroll-target="#overview">Overview</a>
  <ul class="collapse">
  <li><a href="#restriction-enzymes" id="toc-restriction-enzymes" class="nav-link" data-scroll-target="#restriction-enzymes">Restriction Enzymes</a></li>
  </ul></li>
  <li><a href="#protocol" id="toc-protocol" class="nav-link" data-scroll-target="#protocol">Protocol</a>
  <ul class="collapse">
  <li><a href="#before-you-start" id="toc-before-you-start" class="nav-link" data-scroll-target="#before-you-start">Before you start</a></li>
  <li><a href="#restriction-digest" id="toc-restriction-digest" class="nav-link" data-scroll-target="#restriction-digest">Restriction Digest</a></li>
  <li><a href="#dna-quantification-2" id="toc-dna-quantification-2" class="nav-link" data-scroll-target="#dna-quantification-2">DNA Quantification</a></li>
  <li><a href="#adaptor-ligation-approx.-7-hours---proceed-directly-to-pooling-after" id="toc-adaptor-ligation-approx.-7-hours---proceed-directly-to-pooling-after" class="nav-link" data-scroll-target="#adaptor-ligation-approx.-7-hours---proceed-directly-to-pooling-after">Adaptor Ligation ( approx. 7 hours) - proceed directly to pooling after</a></li>
  <li><a href="#pooling-and-beadwash" id="toc-pooling-and-beadwash" class="nav-link" data-scroll-target="#pooling-and-beadwash">Pooling and Beadwash</a></li>
  <li><a href="#size-selection-approx.-2-hours" id="toc-size-selection-approx.-2-hours" class="nav-link" data-scroll-target="#size-selection-approx.-2-hours">Size selection (approx. 2 hours)</a></li>
  <li><a href="#targeted-index-pcr-ampli-cation-8-hours-including-checking-on-gel-and" id="toc-targeted-index-pcr-ampli-cation-8-hours-including-checking-on-gel-and" class="nav-link" data-scroll-target="#targeted-index-pcr-ampli-cation-8-hours-including-checking-on-gel-and">Targeted Index PCR Amplication (8 hours including checking on gel and</a></li>
  <li><a href="#final-beadwash-and-pooling-of-pcr-products" id="toc-final-beadwash-and-pooling-of-pcr-products" class="nav-link" data-scroll-target="#final-beadwash-and-pooling-of-pcr-products">Final beadwash and pooling of PCR products</a></li>
  <li><a href="#check-libraries-on-bioanalyzer" id="toc-check-libraries-on-bioanalyzer" class="nav-link" data-scroll-target="#check-libraries-on-bioanalyzer">Check libraries on Bioanalyzer</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">



<section id="ddrda-library-prep" class="level1">
<h1>ddRDA Library Prep</h1>
<p><a href="https://github.com/DrK-Lo/lotterhoslabprotocols/edit/main/molecprot_ddradlibprep.qmd">Edit this page</a></p>
<p>The protocol for doing a library prep for ddRAD.</p>
<hr>
<p><em>Note: This protocol is outdated</em></p>
</section>
<section id="background" class="level1">
<h1>Background</h1>
<p><strong>D</strong>ouble <strong>d</strong>igest <strong>R</strong>estriction-site <strong>A</strong>ssociated <strong>D</strong>NA <strong>seq</strong>uencing (ddRADseq; Peterson et al.&nbsp;2012) is a popular approach which uses two restriction enzymes in combination with next generation sequencing to produce inexpensive genome-wide datasets. It does this by; first, using two separate restriction enyzmes that cleave DNA at designated Restriction sites located haphazardly along the genome (F.1A). Second, using a series of steps, labels are attached only to DNA fragments that were specifically cut with each enzyme (F.1B). Lastly, the remaining labeled fragments can be pooled (combined with other similarly treated DNA samples) then be sent away for sequencing (F.1C). Pooling is possible through the use of unique barcodes, which can be used to differentiate sequenced fragments by sample using downstream bioinformatics tools. FIGURE HERE</p>
<p>Important Considerations:</p>
<p>LIST HERE</p>
<p>Applications:</p>
<p>LIST HERE</p>
<p>Examples:</p>
<p>LIST HERE</p>
</section>
<section id="materials-and-equipment" class="level1">
<h1>Materials and Equipment</h1>
<section id="reagents-and-consumables" class="level2">
<h2 class="anchored" data-anchor-id="reagents-and-consumables">Reagents and Consumables</h2>
<ul>
<li>Note that we chose these enzymes because they were not sensitive to methylation.</li>
<li>Restriction Enzyme 1: (depends) SphI-HF (NEB, 20,000 units/ml) R3182S</li>
<li>Restriction Enyzme 2: (depends) MspI (R0106S), HpaII (R0171S) or (MluCI NEB, 10,000 units/mL R0538S)</li>
<li>10x Concentrate Enzyme Buffer (part of the enzyme package)</li>
<li>T4 DNA Ligase (NEB, 400,000 units/mL) M0202L</li>
<li>10x T4 buffer (part of the Ligase package)</li>
<li>BioRad Iproof High Fidelity DNA polymerase Cat # 172-5301 (424 for 250 uL [500 units])</li>
<li>DMSO (part of the package of the Iproof polymerase)</li>
<li>MgCl2 (part of the package of the Iproof polymerase)</li>
<li>1 mg/mL BSA (BSA may not be needed, it is now incorporated in the enzyme buffer, but we keep it in anyway) Cat # BP675-1 Fisher</li>
<li>1 M NaCl (follow reagent recipe)</li>
<li>dNTP Cat # R0181 Thermo Scientific</li>
<li>SphI-HF adaptors (add page number for ordering adaptors)</li>
<li>MluCI adaptors (add page number for ordering adaptors)</li>
<li>Milli-Q Water</li>
<li>96 well plates and strip caps or PCR strip tubes Cat # AB-0600 Thermo Scientific</li>
<li>1.5 microcentrifuge tubes for PCR mixes (any 1.5 microcentrifuge would work)</li>
<li>0.1X TE</li>
<li>SPRI or Homemade DNA beads (please see the Home made bead protocol if you need to make them)</li>
<li>EtOH molecular grade</li>
</ul>
<section id="equipment" class="level3">
<h3 class="anchored" data-anchor-id="equipment">Equipment</h3>
<ul>
<li>Vortexer</li>
<li>Pipets (ranges from 2uL-1000uL)</li>
<li>Thermal Cyclers</li>
<li>Full plate centrifuge (if using a full plate)</li>
<li>Qubit (we used the Qubit 3.0 Cat #Q33216) and special tubes Qubit Cat # Q32856 or VWR Cat # PCR-05-C</li>
</ul>
</section>
</section>
<section id="dna-quantification" class="level2">
<h2 class="anchored" data-anchor-id="dna-quantification">DNA quantification</h2>
<ol type="1">
<li>Qubit Reagents</li>
<li>Bioanalyzer chip</li>
</ol>
<section id="dna-quantification-1" class="level3">
<h3 class="anchored" data-anchor-id="dna-quantification-1">DNA quantification</h3>
<ol type="1">
<li>see ““Link””</li>
</ol>
</section>
</section>
</section>
<section id="overview" class="level1">
<h1>Overview</h1>
<p>Genomic DNA is first digested with two restriction enzymes. This results in a pool of fragments that have the sticky- ended restriction cut sites on either end, and these ends provide a template for ligation of the customized adaptor sequences. The adaptor sequences contain the Illumina adaptors and primer sequences for multiplexing sequencing (and hence provide a binding site for the Illumina PCR primers), and the first barcodes of 5-8 base pairs are incorporated to the left adaptor. The bases at one of the end of each adaptor oligo correspond to the ligation site; that is, they match the restriction cut site. NOTE THAT WE MAKE SURE OUR BARCODE DESIGN DOES NOT RE-CREATE THE RESTRICTION SITE. This can be important when using MspI, which is not inactivated by heat. The ligation reaction is pooled and concentrated with a bead wash in preparation for the size selection step. Size selection allows the fragments that are too small or oversized to be removed. The fragments of the appropriate size are amplied through a Polymerase Chain Reaction where the fragments with the left and right adapters get amplied (fragments that carry the wrong combination of adapters won`t amplify due to the selectiveness of the right adapter). The amplied products now have 2 barcodes, and they can be pooled and concentrated with a bead wash. The nal library is good for paired-end sequencing. In January 2016 we ordered new adapters and made some modications. The rst modication was to use the Peterson protocol, instead of a mix of dierent protocols. The second modication was the adapters. Both the p5 and p7 adapters were designed to be oset by a 0-2 bp insertion. For both reads, this will oset the sequencer at the restriction site so that it is not blinded by the laser. The p7 adapter (right) was designed to also contain a 4-bp degenerate base sequence that we can use to identify PCR duplicates. This sequence was accompanied by another 4-bp that were designed to make sure the adapters anneal together well. FOR DDRAD, YOU SHOULD CHOOSE ENZYMES THAT ARE NOT SENSITIVE TO DNA METHYLATION.</p>
<section id="restriction-enzymes" class="level2">
<h2 class="anchored" data-anchor-id="restriction-enzymes">Restriction Enzymes</h2>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Enzyme</th>
<th>Sequence</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>SphI</td>
<td><img src="https://www.neb.com/~/media/Catalog/All-Products/52FC70BA01D1413EB2F4396F7375707F/Cut%20Site%20Image/Sph-I-cutsite_1.gif" class="img-fluid" alt="SphICutSite"></td>
</tr>
<tr class="even">
<td>MspI</td>
<td><img src="https://www.neb.com/~/media/Catalog/All-Products/C7394B3BC98246FAAF04C42D33CC646A/Cut%20Site%20Image/Msp-I-cutsite_1.gif" class="img-fluid" alt="MSpICutSite"></td>
</tr>
</tbody>
</table>
</section>
</section>
<section id="protocol" class="level1">
<h1>Protocol</h1>
<section id="before-you-start" class="level2">
<h2 class="anchored" data-anchor-id="before-you-start">Before you start</h2>
<ul>
<li>Make sure the adaptors are annealed, and also make sure the adaptors and the primer mix are easily accessible in plate format. Please see the section above.</li>
<li>Be familiar with the Pippin Prep and how to use it.</li>
<li>Know what size range you want to select - see details in size selection section. To do this, you should have a plan for sequencing.</li>
<li>Make sure all of the supplies needed for the protocol are present and are sufficient for the number of samples that you will be processing. This protocol is long and it has few stopping steps.</li>
<li>Make sure the enzyme combination is the best for your organisms. The types of enzymes and the digestion length should be checked at least one time prior this protocol.</li>
</ul>
</section>
<section id="restriction-digest" class="level2">
<h2 class="anchored" data-anchor-id="restriction-digest">Restriction Digest</h2>
<p>From NEB website: There are several key factors to consider when setting up a restriction endonuclease digestion. Using the proper amounts of DNA, enzyme and buffer components in the correct reaction volume will allow you to achieve optimal digestion. By definition, 1 unit of restriction enzyme will completely digest 1 ug of substrate DNA in a 50 ul reaction in 60 minutes. This enzyme : DNA : reaction volume ratio can be used as a guide when designing reactions. However, most researchers follow the “typical” reaction conditions listed, where a 5:10 fold overdigestion is recommended to overcome variability in DNA source, quantity and purity. NEB offers the following tips to help you to achieve maximal success in your restriction endonuclease reactions. NEB recommends 10 units restriction enzyme, 1g DNA, 1X NEB Buer (5L), and 1 hour incubation time. DNA (measured with the Qubit) should ideally be at a minimum total amoung of 1.5 g and no more than 2.5 g. For example, you could use 10L of 150-250 ng/L of DNA. Keep on ice. Note that your DNA needs to be concentrated at least 125 ng/L for this protocol (2,500 ng desired/125ng/L = 20L + may need 5-7uL master mix depending on enzymes). For each sample prepare a master mix I (Table 1), mix by vortexing and then centrifuge. For these and all other reactions make sure to prepare an excess of mix to accommodate multiple rounds of pipetting, particularly if you are working with whole plates. Because the enzymes are stored in glycerol and other viscous solutions, a substantial volume is lost through adhesion to the outside of pipette tips. We suggest making more than what you think you will need (this may be overkill, if you have fewer columns than just do one extra per column).</p>
<table class="table">
<thead>
<tr class="header">
<th>Reagent</th>
<th>Number of samples 1X (uL)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>20 units of Enzyme 1</td>
<td>(Calc)</td>
</tr>
<tr class="even">
<td>20 units of Enzyme 2</td>
<td>(Calc)</td>
</tr>
<tr class="odd">
<td>(10x Concentrate Enzyme Buffer)</td>
<td>3uL</td>
</tr>
<tr class="even">
<td>Genomic DNA (1.5-2.5 ug)</td>
<td>(Calc)</td>
</tr>
<tr class="odd">
<td>Molecular grade H20 to make a final volume of 30 uL</td>
<td>(Calc)</td>
</tr>
</tbody>
</table>
<p>Note: The original protocol listed 10xT4 Buer instead of the 10X Concentrate Enzyme Buer. We think this may have been a typo. 10xT4 Buer is designed to work with the Ligase (we use Ligase in the following section) and the 10x Concentrate Enzyme Buer is designed to optimize the enzyme. See NEB website. 1. Incubate samples at 37C overnight. NOTE THAT MspI CANNOT BE KILLED BY HEAT. SUGGEST PROCEEDING IMMEDIATELY TO BEAD WASH. 2. After incubation, let reaction cool to room temperature during this time also remove the AMPURE XP beads from the fridge and let them warm to room temperature 3. Add 45 uL of AMPURE XP to each reaction, mix with pipetting 10 times, and then let incubate for 5 mins on the bench 4. Place plate onto the magnet plate and let the beads separate for 5 mins 5. Carefully remove the supernatant from each well and discard 6. Note that you will not be able to remove all of the supernatant without disturbing the beads. It’s best to leave about 5 uL remaining in each well 7. Leaving the plate in the magnet, wash the reaction by adding 200 uL of FRESH 70% ethanol 70% ethanol should be mixed that day! 8. Remove and discard ethanol and repeat with a second ethanol wash. (From Gold Lab Protocol: Puritz prefers to use only 150 uL of ethanol for second wash. Hollenbeck prefers to use only 1 wash step) 9. After nal wash, let plates air dry for 5-10 mins. All ethanol needs to be evaporated, but beads should not be overdried. 10. Remove plate from magnet and add 30 uL of molecular grade H2O to each well and incubate for 5 mins. Make sure that water comes into contact with beads by mixing well with pipet 11. Place plate back on magnet and let it separate for 2 minutes, then remove supernatant and transfer to clean plate 12. If doing a second set of digestion, repeat steps 4-11.</p>
</section>
<section id="dna-quantification-2" class="level2">
<h2 class="anchored" data-anchor-id="dna-quantification-2">DNA Quantification</h2>
<p>Check quantication of DNA on Qubit. For extra accuracy, reread standards every 48 samples. These numbers will be used in next step.</p>
</section>
<section id="adaptor-ligation-approx.-7-hours---proceed-directly-to-pooling-after" class="level2">
<h2 class="anchored" data-anchor-id="adaptor-ligation-approx.-7-hours---proceed-directly-to-pooling-after">Adaptor Ligation ( approx. 7 hours) - proceed directly to pooling after</h2>
<p><strong>Note: if performing bs-ddRADseq please remember to incorporate specific methylated adapters!</strong></p>
<ol type="1">
<li>Thaw adaptors (or the L and R adaptors). Have these adaptors annealed and easily accessible in plate format.</li>
<li>For each sample, calculate the amount of template needed for 100 ng of DNA and calculate the amount of H20 to add to this amount by using the formula 22.2-(ul of Template)= uL of H20 (total final volume should be 22.2 uL). If samples are of low DNA concentration, this can be adjusted to 31.2-(ul of Template)= uL of H20 for all samples.</li>
<li>Place the correct amount of H20 and then DNA into ligation plate A. This will take some time. Take out T4 ligase buffer to thaw during this time.</li>
<li>Transfer 2 uL of ” adapter to each reaction. PAY ATTENTION TO YOUR DESIGN.</li>
<li>Create ligation master mix as follows:</li>
</ol>
<table class="table">
<thead>
<tr class="header">
<th>Reagent</th>
<th>Number of samples 1X (uL)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>DNA template</td>
<td>22.2</td>
</tr>
<tr class="even">
<td>(NEB Buffer #4 10X) ligation buffer</td>
<td>3 uL</td>
</tr>
<tr class="odd">
<td>“right universal adapter</td>
<td>2 uL</td>
</tr>
<tr class="even">
<td>T4 ligase (Add this last</td>
<td>0.8 uL</td>
</tr>
</tbody>
</table>
<ol start="6" type="1">
<li>Add 5.8 uL of master mix to each reaction, as quickly as possible</li>
<li>Incubate on bench top for one hour, then place in thermocycler and heat-kill at 65C for 10 min. After the heat-kill, cool the solution at 2C per 90 seconds until it reaches room temperature. Hold at 4C or place in fridge.</li>
</ol>
</section>
<section id="pooling-and-beadwash" class="level2">
<h2 class="anchored" data-anchor-id="pooling-and-beadwash">Pooling and Beadwash</h2>
<p>Set up your pools for size selection on the Pippin Prep. You should have experience using the Pippin before you perform this step. 1. After ligation, it is safe to pool samples within one index. Collect all 30 uL from each well and place it into a single 1.5 mL tube (Use a 2.0 mL tube if using 40 uL ligations). If you are using 16 barcodes within an index, this is 16*30 = 480 uL. 2. Mix each pool with pipetting or gentle vortexing. If it is a large volume split each into equal aliquots according to the number of replicate size selections. (These replicates will be combined in last step. It doesn’t aect the total amount of beads used, and you can choose to do fewer if you don’t have a lot of product). 3. Add 1.5X uL of AMPURE XP to each aliquot, mix with pipetting 10 times, and then let incubate for 5 mins on the bench. For instance, if you did two replicate size selections of a pool with 16 barcodes, each one would be 480ul/4 = 120 uL of template and 180 uL of beads. 4. Place tube onto the magnet bar and let the beads separate for 5 mins 5. Carefully remove the supernatant from each tube and discard 6. Sometimes pulling tube upwards and tilting bottom towards the magnet allows for complete supernatant removal. If not, leave a few uL. 7. Leaving the tubes in the magnet, wash the reaction by adding 500 uL of FRESH 70% ethanol 70% ethanol should be mixed that day!. Volume is approximate but should cover all beads 8. Remove ethanol, let tubes air dry for 2 mins. All excess ethanol needs to be evaporated, but shiny beads are OK and beads should not be overdried. Pippin is sensitive to EtOH so err on side of more dry. 9. IF DOING REPLICATES: Remove tubes from magnet and add 30 uL of molecular grade H2O to each tube and incubate for 5 mins. Make sure that water comes into contact with beads by mixing well with pipet or gentle vortexing. 10. IF NOT DOING REPLICATES: Remove tubes from magnet and add 30 uL of TE to each tube and incubate for 5 mins. Make sure that water comes into contact with beads by mixing well with pipet or gentle vortexing. (We use TE here because it helps the DNA stay in the well of the Pippin Prep). 11. Place tubes back on magnet and let beads separate for 2 minutes, then remove supernatant and transfer to a clean 1.5 mL tube. If you used replicates, pool within each index. This means going from multiple aliquots to one tube for each index. 12. IF YOU USED REPLICATES, DO THE FOLLOWING STEPS TO POOL. IF NOT, YOU ARE DONE. If you used replicates, add 1.5X uL of AMPURE XP to each tube, mix with pipetting 10 times, and then let incubate for 5 mins on the bench. 13. Place tubes onto the magnet bar and let the beads separate for 5 mins 14. Carefully remove the supernatant from each tube and discard. Sometimes pulling tube upwards and tilting bottom towards the magnet allows for complete supernatant removal. If not, leave a few uL. 15. Leaving the tubes in the magnet, wash the reaction by adding 500 uL of FRESH 70% ethanol. Volume is approximate but should cover all beads 16. Remove and discard ethanol, and let plates air dry for 2 mins 17. All ethanol needs to be evaporated, but beads should not be overdried. Pippin is sensitive to EtOH so err on side of more dry. 18. Remove tubes from magnet and add 20 uL of TE to each tube and incubate for 5 mins A. Make sure that water comes into contact with beads by mixing well with pipet or gentle vortexing. We use TE here because it helps the DNA stay in the well for the pippin. 19. Place tubes back on magnet and let beads separate for 2 minutes, then remove supernatant and transfer to a clean 1.5 mL tube.</p>
<p>SAFE STOP POINT. Store at 4C for a month, or -20C for longer. It is important to save this, as all of this product is not used for the PCR, and if the results of the PCR do not look good, you can start again from this step.</p>
<p><img src="http://rs870.pbsrc.com/albums/ab261/rainbowquery/STOP.gif~c200" width="30"></p>
</section>
<section id="size-selection-approx.-2-hours" class="level2">
<h2 class="anchored" data-anchor-id="size-selection-approx.-2-hours">Size selection (approx. 2 hours)</h2>
<p>We use a Pippin Prep to do size selection. Pippin works in a similar way to typical gel electrophoresis, but uses standardized gel cassettes and automated size selection, rather than a hand-poured gel and manual excision of fragments from a gel. It is necessary to specify a size range of fragments that you would like represented in the nal genomic library for sequencing. The target size range can be adjusted to modify the number of fragments expected for sequencing. Because techniques such as this typically result in a negative relationship between fragment size and number, selecting larger fragments should decrease the nal number of fragments in the template and increase the coverage depth of these regions after sequencing. Similarly, decreasing the size interval selected will also reduce the number of fragments and increase coverage. The number of fragments produced is also in part a function of genome size, so awareness of your organism<code>s genome size is helpful in making an appropriate choice for size selection (see Fig. 3 of Alex</code>s protocol, and Figure 1 in SCHWEYEN 2014). We have typically used a 100-150 bp range somewhere between 250 and 500 bp in total fragment length (meaning that we might select, as an example, fragments that are 300 - 400 bp in length). Dr Lotterhos has an R script that can be used to do an in silico digest of a genome. It is called ddRADoysterdigestion.R (for C. gigas oyster) but can be applied to any genome. The basic code is in a post on her blog. You should calculate your size range based on the the number of fragments in your selected size range, and your plan for sequencing (the number of paired-end reads and their read length and the total number of bases captured).</p>
</section>
<section id="targeted-index-pcr-ampli-cation-8-hours-including-checking-on-gel-and" class="level2">
<h2 class="anchored" data-anchor-id="targeted-index-pcr-ampli-cation-8-hours-including-checking-on-gel-and">Targeted Index PCR Amplication (8 hours including checking on gel and</h2>
<p>nal beadwash) Following Ligation, pooled groups of samples with individual left adapters are amplied with a unique right adapter. The right adapter would add a unique barcode to the right side that would allow all of the pools to be pooled in one lane for sequencing. To ameliorate stochastic dierences in PCR production of fragments in reactions, we run 3-8 replicate reactions per right barcode group, and later combine them. NOTE THAT WE USE IPROOF HF-TAQ, which is cheaper than the Phusion Taq that most people are using. There are alternative polymerases for ddRAD-seq libraries that may be equally or more eective than Iproof Taq. NEB has good alternatives. 1. Pool all the individuals that have individual left barcodes (the samples would most likely be already pooled from the size selection step). This protocol was original made to amplify a total of 10-72ng NEED TO DOUBLE CHECK THIS of DNA, so grab a representative amount of DNA within this size range. 2. Prepare the following Master Mix. Prepare an individual master mix for each individual right adapter according to the number of replicate PCRs that you want to run. WE SUGGEST 6 REPLICATES.</p>
<table class="table">
<thead>
<tr class="header">
<th>Reagent</th>
<th>Number of samples 1X (uL)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ddH2O</td>
<td>1.75 ( or calc )</td>
</tr>
<tr class="even">
<td>5x Iproof buffer</td>
<td>4</td>
</tr>
<tr class="odd">
<td>dNTP (10 nM)</td>
<td>0.4</td>
</tr>
<tr class="even">
<td>Mg Cl2 (50 mM)</td>
<td>0.5</td>
</tr>
<tr class="odd">
<td>PCR Primer 1 (10uM) universal</td>
<td>2</td>
</tr>
<tr class="even">
<td>PCR Primer 2 (10uM) index specific</td>
<td>2</td>
</tr>
<tr class="odd">
<td>Iproof Taq</td>
<td>0.2</td>
</tr>
<tr class="even">
<td>DMSO</td>
<td>0.15</td>
</tr>
<tr class="odd">
<td>DNA pooled template (after size selection)</td>
<td>5 (or calc)</td>
</tr>
</tbody>
</table>
<ol start="3" type="1">
<li>Thermocycler prole for this PCR: 98C for 1min; 12 cycles of: 98C for 10s, 62C for 30s, 72C for 30s; nal extension at 72C for 10 min. Hold at 4C.</li>
<li>Check if the amplication of the PCR product has gone well by running 2ul of the PCR product in a 2% agarose gel (you can use the same gel to run the nal pooled PCR product cleaned made in the following step).</li>
</ol>
</section>
<section id="final-beadwash-and-pooling-of-pcr-products" class="level2">
<h2 class="anchored" data-anchor-id="final-beadwash-and-pooling-of-pcr-products">Final beadwash and pooling of PCR products</h2>
<ol type="1">
<li>Combine the replicate PCR reactions into a single 1.5 mL tube (6 * 16ul = 96 uL)</li>
<li>Add 144 uL (96*1.5 uL) of AMPURE XP to each aliquot, mix with pipetting 10 times, and then let incubate for 5 mins on the bench.</li>
<li>Place tube onto the magnet bar and let the beads separate for 5 mins</li>
<li>Carefully remove the supernatant from each tube and discard. Sometimes pulling tube upwards and tilting bottom towards the magnet allows for complete supernatant removal. If not, leave a few uL.</li>
<li>Leaving the tubes in the magnet, wash the reaction by adding 500 uL of FRESH 70% ethanol A. 70% ethanol should be mixed that day! Volume is approximate but should cover all beads</li>
<li>Remove ethanol and repeat wash</li>
<li>Remove ethanol, let tubes air dry for 5-10 mins. All ethanol needs to be evaporated, but beads should not be overdried</li>
<li>Remove tubes from magnet and add 30 uL of molecular grade H2O to each tube and incubate for 5 mins</li>
<li>Make sure that water comes into contact with beads by mixing well with pipet or gentle vortexing</li>
<li>Place tubes back on magnet and let beads separate for 2 minutes, then remove supernatant and transfer each index to a clean 1.5 mL tube.</li>
<li>These four tubes are your nal library</li>
<li>Prepare working solution for the Qubit by combining 199 uL of stock solution with 1 uL of dye for every reaction.</li>
<li>Collect 0.5 ml sample tubes, one for every reaction and two for standards</li>
<li>Prepare standards by combing 190 uL of working stock with 10 uL of standard (stored in refrigerator)</li>
<li>Turn on Qubit by touching screen and press button to read new standards. Follow prompts on screen.</li>
<li>Prepare 6 samples for quantication by putting 199 uL of working stock with 1 uL of sample in a 0.5ml tube</li>
<li>Use Qubit to measure DNA content, remembering to press button to calculate stock concentration and to press button to save data before switching tube and pressing new sample”</li>
<li>If each index is between 5-100 ng/uL, celebrate. You are finished.</li>
</ol>
</section>
<section id="check-libraries-on-bioanalyzer" class="level2">
<h2 class="anchored" data-anchor-id="check-libraries-on-bioanalyzer">Check libraries on Bioanalyzer</h2>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>