<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.251">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>LotterhosLabProtocols – code_bioinformatics</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar docked nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">LotterhosLabProtocols</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html">Home</a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./about.html">About</a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title">(Untitled)</h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation docked overflow-auto">
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">Coding Protocols</a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./code_bioinformatics.html" class="sidebar-item-text sidebar-link active">Bioinformatics Resources</a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#bioinformatics-resources" id="toc-bioinformatics-resources" class="nav-link active" data-scroll-target="#bioinformatics-resources">Bioinformatics Resources</a>
  <ul class="collapse">
  <li><a href="#data-skills" id="toc-data-skills" class="nav-link" data-scroll-target="#data-skills"><strong>Data Skills</strong></a></li>
  <li><a href="#rad-seq" id="toc-rad-seq" class="nav-link" data-scroll-target="#rad-seq"><strong>RAD seq</strong></a></li>
  <li><a href="#vcf-files-and-population-structure" id="toc-vcf-files-and-population-structure" class="nav-link" data-scroll-target="#vcf-files-and-population-structure"><strong>VCF files and population structure</strong></a>
  <ul class="collapse">
  <li><a href="#filter-vcf-table-and-produce-a-pca-plot" id="toc-filter-vcf-table-and-produce-a-pca-plot" class="nav-link" data-scroll-target="#filter-vcf-table-and-produce-a-pca-plot"><strong>Filter VCF table and produce a PCA plot</strong></a></li>
  <li><a href="#structure-plot" id="toc-structure-plot" class="nav-link" data-scroll-target="#structure-plot"><strong>Structure plot</strong></a></li>
  <li><a href="#run-an-rda" id="toc-run-an-rda" class="nav-link" data-scroll-target="#run-an-rda"><strong>Run an RDA</strong></a></li>
  <li><a href="#run-a-gwas-or-gea" id="toc-run-a-gwas-or-gea" class="nav-link" data-scroll-target="#run-a-gwas-or-gea"><strong>Run a GWAS or GEA</strong></a></li>
  <li><a href="#other-resources" id="toc-other-resources" class="nav-link" data-scroll-target="#other-resources"><strong>Other resources</strong></a></li>
  </ul></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">



<section id="bioinformatics-resources" class="level1">
<h1>Bioinformatics Resources</h1>
<hr>
<section id="data-skills" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="data-skills"><strong>Data Skills</strong></h2>
<p>Vince Buffalo’s <em>Bioinformatic Data Skills</em> is a great book to build skills. Katie has a set of great readings for learning, contact her for more info.</p>
</section>
<section id="rad-seq" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="rad-seq"><strong>RAD seq</strong></h2>
<p>https://radcamp.github.io/NYC2019/RADCamp-PartII-Day1-AM.html</p>
<p>https://github.com/dereneaton/ipyrad/blob/master/newdocs/assembly_guidelines.rst</p>
<p>http://www.popgen.dk/software/index.php/PCAngsd handle genotype likelihoods</p>
</section>
<section id="vcf-files-and-population-structure" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="vcf-files-and-population-structure"><strong>VCF files and population structure</strong></h2>
<p>What is a VCF? https://samtools.github.io/hts-specs/VCFv4.2.pdf</p>
<p>Katie also has some lectures on VCF files.</p>
<section id="filter-vcf-table-and-produce-a-pca-plot" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="filter-vcf-table-and-produce-a-pca-plot"><strong>Filter VCF table and produce a PCA plot</strong></h3>
<p>When getting started, it’s best to filter for relatedness and linkage disequilibrium, so that you have a quasi-independent set of individuals and SNPs.</p>
<p>Check related individuals with vcf using vcftools (–relatedness)</p>
<p>Filter the VCF file. Use VCF tools to do this: http://vcftools.sourceforge.net/</p>
<ul>
<li><p>for minor alleles &lt; 0.05 (if you have a lot of SNPs, or 0.01 if you don’t have a lot of SNPs)</p></li>
<li><p>for missing data at SNPs (keep SNPs in at least X% of individuals in all dataset)</p></li>
<li><p>for missing data at population level (keeps SNPs with calls in X% or higher of individuals per population)</p></li>
<li><p>for individuals with low SNP count (keep individuals with higher than X% of SNPs called)</p></li>
<li><p>Depending on the context, X% might be 80% or might be 100%</p></li>
</ul>
<p>To use the bigsnpr R package, you will have to change your data from the VCF format to a genotype matrix with 0/1/2 counts of the alternate allele. <a href="https://github.com/laurabenestan/RDA_outlier">Transform from vcf to (plink to) raw</a>. See the first steps in this tutorial.</p>
<p>The presence of LD can bias principal components. This <a href="https://academic.oup.com/bioinformatics/article/36/16/4449/5838185?login=true">paper</a> explains the problem. Read about <a href="https://privefl.github.io/bigsnpr/articles/pruning-vs-clumping.html">filtering for LD with pruning and clumping</a></p>
<p><a href="https://privefl.github.io/bigsnpr/articles/how-to-PCA.html">Filter the raw data using snp_autoSVD (it’s at the end of the tutorial)</a> This tutorial also teaches you how to do a PCA.</p>
</section>
<section id="structure-plot" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="structure-plot"><strong>Structure plot</strong></h3>
<p>For Structure use LEA program sNMF package vcf input in R. (http://membres-timc.imag.fr/Olivier.Francois/LEA/tutorial.htm)</p>
</section>
<section id="run-an-rda" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="run-an-rda"><strong>Run an RDA</strong></h3>
<p>Before you run an RDA and do some of the things in these tutorials, talk to Dr.&nbsp;L about your plan. Outlier SNP analysis with RDA have a lot of false positives, but the individual scores are very useful.</p>
<p>https://popgen.nescent.org/2018-03-27_RDA_GEA.html</p>
<p>https://github.com/laurabenestan/RDA_outlier</p>
<p>Katie also has some RDA code on simulations and oyster data.</p>
</section>
<section id="run-a-gwas-or-gea" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="run-a-gwas-or-gea"><strong>Run a GWAS or GEA</strong></h3>
<p>Genotype-phenotype association: LFMM2. <a href="https://academic.oup.com/mbe/article/36/4/852/5290100">Read the paper lfmm2</a> and <a href="https://onlinelibrary.wiley.com/doi/10.1111/1755-0998.13366">LEA3</a></p>
<p><a href="https://rdrr.io/bioc/LEA/man/lfmm2.html">LFMM2 functions</a></p>
<p><a href="http://www.bioconductor.org/packages/3.3/bioc/html/LEA.html">LEA installation</a> Check with Katie this is correct before running</p>
</section>
<section id="other-resources" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="other-resources"><strong>Other resources</strong></h3>
<p><a href="https://marineomics.github.io/index.html">Marine Omics working group</a></p>


</section>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>